<div class="container">

<table style="width: 100%;"><tr>
<td>g3Lollipop</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Render g3lollipop diagram for the given mutation data</h2>

<h3>Description</h3>

<p>Render g3lollipop diagram for the given mutation data
</p>


<h3>Usage</h3>

<pre><code class="language-R">g3Lollipop(
  mutation.dat,
  gene.symbol,
  uniprot.id = NA,
  gene.symbol.col = "Hugo_Symbol",
  aa.pos.col = "AA_Position",
  protein.change.col = c("Protein_Change", "HGVSp_Short"),
  factor.col = "Mutation_Class",
  plot.options = g3Lollipop.options(),
  save.png.btn = TRUE,
  save.svg.btn = TRUE,
  btn.style = NA,
  output.filename = "output"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mutation.dat</code></td>
<td>
<p>Input genomic mutation data frame</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.symbol</code></td>
<td>
<p>HGNC primary gene symbol</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>uniprot.id</code></td>
<td>
<p>UniProt ID, in case that the specified gene symbol links to
multiple UniProt entries (isoforms). For example, <em>AKAP7</em> gene has two
isoforms in <a href="https://www.uniprot.org/">UniProt</a>,
<a href="https://www.uniprot.org/uniprot/O43687">O43687</a> and
<a href="https://www.uniprot.org/uniprot/Q9P0M2">Q9P0M2</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.symbol.col</code></td>
<td>
<p>Column name of Hugo gene symbols (e.g., TP53). Default
<em>Hugo_Symbol</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aa.pos.col</code></td>
<td>
<p>Column name of the parsed amino-acid change position.
Default <em>AA_Position</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein.change.col</code></td>
<td>
<p>Column name of protein change information (e.g.,
p.K960R, G658S, L14Sfs*15). Default is a list of <em>Protein_Change</em>,
<em>HGVSp_Short</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>factor.col</code></td>
<td>
<p>column of classes in the plot legend. IF <code>NA</code>, use
parsed <em>Mutation_Class</em> column, otherwise, use specified.  Default
<code>NA</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.options</code></td>
<td>
<p>g3lollipop diagram options in list format. Check
<code>g3Lollipop.options</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save.png.btn</code></td>
<td>
<p>If add <em>save-as-png</em> button to the diagram. Default
<code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save.svg.btn</code></td>
<td>
<p>If add <em>save-as-svg</em> button to the diagram. Default
<code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>btn.style</code></td>
<td>
<p>button style, including browser default button style, and
two built-in styles, <em>blue</em> or <em>gray</em>. Default <code>NA</code>,
indicating browser default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.filename</code></td>
<td>
<p>Specify output file name.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>lollipop diagram for the given mutation data. The chart is interactive
within either Shiny applications or Rmd documents under the bindings.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# system mutation data
maf.file &lt;- system.file("extdata", "TCGA.BRCA.varscan.somatic.maf.gz", package = "g3viz")
# read in MAF file
mutation.dat &lt;- readMAF(maf.file)

# use built-in chart theme
chart.options &lt;- g3Lollipop.theme(theme.name = "default",
                                  title.text = "PIK3CA gene (default theme)")
# generate chart
g3Lollipop(mutation.dat,
           gene.symbol = "PIK3CA",
           plot.options = chart.options,
           btn.style = "blue",
           output.filename = "default_theme")

</code></pre>


</div>