<div class="container">

<table style="width: 100%;"><tr>
<td>getMutationsFromCbioportal</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Query cancer genomic mutation data from cBioPortal</h2>

<h3>Description</h3>

<p>Retrieve and parse mutation data from cBioPortal by the given
cBioPortal cancer study ID and the gene symbol.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getMutationsFromCbioportal(
  study.id,
  gene.symbol,
  output.file = NA,
  mutation.type.to.class.df = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>study.id</code></td>
<td>
<p>cbioprotal study ID</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.symbol</code></td>
<td>
<p>HGNC gene symbol.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.file</code></td>
<td>
<p>if specified, output to a file in <em>CSV</em> format.
Default is <code>NA</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutation.type.to.class.df</code></td>
<td>
<p>mapping table from mutation type to class.
See <code>mapMutationTypeToMutationClass</code> for details. Default
<code>NA</code>, which indicates to use default mappings.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a data frame with columns
</p>

<dl>
<dt>Hugo_Symbol</dt>
<dd>
<p>Hugo gene symbol</p>
</dd>
<dt>Protein_Change</dt>
<dd>
<p>Protein change information (cBioportal uses <em>HGVSp</em> format)</p>
</dd>
<dt>Sample_ID</dt>
<dd>
<p>Sample ID</p>
</dd>
<dt>Mutation_Type</dt>
<dd>
<p>mutation type, aka, variant classification.</p>
</dd>
<dt>Chromosome</dt>
<dd>
<p>chromosome</p>
</dd>
<dt>Start_Position</dt>
<dd>
<p>start position</p>
</dd>
<dt>End_Position</dt>
<dd>
<p>end position</p>
</dd>
<dt>Reference_Allele</dt>
<dd>
<p>reference allele</p>
</dd>
<dt>Variant_Allele</dt>
<dd>
<p>variant allele</p>
</dd>
<dt>Mutation_Class</dt>
<dd>
<p>mutation class (e.g., Truncating/Missense/Inframe/Other)</p>
</dd>
<dt>AA_Position</dt>
<dd>
<p>amino-acid position of the variant; if the variant is not in
protein-coding region, <code>NA</code></p>
</dd>
</dl>
<h3>Examples</h3>

<pre><code class="language-R">
# Usage:

</code></pre>


</div>