<div class="container">

<table style="width: 100%;"><tr>
<td>find_genes_qtls_around_markers</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search genes and QTLs around candidate regions</h2>

<h3>Description</h3>

<p>Takes a list of candidate markers and or regions (haplotypes, CNVs, windows, etc.) and search for genes or QTLs in a determined interval
</p>


<h3>Usage</h3>

<pre><code class="language-R">find_genes_qtls_around_markers(
  db_file,
  marker_file,
  method = c("gene", "qtl"),
  marker = c("snp", "haplotype"),
  interval = 0,
  nThreads = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>db_file</code></td>
<td>
<p>The data frame obtained using the import_gff_gtf() function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker_file</code></td>
<td>
<p>The file with the SNP or haplotype positions. Detail: For SNP files, the columns “CHR” and “BP” with the chromosome and base pair position, respectively, are mandatory. For the haplotype, the following columns are mandatory: “CHR”, “BP1” and “BP2”</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>“gene” or “qtl”</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker</code></td>
<td>
<p>"snp" or "haplotype"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interval</code></td>
<td>
<p>The interval in base pair which can be included upstream and downstream from the markers or haplotype coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nThreads</code></td>
<td>
<p>Number of threads to be used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical value defining if messages should of not be printed during the analysis (default=TRUE)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A dataframe with the genes or QTLs mapped within the specified intervals
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(QTLmarkers)
data(gffQTLs)
out.qtls&lt;-find_genes_qtls_around_markers(db_file=gffQTLs, marker_file=QTLmarkers,
method = "qtl", marker = "snp",
interval = 500000, nThreads = 1)
</code></pre>


</div>