<div class="container">

<table style="width: 100%;"><tr>
<td>read.vcf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Create a <code>bed.matrix</code> from VCF files </h2>

<h3>Description</h3>

<p> Create a <code>bed.matrix</code> from a <code>.vcf</code> file.</p>


<h3>Usage</h3>

<pre><code class="language-R"> read.vcf(file, max.snps, get.info = FALSE, convert.chr = TRUE, 
                 verbose = getOption("gaston.verbose",TRUE)) </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p> The name of the VCF file to read</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.snps</code></td>
<td>
<p> The maximal number of SNPs to read </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>get.info</code></td>
<td>
<p> If <code>TRUE</code>, the INFO field from the VCF file will integrated
in <code>@ped$info</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>convert.chr</code></td>
<td>
<p> If <code>TRUE</code>, chromosomes ids <code>"X"</code>, <code>"Y"</code> and <code>"MT"</code>
will be converted in their numeric equivalents</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> If <code>TRUE</code>, display information on the function progress </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The vcf format is described in <a href="https://github.com/samtools/hts-specs">https://github.com/samtools/hts-specs</a>
</p>
<p>In addition to the usual data in the slot <code>@snps</code>, the bed.matrices produced by <code>read.vcf</code> have
<code>@snps$quality</code> and <code>@snps$filter</code> columns corresponding to the QUAL and FILTER fields in the VCF
file. If <code>get.info = TRUE</code>, an additionnal column <code>@snps$info</code> is added, corresponding to the
INFO field.
</p>
<p>The information about individuals in VCF files is incomplete: in the slot <code>@ped</code>, the columns
<code>@ped$famid</code> and <code>@ped$id</code> will both contain the sample id; sex and phenotypes will be set
to unknown.
</p>
<p>The function currently assumes that the <code>GT</code> field is the first field in the genotypes format.
If it is not the case, the variants are discarded.
</p>


<h3>Value</h3>

<p> A <code>bed.matrix</code> </p>


<h3>Author(s)</h3>

<p> Herv√© Perdry and Claire Dandine-Roulland </p>


<h3>See Also</h3>

 <p><code>read.bed.matrix</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">## Read vcf file (from file name)
filepath &lt;-system.file("extdata", "LCT.vcf.gz", package="gaston")
x1 &lt;- read.vcf( filepath )
x1
</code></pre>


</div>