<div class="container">

<table style="width: 100%;"><tr>
<td>dupli</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Small data set to illustrate <code>SNP.rm.duplicates</code>
</h2>

<h3>Description</h3>

<p>The SNPs in this data frame are as follows:
</p>

<dl>
<dt>SNP 1.</dt>
<dd>
<p> Unduplicated SNP </p>
</dd>
<dt>SNPs 2a and 2b.</dt>
<dd>
<p> Two duplicated SNPs with identical alleles </p>
</dd>
<dt>SNPs 3a and 3b.</dt>
<dd>
<p> Two duplicated SNPs with swapped alleles </p>
</dd>
<dt>SNPs 4a and 4b.</dt>
<dd>
<p> Two duplicated SNPs with flipped reference strand </p>
</dd>
<dt>SNPs 5a and 5b.</dt>
<dd>
<p> Two duplicated SNPs with swapped alleles and flipped reference strand </p>
</dd>
<dt>SNPs 6a and 6b.</dt>
<dd>
<p> Two duplicated SNPs with incompatible alleles </p>
</dd>
<dt>SNPs 7a and 7b.</dt>
<dd>
<p> Two duplicated SNPs including one monomorphic SNP (one allele set to <code>"0"</code>) </p>
</dd>
<dt>SNPs 8a, 8b and 8c.</dt>
<dd>
<p> Three duplicated SNPs </p>
</dd>
<dt>SNPs 9a, 9b and 9c.</dt>
<dd>
<p> Three duplicated SNPs with incompatible alleles </p>
</dd>
</dl>
<h3>Usage</h3>

<pre><code class="language-R"> data(dupli) </code></pre>


<h3>Format</h3>

<p>There are three data objects in the dataset:
</p>

<dl>
<dt><code>dupli.gen</code></dt>
<dd>
<p> Genotype matrix </p>
</dd>
<dt><code>dupli.ped</code></dt>
<dd>
<p> Data frame containing all variables corresponding to a <code>.fam</code> file  </p>
</dd>
<dt><code>dupli.bim</code></dt>
<dd>
<p> Data frame containing all variables corresponding to a <code>.bim</code> file </p>
</dd>
</dl>
<h3>See Also</h3>

 <p><code>SNP.rm.duplicates</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(dupli)
x &lt;- as.bed.matrix(dupli.gen, fam = dupli.ped, bim = dupli.bim)
</code></pre>


</div>