<div class="container">

<table style="width: 100%;"><tr>
<td>set.hwe</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Hardy-Weinberg Equilibrium </h2>

<h3>Description</h3>

<p>Returns an updated <code>bed.matrix</code> with a new variable for the <code class="reqn">p</code>-values of an
Hardy-Weinberg Equilibrium test.
</p>


<h3>Usage</h3>

<pre><code class="language-R"> set.hwe(x, method = c("chisquare", "exact"),
           verbose = getOption("gaston.verbose", TRUE))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A <code>bed.matrix</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p> The method to use, either "chisquare" or "exact"  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> If <code>TRUE</code>, display information on the function actions </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Two tests of Hardy-Weinberg Equilibrium are proposed:
</p>

<ul>
<li>
<p> if <code>method = "chisquare"</code>, the good old Chi-square test
</p>
</li>
<li>
<p> if <code>method = "exact"</code>, Haldane's exact test (see Wigginton et al)
</p>
</li>
</ul>
<p>The function <code>set.stats</code> will be called first if necessary.
</p>
<p>The <code class="reqn">p</code>-value is set to <code class="reqn">1.0</code> for SNPs on chromosomes Y and MT. For SNPs on
chromosomes X, currently, the test is performed using only the genotypic counts of women.
</p>


<h3>Value</h3>

<p>A <code>bed.matrix</code> similar to <code>x</code>, with a new variable <code>x@snps$hwe</code>
containing the <code class="reqn">p</code>-values for each SNP.
</p>


<h3>Author(s)</h3>

<p>Herv√© Perdry and Claire Dandine-Roulland</p>


<h3>References</h3>

<p> Wigginton, J. E., Cutler, D. J., &amp; Abecasis, G. R. (2005). <em>A note on exact tests of Hardy-Weinberg equilibrium</em>. The American Journal of Human Genetics, <b>76(5)</b>, <b>887-893</b> </p>


<h3>See Also</h3>

 <p><code>set.stats</code>, <code>set.genomic.sex</code> </p>


<h3>Examples</h3>

<pre><code class="language-R"># Load data
data(LCT)
x &lt;- as.bed.matrix(LCT.gen, LCT.fam, LCT.bim)

# Compute Hardy-Weinberg p-values
x &lt;- set.hwe(x)
head( x@snps[,c("id","hwe")] )
</code></pre>


</div>