<div class="container">

<table style="width: 100%;"><tr>
<td>set.dist</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Set Genetic Distance </h2>

<h3>Description</h3>

<p>Returns an updated <code>bed.matrix</code> with genetic 
distances in centimorgan computed from the variant positions
</p>


<h3>Usage</h3>

<pre><code class="language-R"> set.dist(x, map, verbose = getOption("gaston.verbose", TRUE)) </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A <code>bed.matrix</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p> The genetic map, given by a list of data frames (see Details) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> If <code>TRUE</code>, display information on the function actions </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>A map is a list of data frames, with names corresponding to chromosomes.
Each of these data frames must have columns <code>pos</code> and <code>dist</code> corresponding
to positions in bp and cM, respectively.
</p>
<p>Such maps are too large to be included in a CRAN package. You can get two genetic
maps for the Human Genome (build 36 and 37) in the package <code>HumanGeneticMap</code>  
on GitHub.
</p>
<p>To install this package, run
</p>
<p><code> install.packages("HumanGeneticMap", repos="https://genostats.github.io/R/") </code>
</p>
<p>You can then use this function with <code>set.dist(x, HumanGeneticMap::genetic.map.b36)</code>
for example, for positions on the build 36. Use <code>map = HumanGeneticMap::genetic.map.b37</code>) 
for the build 37.
</p>


<h3>Value</h3>

<p>A <code>bed.matrix</code> similar to <code>x</code>, with updated values in <code>x@snps$dist</code>.
</p>


</div>