<div class="container">

<table style="width: 100%;"><tr>
<td>set.stats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Basic statistics for a <code>bed.matrix</code> </h2>

<h3>Description</h3>

<p>Return an updated <code>bed.matrix</code> with new variables for 
several basic statistics.
</p>


<h3>Usage</h3>

<pre><code class="language-R">set.stats(x, set.p = TRUE, set.mu_sigma = TRUE, 
          verbose = getOption("gaston.verbose",TRUE))

set.stats.snps(x, set.p = TRUE, set.mu_sigma = TRUE, 
               verbose = getOption("gaston.verbose",TRUE)) 

set.stats.ped(x, verbose = getOption("gaston.verbose",TRUE))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A <code>bed.matrix</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>set.p</code></td>
<td>
<p> If <code>TRUE</code>, <code>x@p</code> is updated </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>set.mu_sigma</code></td>
<td>
<p> If <code>TRUE</code>, <code>x@mu</code> and <code>x@sigma</code> are updated </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> If <code>TRUE</code>, display information on the function actions </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>set.stats</code> is called by default by all functions that create a bed.matrix, unless
the global option <code>gaston.auto.set.stats</code> is <code>FALSE</code> (cf example below).
</p>
<p><code>set.stats</code> and <code>set.stats.ped</code> update <code>x@ped</code>, adding the following variables: 
</p>

<ul>
<li> <p><code>N0</code>, <code>N1</code>, <code>N2</code> and <code>NAs</code> give for each individual the number of 
autosomal SNPs with a genotype equal to 0, 1, 2 and missing, respectively
</p>
</li>
<li> <p><code>N0.x</code>, <code>N1.x</code>, <code>N2.x</code> and <code>NAs.x</code> idem for chromosome X
</p>
</li>
<li> <p><code>N0.y</code>, <code>N1.y</code>, <code>N2.y</code> and <code>NAs.y</code> idem for chromosome Y
</p>
</li>
<li> <p><code>N0.mt</code>, <code>N1.mt</code>, <code>N2.mt</code> and <code>NAs.mt</code> idem for mitochondrial SNPs
</p>
</li>
<li> <p><code>callrate</code>, <code>callrate.x</code>, <code>callrate.y</code>, <code>callrate.mt</code> 
is the individual callrate for autosomal, X, Y, mitochondrial SNPs
</p>
</li>
<li> <p><code>hz</code>, <code>hz.x</code>, <code>hz.y</code>, <code>hz.mt</code> is the individual heterozygosity 
for autosomal, X, Y, mitochondrial SNPs
</p>
</li>
</ul>
<p><code>set.stats</code> and <code>set.stats.snps</code> update <code>x@snps</code>, adding the following variables:
</p>

<ul>
<li> <p><code>N0</code>, <code>N1</code>, <code>N2</code> and <code>NAs</code> give for each SNP the number of individuals
with a genotype equal to 0, 1, 2 and missing, respectively
</p>
</li>
<li> <p><code>N0.f</code>, <code>N1.f</code>, <code>N2.f</code> and <code>NAs.f</code> give, only for SNPs on chromosome X, 
the number of female individuals with a genotype equal to 0, 1, 2 and missing, respectively
</p>
</li>
<li> <p><code>callrate</code> is the SNP callrate (for Y linked SNPs, the callrate is computed usin
males only).
</p>
</li>
<li> <p><code>maf</code> is the Minor Allele Frequency 
</p>
</li>
<li> <p><code>hz</code> is the SNP heterozygosity (for X linked SNPs, the heterozygosity is computed
using females only).
</p>
</li>
</ul>
<p>If <code>set.p = TRUE</code>, <code>x@p</code> is updated with the alternate allele frequency.
</p>
<p>If <code>set.mu_sigma = TRUE</code>, <code>x@mu</code> is updated with the genotype mean (equal to <code>2*x@p</code>)
and <code>x@sigma</code> with the genotype standard deviation (should be approximately <code>sqrt(2*x@p*(1-x@p))</code>
under Hardy-Weinberg Equilibrium).
</p>


<h3>Value</h3>

 
<p>A <code>bed.matrix</code> similar to <code>x</code>, with slots updated as described above.
</p>


<h3>Author(s)</h3>

<p> Herv√© Perdry and Claire Dandine-Roulland </p>


<h3>See Also</h3>

 <p><code>set.hwe</code>, <code>set.genomic.sex</code> </p>


<h3>Examples</h3>

<pre><code class="language-R"># Disable auto set stats :
options(gaston.auto.set.stats = FALSE)

# Load data
data(TTN)
x &lt;- as.bed.matrix(TTN.gen, TTN.fam, TTN.bim)
str(x@ped)
str(x@snps)

# Compute statistics
x &lt;- set.stats(x)
str(x@ped)
str(x@snps)

# restore default behavior
options(gaston.auto.set.stats = TRUE)
</code></pre>


</div>