<div class="container">

<table style="width: 100%;"><tr>
<td>SNP.rm.duplicates</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Remove duplicated SNPs </h2>

<h3>Description</h3>

<p> Remove duplicated SNPs, taking into account possible genotype mismatches </p>


<h3>Usage</h3>

<pre><code class="language-R"> SNP.rm.duplicates(x, by = "chr:pos", na.keep = TRUE, incomp.rm = TRUE) </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by</code></td>
<td>
<p> The criterium used to determine duplicates </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.keep</code></td>
<td>
<p> If <code>TRUE</code>, duplicated genotypes which are missing for at 
least one SNP are set to <code>NA</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>incomp.rm</code></td>
<td>
<p> If <code>TRUE</code>, duplicated SNPs with allele incompatibility are
removed.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Positions of duplicated SNPs are determined using <code>SNP.duplicated</code>
using parameter <code>by</code> (we recommend to use <code>"chr:pos"</code>, the default).
</p>
<p>Then the function considers the possibility of alleles swaps or reference strand flips.
In case of allele incompatibility, the SNPs can be removed or not (according to <code>incomp.rm</code>
parameter).
</p>
<p>When alleles can be matched, only one of the two SNPs is conserved. If there are 
genotype incompatibilities between the duplicates for some individuals, these genotypes are set 
to <code>NA</code>. The parameter <code>na.keep</code> settles the case of genotypes missing in one
of the SNPs.
</p>
<p>Moreover the function takes special care of SNP with possible alleles <code>"0"</code>. 
This case occurs for monomorphic SNPs, when data are read from a <code>.ped</code> file; for
example, a whole column of <code>A A</code>'s will result in a SNP with alleles <code>"A"</code> and
<code>"0"</code>. If there's a duplicate of the SNP with a few, says, <code>A C</code>'s in it,
it will have alleles <code>"A"</code> and <code>"C"</code>. In that case, <code>SNP.duplicated</code> 
with <code>by = "chr:pos:alleles"</code> will not consider these SNPs as duplicates. 
</p>


<h3>Value</h3>

<p>A bed.matrix without duplicated SNPs. </p>


<h3>See Also</h3>

 <p><code>SNP.match</code>, <code>SNP.duplicated</code>, <code>dupli</code> </p>


<h3>Examples</h3>

<pre><code class="language-R"># Use example data of 10 individuals with 7 duplicated SNPs
data(dupli)
x &lt;- as.bed.matrix(dupli.gen, fam = dupli.ped, bim = dupli.bim)

# There are any duplicated positions:
dupli.bim

x1 &lt;- SNP.rm.duplicates(x)
# By default (na.keep = TRUE), as soon as the genotype is missing
# in one of the SNPs it is set to missing 
# (here looking at duplicated SNPs 2a and 2b)
as.matrix(x[,2:3])
as.matrix(x1[,2])

# With na.keep = FALSE 
x2 &lt;- SNP.rm.duplicates(x, na.keep = FALSE)
as.matrix(x2[,2])

# Let's examinate SNP 3.a and 3.b (swapped alleles)
as.matrix(x[,4:5])
as.matrix(x1[,3])
as.matrix(x2[,3])

# and so on... (see also ?dupli)
</code></pre>


</div>