<div class="container">

<table style="width: 100%;"><tr>
<td>DM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Dominance Matrix</h2>

<h3>Description</h3>

<p>Compute the Dominance Matrix</p>


<h3>Usage</h3>

<pre><code class="language-R"> DM(x, which.snps, autosome.only = TRUE, chunk = 1L) </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A <code>bed.matrix</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>which.snps</code></td>
<td>
<p> Logical vector, giving which snps to use in the computation.  The default is to use all autosomal SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>autosome.only</code></td>
<td>
<p> If <code>TRUE</code>, only autosomal SNPs will be considered. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chunk</code></td>
<td>
<p> Parameter for the parallelization: how many SNPs are treated by each task</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The Dominance Matrix (DM) gives for each pair of individuals an estimation of 
their probability of sharing two alleles Identical By Descent. 
</p>
<p>It is computed by a moment estimator,
<code class="reqn">{1\over q} ZZ'</code> with <code class="reqn">Z</code> the matrix with entries
<code class="reqn">p \over 1-p</code>, <code class="reqn">-1</code>, <code class="reqn">1-p \over p</code> according to the
values 0, 1, 2 in the genotype matrix <code>x</code> (here <code class="reqn">p</code> is the
frequency of the alternate allele, and <code class="reqn">q</code> is the number of SNPs
(<code>ncol(x)</code>).
</p>


<h3>Value</h3>

<p> A symmetric square matrix of dimension equal to the number of individuals.
Each entry can be interpreted as an estimated probability of sharing two alleles IBD 
â€” as it is a moment estimator, the value can (and will) fall outside of the range (0,1). </p>


<h3>See Also</h3>

 <p><code>GRM</code>, <code>reshape.GRM</code> </p>


<h3>Examples</h3>

<pre><code class="language-R"># load chr2 data set (~10k SNPs in low LD)
x &lt;- read.bed.matrix( system.file("extdata", "chr2.bed", package="gaston") )

# Compute Dominance Matrix
D &lt;- DM(x)
dim(D)

</code></pre>


</div>