<div class="container">

<table style="width: 100%;"><tr>
<td>bed.matrix-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class <code>"bed.matrix"</code>
</h2>

<h3>Description</h3>

<p> S4 class for SNP genotype matrices </p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("bed.matrix", ...)</code>.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>ped</code>:</dt>
<dd> <p><code>data.frame</code> containing information for each individual: <code>famid</code> = Family ID, 
<code>id</code> = Individual ID, <code>father</code> = Father ID, <code>mother</code> = Mother ID, <code>sex</code> = Sex and <code>pheno</code> = Phenotype. 
Can also contain individuals statistic, for example: <code>N0</code>, <code>N1</code> and <code>N2</code> = Number of genotypes equal to 0, 1 and 2 respectively, 
<code>NAs</code> = Number of missing genotypes, <code>callrate</code> = Individual callrate.</p>
</dd>
<dt>
<code>snps</code>:</dt>
<dd> <p><code>data.frame</code> containing information for each SNP: <code>chr</code> = Chromosome, <code>id</code> = SNP ID,
<code>dist</code> = Genetic Distance, <code>pos</code> = Physical position, <code>A1</code> = Reference Allele, <code>A2</code> = Alternative Allele.
Can also contain SNPs statistic, for example: <code>N0</code>, <code>N1</code> and <code>N2</code> = Number of genotypes equal to 0, 1 and 2 repectively,
<code>NAs</code> = Number of missing genotypes, <code>callrate</code> = SNP callrate, <code>maf</code> = Minor allele frequency), <code>hz</code> = heterozygosity</p>
</dd>
<dt>
<code>bed</code>:</dt>
<dd> <p><code>externalptr</code> (pointing to the genotype matrix). </p>
</dd>
<dt>
<code>p</code>:</dt>
<dd> <p><code>vector</code> or <code>NULL</code> for allelic frequencies (allèle <code>A2</code>).</p>
</dd>
<dt>
<code>mu</code>:</dt>
<dd> <p><code>vector</code> or <code>NULL</code> for genotype means (usually <code>mu = 2*p</code>).</p>
</dd>
<dt>
<code>sigma</code>:</dt>
<dd> <p><code>vector</code> or <code>NULL</code> for genotypic standard deviation</p>
</dd>
<dt>
<code>standardize_p</code>:</dt>
<dd> <p><code>logical</code>. If <code>TRUE</code>, the genotype matrix is standardized using means <code>2*p</code> 
and genotypic standard deviation <code>sqrt(2*p*(1-p))</code></p>
</dd>
<dt>
<code>standardize_mu_sigma</code>:</dt>
<dd> <p><code>logical</code>. If <code>TRUE</code>, the genotype matrix is standardize using means
<code>mu</code> and genotypic standard deviation <code>sigma</code>.</p>
</dd>
</dl>
<p>For more details please check the vignette.
</p>


<h3>Methods</h3>


<dl>
<dt>[</dt>
<dd>
<p><code>signature(x = "bed.matrix", i = "numeric" or "logical" or "missing",</code><br><code>j = "numeric" or "logical" or "missing", drop = "missing")</code>: 
<br> Extract a sub-matrix (a new <code>bed.matrix</code>). </p>
</dd>
<dt>%*%</dt>
<dd>
<p><code>signature(x = "bed.matrix", y = "matrix" or "vector")</code>: 
<br> Right matrix multiplication of the genotype matrix (possibly centered and reduced) with a <code>matrix</code> or a <code>vector</code>. </p>
</dd>
<dt>%*%</dt>
<dd>
<p><code>signature(x = "matrix" or "vector", y = "bed.matrix")</code>: 
<br> Left matrix multiplication of the genotype matrix with a <code>matrix</code> or a <code>vector</code>. </p>
</dd>
<dt>as.matrix</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>: 
<br> Convert a <code>bed.matrix</code> into a <code>matrix</code>.
The resulting matrix can be huge, use this method only for a small bed.matrix! </p>
</dd>
<dt>standardize</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>: 
<br> Get the standardize parameter of <code>bed.matrix</code>. Can be "none", "p" or "mu_sigma". </p>
</dd>
<dt>standardize&lt;-</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>:
<br> Set the standardize parameter of a <code>bed.matrix</code>. </p>
</dd>
<dt>dim</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>: 
<br> Get the number of individuals (rows) and the number of SNPs (columns). </p>
</dd>
<dt>head</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>: 
<br> Print the head of the genotype matrix of a <code>bed.matrix</code> object. </p>
</dd>
<dt>mu</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>: 
<br> Get the <code>mu</code> slot of a <code>bed.matrix</code>. </p>
</dd>
<dt>mu&lt;-</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>: 
<br> Set the <code>mu</code> slot of a <code>bed.matrix</code>. </p>
</dd>
<dt>p</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>: 
<br> Get the <code>p</code> slot of a <code>bed.matrix</code>. </p>
</dd>
<dt>p&lt;-</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>: 
<br> Set the <code>p</code> slot of a <code>bed.matrix</code>. </p>
</dd>
<dt>show</dt>
<dd>
<p><code>signature(object = "bed.matrix")</code>: 
<br> Displays basic information about a <code>bed.matrix</code>. </p>
</dd>
<dt>sigma</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>: 
<br> Get the <code>sigma</code> slot of a <code>bed.matrix</code>. </p>
</dd>
<dt>sigma&lt;-</dt>
<dd>
<p><code>signature(x = "bed.matrix")</code>: 
<br> Set the <code>sigma</code> slot of a <code>bed.matrix</code>. </p>
</dd>
<dt>cbind</dt>
<dd>
<p><code>signature(... = "bed.matrix")</code>: 
<br> Combine a sequence of <code>bed.matrix</code> by columns. </p>
</dd>
<dt>rbind</dt>
<dd>
<p><code>signature(... = "bed.matrix")</code>: 
<br> Combine a sequence of <code>bed.matrix</code> by rows. </p>
</dd>
</dl>
<h3>Author(s)</h3>

<p> Hervé Perdry and Claire Dandine-Roulland </p>


<h3>See Also</h3>

 <p><code>read.bed.matrix</code>, <code>write.bed.matrix</code>,
<code>set.stats</code>, <code>select.snps</code>, <code>select.inds</code>, <code>GRM</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">showClass("bed.matrix")

# Conversion example
data(LCT)
x1 &lt;- as(LCT.gen, "bed.matrix")
x1
head(x1@ped)
head(x1@snps)

# the function as.bed.matrix is an alternative
x2 &lt;- as.bed.matrix(LCT.gen, LCT.fam, LCT.bim)
x2
head(x2@ped)
head(x2@snps)
</code></pre>


</div>