<div class="container">

<table style="width: 100%;"><tr>
<td>set.genomic.sex</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Genomic Sex </h2>

<h3>Description</h3>

<p>Returns an updated <code>bed.matrix</code> with a new variable for the genomic sex
of each individual.
</p>


<h3>Usage</h3>

<pre><code class="language-R"> set.genomic.sex(x, plot = FALSE, verbose = getOption("gaston.verbose",TRUE)) </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A <code>bed.matrix</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p> If <code>TRUE</code>, plots the variables used for the clustering </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> If <code>TRUE</code>, displays information on the function actions </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For each individual, the function uses the hetorozygosity rate for SNPs on X chromosome, 
and the call rate for SNPs on the Y chromosomes (both statistics computed by <code>set.stats</code>), 
to cluster the individuals using <code>kmeans</code>. 
</p>
<p>If <code>plot = TRUE</code>, a plot is produced with the two variables used and the clusters
determined by <code>kmeans</code>.
</p>


<h3>Value</h3>

<p>A <code>bed.matrix</code> similar to <code>x</code>, with a new variable <code>x@ped$genomic.sex</code>
containing the genomic sex for each individual.
</p>


<h3>Author(s)</h3>

<p>Herv√© Perdry</p>


<h3>See Also</h3>

 <p><code>set.stats</code>, <code>set.hwe</code> </p>


</div>