<div class="container">

<table style="width: 100%;"><tr>
<td>SNP.match</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> SNP matching </h2>

<h3>Description</h3>

<p> Returns a vector of the positions of (first) SNP matching of
its first argument in its second. </p>


<h3>Usage</h3>

<pre><code class="language-R"> SNP.match(x, table, by = "chr:pos:alleles") </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A bed.matrix or a data.frame</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>table</code></td>
<td>
<p> A bed.matrix or a data.frame</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by</code></td>
<td>
<p> The criterium used to matchSNPs </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>When <code>x</code> is a bed.matrix, the data.frame <code>x@bed</code> will be used; the 
same holds for <code>table</code>. The columns that will be taken in consideration
are <code>id</code>, <code>chr</code>, <code>pos</code>, <code>A1</code>, and <code>A2</code>. Not all columns
are mandatory (see below). 
</p>
<p>The matching criterium is specified by parameter <code>by</code>.
There are 5 possible criteria : (i) matching by chromosome and position 
with <code>by = "chr:pos"</code>, (ii) matching by chromosome, position, and alleles
with <code>by = "chr:pos:alleles"</code>, (iii) matching by id with <code>by = "id"</code>,
(iv) matching by id, chromosome and position with <code>by  = "id:chr:pos"</code>, 
and (v) matching by id, chromosome, position and alleles with <code>by = "id:chr:pos:alleles"</code>.
</p>
<p>For each SNP in <code>x</code>, the function looks for the position of the first
matching SNP in <code>table</code>. If alleles are included in the matching criterium
(ie if allele columns <code>A1</code> and <code>A2</code> are present in x), the function 
also checks for SNP matching with swapped alleles (a SNP A/C would match a 
SNP C/A), or with reference strand flipped (i.e. a SNP A/C would match a SNP T/G) 
or both (a SNP A/C would match a SNP G/T).
</p>
<p>This function should prove useful for data set merging.
</p>


<h3>Value</h3>

<p>A named list with one or three members, depending on whether alleles are included
in the matching criterium.
</p>
<table>
<tr style="vertical-align: top;">
<td><code>index</code></td>
<td>
<p>An integer vector giving the position of first match in <code>table</code>, or <code>NA</code> if there is no match</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>swap</code></td>
<td>
<p>A logical vector indicating whether the match is with swapped alleles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flip</code></td>
<td>
<p>A logical vector indicating whether the match is with flipped strand</p>
</td>
</tr>
</table>
<h3>See Also</h3>

 <p><code>SNP.duplicated</code> </p>


</div>