<div class="container">

<table style="width: 100%;"><tr>
<td>manhattan</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Manhattan plot </h2>

<h3>Description</h3>

<p> Draws a Manhattan plot </p>


<h3>Usage</h3>

<pre><code class="language-R"> manhattan(x, bty = "n", chrom.col = c("black", "gray50"), thinning = TRUE, ... ) </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> A data.frame with columns named <code>chr</code>, <code>pos</code> and <code>p</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bty</code></td>
<td>
<p> Type of box to draw about the plot. Default is to draw none. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thinning</code></td>
<td>
 <p><code>Logical</code>. If <code>TRUE</code>, not all points are displayed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom.col</code></td>
<td>
<p> Alternating colors for chromosomes. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> Graphical parameters to be passed to <code>plot</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If there is only one chromosome value in <code>x$chr</code>, the x-axis will be labeled with the SNP
position. In the general case, the x-axis is labeled with the chromosome name and the color
of the points alternates between the colors in <code>chrom.col</code>.
</p>
<p>The default value <code>bty = "n"</code> should give the best result for GWAS Manhattan plots. 
See <code>par</code> for other possible values of <code>bty</code> and their meaning.
</p>
<p>The thinning procedure suppress some points to avoid generating too heavy graphs. The user 
should check that setting <code>thinning = FALSE</code> does not change the final aspect of the 
plot. 
</p>


<h3>Value</h3>

<p>An invisible copy of <code>x</code> is returned, in which a column <code>coord</code> has been added  
if there is more than one chromosome value in <code>x$chr</code>. This column contains the x-coordinates of each
SNP on the plot, and should prove helpful to annotate it. </p>


<h3>See Also</h3>

 <p><code>association.test</code>, <code>qqplot.pvalues</code>, 
<code>par</code>, <code>plot.default</code>, <code>points.default</code> </p>


</div>