<div class="container">

<table style="width: 100%;"><tr>
<td>GEEaSPUset</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
An adaptive sum of powered score (SPU) test for SNP-set (gene) and mutiple phenotypes
</h2>

<h3>Description</h3>

<p>Tests whether SNP-set (gene) is associated with multivariate outcomes;
provides a series of p-values of GEE-SPU(pow, pow2) and GEEaSPUset tests.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GEEaSPUset(pheno, geno, Z = NULL, model = "gaussian", corstr = "independence", 
          pow = c(1,2,4,8), pow2 = c(1,2,4,8), n.perm = 1000, 
          null.type = "perm", score.test = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>a numeric phenotype matrix with each row as a different individual and each column as a separate phenotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>a numeric genotype matrix with each row as a different individual and each column as a snp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>
<p>a numeric covariate matrix with each row as a different individual and each column as a covariated to be adjested.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>a character string specifying the model of the phenotypes. 
Models supported are "gaussian" for a quantitative trait and "binomial" for a binary trait (default = "gaussian").
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>corstr</code></td>
<td>
<p>a character string specifying the correlation structure of phenotypes. The following are permitted: "independence", "fixed", "stat_M_dep", "non_stat_M_dep", "exchangeable", "AR-M" and "unstructured" (default = "independence").
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow</code></td>
<td>
<p>a vector of the power weight to be used at a SNP level (default = c(1,2,4,8)).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow2</code></td>
<td>
<p>a vector of the power weight to be used at a trait level (default = c(1,2,4,8)).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.perm</code></td>
<td>
<p>a numeric value of number of null statistics (default = 1000).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>null.type</code></td>
<td>
<p>a character string specifying how to generate null statistics; "perm" is used when null statistics are generated using permutations and  
"sim" is used when null statistics are generated using simulations (default = "perm").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>score.test</code></td>
<td>
<p>a logical value indicating whether to include GEEaSPU-Score test along with GEE-Score test (default = FALSE). If TRUE, it computes p-values of GEEaSPU-Score and GEE-Score as well as GEEaSPU test.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Adaptive association tests for SNP-set (gene) and multiple phenotypes using GEE.
</p>


<h3>Value</h3>

<p>a vector of p-values from GEE-SPU(pow, pow2) tests and GEE-aSPUset test.
</p>


<h3>Note</h3>

<p>When large SNP-set (namely large gene size) or large number of phenotypes are included, 
the permuation based test (null.type = "perm") is recommended.
</p>
<p>An option "binomial" model only supports the option, null.type="sim".
</p>


<h3>Author(s)</h3>

<p>Junghi Kim and Wei Pan
</p>


<h3>References</h3>

<p>Kim J, Zhang Y, and Pan W (2016). "Powerful and adaptive testing for multi-trait and multi-SNP associations with GWAS and sequencing data." <em>Genetics</em>, 203(2), 715â€“731.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># -- simulating phenotypes
# -- n.subjects: number of subjects
# -- n.traits: number of phenotypes
# -- Sigma: covariance matrix of phenotypes (e.g. AR(1))

set.seed(136)
n.subjects &lt;- 100
n.traits &lt;- 3
sigma &lt;- 2; rho &lt;- 0.5
Sigma0 &lt;- diag(n.traits); 
Sigma &lt;- sigma * rho^abs(row(Sigma0) - col(Sigma0))
eS &lt;- eigen(Sigma, symmetric = TRUE)
ev &lt;- eS$values
X &lt;- matrix(rnorm(n.subjects * n.traits), n.subjects)
pheno &lt;- X %*% diag(sqrt(pmax(ev, 0)), ncol(Sigma)) %*% eS$vectors


# -- simulating genotype
# -- n.geno: number of SNPs included in the SNP set/gene

n.geno &lt;- 3
maf &lt;- 0.2
geno &lt;- matrix(rbinom(n = n.subjects * n.geno, size = 2, prob = maf), ncol = n.geno)


# -- Computing the p-value of GEEaSPUpath test with the permutation based method
Pvl &lt;- GEEaSPUset(pheno = pheno, geno = geno, Z = NULL, model = "gaussian", 
        corstr = "independence", pow = c(1,4,Inf), pow2 = c(1,4,Inf), n.perm = 1000,         
        null.type = "perm", score.test = FALSE)


# -- Each element of Pvl is a p value of GEE-SPU(pow,pow2) in order
# -- The last element of Pvl is a p value of GEE-aSPUset test
Pvl
Pvl[length(Pvl)]


# &gt; Pvl
#     SPU.1.1     SPU.1.4   SPU.1.Inf     SPU.4.1     SPU.4.4   SPU.4.Inf   SPU.Inf.1 
#  0.01400000  0.08800000  0.07200000  0.53000000  0.41000000  0.32100000  0.55100000 
#   SPU.Inf.4 SPU.Inf.Inf     aSPUset 
#  0.48700000  0.41000000  0.04095904 
 
 
# &gt; Pvl[length(Pvl)]
#    aSPUset 
# 0.04095904
</code></pre>


</div>