<div class="container">

<table style="width: 100%;"><tr>
<td>order.genotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Order/sort genotype/haplotype object</h2>

<h3>Description</h3>

<p>Order/sort genotype or haplotype object according
to order of allele names or genotypes</p>


<h3>Usage</h3>

<pre><code class="language-R">
## S3 method for class 'genotype'
order(..., na.last=TRUE, decreasing=FALSE,
  alleleOrder=allele.names(x), genotypeOrder=NULL)

## S3 method for class 'genotype'
sort(x, decreasing=FALSE, na.last=NA, ...,
  alleleOrder=allele.names(x), genotypeOrder=NULL)

genotypeOrder(x)
genotypeOrder(x) &lt;- value

</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>genotype or haplotype in <code>order</code> method; not used
for <code>sort</code> method</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>genotype or haplotype in <code>sort</code> method</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.last</code></td>
<td>
<p>as in default <code>order</code> or <code>sort</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>decreasing</code></td>
<td>
<p>as in default <code>order</code> or <code>sort</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleleOrder</code></td>
<td>
<p>character, vector of allele names in wanted order</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypeOrder</code></td>
<td>
<p>character, vector of genotype/haplotype names in
wanted order</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>value</code></td>
<td>
<p>the same as in argument <code>order.genotype</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Argument <code>genotypeOrder</code> can be usefull, when you want that some
genotypes appear "together", whereas they are not "together" by allele
order.
</p>
<p>Both methods (<code>order</code> and <code>sort</code>) work with genotype and
haplotype classes.
</p>
<p>If <code>alleleOrder</code> is given, <code>genotypeOrder</code> has no effect.
</p>
<p>Genotypes/haplotypes, with missing alleles in <code>alleleOrder</code> are
treated as <code>NA</code> and ordered according to <code>order</code>
arguments related to <code>NA</code> values. In such cases a warning is issued
("Found data values not matching specified alleles. Converting to NA.")
and can be safely ignored. Genotypes present in <code>x</code>, but not
specified in <code>genotypeOrder</code>, are also treated as <code>NA</code>.
</p>
<p>Value of <code>genotypeOrder</code> such as "B/A" matches also "A/B" in case
of genotypes.
</p>
<p>Only unique values in argument <code>alleleOrder</code> or
<code>genotypeOrder</code> are used i.e. first occurrence prevails.
</p>


<h3>Value</h3>

<p>The same as in <code>order</code> or <code>sort</code></p>


<h3>Author(s)</h3>

<p>Gregor Gorjanc</p>


<h3>See Also</h3>

<p><code>genotype</code>,
<code>allele.names</code>,
<code>order</code>, and
<code>sort</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  x &lt;- c("C/C", "A/C", "A/A", NA, "C/B", "B/A", "B/B", "B/C", "A/C")
  alleles &lt;- c("A", "B", "C")

  g &lt;- genotype(x, alleles=alleles, reorder="yes")
  ## "C/C" "A/C" "A/A" NA    "B/C" "A/B" "B/B" "B/C" "A/C"

  h &lt;- haplotype(x, alleles=alleles)
  ## "C/C" "A/C" "A/A" NA    "C/B" "B/A" "B/B" "B/C" "A/C"

  ## --- Standard usage ---

  sort(g)
  ## "A/A" "A/B" "A/C" "A/C" "B/B" "B/C" "B/C" "C/C" NA

  sort(h)
  ## "A/A" "A/C" "A/C" "B/A" "B/B" "B/C" "C/B" "C/C" NA

  ## --- Reversed order of alleles ---

  sort(g, alleleOrder=c("B", "C", "A"))
  ## "B/B" "B/C" "B/C" "A/B" "C/C" "A/C" "A/C" "A/A" NA
  ## note that A/B comes after B/C since it is treated as B/A;
  ## order of alleles (not in alleleOrder!) does not matter for a genotype

  sort(h, alleleOrder=c("B", "C", "A"))
  ## "B/B" "B/C" "B/A" "C/B" "C/C" "A/C" "A/C" "A/A" NA

  ## --- Missing allele(s) in alleleOrder ---

  sort(g, alleleOrder=c("B", "C"))
  ## "B/B" "B/C" "B/C" "C/C" "A/C" "A/A" NA    "A/B" "A/C"

  sort(g, alleleOrder=c("B"))
  ## "B/B" "C/C" "A/C" "A/A" NA    "B/C" "A/B" "B/C" "A/C"
  ## genotypes with missing allele are treated as NA

  sort(h, alleleOrder=c("B", "C"))
  ## "B/B" "B/C" "C/B" "C/C" "A/C" "A/A" NA    "B/A" "A/C"

  sort(h, alleleOrder=c("B"))
  ## "B/B" "C/C" "A/C" "A/A" NA    "C/B" "B/A" "B/C" "A/C"

  ## --- Use of genotypeOrder ---

  sort(g, genotypeOrder=c("A/A", "C/C", "B/B", "A/B", "A/C", "B/C"))
  ## "A/A" "C/C" "B/B" "A/B" "A/C" "A/C" "B/C" "B/C" NA

  sort(h, genotypeOrder=c("A/A", "C/C", "B/B",
                          "A/C", "C/B", "B/A", "B/C"))
  ## "A/A" "C/C" "B/B" "A/C" "A/C" "C/B" "B/A" "B/C" NA

  ## --- Missing genotype(s) in genotypeOrder ---

  sort(g, genotypeOrder=c(       "C/C",        "A/B", "A/C", "B/C"))
  ## "C/C" "A/B" "A/C" "A/C" "B/C" "B/C" "A/A" NA    "B/B"

  sort(h, genotypeOrder=c(       "C/C",        "A/B", "A/C", "B/C"))
  ## "C/C" "A/C" "A/C" "B/C" "A/A" NA    "C/B" "B/A" "B/B"
</code></pre>


</div>