<div class="container">

<table style="width: 100%;"><tr>
<td>filter_vcf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Filter VCF</h2>

<h3>Description</h3>

<p>filter VCF by GFF annotation or by position or both
</p>


<h3>Usage</h3>

<pre><code class="language-R">filter_vcf(vcf, gff = gff,
           mode = c("POS", "type", "both"),
           Chr = Chr, start = start, end = end,
           type = c("CDS", "exon", "gene", "genome", "custom"),
           cusTyp = c("CDS", "five_prime_UTR", "three_prime_UTR"),
           geneID = geneID)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>object of vcfR class, VCF file imported by <code>import_vcf()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gff</code></td>
<td>
<p>object of GRanges class, genome annotations imported by
<code>import_gff()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>
<p>filter mode, one of "POS", "type", "both"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Chr</code></td>
<td>
<p>chromosome name, needed if mode set to "POS" or "both"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>start position, needed if mode set to "POS" or "both"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>end position, needed if mode set to "POS" or "both"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>filter type, needed if mode set to "type" or "both",
one of "CDS", "exon", "gene", "genome", "custom",
if <code>type</code> was set to "custom", then <code>custom_type</code> is needed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cusTyp</code></td>
<td>
<p>character vector, custom filter type,
needed if <code>type</code> set to "custom"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneID</code></td>
<td>
<p>gene ID</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>vcfR
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# filtet vcf
data("geneHapR_test")
vcf_f1 &lt;- filter_vcf(vcf, mode = "POS",
                    Chr = "scaffold_1",
                    start = 4300, end = 5890)

vcf_f2 &lt;- filter_vcf(vcf, mode = "type",
                    gff = gff,
                    type = "CDS")
vcf_f3 &lt;- filter_vcf(vcf, mode = "both",
                    Chr = "scaffold_1",
                    start = 4300, end = 5890,
                    gff = gff,
                    type = "CDS")

</code></pre>


</div>