<div class="container">

<table style="width: 100%;"><tr>
<td>run_simplify_enrichment</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run simplifyGOFromMultipleLists with specified measure and method</h2>

<h3>Description</h3>

<p>This function is simply a wrapper for the simplifyGOFromMultipleLists function in the simplifyEnrichment package,
created for the ease of data input. All credit for the underlying functionality goes to the authors
of the simplifyEnrichment package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_simplify_enrichment(
  fs_GO_results,
  padj_column = "p.adjust",
  padj_cutoff,
  ont,
  measure,
  method,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fs_GO_results</code></td>
<td>
<p>A list of dataframes containing GO enrichment results for feature selection methods. The GO object for the simplifyGOFromMultipleLists function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>padj_column</code></td>
<td>
<p>Character. The column name for the p-value adjustment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>padj_cutoff</code></td>
<td>
<p>Numeric. The cutoff for the p-value adjustment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ont</code></td>
<td>
<p>Character. The ontology for the simplifyGOFromMultipleLists function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>measure</code></td>
<td>
<p>Character. The semantic similarity measure for the simplifyGOFromMultipleLists function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Character. The clustering method for the simplifyGOFromMultipleLists function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other parameters that can be passed to simplifyGOFromMultipleLists</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The result of the <code>simplifyGOFromMultipleLists</code> function, typically comprising a heatmap or other
visualization that displays the simplified GO enrichment results. The specific output format depends
on the chosen semantic similarity measure and clustering method.
</p>


<h3>References</h3>

<p>For more information on the simplifyEnrichment package, see the original publication:
Gu Z, HÃ¼bschmann D. simplifyEnrichment: A Bioconductor Package for Clustering and Visualizing Functional Enrichment Results.
Genomics Proteomics Bioinformatics. 2023 Feb;21(1):190-202. doi: 10.1016/j.gpb.2022.04.008.
Epub 2022 Jun 6. PMID: 35680096; PMCID: PMC10373083.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Mock GO enrichment results for two feature selection methods
fs_GO_results &lt;- list(
  method1 = list(result = data.frame(GO_ID = c("GO:0008150", "GO:0009987"),
                                    Description = c("Biological Process 1", "Biological Process 2"),
                                    'p.adjust' = c(0.01, 0.02))),
  method2 = list(result = data.frame(GO_ID = c("GO:0008150", "GO:0008152"),
                                    Description = c("Biological Process 1", "Biological Process 3"),
                                    'p.adjust' = c(0.03, 0.04)))
)

# Run the wrapper function with mock data
enrichment_result &lt;- run_simplify_enrichment(fs_GO_results,
                                             padj_column = 'p.adjust',
                                             padj_cutoff = 0.05,
                                             ont = "BP",
                                             measure = "Wang",
                                             method = "kmeans")
print(enrichment_result)


</code></pre>


</div>