<div class="container">

<table style="width: 100%;"><tr>
<td>import_bed</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>import annotation files in BED format</h2>

<h3>Description</h3>

<p>import bed files contains annotations into R as GRanges object
</p>


<h3>Usage</h3>

<pre><code class="language-R">import_bed(con, quite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>con</code></td>
<td>
<p>A path, URL, connection or <code>BEDFile</code> object. For the
functions ending in <code>.bed</code>, <code>.bedGraph</code> and <code>.bed15</code>,
the file format is indicated by the function name. For the base
<code>export</code> and <code>import</code> functions, the format must be
indicated another way. If <code>con</code> is a path, URL or connection,
either the file extension or the <code>format</code> argument needs to be
one of “bed”, “bed15”, “bedGraph”,
“bedpe”, “narrowPeak”, or
“broadPeak”. Compressed
files (“gz”, “bz2” and
“xz”) are handled transparently.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quite</code></td>
<td>
<p>whether show message</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If there is no genome annotation file in GFF format for your interest species,
a BED file is convenient to custom a simple annotation file for a gene.
Here we suggest two type of BED format: BED6 and BED4.
</p>
<p>As the definition of
<a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1">UCSC</a>.
The BED6 contains 6 columns, which are 1) chrom, 2) chromStart, 3) chromEnd,
4) name, 5) score and 6) strand. The BED4 format contains the first 4 column
of BED6 format.
</p>
<p>However, in gene haplotype statistics, we only care about the type of each site.
Thus we use the fourth column to definition the
transcripts name and "CDS" or "URTs", separated by a space, eg.:
</p>
<p><code>Chr8   678   890   HD1.1 CDS   .   -</code>
</p>
<p><code>Chr8   891   989   HD1.1 five_prime_UTR   .   -</code>
</p>
<p><code>Chr8   668   759   HD1.2 CDS   .   -</code>
</p>
<p><code>Chr8   908   989   HD1.2 CDS   .   -</code>
</p>
<p>This example indicate a small gene named as HD1 have two transcripts,
named as HD1.1 and HD1.2, separately. HD1 has a CDS and a UTR region;
while HD1.2 has two CDS region.
</p>


<h3>Value</h3>

<p>GRange object
</p>


<h3>Examples</h3>

<pre><code class="language-R">
bed.Path &lt;- system.file("extdata", "annotation.bed6", package = "geneHapR")
bed &lt;- import_bed(bed.Path)
bed

</code></pre>


</div>