<div class="container">

<table style="width: 100%;"><tr>
<td>vcf2hap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generat Haps from VCF</h2>

<h3>Description</h3>

<p>Generate hapResult from vcfR object
A simple filter by position was provided in this function,
however it's prefer to filter VCF (vcfR object) through
<code>filter_vcf()</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">vcf2hap(
  vcf,
  hapPrefix = "H",
  filter_Chr = FALSE,
  filter_POS = FALSE,
  pad = 3,
  hetero_remove = TRUE,
  na_drop = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>vcfR object imported by <code>import_vcf()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hapPrefix</code></td>
<td>
<p>prefix of hap names, default as "H"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_Chr</code></td>
<td>
<p>not used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_POS</code></td>
<td>
<p>not used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pad</code></td>
<td>
<p>The number length in haplotype names should be extend to.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hetero_remove</code></td>
<td>
<p>whether remove accessions contains hybrid site or not.
Default as <code>TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na_drop</code></td>
<td>
<p>whether remove accessions contains unknown allele site or not
Default as <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Parameters not used</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>object of hapResult class
</p>


<h3>Author(s)</h3>

<p>Zhangrenl
</p>


<h3>See Also</h3>

<p>extract genotype from vcf:
<code>vcfR::extract_gt_tidy()</code>,
import vcf files:
<code>import_vcf()</code> (preferred) and
<code>vcfR::read.vcfR()</code>,
filter vcf according <strong>position</strong> and <strong>annotations</strong>:
<code>filter_vcf()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("geneHapR_test")
hapResult &lt;- vcf2hap(vcf)
</code></pre>


</div>