<div class="container">

<table style="width: 100%;"><tr>
<td>Hardy-Weinberg</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Tests of Hardy-Weinberg genotypic proportions</h2>

<h3>Description</h3>

<p>Compute variants of the exact conditional test for Hardy-Weinberg genotypic proportions.  The tests differ by their test statistics. <code>HWtable_analysis</code> handles a single table of genotype counts, and <code>test_HW</code> requires a standard genepop input file. See <a href="../doc/all-menu-options.html#option-1-hardy-weinberg-hw-exact-tests">this section</a> of the Genepop-executable documentation for more information on the statistical methods.
</p>


<h3>Usage</h3>

<pre><code class="language-R">test_HW(
  inputFile,
  which = "Proba",
  outputFile = "",
  settingsFile = "",
  enumeration = FALSE,
  dememorization = 10000,
  batches = 20,
  iterations = 5000,
  verbose = interactive()
)

HWtable_analysis(
  inputFile,
  which = "Proba",
  settingsFile = "",
  enumeration = FALSE,
  dememorization = 10000,
  batches = 20,
  iterations = 5000,
  verbose = interactive()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>inputFile</code></td>
<td>
<p>character: The path of the input file. For <code>test_HW</code>, this file should be in Genepop format. For <code>HWtable_analysis</code>, it should be in ad hoc format illustrated by sample file <code>Rhesus.txt</code> used in the Examples section, and further detailed in <a href="../doc/all-menu-options.html#analyzing-a-single-genotypic-matrix">this section</a> of the Genepop-executable documentation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>which</code></td>
<td>
<p>character: <code>'Proba'</code>, <code>'excess'</code>, and <code>'deficit'</code> to perform the probability test, score test for excess, and score tests for deficit, respectively, in each population and for each locus. <code>test_HW</code> additionally handles <code>'global excess'</code> and  <code>'global deficit'</code> for global tests for all loci and/or all populations, and <code>HWtable_analysis</code> additionally handles <code>'Fis'</code> to report basic information (allele frequencies and Fis).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputFile</code></td>
<td>
<p>character: The path of the output file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>settingsFile</code></td>
<td>
<p>character: The path of the settings file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>enumeration</code></td>
<td>
<p>logical: whether to compute the complete enumeration test for samples with less than 5 alleles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dememorization</code></td>
<td>
<p>integer: length of dememorization step of Markov chain algorithm</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batches</code></td>
<td>
<p>integer: Number of batches</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iterations</code></td>
<td>
<p>integer: Iterations per batch</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical: whether to print some information</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The path of the output file is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class="language-R">locinfile &lt;- genepopExample('sample.txt')
test_HW(locinfile, which='deficit', 'sample.txt.D')
if ( ! interactive()) clean_workdir(otherfiles='sample.txt')
# Example in Guo &amp; Thompson 1992 Table 5
locinfile &lt;- genepopExample('Rhesus.txt')
outfile &lt;- HWtable_analysis(locinfile,which='Proba',batches = 1000,iterations = 1000)
readLines(outfile)[21]
#clean_workdir(otherfiles='Rhesus.txt')
</code></pre>


</div>