<div class="container">

<table style="width: 100%;"><tr>
<td>LDheatmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>This function produces a pairwise LD plot.</h2>

<h3>Description</h3>

<p><code>LDheatmap()</code> is used to produce a graphical display, as a heat map,
of pairwise linkage disequilibrium (LD) measurements for SNPs.
The heat map is a false color image in the upper-left diagonal of a square plot.
Optionally, a line parallel to the diagonal of the image indicating
the physical or genetic map positions of the SNPs may be added, along
with text reporting the total length of the genomic region considered.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_LDheatmap(
  hap,
  gff,
  Chr,
  start,
  end,
  geneID = NULL,
  distances = "physical",
  LDmeasure = "r",
  title = "Pairwise LD",
  add.map = TRUE,
  map.height = 1,
  colorLegend = TRUE,
  geneMapLocation = 0.15,
  geneMapLabelX = NULL,
  geneMapLabelY = NULL,
  SNP.name = TRUE,
  color = NULL,
  color_gmodel = "grey",
  color_snp = "grey",
  color_snpname = "grey40",
  cex_snpname = 0.8,
  snpmarks_height = NULL,
  newpage = TRUE,
  name = "ldheatmap",
  vp.name = NULL,
  pop = FALSE,
  text = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hap</code></td>
<td>
<p>R object of hapSummary or hapResult class.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gff</code></td>
<td>
<p>annotations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Chr, start, end, geneID</code></td>
<td>
<p>chromosome, start and end position, gene ID for
extract annotation in target range.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distances</code></td>
<td>
<p>A character string to specify whether the provided map locations
are in physical or genetic distances.
If <code>distances = "physical"</code> (default), the text describing the total
length of the region will be “Physical Length:XXkb” where XX is the
length of the region in kilobases. If <code>distances = "genetic"</code>, the
text will be “Genetic Map Length:YYcM” where YY is
the length of the region in centiMorgans.
If <code>gdat</code> is an object of class LDheatmap,
<code>distances</code> is taken from <code>gdat</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LDmeasure</code></td>
<td>
<p>A character string specifying the measure of LD
</p>

<ul><li>
<p> either allelic correlation <code class="reqn">r^2</code> or Lewontin's
|D<code class="reqn">'</code>|; default = <code>"r"</code> for <code class="reqn">r^2</code>;
type <code>"D'"</code> for |D<code class="reqn">'</code>|. This argument is ignored when the user has already
supplied calculated LD measurements through <code>gdat</code> (i.e., when <code>gdat</code>
is a matrix of pairwise LD measurements or an object of class <code>"LDheatmap"</code>).
</p>
</li></ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>A character string for the main title of the plot.
Default is “Pairwise LD”.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add.map</code></td>
<td>
<p>If <code>TRUE</code> (default) a diagonal line indicating
the physical or genetic map positions of the SNPs will be added to
the plot, along with text indicating the total length of the
genetic region.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map.height</code></td>
<td>
<p>the height of gene map, default is 0.02</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colorLegend</code></td>
<td>
<p>If <code>TRUE</code> (default) the color legend is drawn.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneMapLocation</code></td>
<td>
<p>A numeric value specifying the position of the line
parallel to the diagonal of the matrix; the larger the value, the
farther it lies from the matrix diagonal. Ignored when <code>add.map = FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneMapLabelX</code></td>
<td>
<p>A numeric value specifying the x-coordinate
of the text indicating the total length of the genomic region
being considered. Ignored when <code>add.map = FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneMapLabelY</code></td>
<td>
<p>A numeric value specifying the y-coordinate
of the text indicating the total length of the genomic region
being considered. Ignored when <code>add.map = FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNP.name</code></td>
<td>
<p>a logical vector indicated wherther display SNP names using positions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color</code></td>
<td>
<p>A range of colors to be used for drawing the heat map. Default
is <code>grDevices::colorRampPalette(c("red", "grey"))(30)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color_gmodel, color_snp, color_snpname</code></td>
<td>
<p>the color of gene model and snp and snp names respectively, default as grey80.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex_snpname</code></td>
<td>
<p>the size of snp names/labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpmarks_height</code></td>
<td>
<p>the height of snp marks, if set as NULL, nothing will display on gene model
where the heat map is going to be drawn.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>newpage</code></td>
<td>
<p>If <code>TRUE</code> (default), the heat map will be drawn on a new page.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>A character string specifying the name of the LDheatmap
graphical object (<code>grob</code>) to be produced.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vp.name</code></td>
<td>
<p>A character string specifying the name of the viewport</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>If <code>TRUE</code>, the viewport where the heat map is drawn is
<code>pop</code>ped (i.e. removed) from the viewport tree after drawing.
Default = <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>text</code></td>
<td>
<p>If <code>TRUE</code>, the LD measurements are printed on each cell.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The input object <code>gdat</code> can be a data frame of <code>genotype</code> objects
(a data structure from the <span class="pkg">genetics</span> package), a <code>SnpMatrix</code> object
(a data structure from the <span class="pkg">snpStats</span> package), or
any square matrix with values between 0 and 1 inclusive.
LD computation is much faster for <code>SnpMatrix</code> objects than for
<code>genotype</code> objects.
In the case of a matrix of LD values between 0 and 1,
the values above the diagonal will be plotted.
In the display of LD, SNPs appear in the order supplied by the
user as the horizontal and vertical coordinates are increased
and one moves along the off-diagonal line, from the bottom-left
to the top-right corner. To achieve this, the conventions of
the <code>image()</code> function have been adopted, in which horizontal
coordinates correspond to the rows of the matrix and vertical coordinates
correspond to columns, and vertical coordinates are indexed in increasing
order from bottom to top.
See the package vignette <code>LDheatmap</code> for more examples and details
of the implementation. Examples of adding “tracks” of genomic
annotation above a flipped heatmap are in the package vignette
<code>addTracks</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"LDheatmap"</code> which contains the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>LDmatrix</code></td>
<td>
<p> The matrix of pairwise LD measurements plotted in the heat map. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LDheatmapGrob</code></td>
<td>
<p> A grid graphical object (grob) representing the produced heat map. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>heatmapVP</code></td>
<td>
<p> The viewport in which the heat map is drawn. See viewport.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genetic.distances</code></td>
<td>
<p>The vector of the supplied physical or
genetic map locations, or the vector of equispaced marker distances
when no distance vector is supplied.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distances</code></td>
<td>
<p> A character string specifying whether the provided map
distances are physical or genetic. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color</code></td>
<td>
<p> The range of colors used for drawing the heat map. </p>
</td>
</tr>
</table>
<p>The <code>grob</code> <code>LDheatmapGrob</code> has three <code>grob</code>s as its children (components).
They are listed below along with their own children and respectively represent
the color image with main title, genetic map and color key of the heat map:
<code>"heatMap"</code> - <code>"heatmap"</code>, <code>"title"</code>;
<code>"geneMap"</code> - <code>"diagonal"</code>, <code>"segments"</code>,
<code>"title"</code>, <code>"symbols"</code>, <code>"SNPnames"</code>; and
<code>"Key"</code> - <code>"colorKey"</code>, <code>"title"</code>, <code>"labels"</code>,
<code>"ticks"</code>, <code>"box"</code>.
</p>


<h3>Note</h3>

<p>The produced heat map can be modified in two ways.
First, it is possible to edit <em>interactively</em> the grob components of the heat map,
by using the function <code>grid.edit</code>;
the function will not work if there is no
open graphical device showing the heat map.
Alternatively, the user can use the function
<code>editGrob</code> and work with
the grob <code>LDheatmapGrob</code> returned by <code>LDheatmap</code>.
See Examples for usage.
<code>LDheatmap()</code> uses <code>Grid</code>, which
does not respond to <code>par()</code> settings.
Hence modifying <code>par()</code> settings of <code>mfrow</code> and <code>mfcol</code>
will not work with <code>LDheatmap()</code>. The Examples section shows how to
display multiple heat maps on one plot without the use
of <code>par()</code>.
</p>


<h3>References</h3>

<p>Shin J-H, Blay S, McNeney B and Graham J (2006). LDheatmap:
An R Function for Graphical Display of Pairwise Linkage
Disequilibria Between Single Nucleotide Polymorphisms.
Journal of Statistical Software, <b>16</b> Code Snippet 3
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Pass LDheatmap a SnpMatrix object
data(geneHapR_test)
plot_LDheatmap(hap = hapResult,
               gff = gff,
               Chr = hapResult[1,2],
               start = 4000, end = 8200)
</code></pre>


</div>