<div class="container">

<table style="width: 100%;"><tr>
<td>HWE.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimate Disequilibrium and Test for Hardy-Weinberg Equilibrium</h2>

<h3>Description</h3>

<p>Estimate disequilibrium parameter and test the null hypothesis that 
Hardy-Weinberg equilibrium holds. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">HWE.test(x, ...)
## S3 method for class 'genotype'
HWE.test(x, exact = nallele(x)==2, simulate.p.value=!exact,
         B=10000, conf=0.95, ci.B=1000, ... )
## S3 method for class 'data.frame'
HWE.test(x, ..., do.Allele.Freq=TRUE, do.HWE.test=TRUE)
## S3 method for class 'HWE.test'
print(x, show=c("D","D'","r","table"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>genotype or haplotype object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exact</code></td>
<td>
<p>a logical value indicated whether the p-value should be
computed using the exact method, which is only available for 2
allele genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simulate.p.value</code></td>
<td>
<p>a logical value indicating whether the p-value
should be computed using simulation instead of using the
<code class="reqn">\chi^2</code> approximation. Defaults to <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>
<p>Number of simulation iterations to use when
<code>simulate.p.value=TRUE</code>. Defaults to 10000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conf</code></td>
<td>
<p>Confidence level to use when computing the confidence
level for D-hat.  Defaults to 0.95, should be in (0,1). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ci.B</code></td>
<td>
<p>Number of bootstrap iterations to use when computing the
confidence interval. Defaults to 1000.</p>
</td>
</tr>
</table>
<table>
<tr style="vertical-align: top;">
<td><code>show</code></td>
<td>
<p>a character vector containing the names of HWE test
statistics to display from the set of "D", "D'", "r", and "table".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> optional parameters passed to <code>HWE.test</code> (data.frame
method) or <code>chisq.test</code> (base method).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.Allele.Freq</code></td>
<td>
<p>logicial indication whether to summarize allele
frequencies.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.HWE.test</code></td>
<td>
<p>logicial indication whether to perform HWE tests</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>HWE.test calls <code>diseq</code> to computes the Hardy-Weinberg
(dis)equilibrium statistics D, D', and r (correlation coefficient).
Next it calls <code>diseq.ci</code> to compute a bootstrap confidence
interval for these estimates.  Finally, it calls
<code>chisq.test</code> to compute a p-value for Hardy-Weinberg
Equilibrium using a simulation/permutation method.
</p>
<p>Using bootstrapping for the confidence interval and simulation for the
p-value avoids reliance on the assumptions the underlying Chi-square
approximation.  This is particularly important when some allele pairs
have small counts.
</p>
<p>For details on the definition of D, D', and r, see the help page for
<code>diseq</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>HWE.test</code> with components
</p>
<table>
<tr style="vertical-align: top;">
<td><code>diseq</code></td>
<td>
<p>A <code>diseq</code> object providing details on the
disequilibrium estimates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ci</code></td>
<td>
<p>A <code>diseq.ci</code> object providing details on the
bootstrap confidence intervals for the disequilibrium estimates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test</code></td>
<td>
<p>A <code>htest</code> object providing details on the
permutation based Chi-square test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p>function call used to creat this object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conf, B, ci.B, simulate.p.value</code></td>
<td>
<p>values used for these arguments.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Gregory R. Warnes <a href="mailto:greg@warnes.net">greg@warnes.net</a>
</p>


<h3>See Also</h3>

<p><code>genotype</code>,
<code>diseq</code>,
<code>diseq.ci</code>,
<code>HWE.chisq</code>,
<code>HWE.exact</code>,
<code>chisq.test</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Marker with two alleles:
example.data   &lt;- c("D/D","D/I","D/D","I/I","D/D",
                    "D/D","D/D","D/D","I/I","")
g1  &lt;- genotype(example.data)
g1

HWE.test(g1)

## Compare with individual calculations:
diseq(g1)
diseq.ci(g1)
HWE.chisq(g1)
HWE.exact(g1)


## Marker with three alleles: A, C, and T
three.data   &lt;- c(rep("A/A",16),
                  rep("C/A",40),
                  rep("C/T",40),
                  rep("C/C",20),
                  rep("T/T",6))

g3  &lt;- genotype(three.data)
g3

HWE.test(g3, ci.B=10000)
</code></pre>


</div>