<div class="container">

<table style="width: 100%;"><tr>
<td>avgp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Average Periodogram for Multiple (Genetic) Time Series</h2>

<h3>Description</h3>

<p><code>avgp</code> calculates and plots the average periodogram as described in
Wichert, Fokianos and Strimmer (2004).
</p>


<h3>Usage</h3>

<pre><code class="language-R">avgp(x, title = deparse(substitute(x)), plot = TRUE, angular = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>multiple (genetic) time series data. Each <b>column</b> of this
matrix corresponds to a separate variable/time series</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>name of the data set (default is the name of the data object)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>plot the average periodogram?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>angular</code></td>
<td>
<p>convert frequencies to angular frequencies?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>arguments passed to <code>plot</code>
and to <code>periodogram</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The average periodogram is simply the frequency-wise average of the spectral density (as estimated
by the Fourier transform) over all times series. 
To calculate the average periodogram the function <code>periodogram</code>
is used. See Wichert, Fokianos and Strimmer (2004)
for more details.
</p>


<h3>Value</h3>

<p>A list object with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>
<p>A vector with the discrete Fourier frequencies (see <code>periodogram</code>).
If the option angular=TRUE  then the output are angular frequencies (2*pi*f).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>avg.spec</code></td>
<td>
<p>A vector with the average power spectral density at each frequency.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>Name of the data set underlying the average periodogram.</p>
</td>
</tr>
</table>
<p>The result is returned invisibly if <code>plot</code> is true.
</p>


<h3>Author(s)</h3>

<p>Konstantinos Fokianos and
Korbinian Strimmer (<a href="https://www.strimmerlab.org/">https://www.strimmerlab.org/</a>).
</p>


<h3>References</h3>

<p>Wichert, S., Fokianos, K., and Strimmer, K. (2004).  Identifying periodically expressed transcripts
in microarray time series data. <em>Bioinformatics</em> <b>20</b>:5-20.
</p>


<h3>See Also</h3>

<p><code>periodogram</code>, <code>spectrum</code>.</p>


<h3>Examples</h3>

<pre><code class="language-R"># load GeneCycle library
library("GeneCycle")

# load data set
data(caulobacter)

# how many samples and how many genes?
dim(caulobacter)

# average periodogram
avgp.caulobacter &lt;- avgp(caulobacter, "Caulobacter")
avgp.caulobacter

# just compute and don't plot 
avgp(caulobacter, "Caulobacter", plot=FALSE)
</code></pre>


</div>