<div class="container">

<table style="width: 100%;"><tr>
<td>HWE.chisq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform Chi-Square Test for Hardy-Weinberg Equilibrium</h2>

<h3>Description</h3>

<p>Test the null hypothesis that Hardy-Weinberg equilibrium holds using
the Chi-Square method.
</p>


<h3>Usage</h3>

<pre><code class="language-R">HWE.chisq(x, ...)
## S3 method for class 'genotype'
HWE.chisq(x, simulate.p.value=TRUE, B=10000, ...)
          
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>genotype or haplotype object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simulate.p.value</code></td>
<td>
<p>a logical value indicating whether the p-value
should be computed using simulation instead of using the
<code class="reqn">\chi^2</code> approximation. Defaults to <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>
<p>Number of simulation iterations to use when
<code>simulate.p.value=TRUE</code>. Defaults to 10000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> optional parameters passed to <code>chisq.test</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function generates a 2-way table of allele counts, then calls
<code>chisq.test</code> to compute a p-value for Hardy-Weinberg
Equilibrium.  By default, it uses an unadjusted Chi-Square test
statistic and computes the p-value using a simulation/permutation
method.  When <code>simulate.p.value=FALSE</code>, it computes the test
statistic using the Yates continuity correction and tests it against
the asymptotic Chi-Square distribution with the approproate degrees of
freedom.
</p>
<p>Note: The Yates continuty correction is applied *only* when
<code>simulate.p.value=FALSE</code>, so that the reported test statistics
when <code>simulate.p.value=FALSE</code> and <code>simulate.p.value=TRUE</code>
will differ.
</p>


<h3>Value</h3>

<p>An object of class <code>htest</code>.
</p>


<h3>See Also</h3>

<p><code>HWE.exact</code>,
<code>HWE.test</code>,
<code>diseq</code>,
<code>diseq.ci</code>,
<code>allele</code>,
<code>chisq.test</code>,
<code>boot</code>,
<code>boot.ci</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
example.data   &lt;- c("D/D","D/I","D/D","I/I","D/D",
                    "D/D","D/D","D/D","I/I","")
g1  &lt;- genotype(example.data)
g1

HWE.chisq(g1)
# compare with
HWE.exact(g1)
# and 
HWE.test(g1)

three.data   &lt;- c(rep("A/A",8),
                  rep("C/A",20),
                  rep("C/T",20),
                  rep("C/C",10),
                  rep("T/T",3))

g3  &lt;- genotype(three.data)
g3

HWE.chisq(g3, B=10000)


</code></pre>


</div>