<div class="container">

<table style="width: 100%;"><tr>
<td>homozygote</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Extract Features of Genotype objects</h2>

<h3>Description</h3>

<p><code>homozygote</code> creates an vector of logicals that are true when the
alleles of the corresponding observation are the identical.
</p>
<p><code>heterozygote </code> creates an vector of logicals that are true when the
alleles of the corresponding observation differ.
</p>
<p><code>carrier</code> create a logical vector or matrix of logicals
indicating whether the specified alleles are present.
</p>
<p><code>allele.count</code> returns the number of copies of the specified
alleles carried by each observation.
</p>
<p><code>allele</code> extract the specified allele(s) as a character vector
or a 2 column matrix.
</p>
<p><code>allele.names</code> extract the set of allele names.
</p>


<h3>Usage</h3>

<pre><code class="language-R">homozygote(x,  allele.name, ...)
heterozygote(x, allele.name, ...)
carrier(x, allele.name, ...)
## S3 method for class 'genotype'
carrier(x, allele.name=allele.names(x),
        any=!missing(allele.name), na.rm=FALSE, ...)
allele.count(x, allele.name=allele.names(x),any=!missing(allele.name),
             na.rm=FALSE)
allele(x, which=c(1,2) )
allele.names(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
 <p><code>genotype</code> object </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> optional parameters (ignored) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele.name</code></td>
<td>
<p> character value or vector of allele names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>any</code></td>
<td>
<p> logical value.  When <code>TRUE</code>, a single count or
indicator is returned by combining the results for all of the
elements of <code>allele</code>. If <code>FALSE</code> separate counts or
indicators should be returned for each element of
<code>allele</code>.  Defaults to <code>FALSE</code> if <code>allele</code> is
missing. Otherwise defaults to <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.rm</code></td>
<td>
<p> logical value indicating whether to remove missing
values.  When true, any <code>NA</code> values will be replaced by
<code>0</code> or <code>FALSE</code> as appropriate.  Defaults to <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>which</code></td>
<td>
<p> selects which allele to return. For first allele use
<code>1</code>.  For second allele use <code>2</code>.  For both (the default)
use <code>c(1,2)</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>When the <code>allele.name</code> argument is given, heterozygote and
homozygote return <code>TRUE</code> if <em>exactly</em> one or both alleles,
respectively, match the specified allele.name.
</p>


<h3>Value</h3>

<p><code>homozygote</code> and <code>heterozygote </code> return a vector of
logicals.
</p>
<p><code>carrier</code> returns a logical vector if only one allele is
specified, or if <code>any</code> is <code>TRUE</code>.  Otherwise, it returns
matrix of logicals with one row for each element of <code>allele</code>.
</p>
<p><code>allele.count</code> returns a  vector of counts if only one allele is
specified, or if <code>any</code> is <code>TRUE</code>.  Otherwise, it returns
matrix of counts with one row for each element of <code>allele</code>.
</p>
<p><code>allele</code> returns a character vector when one allele is
specified.  When 2 alleles are specified, it returns a 2 column
character matrix.
</p>
<p><code>allele.names</code> returns a character vector containing the set of
allele names.
</p>


<h3>Author(s)</h3>

<p> Gregory R. Warnes <a href="mailto:greg@warnes.net">greg@warnes.net</a> </p>


<h3>See Also</h3>

<p><code>genotype</code>,
<code>HWE.test</code>,




<code>summary.genotype</code>,

<code>locus</code>
<code>gene</code>
<code>marker</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
example.data   &lt;- c("D/D","D/I","D/D","I/I","D/D","D/D","D/D","D/D","I/I","")
g1  &lt;- genotype(example.data)
g1

heterozygote(g1)
homozygote(g1)

carrier(g1,"D")
carrier(g1,"D",na.rm=TRUE)

# get count of one allele 
allele.count(g1,"D")

# get count of each allele
allele.count(g1)  # equivalent to
allele.count(g1, c("D","I"), any=FALSE)

# get combined count for both alleles
allele.count(g1,c("I","D"))

# get second allele
allele(g1,2)

# get both alleles
allele(g1)

</code></pre>


</div>