<div class="container">

<table style="width: 100%;"><tr>
<td>plotHapTable2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>plotHapTable2</h2>

<h3>Description</h3>

<p>plot the hapResult in table like style using grid system.
This function is under development and may not stable.
Some parameters may deleted or renamed in future.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotHapTable2(
  hapSummary,
  show_indel_size = 1,
  replaceMultiAllele = TRUE,
  angle = 0,
  show_INFO = FALSE,
  INFO_split = c(";", ",", "\\|"),
  INFO_tag = "ANN",
  geneID = NA,
  tag_field = -1,
  title = "",
  gff = NULL,
  show_chr_name = TRUE,
  Chr = NULL,
  start = NULL,
  end = NULL,
  model_rect_col = "black",
  model_rect_fill = "grey80",
  model_line_col = "black",
  model_anno_txt = NULL,
  model_anno_col = "black",
  model_anno_cex = 1,
  table_txt_col = "black",
  table_txt_cex = 1,
  model_anno_pos = c(-1, -1),
  model_anno_adj = c(0, 1),
  gene_model_height = 0.2,
  space_height = 0.1,
  table_height = NULL,
  CDS_height = 0.3,
  link_line_type = 3,
  headrows = 1,
  equal_col_width = FALSE,
  head_anno = 1,
  col_annots = 0,
  row_labels = 1,
  row_annots = 1,
  table_line_col = "white",
  annot_for_each_transcrips = TRUE,
  labels_fill = "white",
  annot_fill = "grey90",
  head_fill = NULL,
  cell_fill = NULL,
  style = gpar(fontfamily = "sans", fontface = 1, cex = 0.7),
  footbar = "",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hapSummary</code></td>
<td>
<p>the hapSummary or hapResult object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_indel_size</code></td>
<td>
<p>the Indel length longer will be replaced by "i1,i2,i3,..."</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replaceMultiAllele</code></td>
<td>
<p>replace multi-allele title by 'T1,T2,...' or not</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>angle</code></td>
<td>
<p>the angle of number positions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_INFO</code></td>
<td>
<p>show annotation field or not, default as <code>FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>INFO_split, INFO_tag, geneID, tag_field</code></td>
<td>
<p>used to set annotation in haplotype table.
And the geneID was used to fileter annotation in INFO field.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>title of plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gff</code></td>
<td>
<p>gff or bed annotation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_chr_name</code></td>
<td>
<p>show chromosome name at left-top cell or not</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Chr, start, end</code></td>
<td>
<p>which range should be plotted in gene model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model_rect_col, model_rect_fill, model_line_col</code></td>
<td>
<p>a string specified the color for line/rectangle in gene model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>table_txt_col, table_txt_cex</code></td>
<td>
<p>controls the color and size of texts in genotype table</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model_anno_pos, model_anno_adj, model_anno_cex, model_anno_col, model_anno_txt</code></td>
<td>
<p>the position (x,y), just (hjust, vjust), color, size and content of annotation text in gene model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_model_height, table_height, space_height</code></td>
<td>
<p>the plotting range height of gene model, table and spacer</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CDS_height</code></td>
<td>
<p>a numeric vector specified the height of CDS, and the height of utr is half of that,
only useful when gff is provided,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>link_line_type</code></td>
<td>
<p>the type of link lines for mutations in gene model and genotype table</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>equal_col_width</code></td>
<td>
<p>a bool or numeric vector specified whether column with should be equal</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col_annots, head_anno, headrows, row_annots, row_labels</code></td>
<td>
<p>the column or row number of annotation or labels or heads</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>table_line_col</code></td>
<td>
<p>the line color in genotype table</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annot_for_each_transcrips</code></td>
<td>
<p>mark the strand and trancripts name for each gene modle</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annot_fill, head_fill, labels_fill</code></td>
<td>
<p>the fill color of annotation, head and label row or columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell_fill</code></td>
<td>
<p>a color vector or function or named vector specified cell fill color</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>style</code></td>
<td>
<p>see help(gpar)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>footbar</code></td>
<td>
<p>the foot notes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>param not used</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">#
data(geneHapR_test)
plotHapTable2(hapResult)
plotHapTable2(hapResult, gff = gff)
</code></pre>


</div>