<div class="container">

<table style="width: 100%;"><tr>
<td>GeneScape</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GeneScape</h2>

<h3>Description</h3>

<p>This function simulate single cell RNAseq data with complicated differential expression and correlation structure.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GeneScape(
  nCells = 6000,
  nGroups = NULL,
  groups = NULL,
  lib.size.loc = 9.3,
  lib.size.scale = 0.25,
  de.fc.mat = NULL,
  nGenes = 5000,
  gene.mean.shape = 0.3,
  gene.mean.rate = 0.15,
  gene.means = NULL,
  de.n = 50,
  de.share = NULL,
  de.id = NULL,
  de.fc.loc = 0.7,
  de.fc.scale = 0.2,
  add.sub = FALSE,
  sub.major = NULL,
  sub.prop = 0.1,
  sub.group = NULL,
  sub.de.n = 20,
  sub.de.id = NULL,
  sub.de.common = FALSE,
  sub.de.fc.loc = 0.7,
  sub.de.fc.scale = 0.2,
  add.cor = FALSE,
  cor.n = 4,
  cor.size = 20,
  cor.cor = 0.7,
  cor.id = NULL,
  band.width = 10,
  add.hub = FALSE,
  hub.n = 10,
  hub.size = 20,
  hub.cor = 0.4,
  hub.id = NULL,
  hub.fix = NULL,
  drop = FALSE,
  dropout.location = -2,
  dropout.slope = -1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nCells</code></td>
<td>
<p>number of cells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nGroups</code></td>
<td>
<p>number of cell groups</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groups</code></td>
<td>
<p>group information for cells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lib.size.loc</code></td>
<td>
<p>location parameter for library size (log-normal distribution)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lib.size.scale</code></td>
<td>
<p>scale parameter for library size (log-normal distribution)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de.fc.mat</code></td>
<td>
<p>differential expression fold change matrix, could be generated by this function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nGenes</code></td>
<td>
<p>number of genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.mean.shape</code></td>
<td>
<p>shape parameter for mean expression level (Gamma distribution)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.mean.rate</code></td>
<td>
<p>rate parameter for mean expression level (Gamma distribution)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.means</code></td>
<td>
<p>mean gene expression levels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de.n</code></td>
<td>
<p>number of differentially expressed genes in each cell type. Should be a integer or a vector of length nGroups</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de.share</code></td>
<td>
<p>number of shared DE genes between neighbor cell types. Should be a vector of length (nGroups - 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de.id</code></td>
<td>
<p>the index of genes that are DE across cell types. Should be a list of vectors. 
Each vector corresponds to a cell type. With non-null value of de.id, de.n and de.share would be ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de.fc.loc</code></td>
<td>
<p>the location parameter for the fold change of DE genes. Should be a number, a vector of length nGroups</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de.fc.scale</code></td>
<td>
<p>the scale parameter for fold change (log-normal distribution). Should be a number or a vector of length nGroups</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add.sub</code></td>
<td>
<p>whether to add sub-cell-types</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sub.major</code></td>
<td>
<p>the major cell types correspond to the sub-cell-types</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sub.prop</code></td>
<td>
<p>proportion of sub-cell-types in the corresponding major cell type</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sub.group</code></td>
<td>
<p>cell index for sub-cell-types. With non-null sub.group specified, sub.prop would be ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sub.de.n</code></td>
<td>
<p>number of differentially expressed genes in each sub-cell-type compared to the corresponding major cell type. Should be a integer or a vector of length sub.major</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sub.de.id</code></td>
<td>
<p>the index of additional differentially expressed genes between sub-cell-types and the corresponding major cell types</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sub.de.common</code></td>
<td>
<p>whether the additional differential expression structure should be same for all sub-cell-types</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sub.de.fc.loc</code></td>
<td>
<p>similar to de.fc.loc, but for addtional differentially expressed genes in sub-cell-types</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sub.de.fc.scale</code></td>
<td>
<p>similar to de.fc.scale, but for addtional differentially expressed genes in sub-cell-types</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add.cor</code></td>
<td>
<p>whether to add pathways (correlated genes)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor.n</code></td>
<td>
<p>number of pathways included. Should be a integer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor.size</code></td>
<td>
<p>number of correlated genes (length of pathway). Should be a number or a vector of length cor.n</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor.cor</code></td>
<td>
<p>correlation parameters</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor.id</code></td>
<td>
<p>gene index of correlated (pathway) genes. Should be a list of vectors, with each vector represents a pathway. With non-null value of cor.id, cor.n would be ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>band.width</code></td>
<td>
<p>No correlation exists if distance of 2 genes are further than band_width in a pathway</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add.hub</code></td>
<td>
<p>whether to add hub genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hub.n</code></td>
<td>
<p>number of hub genes included. Should be a integer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hub.size</code></td>
<td>
<p>number of genes correlated to the hub gene. Should be a number or a vector of length hub.n</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hub.cor</code></td>
<td>
<p>correlation parameters between hub genes and their correlated genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hub.id</code></td>
<td>
<p>gene index of hub genes. Should be a list of vectors. With non-null value of hub.id, hub.n would be ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hub.fix</code></td>
<td>
<p>user defined genes correlated to hub genes (others are randomly selected). Should be a list of vectors of length hub.n or same as hub.id.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drop</code></td>
<td>
<p>whether to add dropout</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dropout.location</code></td>
<td>
<p>dropout mid point (the mean expression level at which the probability is equal to 0.5, same as splat. Could be negative)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dropout.slope</code></td>
<td>
<p>how dropout proportion changes with increasing expression</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Compared to splat method in Splatter R package, this function can fix the number and position of differentially expressed genes, 
have more complicated differential expression structure, add sub-cell-types, correlated genes (AR(1) correlation structure with bound, mimicking pathways) and hub genes.
</p>


<h3>Value</h3>

<p>A list of observed data, true data (without dropout), differential expression rate and hub gene indices.
</p>


<h3>References</h3>

<p>Zappia, L., Phipson, B., &amp; Oshlack, A. (2017). Splatter: Simulation of single-cell RNA sequencing data. Genome Biology, 18(1). https://doi.org/10.1186/s13059-017-1305-0
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(1)
data &lt;- GeneScape()
</code></pre>


</div>