<div class="container">

<table style="width: 100%;"><tr>
<td>ashaplotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>as.haplotype</h2>

<h3>Description</h3>

<p>convert <code>hapSummary</code> or <code>hapResult</code> class into <code>haplotype</code> class (pegas)
</p>


<h3>Usage</h3>

<pre><code class="language-R">as.haplotype(hap)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>hap</code></td>
<td>
<p>object of <code>hapSummary</code> or <code>hapResult</code> class</p>
</td>
</tr></table>
<h3>Value</h3>

<p>haplotype class
</p>


<h3>Note</h3>

<p>It's not advised for <code>hapSummary</code> or <code>hapResult</code> with indels, due to indels will
convert to SNPs with equal length of each indel.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("geneHapR_test")
hap &lt;- as.haplotype(hapResult)
hapSummary &lt;- hap_summary(hapResult)
hap &lt;- as.haplotype(hapSummary)
</code></pre>


</div>