<div class="container">

<table style="width: 100%;"><tr>
<td>summary.genotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Allele and Genotype Frequency from a Genotype or
Haplotype Object</h2>

<h3>Description</h3>

<p><code>summary.genotype</code> creates an object containing allele and
genotype frequency from a <code>genotype</code> or <code>haplotype</code>
object.  <code>print.summary.genotype</code> displays a
<code>summary.genotype</code> object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  ## S3 method for class 'genotype'
summary(object, ..., maxsum)
  ## S3 method for class 'summary.genotype'
print(x,...,round=2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object, x</code></td>
<td>
<p> an object of class <code>genotype</code> or <code>haplotype</code> (for
<code>summary.genotype</code>) or an object of class
<code>summary.genotype</code> (for <code>print.summary.genotype</code>) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> optional parameters.  Ignored by <code>summary.genotype</code>,
passed to <code>print.matrix</code> by <code>print.summary,genotype</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxsum</code></td>
<td>
<p> specifying any value for the parameter
maxsum will cause <code>summary.genotype</code> to fall back to
<code>summary.factor</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>round</code></td>
<td>
<p> number of digits to use when displaying proportions.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Specifying any value for the parameter <code>maxsum</code> will cause fallback
to <code>summary.factor</code>.  This is so that the function
<code>summary.dataframe</code> will give reasonable output when it contains a
genotype column.  (Hopefully we can figure out something better to do
in this case.)
</p>


<h3>Value</h3>

<p>The returned value of <code>summary.genotype</code> is an object of class
<code>summary.genotype</code> which
is a list with the following components:
</p>
<table><tr style="vertical-align: top;">
<td><code>locus </code></td>
<td>
<p>locus information field (if present) from <code>x</code></p>
</td>
</tr></table>
<p>.
</p>
<table>
<tr style="vertical-align: top;">
<td><code>allele.names</code></td>
<td>
<p> vector of allele names </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele.freq </code></td>
<td>

<p>A two column matrix with one row for each allele, plus one row for
<code>NA</code> values (if present).  The first column, <code>Count</code>,
contains the frequency of the corresponding allele value.  The
second column, <code>Proportion</code>, contains the fraction of alleles
with the corresponding allele value.  Note each observation contains
two alleles, thus the <code>Count</code> field sums to twice the number of
observations.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype.freq</code></td>
<td>

<p>A two column matrix with one row for each genotype, plus one row for
<code>NA</code> values (if present). The first column, <code>Count</code>, contains the
frequency of the corresponding genotype.  The second column,
<code>Proportion</code>, contains the fraction of genotypes with the
corresponding value.
</p>
</td>
</tr>
</table>
<p><code>print.summary.genotype</code> silently returns the object <code>x</code>.
</p>


<h3>Author(s)</h3>

<p> Gregory R. Warnes <a href="mailto:greg@warnes.net">greg@warnes.net</a> </p>


<h3>See Also</h3>

<p><code>genotype</code>,
<code>HWE.test</code>,
<code>allele</code>,
<code>homozygote</code>,
<code>heterozygote</code>,
<code>carrier</code>,

<code>allele.count</code>
<code>locus</code>
<code>gene</code>
<code>marker</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
example.data   &lt;- c("D/D","D/I","D/D","I/I","D/D",
                    "D/D","D/D","D/D","I/I","")
g1  &lt;- genotype(example.data)
g1

summary(g1)
</code></pre>


</div>