<div class="container">

<table style="width: 100%;"><tr>
<td>filterLargep.link</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pre-process of Large VCF File(s)</h2>

<h3>Description</h3>

<p>Filter/extract one or multiple gene(s)/range(s) from a large
<code>p.link</code> file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">filterLargeP.link(
  root,
  rootOut = rootOut,
  Chr = Chr,
  POS = NULL,
  start = start,
  end = end,
  override = TRUE,
  sep = "\t"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>root</code></td>
<td>
<p>The file name without suffix. This function only support p.link file
format stored in "map" and "ped" format, the file names after removed suffix
should be same with each other.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rootOut</code></td>
<td>
<p>Path(s) of output <code>p.link</code> file stored in "ped&amp;map" format.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Chr</code></td>
<td>
<p>a single CHROM name or CHROM names vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>POS, start, end</code></td>
<td>
<p>provide the chromosome name should be extract from orignal p.link dataset.
<code>POS</code>: a vector consist with start and end position, eg.: <code>c(1,200)</code>
indicates 3 ranges (1~200, 300~500 and 300~400).
if <code>POS</code> is <code>NULL</code>, <code>start</code> and <code>end</code> are needed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>override</code></td>
<td>
<p>whether override existed file or not, default as <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>a character indicate the separation of map and ped file, default is <code style="white-space: pre;">⁠\t⁠</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This package import P.link files. However, import a large P.link file is time and
memory consuming. It's suggested that extract variants in target
range with <code>filterLargeP.link()</code> before identification of haplotype.
</p>
<p>When filter/extract multi genes/ranges, the parameter of <code>Chr</code> and <code>POS</code>
must have equal length. Results will save to a single file if the user
provide a single file path or save to multiple P.link file(s) when a equal length
vector consist with file paths is provided.
</p>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class="language-R">
 # The filteration of P.link of regular size should be done with `filter_plink.pedmap()`.
 # however, here, we use a mini vcf instead just for example and test

 pedfile &lt;- system.file("extdata",
                        "snp3kvars-CHR8-25947258-25951166-plink.ped",
                        package = "geneHapR")
 mapfile &lt;- system.file("extdata",
                        "snp3kvars-CHR8-25947258-25951166-plink.map",
                        package = "geneHapR")
 oldDir &lt;- getwd()
 temp_dir &lt;- tempdir()
 if(! dir.exists(temp_dir))
   dir.create(temp_dir)
 setwd(temp_dir)
 file.copy(pedfile, "test.ped")
 file.copy(mapfile, "test.map")

 # extract a single gene/range from large vcf
 filterLargeP.link(root = "test",
                   rootOut = "filtered_test",
                   Chr = "scaffold_1", POS = c(4300,5000), override = TRUE)

setwd(oldDir)

# delete temp_dir
unlink(temp_dir, recursive = TRUE)

</code></pre>


</div>