<div class="container">

<table style="width: 100%;"><tr>
<td>genotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genotype or Haplotype Objects.</h2>

<h3>Description</h3>

<p><code>genotype</code> creates a genotype object.
</p>
<p><code>haplotype</code> creates a haplotype object.
</p>
<p><code>is.genotype</code> returns <code>TRUE</code> if <code>x</code> is of class
<code>genotype</code>
</p>
<p><code>is.haplotype</code> returns <code>TRUE</code> if <code>x</code> is of class
<code>haplotype</code>
</p>
<p><code>as.genotype</code> attempts to coerce its argument into an object of
class <code>genotype</code>.
</p>
<p><code>as.genotype.allele.count</code> converts allele counts (0,1,2) into
genotype pairs ("A/A", "A/B", "B/B").
</p>
<p><code>as.haplotype</code> attempts to coerce its argument into an object of
class <code>haplotype</code>.
</p>
<p><code>nallele</code> returns the number of alleles in an object of class
<code>genotype</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  genotype(a1, a2=NULL, alleles=NULL, sep="/", remove.spaces=TRUE,
           reorder = c("yes", "no", "default", "ascii", "freq"),
           allow.partial.missing=FALSE, locus=NULL,
           genotypeOrder=NULL)

  haplotype(a1, a2=NULL, alleles=NULL, sep="/", remove.spaces=TRUE,
            reorder="no", allow.partial.missing=FALSE, locus=NULL,
            genotypeOrder=NULL)

  is.genotype(x)

  is.haplotype(x)

  as.genotype(x, ...)

  ## S3 method for class 'allele.count'
as.genotype(x, alleles=c("A","B"), ... )

  as.haplotype(x, ...)

  ## S3 method for class 'genotype'
print(x, ...)

  nallele(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> either an object of class <code>genotype</code> or
<code>haplotype</code> or an object to be converted to class <code>genotype</code> or
<code>haplotype</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a1,a2</code></td>
<td>
<p> vector(s) or matrix containing two alleles
for each individual. See details, below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p> names (and order if <code>reorder="yes"</code>) of possible
alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p> character separator or column number used to divide
alleles when <code>a1</code> is a vector of strings where each string
holds both alleles. See below for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove.spaces</code></td>
<td>
<p> logical indicating whether spaces and tabs will
be removed from a1 and a2  before processing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reorder</code></td>
<td>
<p>how should alleles within an individual be reordered.
If <code>reorder="no"</code>, use the order specified by the alleles
parameter.  If <code>reorder="freq"</code> or
<code>reorder="yes"</code>, sort alleles within each individual by
observed frequency.  If <code>reorder="ascii"</code>, reorder alleles in
ASCII order (alphabetical, with all upper case before
lower case). The default value for <code>genotype</code>
is <code>"freq"</code>.  The default value for <code>haplotype</code> is
<code>"no"</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allow.partial.missing</code></td>
<td>
<p>logical indicating whether one allele is
permitted to be missing.  When set to <code>FALSE</code> both alleles
are set to <code>NA</code> when either is missing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus</code></td>
<td>
<p> object of class locus, gene, or marker, holding
information about the source of this genotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypeOrder</code></td>
<td>
<p>character, vector of genotype/haplotype names so
that further functions can sort genotypes/haplotypes in wanted
order</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>optional arguments</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Genotype objects hold information on which gene or marker alleles were
observed for different individuals.  For each individual, two alleles
are recorded.
</p>
<p>The genotype class considers the stored alleles to be unordered, i.e., "C/T"
is equivalent to "T/C".  The haplotype class considers the order of the
alleles to be significant so that "C/T" is distinct from "T/C".
</p>
<p>When calling <code>genotype</code> or <code>haplotype</code>:
</p>

<ul>
<li>
<p> If only <code>a1</code> is provided and is a character vector, it is
assumed that each element encodes both alleles. In this case, if
<code>sep</code> is a character string, <code>a1</code> is assumed to be coded
as "Allele1&lt;sep&gt;Allele2".  If <code>sep</code> is a numeric value, it is
assumed that character locations <code>1:sep</code> contain allele 1 and
that remaining locations contain allele 2.
</p>
</li>
<li>
<p> If <code>a1</code> is a matrix, it is assumed that column 1 contains
allele 1 and column 2 contains allele 2.
</p>
</li>
<li>
<p> If <code>a1</code> and <code>a2</code> are both provided, each is assumed to
contain one allele value so that the genotype for an individual is
obtained by <code>paste(a1,a2,sep="/")</code>.
</p>
</li>
</ul>
<p>If <code>remove.spaces</code> is TRUE, (the default) any whitespace
contained in <code>a1</code> and <code>a2</code> is removed when the genotypes are
created.  If whitespace is used as the separator, (eg "C C", "C T",
...), be sure to set remove.spaces to FALSE.
</p>
<p>When the alleles are explicitly specified using the <code>alleles</code>
argument, all potential alleles not present in the list will be
converted to <code>NA</code>.
</p>
<p>NOTE: <code>genotype</code> assumes that the order of the alleles is not important
(E.G., "A/C" == "C/A").  Use class <code>haplotype</code> if order is significant.
</p>
<p>If <code>genotypeOrder=NULL</code> (the default setting), then
<code>expectedGenotypes</code> is used to get standard sorting order.
Only unique values in <code>genotypeOrder</code> are used, which in turns
means that the first occurrence prevails. When <code>genotypeOrder</code> is
given some genotype names, but not all that appear in the data, the
rest (those in the data and possible combinations based on allele
variants) is automatically added at the end of
<code>genotypeOrder</code>. This puts "missing" genotype names at the end of
sort order. This feature is especially useful when there are a lot of
allele variants and especially in haplotypes. See examples.
</p>


<h3>Value</h3>

<p>The genotype class extends "factor" and haplotype extends
genotype. Both classes have the following attributes:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>levels</code></td>
<td>
<p> character vector of possible genotype/haplotype values
stored coded by <code>paste( allele1, "/", allele2, sep="")</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele.names</code></td>
<td>
<p> character vector of possible alleles. For a SNP,
these might be c("A","T").   For a variable length dinucleotyde
repeat this might be c("136","138","140","148"). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele.map</code></td>
<td>
<p> matrix encoding how the factor levels correspond to
alleles.  See the source code to <code>allele.genotype()</code> for how to
extract allele values using this matrix.  Better yet, just use
<code>allele.genotype()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypeOrder</code></td>
<td>
<p>character, genotype/haplotype names in
defined order that can used for sorting in various functions. Note
that this slot stores both ordered and unordered genotypes
i.e. "A/B" and "B/A".</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Gregory R. Warnes <a href="mailto:greg@warnes.net">greg@warnes.net</a> and
Friedrich Leisch.</p>


<h3>See Also</h3>

<p><code>HWE.test</code>,
<code>allele</code>,
<code>homozygote</code>,
<code>heterozygote</code>,
<code>carrier</code>,
<code>summary.genotype</code>,
<code>allele.count</code>,
<code>sort.genotype</code>,
<code>genotypeOrder</code>,
<code>locus</code>,
<code>gene</code>,
<code>marker</code>, and
<code>%in%</code> for default %in% method
</p>


<h3>Examples</h3>

<pre><code class="language-R"># several examples of genotype data in different formats
example.data   &lt;- c("D/D","D/I","D/D","I/I","D/D",
                    "D/D","D/D","D/D","I/I","")
g1  &lt;- genotype(example.data)
g1

example.data2  &lt;- c("C-C","C-T","C-C","T-T","C-C",
                    "C-C","C-C","C-C","T-T","")
g2  &lt;- genotype(example.data2,sep="-")
g2


example.nosep  &lt;- c("DD", "DI", "DD", "II", "DD",
                    "DD", "DD", "DD", "II", "")
g3  &lt;- genotype(example.nosep,sep="")
g3

example.a1 &lt;- c("D",  "D",  "D",  "I",  "D",  "D",  "D",  "D",  "I",  "")
example.a2 &lt;- c("D",  "I",  "D",  "I",  "D",  "D",  "D",  "D",  "I",  "")
g4  &lt;- genotype(example.a1,example.a2)
g4

example.mat &lt;- cbind(a1=example.a1, a1=example.a2)
g5  &lt;- genotype(example.mat)
g5

example.data5  &lt;- c("D   /   D","D   /   I","D   /   D","I   /   I",
                    "D   /   D","D   /   D","D   /   D","D   /   D",
                    "I   /   I","")
g5  &lt;- genotype(example.data5,rem=TRUE)
g5

# show how genotype and haplotype differ
data1 &lt;- c("C/C", "C/T", "T/C")
data2 &lt;- c("C/C", "T/C", "T/C")

test1  &lt;- genotype( data1 )
test2  &lt;- genotype( data2 )

test3  &lt;-  haplotype( data1 )
test4  &lt;-  haplotype( data2 )

test1==test2
test3==test4

test1=="C/T"
test1=="T/C"

test3=="C/T"
test3=="T/C"

## also
test1 
test1 
test3 

test1 
test1 

test3 
test3 

## "Messy" example

m3  &lt;-  c("D D/\t   D D","D\tD/   I",  "D D/   D D","I/   I",
          "D D/   D D","D D/   D D","D D/   D D","D D/   D D",
          "I/   I","/   ","/I")

genotype(m3)
summary(genotype(m3))

m4  &lt;-  c("D D","D I","D D","I I",
          "D D","D D","D D","D D",
          "I I","   ","  I")

genotype(m4,sep=1)
genotype(m4,sep=" ",remove.spaces=FALSE)
summary(genotype(m4,sep=" ",remove.spaces=FALSE))

m5  &lt;-  c("DD","DI","DD","II",
          "DD","DD","DD","DD",
          "II","   "," I")
genotype(m5,sep=1)
haplotype(m5,sep=1,remove.spaces=FALSE)

g5  &lt;- genotype(m5,sep="")
h5  &lt;- haplotype(m5,sep="")

heterozygote(g5)
homozygote(g5)
carrier(g5,"D")

g5[9:10]  &lt;- haplotype(m4,sep=" ",remove=FALSE)[1:2]
g5

g5[9:10]
allele(g5[9:10],1)
allele(g5,1)[9:10]

# drop unused alleles
g5[9:10,drop=TRUE]
h5[9:10,drop=TRUE]

# Convert allele.counts into genotype

x &lt;- c(0,1,2,1,1,2,NA,1,2,1,2,2,2)
g &lt;- as.genotype.allele.count(x, alleles=c("C","T") )
g

# Use of genotypeOrder
example.data   &lt;- c("D/D","D/I","I/D","I/I","D/D",
                    "D/D","D/I","I/D","I/I","")
summary(genotype(example.data))
genotypeOrder(genotype(example.data))

summary(genotype(example.data, genotypeOrder=c("D/D", "I/I", "D/I")))
summary(genotype(example.data, genotypeOrder=c(              "D/I")))
summary(haplotype(example.data, genotypeOrder=c(             "I/D", "D/I")))
example.data &lt;- genotype(example.data)
genotypeOrder(example.data) &lt;- c("D/D", "I/I", "D/I")
genotypeOrder(example.data)

</code></pre>


</div>