<div class="container">

<table style="width: 100%;"><tr>
<td>expectedGenotypes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Construct expected genotypes/haplotypes according to known allele
variants</h2>

<h3>Description</h3>

<p><code>expectedGenotypes</code> constructs expected genotypes according to
known allele variants, which can be quite tedious with large number of
allele variants. It can handle different level of ploidy.
</p>


<h3>Usage</h3>

<pre><code class="language-R">expectedGenotypes(x, alleles=allele.names(x), ploidy=2, sort=TRUE,
                  haplotype=FALSE)
expectedHaplotypes(x, alleles=allele.names(x), ploidy=2, sort=TRUE,
                   haplotype=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>genotype or haplotype</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>character, vector of allele names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>numeric, number of chromosome sets i.e. 2 for human autosomal
genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort</code></td>
<td>
<p>logical, sort genotypes according to order of alleles in
<code>alleles</code> argument</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haplotype</code></td>
<td>
<p>logical, construct haplotypes i.e. ordered genotype</p>
</td>
</tr>
</table>
<p>At least one of <code>x</code> or <code>alleles</code> must be given.
</p>


<h3>Details</h3>

<p><code>expectedHaplotypes()</code> just calls <code>expectedGenotypes()</code> with
argument <code>haplotype=TRUE</code>.
</p>


<h3>Value</h3>

<p>A character vector with genotype names as "alele1/alele2" for diploid
example. Length of output is <code class="reqn">(n*(n+1))/2</code> for genotype (unordered
genotype) and <code class="reqn">n*n</code> for haplotype (ordered genotype) for <code class="reqn">n</code>
allele variants.
</p>


<h3>Author(s)</h3>

<p> Gregor Gorjanc </p>


<h3>See Also</h3>

 <p><code>allele.names</code>, <code>genotype</code></p>


<h3>Examples</h3>

<pre><code class="language-R">  ## On genotype
  prp &lt;- c("ARQ/ARQ", "ARQ/ARQ", "ARR/ARQ", "AHQ/ARQ", "ARQ/ARQ")
  alleles &lt;- c("ARR", "AHQ", "ARH", "ARQ", "VRR", "VRQ")
  expectedGenotypes(as.genotype(prp))
  expectedGenotypes(as.genotype(prp, alleles=alleles))
  expectedGenotypes(as.genotype(prp, alleles=alleles, reorder="yes"))

  ## Only allele names
  expectedGenotypes(alleles=alleles)
  expectedGenotypes(alleles=alleles, ploidy=4)

  ## Haplotype
  expectedHaplotypes(alleles=alleles)
  expectedHaplotypes(alleles=alleles, ploidy=4)[1:20]
</code></pre>


</div>