<div class="container">

<table style="width: 100%;"><tr>
<td>filter_plink.pedmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>filter_plink.pedmap</h2>

<h3>Description</h3>

<p>used for filtration of p.link
</p>


<h3>Usage</h3>

<pre><code class="language-R">filter_plink.pedmap(x,
                    mode = c("POS", "type", "both"),
                    Chr = Chr, start = start, end = end,
                    gff = gff, type = type, cusTyp = cusTyp,
                    geneID = geneID)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a list stored the p.link information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>
<p>filtration mode, one of c("POS", "type", "both")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Chr</code></td>
<td>
<p>the chromosome name, need if mode set as POS or both</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start, end</code></td>
<td>
<p>numeric, the range of filtration, and the start should smaller than end</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gff</code></td>
<td>
<p>the imported gff object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>should be in <code>unique(gff$type)</code>, usually as "CDS", "genome".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cusTyp</code></td>
<td>
<p>if <code>type</code> set as custom, then <code>cusTyp</code> is needed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneID</code></td>
<td>
<p>geneID</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list, similar with <code>x</code>, but filtered
</p>


<h3>Examples</h3>

<pre><code class="language-R">
   pedfile &lt;- system.file("extdata",
                          "snp3kvars-CHR8-25947258-25951166-plink.ped",
                          package = "geneHapR")
   mapfile &lt;- system.file("extdata",
                          "snp3kvars-CHR8-25947258-25951166-plink.map",
                          package = "geneHapR")
   p.link &lt;- import_plink.pedmap(pedfile = pedfile, mapfile = mapfile,
                                 sep_map = "\t", sep_ped = "\t")
   p.link &lt;- filter_plink.pedmap(p.link, mode = "POS",
                                 Chr = "chr08", start = 25948004,
                                 end = 25949944)
   hapResult &lt;- plink.pedmap2hap(p.link, hapPrefix = "H",
                                 hetero_remove = TRUE,
                                 na_drop = TRUE)

</code></pre>


</div>