<div class="container">

<table style="width: 100%;"><tr>
<td>locus</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Create and Manipulate Locus, Gene, and Marker Objects</h2>

<h3>Description</h3>

<p><code>locus</code>, <code>gene</code>, and <code>marker</code> create objects to store
information, respectively, about genetic loci, genes, and markers.
</p>
<p><code>is.locus</code>, <code>is.gene</code>, and <code>ismarker</code> test whether an
object is a member of the respective class.
</p>
<p><code>as.character.locus</code>, <code>as.character.gene</code>,
<code>as.character.marker</code> return a character string containing a
compact encoding the object.
</p>
<p><code>getlocus</code>, <code>getgene</code>, <code>getmarker</code> extract locus data
(if present) from another object.
</p>
<p><code>locus&lt;-</code>, <code>marker&lt;-</code>, and <code>gene&lt;-</code> adds locus data to
an object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  locus(name, chromosome, arm=c("p", "q", "long", "short", NA),
        index.start, index.end=NULL)

  gene(name, chromosome, arm=c("p", "q", "long", "short"),
       index.start, index.end=NULL)

  marker(name, type, locus.name, bp.start, bp.end = NULL,
         relative.to = NULL, ...)

  is.locus(x)

  is.gene(x)

  is.marker(x)

  ## S3 method for class 'locus'
as.character(x, ...)

  ## S3 method for class 'gene'
as.character(x, ...)

  ## S3 method for class 'marker'
as.character(x, ...)

  getlocus(x, ...)

  locus(x) &lt;- value

  marker(x) &lt;- value

  gene(x) &lt;- value

</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>character string giving locus, gene, or marker name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome</code></td>
<td>
<p>integer specifying chromosome number (1:23 for humans).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arm</code></td>
<td>
<p>character indicating long or short arm of the chromosome.
Long is be specified by "long" or "p".  Short is specified by
"short" or "q".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index.start</code></td>
<td>
<p>integer specifying location of start of locus or
gene on the chromosome.  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index.end</code></td>
<td>
<p>optional integer specifying location of end of locus or
gene on the chromosome.  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>character string indicating marker type, e.g. "SNP"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus.name</code></td>
<td>
<p>either a character string giving the name of the
locus or gene (other details may be specified using <code>...</code>) or a
<code>locus</code> or <code>gene</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp.start</code></td>
<td>
<p>start location of marker, in base pairs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp.end</code></td>
<td>
<p>end location of marker, in base pairs (optional)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>relative.to</code></td>
<td>
<p>location (optional) from which <code>bp.start</code> and
<code>bp.end</code> are calculated. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>parameters for <code>locus</code> used to fill in additional
details on the locus or gene within which the marker is located. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>locus</code>, <code>gene</code>, or
<code>marker</code>, or (for <code>getlocus</code>, <code>locus&lt;-</code>,
<code>marker&lt;-</code>, and <code>gene&lt;-</code>) an object that may contain a locus
attribute or field, notably a <code>genotype</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>value</code></td>
<td>
<p><code>locus</code>, <code>marker</code>, or <code>gene</code> object</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Object of class <code>locus</code> and <code>gene</code>are lists with the
elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>character string giving locus, gene, or marker name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome</code></td>
<td>
<p>integer specifying chromosome number (1:23 for humans).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arm</code></td>
<td>
<p>character indicating long or short arm of the chromosome.
Long is be specified by "long" or "p".  Short is specified by
"short" or "q".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index.start</code></td>
<td>
<p>integer specifying location of start of locus or
gene on the chromosome.  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index.end</code></td>
<td>
<p>optional integer specifying location of end of locus or
gene on the chromosome.  </p>
</td>
</tr>
</table>
<p>Objects of class <code>marker</code> add the additional fields:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>marker.name</code></td>
<td>
<p>character string giving the name of the marker</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp.start</code></td>
<td>
<p>start location of marker, in base pairs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp.end</code></td>
<td>
<p>end location of marker, in base pairs (optional)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>relative.to</code></td>
<td>
<p>location (optional) from which <code>bp.start</code> and
<code>bp.end</code> are calculated. </p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Gregory R. Warnes <a href="mailto:greg@warnes.net">greg@warnes.net</a> </p>


<h3>See Also</h3>

<p><code>genotype</code>,
</p>


<h3>Examples</h3>

<pre><code class="language-R">ar2  &lt;- gene("AR2",chromosome=7,arm="q",index.start=35)
ar2

par  &lt;- locus(name="AR2 Psedogene",
              chromosome=1,
              arm="q",
              index.start=32,
              index.end=42)
par

c109t  &lt;- marker(name="C-109T",
                 type="SNP",
                 locus.name="AR2",
                 chromosome=7,
                 arm="q",
                 index.start=35,
                 bp.start=-109,
                 relative.to="start of coding region")
c109t

c109t  &lt;- marker(name="C-109T",
                 type="SNP",
                 locus=ar2,
                 bp.start=-109,
                 relative.to="start of coding region")
c109t




example.data   &lt;- c("D/D","D/I","D/D","I/I","D/D",
                    "D/D","D/D","D/D","I/I","")
g1  &lt;- genotype(example.data, locus=ar2)
g1

getlocus(g1)

summary(g1)
HWE.test(g1)

g2  &lt;- genotype(example.data, locus=c109t)
summary(g2)

getlocus(g2)

heterozygote(g2)
homozygote(g1)

allele(g1,1)

carrier(g1,"I")

heterozygote(g2)
</code></pre>


</div>