<div class="container">

<table style="width: 100%;"><tr>
<td>network.test.edges</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Graphical Gaussian Models: Assess Significance of Edges (and Directions)</h2>

<h3>Description</h3>

<p><code>network.test.edges</code> returns a data frame containing all edges listed
in order of the magnitude of the partial correlation associated with each edge.
If <code>fdr=TRUE</code> then in addition the
p-values, q-values and posterior probabilities (=1 - local fdr) for each potential 
edge are computed.
</p>
<p><code>extract.network</code> returns a data frame with a subset of significant
edges. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">network.test.edges(r.mat, fdr=TRUE, direct=FALSE, plot=TRUE, ...)
extract.network(network.all, method.ggm=c("prob", "qval","number"), 
      cutoff.ggm=0.8, method.dir=c("prob","qval","number", "all"), 
      cutoff.dir=0.8, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>r.mat</code></td>
<td>
<p>matrix of partial correlations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr</code></td>
<td>
<p>estimate q-values and local fdr</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>direct</code></td>
<td>
<p>compute additional statistics for obtaining a partially directed network</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>plot density and distribution function and (local) fdr values</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>parameters passed on to <code>fdrtool</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>network.all</code></td>
<td>
<p>list with partial correlations and fdr values for all potential edges
(i.e. the output of <code>network.test.edges</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method.ggm</code></td>
<td>
<p>determines which criterion is used to select significant partial correlations (default: prob)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff.ggm</code></td>
<td>
<p>default cutoff for significant partial correlations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method.dir</code></td>
<td>
<p>determines which criterion is used to select significant directions (default: prob) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff.dir</code></td>
<td>
<p>default cutoff for significant directions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>print information on the number of significant edges etc.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For assessing the significance of edges in the GGM 
a mixture model is fitted to the partial correlations using <code>fdrtool</code>. 
This results in (i)  two-sided p-values for the test of non-zero correlation,
(ii) corresponding posterior probabilities (= 1- local fdr), as well as (iii) tail
area-based q-values.  See  Sch\"afer and Strimmer (2005) for details.
</p>
<p>For determining putatative directions on this GGM log-ratios of standardized partial 
variances re estimated, and subsequently the corresponding (local) fdr values are
computed - see Opgen-Rhein and Strimmer (2007).
</p>


<h3>Value</h3>

<p><code>network.test.edges</code> returns a
data frame with the following columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>pcor</code></td>
<td>
<p>correlation (from r.mat)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>node1</code></td>
<td>
<p>first node connected to edge</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>node2</code></td>
<td>
<p>second node connected to edge</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval</code></td>
<td>
<p>p-value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qval</code></td>
<td>
<p>q-value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prob</code></td>
<td>
<p>probability that edge is nonzero (= 1-local fdr</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.spvar</code></td>
<td>
<p>log ratio of standardized partial variance (determines direction)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval.dir</code></td>
<td>
<p>p-value (directions)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qval.dir</code></td>
<td>
<p>q-value (directions)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prob.dir</code></td>
<td>
<p>1-local fdr (directions) </p>
</td>
</tr>
</table>
<p>Each row in the data frame corresponds to one edge, and the rows are sorted
according the absolute strength of the correlation (from strongest to weakest)
</p>
<p><code>extract.network</code> processes the above data frame containing all potential edges,
and returns a dataframe with a subset of edges.  If applicable, an additional
last column (11) contains additional information on the directionality of an edge.
</p>


<h3>Author(s)</h3>

<p>Rainer Opgen-Rhein,
Juliane Sch\"afer,
Korbinian Strimmer (<a href="https://strimmerlab.github.io">https://strimmerlab.github.io</a>).
</p>


<h3>References</h3>

<p>Sch\"afer, J., and Strimmer, K. (2005).  An empirical Bayes approach to inferring
large-scale gene association networks. <em>Bioinformatics</em> <b>21</b>:754-764.
</p>
<p>Opgen-Rhein, R., and K. Strimmer. (2007). From correlation to causation networks: 
a simple approximate learning algorithm and its application to high-dimensional 
plant gene expression data. <em>BMC Syst. Biol.</em> <b>1</b>:37.
</p>


<h3>See Also</h3>

<p><code>cor0.test</code>,
<code>fdrtool</code>,
<code>ggm.estimate.pcor</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># load GeneNet library
library("GeneNet")
 
# ecoli data 
data(ecoli)

# estimate partial correlation matrix 
inferred.pcor &lt;- ggm.estimate.pcor(ecoli)

# p-values, q-values and posterior probabilities for each potential edge 
#
test.results &lt;- network.test.edges(inferred.pcor)

# show best 20 edges (strongest correlation)
test.results[1:20,]

# extract network containing edges with prob &gt; 0.9 (i.e. local fdr &lt; 0.1)
net &lt;- extract.network(test.results, cutoff.ggm=0.9)
net

# how many are significant based on FDR cutoff Q=0.05 ?
num.significant.1 &lt;- sum(test.results$qval &lt;= 0.05)
test.results[1:num.significant.1,]

# how many are significant based on "local fdr" cutoff (prob &gt; 0.9) ?
num.significant.2 &lt;- sum(test.results$prob &gt; 0.9)
test.results[test.results$prob &gt; 0.9,]

# parameters of the mixture distribution used to compute p-values etc.
c &lt;- fdrtool(sm2vec(inferred.pcor), statistic="correlation")
c$param

</code></pre>


</div>