<div class="container">

<table style="width: 100%;"><tr>
<td>plotEFF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>plotEFF</h2>

<h3>Description</h3>

<p>plotEFF
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotEFF(
  siteEFF,
  gff = gff,
  Chr = Chr,
  start = start,
  end = end,
  showType = c("five_prime_UTR", "CDS", "three_prime_UTR"),
  CDS.height = CDS.height,
  cex = 0.1,
  col = c("red", "yellow"),
  pch = 20,
  main = main,
  legend.cex = 0.8,
  gene.legend = TRUE,
  markMutants = TRUE,
  mutants.col = 1,
  mutants.type = 1,
  y = c("pvalue", "effect"),
  ylab = ylab,
  legendtitle = legendtitle,
  par.restore = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>siteEFF</code></td>
<td>
<p>matrix, column name are pheno names and row name are site position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gff</code></td>
<td>
<p>gff annotation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Chr</code></td>
<td>
<p>the chromosome name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>start position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>end position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showType</code></td>
<td>
<p>character vector, eg.: "CDS", "five_prime_UTR",
"three_prime_UTR"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CDS.height</code></td>
<td>
<p>numeric indicate the height of CDS in gene model,
range: <code style="white-space: pre;">⁠[0,1]⁠</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>a numeric control the size of point</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>vector specified the color bar</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pch</code></td>
<td>
<p>vector controls points type, see
<code>par()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>main title</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend.cex</code></td>
<td>
<p>a numeric control the legend size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.legend</code></td>
<td>
<p>whether add legend for gene model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markMutants</code></td>
<td>
<p>whether mark mutants on gene model, default as <code>TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutants.col</code></td>
<td>
<p>color of lines which mark mutants</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutants.type</code></td>
<td>
<p>a vector of line types</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y, ylab, legendtitle</code></td>
<td>
<p><em>y:</em> indicate either pvalue or effect should be used as y axix,
<strong>ylab,legendtitle:</strong>,character, if missing, the value will be decide by y.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>par.restore</code></td>
<td>
<p>default as <code>TRUE</code>, wether restore the origin par after ploted EFF.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data("geneHapR_test")

# calculate site functional effect
# siteEFF &lt;- siteEFF(hapResult, pheno, names(pheno))
# plotEFF(siteEFF, gff = gff, Chr = "scaffold_1")

</code></pre>


</div>