<div class="container">

<table style="width: 100%;"><tr>
<td>filterLargeVCF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pre-process of Large VCF File(s)</h2>

<h3>Description</h3>

<p>Filter/extract one or multiple gene(s)/range(s) from a large
<code style="white-space: pre;">⁠*.vcf/*.vcf.gz⁠</code> file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">filterLargeVCF(VCFin = VCFin, VCFout = VCFout,
                Chr = Chr,
                POS = NULL,
                start = start,
                end = end,
                override = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>VCFin</code></td>
<td>
<p>Path of input <code style="white-space: pre;">⁠*.vcf/*.vcf.gz⁠</code> file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>VCFout</code></td>
<td>
<p>Path(s) of output <code style="white-space: pre;">⁠*.vcf/*.vcf.gz⁠</code> file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Chr</code></td>
<td>
<p>a single CHROM name or CHROM names vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>POS, start, end</code></td>
<td>
<p>provide the range should be extract from orignal vcf.
<code>POS</code>: a vector consist with start and end position or a list
with length equal to <code>Chr</code>, eg.: <code>list(c(1,200), c(300,500), c(300,400))</code>
indicates 3 ranges (1~200, 300~500 and 300~400).
if <code>POS</code> is <code>NULL</code>, <code>start</code> and <code>end</code> are needed, eg.:
<code>start = c(1, 30)</code> and <code>end = c(200, 150)</code> indicates 2 ranges
(1~200 and 30~150)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>override</code></td>
<td>
<p>whether override existed file or not, default as <code>TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This package import VCF files with 'vcfR' which is more efficient to
import/manipulate VCF files in 'R'. However, import a large VCF file is time and
memory consuming. It's suggested that filter/extract variants in target
range with <code>filterLargeVCF()</code>.
</p>
<p>When filter/extract multi genes/ranges, the parameter of <code>Chr</code> and <code>POS</code>
must have equal length. Results will save to a single file if the user
provide a single file path or save to multiple VCF file(s) when a equal length
vector consist with file paths is provided.
</p>
<p>However, if you have hundreds gene/ranges need to extract from very
large VCF file(s), it's prefer to process with other linux tools in a script
on server, such as: 'vcftools' and 'bcftools'.
</p>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class="language-R">
 # The filteration of small vcf should be done with `filter_vcf()`.
 # however, here, we use a mini vcf instead just for example and test.

 vcfPath &lt;- system.file("extdata", "var.vcf.gz", package = "geneHapR")

 oldDir &lt;- getwd()
 temp_dir &lt;- tempdir()
 if(! dir.exists(temp_dir))
   dir.create(temp_dir)
 setwd(temp_dir)
 # extract a single gene/range from large vcf
 filterLargeVCF(VCFin = vcfPath, VCFout = "filtered.vcf.gz",
                Chr = "scaffold_1", POS = c(4300,5000), override = TRUE)

 # extract multi genes/ranges from large vcf
 filterLargeVCF(VCFin = vcfPath,
                VCFout = c("filtered1.vcf.gz",
                           "filtered2.vcf.gz",
                           "filtered3.vcf.gz"),
                Chr = rep("scaffold_1", 3),
                POS = list(c(4300, 5000),
                           c(5000, 6000),
                           c(5000, 7000)),
                override = TRUE)

setwd(oldDir)

</code></pre>


</div>