<div class="container">

<table style="width: 100%;"><tr>
<td>hap2hmp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert hapResult object to hapmap (hmp) format,
for interact with other packages</h2>

<h3>Description</h3>

<p>Convert hapResult object to hapmap (hmp) format,
for interact with other packages
</p>


<h3>Usage</h3>

<pre><code class="language-R">hap2hmp(hap)

hmp2hap(hmp, hapPrefix = "H", hetero_remove = TRUE, na_drop = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hap</code></td>
<td>
<p>object of "hapResult" class</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hmp</code></td>
<td>
<p>object of "data.frame" class in hapmap format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hapPrefix</code></td>
<td>
<p>prefix of haplotype names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hetero_remove</code></td>
<td>
<p>whether remove accessions contains hyb-sites, Character not A T C G</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na_drop</code></td>
<td>
<p>whether drop accessions contains missing data ("N", "NA", ".")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Arguments passed on to <code>table2hap</code>
</p>

<dl>
<dt><code>x</code></dt>
<dd>
<p>a data.frame contains variants information.
The first file column are fix as Chrome name, position, reference nuclieotide,
alter nuclieotide and INFO. Accession genotype should be in followed columns.
"-" will be treated as Indel. "." and "N" will be treated as missing data.
Heterozygotes should be "A/T", "AAA/A"</p>
</dd>
<dt><code>pad</code></dt>
<dd>
<p>The number length in haplotype names should be extend to.</p>
</dd>
</dl>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a data.frame in hapmap format.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data("geneHapR_test")
hmp &lt;- hap2hmp(hapResult)
hap &lt;- hmp2hap(hmp)

</code></pre>


</div>