<div class="container">

<table style="width: 100%;"><tr>
<td>table2hap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>table2hap</h2>

<h3>Description</h3>

<p>convert variants stored in table format into hapResult
</p>


<h3>Usage</h3>

<pre><code class="language-R">table2hap(x, hapPrefix = "H", pad = 3, hetero_remove = TRUE, na_drop = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a data.frame contains variants information.
The first file column are fix as Chrome name, position, reference nuclieotide,
alter nuclieotide and INFO. Accession genotype should be in followed columns.
"-" will be treated as Indel. "." and "N" will be treated as missing data.
Heterozygotes should be "A/T", "AAA/A"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hapPrefix</code></td>
<td>
<p>prefix of haplotype names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pad</code></td>
<td>
<p>The number length in haplotype names should be extend to.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hetero_remove</code></td>
<td>
<p>whether remove accessions contains hyb-sites, Character not A T C G</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na_drop</code></td>
<td>
<p>whether drop accessions contains missing data ("N", "NA", ".")</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>object of hapSummary class
</p>


<h3>Examples</h3>

<pre><code class="language-R">
   data("geneHapR_test")
   hapResult &lt;- table2hap(gt.geno, hapPrefix = "H",
                          hetero_remove = TRUE,
                          na_drop = TRUE)

</code></pre>


</div>