<div class="container">

<table style="width: 100%;"><tr>
<td>plotHapTable</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>plotHapTable</h2>

<h3>Description</h3>

<p>display hap result as a table-like figure
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotHapTable(hapSummary,
             hapPrefix = "H",
             title = "",
             geneName = geneName,
             INFO_tag = NULL,
             tag_split = tag_split,
             tag_field = tag_field,
             tag_name = tag_name,
             displayIndelSize = 0, angle = c(0,45,90),
             replaceMultiAllele = TRUE,
             ALLELE.color = "grey90")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hapSummary</code></td>
<td>
<p>object of hapSummary class</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hapPrefix</code></td>
<td>
<p>prefix of haplotype names. Default as "H"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>the main title of the final figure</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneName</code></td>
<td>
<p>character, will be used for filter INFO filed of ANN</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>INFO_tag</code></td>
<td>
<p>The annotations in the INFO field are represented as tag-value
pairs, where the tag and value are separated by an equal sign, ie "=", and
pairs are separated by colons, ie ";".
For more information please see details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tag_split</code></td>
<td>
<p>usually, the value of tag-value contains one information.
However, if a tag contains more than one fields, eg "ANN", then <code>tag_split</code> is
needed. When <code>INFO_tag</code> was set as "ANN" or "SNPEFF", <code>tag_split</code> will be set
as "|" by default, see details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tag_field</code></td>
<td>
<p>integer, if a tag-value contains more than one fields,
user need to specified which field should be display. If <code>tag_field</code> set as
0, the whole contents will be displayed. Default as 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tag_name</code></td>
<td>
<p>tag name is displayed in Hap figure. If <code>tag_name</code> is
missing, will take the value of <code>INFO_tag</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>displayIndelSize</code></td>
<td>
<p>display indels with max size of <code>displayIndelSize</code>,
If set as 0, all indels will convert into "i*" of which "i" represents "indel".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>angle</code></td>
<td>
<p>the angle of coordinates, should be one of 0, 45 and 90</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replaceMultiAllele</code></td>
<td>
<p>whether to replace MultiAllele with "T*",
default as <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ALLELE.color</code></td>
<td>
<p>the color of ALLELE row, default as "grey90"</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In <strong>VCF</strong> files, the INFO field are represented as tag-value
pairs, where the tag and value are separated by an equal sign, ie "=", and
pairs are separated by colons, ie ";".
</p>
<p>If hapSummarys were generated from sequences, INFO row is null.
If hapSummarys were generated from VCF, INFO was take from the INFO column
in the source VCF file.
Some tag-values may contains more than one value separated by
"|", eg.: "ANN" or "snpEFF" added by 'snpeff' or other software. For those
fields we need specified value of <code>tag_field = "ANN"</code> and <code>tag_split = "[\|]"</code>,
it's suggest specified the value of <code>tag_name</code> for display in figure.
</p>
<p>'snpeff', a toolbox for genetic variant annotation and functional effect
prediction, will add annotations to INFO filed in VCF file under a tag
named as "ANN". The annotations contains several fields separated by "|".
eg.:
</p>

<ol>
<li>
<p> Allele
</p>
</li>
<li>
<p> Annotation
</p>
</li>
<li>
<p> Annotation_Impact
</p>
</li>
<li>
<p> Gene_Name
</p>
</li>
<li>
<p> Gene_ID
</p>
</li>
<li>
<p> Feature_Type
</p>
</li>
<li>
<p> Feature_ID
</p>
</li>
<li>
<p> Transcript_BioType
</p>
</li>
<li>
<p> Rank
</p>
</li>
<li>
<p> HGVS.c
</p>
</li>
<li>
<p> HGVS.p
</p>
</li>
<li>
<p> cDNA.pos/cDNA.length
... ...
</p>
</li>
</ol>
<p>However, the INFO in hapResults may missing annotations that we need.
In this case, we can custom INFOs in hapSummarys with <code>addINFO()</code>.
Once the needed annotations were included in hap results, we can display
them with <code>plotHapTable()</code> by specify the value of <code>INFO_tag</code>.
</p>


<h3>Value</h3>

<p>ggplot2 object
</p>


<h3>See Also</h3>

<p><code>addINFO()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("geneHapR_test")
plotHapTable(hapResult)
</code></pre>


</div>