<div class="container">

<table style="width: 100%;"><tr>
<td>read_grm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read GCTA GRM and related plink2 binary files</h2>

<h3>Description</h3>

<p>This function reads a GCTA Genetic Relatedness Matrix (GRM, i.e. kinship) set of files in their binary format, returning the kinship matrix and, if available, the corresponding matrix of pair sample sizes (non-trivial under missingness) and individuals table.
Setting some options allows reading plink2 binary kinship formats such as "king" (see examples).
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_grm(
  name,
  n_ind = NA,
  verbose = TRUE,
  ext = "grm",
  shape = c("triangle", "strict_triangle", "square"),
  size_bytes = 4,
  comment = "#"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>The base name of the input files.
Files with that base, plus shared extension (default "grm", see <code>ext</code> below), plus extensions <code>.bin</code>, <code>.N.bin</code>, and <code>.id</code> are read if they exist.
Only <code style="white-space: pre;">⁠.&lt;ext&gt;.bin⁠</code> is absolutely required; <code style="white-space: pre;">⁠.&lt;ext&gt;.id⁠</code> can be substituted by the number of individuals (see below); <code style="white-space: pre;">⁠.&lt;ext&gt;.N.bin⁠</code> is entirely optional.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_ind</code></td>
<td>
<p>The number of individuals, required if the file with the extension <code style="white-space: pre;">⁠.&lt;ext&gt;.id⁠</code> is missing.
If the file with the <code style="white-space: pre;">⁠.&lt;ext&gt;.id⁠</code> extension is present, then this <code>n_ind</code> is ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If <code>TRUE</code> (default), function reports the path of the files being loaded.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ext</code></td>
<td>
<p>Shared extension for all three inputs (see <code>name</code> above; default "grm").
Another useful value is "king" for KING-robust estimates produced by plink2.
If <code>NA</code>, no extension is added.
If given <code>ext</code> is also present at the end of <code>name</code>, then it is not added again.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shape</code></td>
<td>
<p>The shape of the information to read (may be abbreviated).
Default "triangle" assumes there are <code>n*(n+1)/2</code> values to read corresponding to the upper triangle including the diagonal (required for GCTA GRM).
"strict_triangle" assumes there are <code>n*(n-1)/2</code> values to read corresponding to the upper triangle <em>excluding</em> the diagonal (best for plink2 KING-robust).
Lastly, "square" assumes there are <code>n*n</code> values to read corresponding to the entire square matrix, ignoring symmetry.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size_bytes</code></td>
<td>
<p>The number of bytes per number encoded.
Default 4 corresponds to GCTA GRM and plink2 "bin4", whereas plink2 "bin" requires a value of 8.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comment</code></td>
<td>
<p>Character to start comments in <code style="white-space: pre;">⁠&lt;ext&gt;.id⁠</code> file only.
Default "#" helps plink2 <code>.id</code> files (which have a header that starts with "#", which is therefore ignored) be read just like plink1 and GCTA files (which do not have a header).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with named elements:
</p>

<ul>
<li> <p><code>kinship</code>: The symmetric <code>n</code>-times-<code>n</code> kinship matrix (GRM).  Has IDs as row and column names if the file with extension <code style="white-space: pre;">⁠.&lt;ext&gt;.id⁠</code> exists.  If <code>shape='strict_triangle'</code>, diagonal will have missing values.
</p>
</li>
<li> <p><code>M</code>: The symmetric <code>n</code>-times-<code>n</code> matrix of pair sample sizes (number of non-missing loci pairs), if the file with extension <code style="white-space: pre;">⁠.&lt;ext&gt;.N.bin⁠</code> exists.  Has IDs as row and column names if the file with extension <code style="white-space: pre;">⁠.&lt;ext&gt;.id⁠</code> was available.  If <code>shape='strict_triangle'</code>, diagonal will have missing values.
</p>
</li>
<li> <p><code>fam</code>: A tibble with two columns: <code>fam</code> and <code>id</code>, same as in Plink FAM files.  Returned if the file with extension <code style="white-space: pre;">⁠.&lt;ext&gt;.id⁠</code> exists.
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>write_grm()</code>
</p>
<p>Greatly adapted from sample code from GCTA:
<a href="https://cnsgenomics.com/software/gcta/#MakingaGRM">https://cnsgenomics.com/software/gcta/#MakingaGRM</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># to read "data.grm.bin" and etc, run like this:
# obj &lt;- read_grm("data")
# obj$kinship # the kinship matrix
# obj$M       # the pair sample sizes matrix
# obj$fam     # the fam and ID tibble

# The following example is more awkward
# because package sample data has to be specified in this weird way:

# read an existing set of GRM files
file &lt;- system.file("extdata", 'sample.grm.bin', package = "genio", mustWork = TRUE)
file &lt;- sub('\\.grm\\.bin$', '', file) # remove extension from this path on purpose
obj &lt;- read_grm(file)
obj$kinship # the kinship matrix
obj$M       # the pair sample sizes matrix
obj$fam     # the fam and ID tibble

# Read sample plink2 KING-robust files (several variants).
# Read both base.king.bin and base.king.id files.
# All generated with "plink2 &lt;input&gt; --make-king &lt;options&gt; --out base"
# (replace "base" with actual base name) with these options:
# #1) "triangle bin"
# data &lt;- read_grm( 'base', ext = 'king', shape = 'strict', size_bytes = 8 )
# #2) "triangle bin4"
# data &lt;- read_grm( 'base', ext = 'king', shape = 'strict' )
# #3) "square bin"
# data &lt;- read_grm( 'base', ext = 'king', shape = 'square', size_bytes = 8 )
# #4) "square bin4"
# data &lt;- read_grm( 'base', ext = 'king', shape = 'square' )

</code></pre>


</div>