<div class="container">

<table style="width: 100%;"><tr>
<td>write_bed</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Write a genotype matrix into Plink BED format</h2>

<h3>Description</h3>

<p>This function accepts a standard R matrix containing genotypes (values in <code>c( 0, 1, 2, NA )</code>) and writes it into a Plink-formatted BED (binary) file.
Each genotype per locus (<code>m</code> loci) and individual (<code>n</code> total) counts the number of alternative alleles or <code>NA</code> for missing data.
No *.fam or *.bim files are created by this basic function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">write_bed(file, X, verbose = TRUE, append = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>Output file path.  .bed extension may be omitted (will be added automatically if it is missing).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>The <code>m</code>-by-<code>n</code> genotype matrix.
Row and column names, if present, are ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If <code>TRUE</code> (default), function reports the path of the file being written (after autocompleting the extension).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>append</code></td>
<td>
<p>If <code>TRUE</code>, appends variants onto the file. (Default is <code>FALSE</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Genotypes with values outside of [0, 2] cause an error, in which case the partial output is deleted.
However, beware that decimals get truncated internally, so values that truncate to 0, 1, or 2 will not raise errors.
The BED format does not accept fractional dosages, so such data will not be written as expected.
</p>


<h3>Value</h3>

<p>Nothing
</p>


<h3>See Also</h3>

<p><code>write_plink()</code> for writing a set of BED/BIM/FAM files.
</p>
<p>Plink BED format reference:
<a href="https://www.cog-genomics.org/plink/1.9/formats#bed">https://www.cog-genomics.org/plink/1.9/formats#bed</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># to write an existing matrix `X` into file "data.bed", run like this:
# write_bed("data", X)
# this also works
# write_bed("data.bed", X)

# The following example is more detailed but also more awkward
# because (only for these examples) the package must create the file in a *temporary* location

file_out &lt;- tempfile('delete-me-example', fileext = '.bed') # will also work without extension
# create 10 random genotypes
X &lt;- rbinom(10, 2, 0.5)
# replace 3 random genotypes with missing values
X[sample(10, 3)] &lt;- NA
# turn into 5x2 matrix
X &lt;- matrix(X, nrow = 5, ncol = 2)
# write this data to file in BED format
# (only *.bed gets created, no *.fam or *.bim in this call)
write_bed(file_out, X)
# delete output when done
file.remove(file_out)

</code></pre>


</div>