<div class="container">

<table style="width: 100%;"><tr>
<td>read_plink</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read genotype and sample data in a Plink BED/BIM/FAM file set.</h2>

<h3>Description</h3>

<p>This function reads a genotype matrix (<code>X</code>, encoded as reference allele dosages) and its associated locus (<code>bim</code>) and individual (<code>fam</code>) data tables in the three Plink files in BED, BIM, and FAM formats, respectively.
All inputs must exist or an error is thrown.
This function is a wrapper around the more basic functions
<code>read_bed()</code>,
<code>read_bim()</code>,
<code>read_fam()</code>,
which simplifies data parsing and additionally better guarantees data integrity.
Below suppose there are <code>m</code> loci and <code>n</code> individuals.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_plink(file, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>Input file path, without extensions (each of .bed, .bim, .fam extensions will be added automatically as needed).
Alternatively, input file path may have .bed extension (but not .bim, .fam, or other extensions).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If <code>TRUE</code> (default), function reports the paths of the files being read (after autocompleting the extensions).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A named list with items in this order: <code>X</code> (genotype matrix, see description in return value of <code>read_bed()</code>), <code>bim</code> (tibble, see <code>read_bim()</code>), <code>fam</code> (tibble, see <code>read_fam()</code>).
<code>X</code> has row and column names corresponding to the <code>id</code> values of the <code>bim</code> and <code>fam</code> tibbles.
</p>


<h3>See Also</h3>

<p><code>read_bed()</code>,
<code>read_bim()</code>, and
<code>read_fam()</code>
for individual parsers of each input table, including a description of each object returned.
</p>
<p><code>geno_to_char()</code> for translating numerical genotypes into more human-readable character encodings.
</p>
<p>Plink BED/BIM/FAM format reference:
<a href="https://www.cog-genomics.org/plink/1.9/formats">https://www.cog-genomics.org/plink/1.9/formats</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># to read "data.bed" etc, run like this:
# obj &lt;- read_plink("data")
# this also works
# obj &lt;- read_plink("data.bed")
#
# you get a list with these three items:
# genotypes
# obj$X
# locus annotations
# obj$bim
# individual annotations
# obj$fam

# The following example is more awkward
# because package sample data has to be specified in this weird way:

# first get path to BED file
file &lt;- system.file("extdata", 'sample.bed', package = "genio", mustWork = TRUE)

# read genotypes and annotation tables
plink_data &lt;- read_plink(file)
# genotypes
plink_data$X
# locus annotations
plink_data$bim
# individual annotations
plink_data$fam

# the same works without .bed extension
file &lt;- sub('\\.bed$', '', file) # remove extension
# it works!
plink_data &lt;- read_plink(file)

</code></pre>


</div>