<div class="container">

<table style="width: 100%;"><tr>
<td>gen.simuProb</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gene dropping simulations - Probabilities</h2>

<h3>Description</h3>

<p>Returns the probabilities that specified probands inherit disease alleles from ancestors.</p>


<h3>Usage</h3>

<pre><code class="language-R">gen.simuProb(gen, pro, statePro, ancestors, stateAncestors, simulNo=5000, 
                    probRecomb=c(0,0), probSurvival=1.0)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gen</code></td>
<td>
<p>An object of class GLgen obtained with gen.genealogy, gen.lineages or gen.branching. Required.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pro</code></td>
<td>
<p>Vector of proband id numbers to be included. Required.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>statePro</code></td>
<td>
<p>Required vector indicating, for each proband in pro, the probability to be calculated:
</p>
<p>0 = no disease allele is transmitted
</p>
<p>1 = 1 disease allele is transmitted
</p>
<p>2 = 2 disease alleles are transmitted
</p>
<p>3 = 1 or 2 disease alleles are transmitted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ancestors</code></td>
<td>
<p>Vector of ancestors id numbers to be included. Required.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stateAncestors</code></td>
<td>
<p>Required vector indicating, for each ancestor in ancestors, the genotype state:
</p>
<p>0 = no disease allele present
</p>
<p>1 = 1 disease allele present
</p>
<p>2 = 2 disease alleles present</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simulNo</code></td>
<td>
<p>Number of simulations to perform. Default is 5000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>probRecomb</code></td>
<td>
<p>Recombination probabilities for males and females. Default is no recombination.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>probSurvival</code></td>
<td>
<p>Survival probability for homozygotes. Default is 1.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing the following:
</p>
<p>the joint probability of specified statePro for all probands,
</p>
<p>the probability of specified statePro for each proband,
</p>
<p>the probability that, 0, 1, ..., and all probands inherit the specified number of disease alleles.
</p>


<h3>See Also</h3>

<p><code>gen.genealogy</code>
<code>gen.simuSample</code>
<code>gen.simuSet</code>
<code>gen.simuSampleFreq</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(geneaJi) 
genJi&lt;-gen.genealogy(geneaJi) 
# Probability that subjects 1 and 29 get 1 and 2 alleles from ancestors 20 and 25,
# that have themselves 2 and 1.
gen.simuProb(genJi, pro=c(1,29), statePro=c(1,2), ancestors=c(20,25), stateAncestors=c(2,1),
             simulNo=10000)
# Probability that subjects 1 and 29 get 1 segment from ancestors 25,
# knowing the segment has a male
# recombination rate of 0.02 and a female recombination rate of 0.04.
gen.simuProb(genJi, pro=c(1,29), statePro=c(1,1), ancestors=c(25), stateAncestors=c(1),
             simulNo=10000, probRecomb = c(0.02, 0.04))
# Probability that subjects 1 and 29 get 1 and 2 alleles from ancestors 20 and 25,
# that have themselves 2 and 1 and knowing that homozygous people have a survival rate
# of 0.50.
gen.simuProb(genJi, pro=c(1,29), statePro=c(1,2), ancestors=c(20,25), stateAncestors=c(2,1),
             simulNo=10000, probSurvival=0.5)

</code></pre>


</div>