<div class="container">

<table style="width: 100%;"><tr>
<td>importVCF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>importVCF</h2>

<h3>Description</h3>

<p>Import a VCF function
</p>


<h3>Usage</h3>

<pre><code class="language-R">importVCF(file, na.seq = "./.")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>The file name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.seq</code></td>
<td>
<p>The missing value definition</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function imports a VCF file.
</p>
<p>In case the logicl flag 'phased' is set to TRUE then the genotypes are expected to be in the format 0|0, otherwise they are expected
to be like 0/1 .
</p>
<p>The example file was downloaded from here:
</p>
<p>ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/pilot_data/release/2010_07/exon/snps/
</p>


<h3>Value</h3>

<p>A vcf object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class="language-R">
 # Define here the location on HDD for the example file
   fpath &lt;- system.file("extdata","example.vcf", package="GenomicTools.fileHandler")
 # Import the example vcf file  
   importVCF(fpath)

</code></pre>


</div>