<div class="container">

<table style="width: 100%;"><tr>
<td>read2_paths</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Read to SNPs to sets; Map SNPs to gene-sets/pathways
</h2>

<h3>Description</h3>

<p>Reads the sets/pathways, map the SNPs and genes to the gene-sets/pathways
read2_paths uses the "genome_order" output(ordered_alldata, gs_locs) to assign genomic location indexes
to each element in the gene-set. The permutation method must be defined (i.e. level = "snp" OR level = "gene").
</p>


<h3>Usage</h3>

<pre><code class="language-R">read2_paths(ordered_alldata="",gs_locs="",sets_from="workspace",
sets_prefix="RHSA",level="snp",envir="")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ordered_alldata</code></td>
<td>

<p>Ordered data according to the SNPs genomic location. Traits start at column 7<br>
Return variable from:<br>
genome_results &lt;-genome_order(all_data=all_data)<br>
ordered_alldata &lt;- genome_results$ordered_alldata<br></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gs_locs</code></td>
<td>

<p>Gene annotation,indexes and number of observations<br>
Return variable from genome_order():<br>
genome_results &lt;-genome_order(all_data=all_data)<br>
gs_locs &lt;- genome_results$gs_locs<br></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sets_from</code></td>
<td>

<p>Location of the gene-sets. Default set to "workspace" <br>
sets_from="workspace" OR sets_from="directory"<br>
"directory", only will search for information in the working directory.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sets_prefix</code></td>
<td>

<p>Prefix of the gene-set variables or files.<br>
Default set to sets_prefix= "RHSA" 
e.g. Variables "RHSA164843","RHSA446343","RHSA8876384"<br>
each variable/file contains the list of gene identifiers part of that pathway
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>level</code></td>
<td>

<p>The level at which the permutations will be performed. Assigns the indexes according to snps or genes<br>
Default value "snp" 
level values = "snp" OR "gene"
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>envir</code></td>
<td>

<p>R environment where pathway data is stored. e.g(envir=.GlobalEnv, envir=gper.env)
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pathways</code></td>
<td>

<p>Pathway Id, Description,Number of Genes in the pathway, Number of genes found in the dataset,
Number of SNPs found in the dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>per_ors</code></td>
<td>

<p>A list of identifiers mapped to each pathway
</p>
</td>
</tr>
</table>
<h3>Format</h3>

<pre>
Input: Ordered_alldata
name     Chromosome  Location GENE_ID   Symbol Orientation Trait1 Trait2
rs1001567          1 9194614     &lt;NA&gt;     &lt;NA&gt;        &lt;NA&gt; 0.96 0.89
rs1000313          1 15405489   23254 KIAA1026           + 0.93 0.57
rs1002365          1 19797248    &lt;NA&gt;     &lt;NA&gt;        &lt;NA&gt; 0.68 0.58
rs1002706          1 25051153    &lt;NA&gt;     &lt;NA&gt;        &lt;NA&gt; 0.71 0.02
rs1002487          1 26865971    6195  RPS6KA1           + 0.98 0.78

Input:gs_locs
      Symbol   Chromosome Location    Gene_ID Start_Indx Observations
 [1,] "ACYP2"  "2"        "54399633"  "98"    "35"       "1"         
 [2,] "AMPD3"  "11"       "10514707"  "272"   "898"      "1"         
 [3,] "ANK2"   "4"        "113830885" "287"   "479"      "4"
 
Input:pathway example
RHSA8964572
[1]   1149 128486 161247  29923 345275  63924
          
Output:pathways
     ID            GenesInPath GenesFound SNPsInPath
"RHSA109582"  "681"       "8"        "11"      
"RHSA1474244" "418"       "7"        "10"      
"RHSA164843"  "11"        "0"        "0"       
"RHSA446343"  "4"         "1"        "1"       
"RHSA8876384" "32"        "1"        "1"       
"RHSA8964572" "6"         "1"        "1"       

</pre>


<h3>See Also</h3>

<p><code>genes_permutation</code>
<code>snps_permutation</code>
<code>genome_order</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## DEMO - SNP Level data stored in workspace #######################
# library(genomicper)
data(demo,SNPsAnnotation)
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)
genome_results &lt;-genome_order(all_data=all_data)
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs
	
# Create new environment to save variables (e.g. pathways, permutations):
gper.env &lt;- new.env()

data(RHSA164843,RHSA446343,RHSA8876384,RHSA8964572,RHSA109582,RHSA1474244,envir=gper.env)

paths_res &lt;- read2_paths(ordered_alldata=ordered_alldata,
gs_locs=gs_locs,sets_from="workspace",sets_prefix="RHSA",
level="snp",envir=gper.env)		
	pers_ids &lt;- paths_res$per_ors
	pathways&lt;- paths_res$pathways
####################################################################
</code></pre>


</div>