<div class="container">

<table style="width: 100%;"><tr>
<td>barto</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Comparison of 4 Bartonella genomes</h2>

<h3>Description</h3>

<p>Comparison of 4 Bartonella genomes by BLAST. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(barto)
</code></pre>


<h3>Format</h3>

<p><code>barto</code>, a list of three dataframes, representing the four genomes
and their pairwise comparisons:
</p>

<ul>
<li>
<p><code>dna_segs</code>which is a list of 4 <code>dna_seg</code>
objects, containing all the protein genes for each genome. Obtained
by reading ptt files downloaded from NCBI with
<code>read_dna_seg_from_ptt</code>.
</p>
</li>
<li>
<p><code>comparisons</code>which is a list of 3 <code>comparison</code>
objects, obtained by doing genome-to-genome (fasta files) BLASTS,
and then reading the resulting tab files with
<code>read_comparison_from_blast</code>.
</p>
</li>
<li>
<p><code>rnt_segs</code>which is a list of 4 <code>dna_seg</code> objects,
containing all the RNA genes of the four genomes. Obtained by
reading rnt files downloaded from NCBI with
<code>read_dna_seg_from_ptt</code>.
</p>
</li>
</ul>
<p>A bash script to obtain the same file as in the dataset is
available in the <code>extdata</code> folder of the package. Find its
location by running 
<code>system.file('extdata/barto.sh', package = 'genoPlotR')</code>.
</p>


<h3>References</h3>

<p>BLAST: http://www.ncbi.nlm.nih.gov/blast/
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(barto)
plot_gene_map(barto$rnt_segs, barto$comparisons, gene_type="blocks")
</code></pre>


</div>