<div class="container">

<table style="width: 100%;"><tr>
<td>genoPlotR-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Plot Publication-Grade Gene and Genome Maps
</h2>

<h3>Description</h3>

<p>Draws gene or genome maps and comparisons between these, in a 
   publication-grade manner. Starting from simple, common files, it will 
   draw postscript or PDF files that can be sent as such to journals.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> genoPlotR</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Title: </td>
<td style="text-align: left;"> Plot Publication-Grade Gene and Genome Maps</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 0.8.11</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2021-01-05</td>
</tr>
<tr>
<td style="text-align: left;">
Author: </td>
<td style="text-align: left;"> Lionel Guy &lt;lionel.guy@imbim.uu.se&gt;</td>
</tr>
<tr>
<td style="text-align: left;">
URL: </td>
<td style="text-align: left;"> http://genoplotr.r-forge.r-project.org/</td>
</tr>
<tr>
<td style="text-align: left;">
Depends: </td>
<td style="text-align: left;"> R (&gt;= 2.10.0), ade4, grid</td>
</tr>
<tr>
<td style="text-align: left;">
Imports: </td>
<td style="text-align: left;"> methods</td>
</tr>
<tr>
<td style="text-align: left;">
Maintainer: </td>
<td style="text-align: left;"> Lionel Guy &lt;lionel.guy@imbim.uu.se&gt;</td>
</tr>
<tr>
<td style="text-align: left;">
Description: </td>
<td style="text-align: left;"> Draws gene or genome maps and comparisons between these, in a 
   publication-grade manner. Starting from simple, common files, it will 
   draw postscript or PDF files that can be sent as such to journals.</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
<td style="text-align: left;">
LazyLoad: </td>
<td style="text-align: left;"> yes</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Index of help topics:
</p>
<pre>
annotation              Annotation class and class functions
apply_color_scheme      Apply a color scheme
artemisColors           Artemis Colors
auto_annotate           Auto-annotate dna_segs
barto                   Comparison of 4 Bartonella genomes
c.dna_seg               Concatenate dna_seg objects
chrY_subseg             Comparisons of subsegments of the Y chromosome
                        in human and chimp
comparison              Comparison class and class functions
dna_seg                 DNA segment (dna_seg) class and class functions
gene_types              Gene types
genoPlotR-package       Plot Publication-Grade Gene and Genome Maps
human_nt                Human-readable nucleotide scale
mauve_bbone             Mauve backbone of 4 Bartonella genomes
middle                  Middles of a dna_seg
plot_gene_map           Plot gene and genome maps
range.dna_seg           Range calculation
read_functions          Reading functions
reverse                 Reverse objects
seg_plot                seg_plot class and class functions
three_genes             Three genes data set
trim                    Trimming data frames or more complex objects
                        with &gt;= 2 numeric columns
</pre>
<p>The only plotting function is <code>plot_gene_map</code>, which
produces <code>link[grid]{grid}</code> graphics. Data is composed mainly of
DNA segments (<code>dna_seg</code>) objects, which represent
collections of genes or segments of genomes, and of
<code>comparison</code> objects, which are the pairwise comparisons
between the <code>dna_seg</code>s. Data can be read from files (see
<code>read_functions</code>) or from R objects like
<code>data.frame</code>s or <code>list</code>s, with <code>dna_seg</code> and
<code>comparison</code> conversion functions. 
</p>


<h3>Author(s)</h3>

<p>Lionel Guy &lt;lionel.guy@imbim.uu.se&gt;
</p>
<p>Maintainer: Lionel Guy &lt;lionel.guy@imbim.uu.se&gt;
</p>


<h3>References</h3>

<p>Guy, L., Roat Kultima, J, and Andersson, S.G.E. (2010). genoPlotR:
comparative gene and genome visualization in R. Bioinformatics 26(18):2334-2335.
</p>


<h3>See Also</h3>

<p><code>plot_gene_map</code> for plotting. <code>dna_seg</code> and
<code>comparison</code> for the base objects and conversion functions.
<code>read_dna_seg_from_tab</code>, <code>read_dna_seg_from_ptt</code>, 
<code>read_comparison_from_tab</code> and
<code>read_comparison_from_blast</code> to read from files.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## simple example
## dna segments
## data.frame with several genes
names1 &lt;- c("feat1", "feat2", "feat3")
starts1 &lt;- c(2, 1000, 1050)
ends1 &lt;- c(600, 800, 1345)
strands1 &lt;- c("-", -1, 1)
cols1 &lt;- c("blue", "grey", "red")
df1 &lt;- data.frame(name=names1, start=starts1, end=ends1,
                  strand=strands1, col=cols1)
dna_seg1 &lt;- dna_seg(df1)
is.dna_seg(dna_seg1)

## with only one gene, or two, and merging
gene2a &lt;- dna_seg(list(name="feat1", start=50, end=900, strand="-", col="blue"))
genes2b &lt;- dna_seg(data.frame(name=c("feat2", "feat3"), start=c(800, 1200),
                              end=c(1100, 1322), strand=c("+", 1),
                              col=c("grey", "red")))
dna_seg2 &lt;- c.dna_seg(gene2a, genes2b)
is.dna_seg(dna_seg2)

## reading from file
dna_seg3_file &lt;- system.file('extdata/dna_seg3.tab', package = 'genoPlotR')
dna_seg3 &lt;- read_dna_seg_from_tab(dna_seg3_file)
is.dna_seg(dna_seg3)

## comparison
## from a data.frame
comparison1 &lt;- as.comparison(data.frame(start1=starts1, end1=ends1,
                                        start2=dna_seg2$start,
                                        end2=dna_seg2$end))
is.comparison(comparison1)

## from a file
comparison2_file &lt;- system.file('extdata/comparison2.tab',
                                package = 'genoPlotR')
comparison2 &lt;- read_comparison_from_tab(comparison2_file,
                                        color_scheme="red_blue")
is.comparison(comparison1)

## plot
plot_gene_map(dna_segs=list(dna_seg1, dna_seg2, dna_seg3),
              comparisons=list(comparison1, comparison2)) 
</code></pre>


</div>