<div class="container">

<table style="width: 100%;"><tr>
<td>genes_permutation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Gene-level Permutations 
</h2>

<h3>Description</h3>

<p>Performs gene-level circular genomic permutations. In each permutation,the complete set of<br> 
SNP association p-values are permuted by rotation with respect to the SNPs' genomic locations.<br>
Once these 'simulated' p-values are assigned,the joint gene p-values are calculated using<br>
Fisher's combination test,and pathways' association tested using the hypergeometric test<br></p>


<h3>Usage</h3>

<pre><code class="language-R">genes_permutation(ordered_alldata = "", pers_ids = "", pathways = "", 
ntraits = "", nper = 100, threshold = 0.05, seed=10,saveto = "workspace", 
gs_locs="", envir = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ordered_alldata</code></td>
<td>

<p>Return variable from "genome_order". Ordered genome and trait p-values
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gs_locs</code></td>
<td>

<p>Return variable from "genome_order". SNP indexes
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pers_ids</code></td>
<td>

<p>Return variable "per_ors" from "read2_paths". Gene indexes
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pathways</code></td>
<td>

<p>Return variable "pathways" from "read2_paths"
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ntraits</code></td>
<td>

<p>Traits INDEX to be analysed. Index according to "ordered_alldata". <br>
Trait Columns index must start at 7. Example: ntraits=c(7:9), ntraits=7
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nper</code></td>
<td>

<p>Number of permutations.Example: nper=1000
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>

<p>Threshold to be set by the hypergeometric test. threshold=0.05
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>

<p>Set a number for random sampling
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saveto</code></td>
<td>

<p>Save permutation results to "workspace" OR "directory"
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>envir</code></td>
<td>

<p>R environment to save the data to when saveto is set to "workspace"
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns "Permus_trait" variables or files (permutation datasets). 
</p>


<h3>References</h3>

<p>Imports phyper (from stats)
</p>


<h3>See Also</h3>

<p><code>snps_permutation</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#load data
data(demo,SNPsAnnotation)
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)

# Prepare Genome
genome_results &lt;-genome_order(all_data=all_data)
	# Results from genome_order
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs

# Create new environment to save data:
gper.env &lt;- new.env()

# Get pathways
data(RHSA164843,RHSA446343,RHSA8876384,RHSA8964572,RHSA109582,RHSA1474244,envir=gper.env)

# Map Genes to pathways
paths_res &lt;- read2_paths(ordered_alldata=ordered_alldata,gs_locs=gs_locs,
sets_from="workspace",sets_prefix="RHSA",level="gene",envir=gper.env)
pers_ids &lt;- paths_res$per_ors
pathways&lt;- paths_res$pathways

# Perform Permutations:
genes_permutation(ordered_alldata=ordered_alldata,
pers_ids=pers_ids,pathways=pathways,ntraits=c(7:9),
nper=10,threshold=0.05, saveto="workspace",
gs_locs=gs_locs,envir = gper.env)

# Results
results &lt;- get_results(res_pattern="Permus",level="gene",
from="workspace",threshold=0.05,envir= gper.env)
</code></pre>


</div>