<div class="container">

<table style="width: 100%;"><tr>
<td>separateByPopulations</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Separate Genotype Matrix by Populations</h2>

<h3>Description</h3>

<p>This function separates a genotype matrix into two data frames based on population assignments.
It's designed to work with the batches or windows processed by <code>process_vcf_in_batches</code> and <code>process_vcf_in_windows</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">separateByPopulations(
  sep_gt,
  pop1_names,
  pop2_names,
  ploidy = 2,
  rm_ref_alleles = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sep_gt</code></td>
<td>
<p>A genotype matrix similar to the <code style="white-space: pre;">⁠@sep_gt⁠</code> slot of a <code>vcfR</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop1_names</code></td>
<td>
<p>A character vector of individual names for the first population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop2_names</code></td>
<td>
<p>A character vector of individual names for the second population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rm_ref_alleles</code></td>
<td>
<p>Logical, whether variants that only have the reference allele
should be removed from the respective subpopulations data frame. (Default = TRUE)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing two data frames, one for each population.
</p>


</div>