<div class="container">

<table style="width: 100%;"><tr>
<td>genomicper-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Circular Genomic Permutations
</h2>

<h3>Description</h3>

<p>Description: Circular genomic permutation approach uses genome wide association studies (GWAS) results to establish the significance of pathway/gene-set associations whilst accounting for genomic structure. All single nucleotide polymorphisms (SNPs) in the GWAS are placed in a 'circular genome' according to their location. Then the complete set of SNP association p-values are permuted by rotation with respect to the SNPs' genomic locations. Two testing frameworks are available: permutations at the gene level, and permutations at the SNP level. The permutation at the gene level uses Fisher's combination test to calculate a single gene p-value, followed by the hypergeometric test. The SNP count methodology maps each SNP to pathways/gene-sets and calculates the proportion of SNPs for the real and the permutated datasets above a pre-defined threshold. Genomicper requires a matrix of GWAS association p-values and SNPs annotation to genes. Pathways can be obtained from within the package or can be provided by the user.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> genomicper</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 1.7</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2020-05-06</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Claudia P. Cabrera, Pau Navarro, Chris S. Haley <br>
Maintainer: Claudia Cabrera &lt;c.cabrera@qmul.ac.uk&gt;
</p>


<h3>References</h3>

<p>SNP-level Permutations:<br>
Genomicper: genome-wide association SNP-set analysis<br>
Claudia P. Cabrera*, Pau Navarro*, Jennifer E. Huffman, Alan F. Wright, Caroline Hayward,Harry Campbell, James F. Wilson, Igor Rudan, Nicholas D. Hastie, Veronique Vitart, Chris S. Haley*<br><br>
Gene-level Permutations:<br>
Uncovering Networks from Genome-Wide Association Studies via<br> 
Circular Genomic Permutation. G3: Genes|Genomes|Genetics 2, 1067-1075.<br>
Claudia P. Cabrera*, Pau Navarro*, Jennifer E. Huffman, Alan F. Wright, Caroline Hayward,Harry Campbell, James F. Wilson, Igor Rudan, Nicholas D. Hastie, Veronique Vitart, Chris S. Haley*
</p>


<h3>See Also</h3>

<p>Genomicper functions:	
1)  <code>read_pvals</code>,
2)  <code>genome_order</code>,
3)  <code>get_pathways</code>,
4)  <code>read2_paths</code>,
5A) <code>snps_permutation</code>,
5B) <code>genes_permutation</code>,
6)  <code>get_results</code>,
7)  <code>plot_results</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#############################################################################
#  Genomicper functions                                            ##########
# 1)  read_pvals(data_name="",snps_ann="")
# 2)  genome_order(all_data="")
# 3)  get_pathways(source="reactome",all_paths="",envir="")
# 4)  read2_paths(ordered_alldata="",gs_locs="",sets_from="",sets_prefix="RHSA",level="")
# 5A) snps_permutation(ordered_alldata="",pers_ids="",ntraits="",nper="",saveto="",
#		threshold="",gs_locs=gs_locs,envir = "")	
# 5B) genes_permutation(ordered_alldata="",pers_ids="",pathways="",
#		ntraits="",nper="",threshold="",saveto="",gs_locs=gs_locs,envir = "")
# 6)  get_results(res_pattern="Permus",level="snp",from="workspace",
#		threshold=0.05, envir = "")
# 7) plot_results(results = "", by = "", plot_all = TRUE, var = "", save_plot = TRUE, 
# 						plot_name = "", bf = FALSE, save_qq = TRUE)  
#############################################################################
############## DEMO: #######################################################

#### SNP-level  #############################################################
# SNPs annotation and Pathways provided by user
# all data stored at the WORKSPACE

### Load files for analysis
data(demo,SNPsAnnotation)

# Read &amp; format GWAS pvalues
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)

# Order data according to the genome
genome_results &lt;-genome_order(all_data=all_data)
	# Results from genome_order
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs

# Create new environment to save variables (e.g. pathways, permutations):
gper.env &lt;- new.env()

# Pathways can be downloaded using the function get_pathways()  
# Load example pathways into the new environment. 
data(RHSA164843,RHSA446343,RHSA8876384,RHSA8964572,RHSA109582,RHSA1474244,envir=gper.env)

# Map SNPs to pathways 
paths_res &lt;- read2_paths(ordered_alldata=ordered_alldata,
gs_locs=gs_locs,sets_from="workspace",sets_prefix="RHSA",
level="snp",envir=gper.env)
	# Results from read2_paths:		
	pers_ids &lt;- paths_res$per_ors
	pathways&lt;- paths_res$pathways

# Perform permutations:
snps_permutation(ordered_alldata=ordered_alldata,
pers_ids=pers_ids,ntraits=c(7:13),nper=10,saveto="workspace",
threshold=0.05,gs_locs=gs_locs,envir = gper.env)		  

# Get results						
results &lt;- get_results(res_pattern="Permus",level="snp",
from="workspace",threshold=0.05,envir = gper.env)

# Plot results
## Not run: 
#saves plots to working directory
qq &lt;- plot_results(results=results,by="set",plot_all=TRUE)
qq &lt;- plot_results(results=results,by="trait",
plot_all=FALSE,var="trait1")
# Displays interactive plot. Select a trait/set to plot and 
# set arguments save_plot=FALSE, plot_all = FALSE
# IMPORTANT: to EXIT interactive plot, RIGHT CLICK on the
# plot and STOP.
qq &lt;- plot_results(results=results,by="set",plot_all=FALSE,
var="RHSA109582",save_plot=FALSE)

## End(Not run)
# -- END OF DEMO 
###############################################
</code></pre>


</div>