<div class="container">

<table style="width: 100%;"><tr>
<td>RSCU</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Relative Synonymous Codon Usage
</h2>

<h3>Description</h3>

<p>This function computes the relative frequency of each codon coding for an amino acid.
</p>


<h3>Usage</h3>

<pre><code class="language-R">RSCU(sequence)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>

<p>Input as a nucleic acid sequence of characters from fasta file.
</p>
</td>
</tr></table>
<h3>Details</h3>

<p>RSCU values are the number of times a particular codon is observed, relative to the number of times that the codon would be observed for a uniform synonymous codon usage.
</p>


<h3>Value</h3>

<p>RSCU returns the data frame with all indices.
</p>


<h3>Author(s)</h3>

<p>Dr. Anu Sharma,
Dr. Sudhir Srivastava
</p>


<h3>References</h3>

<p>Elek, A. (2018). coRdon: an R package for codon usage analysis and prediction of gene expressivity (Master's thesis, University of Zagreb. Faculty of Science. Department of Biology).
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(GenomicSig)
RSCU(sequence= system.file("extdata/Nuc_sequence.fasta", package = "GenomicSig"))
</code></pre>


</div>