<div class="container">

<table style="width: 100%;"><tr>
<td>get_results</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Circular Permutation Results
</h2>

<h3>Description</h3>

<p>Creates a summary dataframe of the genomic permutations datasets
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_results(res_pattern="Permus",level="snp",from="workspace",
threshold=0.05,envir = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>res_pattern</code></td>
<td>

<p>Pattern of the Permutation files/variable. eg. res=pattern="Permus"
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>level</code></td>
<td>

<p>Permutation level performed.level values "snp" or "gene"
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>from</code></td>
<td>

<p>Location of the permutation datasets.from values "workspace" or "directory" 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>

<p>Threshold of significance set 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>envir</code></td>
<td>

<p>R environment where save the data to
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table><tr style="vertical-align: top;">
<td><code>results</code></td>
<td>

<p>Data frame with Pathway ID, Trait, Threshold set by permutations,<br> 
Gene results include the theoretical hypergeometric p-value and the,<br> 
observed (Empirical Hypergeometric p-values)<br>
SNP results include the count of significan SNPs and the overall score<br>
Score is the proportion of tests observed with more significant results<br></p>
</td>
</tr></table>
<h3>Format</h3>

<pre>
## SNP level results
     PathID    Trait Threshold RealCount Score
1  hsa00010     abpi         0         0 0.037
2  hsa00010 abpildfa         0         0 0.040
3  hsa04720     abpi         2         0 0.311	
## Gene level results	
     PathID Trait   Threshold     P-Value  Observed
1  hsa00010  abpi 0.040441176 0.058823529 1.0000000
2  hsa00020  abpi 0.000000000 0.000000000 0.1666667
3  hsa00030  abpi 0.040441176 0.058823529 1.0000000
</pre>


<h3>Examples</h3>

<pre><code class="language-R">data(demo,SNPsAnnotation)
all_data &lt;- read_pvals(data_name=demo,snps_ann=SNPsAnnotation)
genome_results &lt;-genome_order(all_data=all_data)
	# Results from genome_order
	ordered_alldata &lt;- genome_results$ordered_alldata
	gs_locs &lt;- genome_results$gs_locs

# Create new environment to save data
gper.env &lt;- new.env()

# Get pathways
data(RHSA164843,RHSA446343,RHSA8876384,RHSA8964572,RHSA109582,RHSA1474244,envir=gper.env)

paths_res &lt;- read2_paths(ordered_alldata=ordered_alldata,gs_locs=gs_locs,
sets_from="workspace",sets_prefix="RHSA",level="snp",envir=gper.env)
		pers_ids &lt;- paths_res$per_ors
		pathways&lt;- paths_res$pathways

snps_permutation(ordered_alldata=ordered_alldata,pers_ids=pers_ids,
ntraits=c(7,9),nper=10,saveto="workspace",threshold=0.05,
gs_locs=gs_locs,envir= gper.env)

results &lt;- get_results(res_pattern="Permus",level="snp",
from="workspace",threshold=0.05,envir = gper.env)
</code></pre>


</div>