<div class="container">

<table style="width: 100%;"><tr>
<td>GenoPop_Impute</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GenoPop-Impute</h2>

<h3>Description</h3>

<p>Performs imputation of missing genomic data in batches using the missForest (Stekhoven &amp; BÃ¼hlmanm, 2012) algorithm. This function reads VCF files, divides it into batches of a fixed number of SNPs, applies the missForest algorithm to each batch, and writes the results to a new VCF file, which will be returned bgzipped and tabix indexed. The choice of the batch size is critical for balancing accuracy and computational demand. We found that a batch size of 500 SNPs is the most accurate for recombination rates typical of mammalians. For on average higher recombination rates (&gt; 5 cM/Mb) we recommend a batch size of 100 SNPs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GenoPop_Impute(
  vcf_path,
  output_vcf,
  batch_size = 1000,
  maxiter = 10,
  ntree = 100,
  threads = 1,
  write_log = FALSE,
  logfile = "log.txt"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf_path</code></td>
<td>
<p>Path to the input VCF file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_vcf</code></td>
<td>
<p>Path for the output VCF file with imputed data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch_size</code></td>
<td>
<p>Number of SNPs to process per batch (default: 500).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxiter</code></td>
<td>
<p>Number of improvement iterations for the random forest algorithm (default: 10).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ntree</code></td>
<td>
<p>Number of decision trees in the random forest (default: 100).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threads</code></td>
<td>
<p>Number of threads used for computation (default: 1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>write_log</code></td>
<td>
<p>If TRUE, writes a log file of the process (advised for large datasets).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logfile</code></td>
<td>
<p>Path to the log file, used if <code>write_log</code> is TRUE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Path to the output VCF file with imputed data.
</p>


<h3>Examples</h3>

<pre><code class="language-R"> vcf_file &lt;- system.file("tests/testthat/sim_miss.vcf.gz", package = "GenoPop")
 index_file &lt;- system.file("tests/testthat/sim_miss.vcf.gz.tbi", package = "GenoPop")
 output_file &lt;- tempfile(fileext = ".vcf")
 GenoPop_Impute(vcf_file, output_vcf = output_file, batch_size = 500)
</code></pre>


</div>