<div class="container">

<table style="width: 100%;"><tr>
<td>PrivateAlleles</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>PrivateAlleles</h2>

<h3>Description</h3>

<p>This function calculates the number of private alleles in two populations from a VCF file. (Alleles which are not present in the other popualtion.)
It processes the file in batches or specified windows across the genome.
For batch processing, it uses <code>process_vcf_in_batches</code>. For windowed analysis, it uses a similar
approach tailored to process specific genomic windows (<code>process_vcf_in_windows</code>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">PrivateAlleles(
  vcf_path,
  pop1_individuals,
  pop2_individuals,
  threads = 1,
  write_log = FALSE,
  logfile = "log.txt",
  batch_size = 10000,
  window_size = NULL,
  skip_size = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf_path</code></td>
<td>
<p>Path to the VCF file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop1_individuals</code></td>
<td>
<p>Vector of individual names belonging to the first population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop2_individuals</code></td>
<td>
<p>Vector of individual names belonging to the second population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threads</code></td>
<td>
<p>Number of threads to use for parallel processing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>write_log</code></td>
<td>
<p>Logical, indicating whether to write progress logs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logfile</code></td>
<td>
<p>Path to the log file where progress will be logged.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch_size</code></td>
<td>
<p>The number of variants to be processed in each batch
(used in batch mode only, default of 10,000 should be suitable for most use cases).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window_size</code></td>
<td>
<p>Size of the window for windowed analysis in base pairs (optional).
When specified, <code>skip_size</code> must also be provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>skip_size</code></td>
<td>
<p>Number of base pairs to skip between windows (optional).
Used in conjunction with <code>window_size</code> for windowed analysis.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>In batch mode (no window_size or skip_size provided): A list containing the number of private alleles for each population.
In window mode (window_size and skip_size provided): A list of data frames, each with columns 'Chromosome', 'Start', 'End', 'PrivateAllelesPop1', and 'PrivateAllelesPop2', representing the count of private alleles within each window for each population.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Batch mode example
vcf_file &lt;- system.file("tests/testthat/sim.vcf.gz", package = "GenoPop")
index_file &lt;- system.file("tests/testthat/sim.vcf.gz.tbi", package = "GenoPop")
pop1_individuals &lt;- c("tsk_0", "tsk_1", "tsk_2")
pop2_individuals &lt;- c("tsk_3", "tsk_4", "tsk_5")
private_alleles &lt;- PrivateAlleles(vcf_file, pop1_individuals, pop2_individuals)

# Window mode example
private_alleles_windows &lt;- PrivateAlleles(vcf_file, pop1_individuals, pop2_individuals,
                                          window_size = 100000, skip_size = 50000)

</code></pre>


</div>