<div class="container">

<table style="width: 100%;"><tr>
<td>importGTF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Import a GTF File</h2>

<h3>Description</h3>

<p>This function imports a gtf file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">importGTF(file, skip = "auto", nrow = -1, use.data.table = TRUE,
  level = "gene", features = NULL, num.features = c("FPKM", "TPM"),
  print.features = FALSE, merge.feature = NULL, merge.all = TRUE,
  class.names = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>file or folder</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>skip</code></td>
<td>
<p>numeric, lines to skip</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nrow</code></td>
<td>
<p>numeric, lines to read</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use.data.table</code></td>
<td>
<p>logical</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>level</code></td>
<td>
<p>Character, read level, default: "gene"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>features to import</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.features</code></td>
<td>
<p>names of the numeric features</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>print.features</code></td>
<td>
<p>Logical, print available features</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>merge.feature</code></td>
<td>
<p>Character, merge multiple samples to dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>merge.all</code></td>
<td>
<p>Logial, shall all samples be merged</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>class.names</code></td>
<td>
<p>Vector with class names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical, verbose function output</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function imports a gtf file. The features names to be imported are defined in <code>features</code>, several features are then
provided as vector. A list of available feature can beprinted, by setting <code>print.features=TRUE</code>.
</p>
<p>The <code>skip</code> option allows to skip a given number of rows, the default is, however, <code>auto</code>. In that case, all rows that
start with the <code>#</code> symbol are skipped.
</p>
<p>In case a set of expression values given in gtf format should be imported and to be merged into a single data table, the feature
that should be used for merging can be provided to the <code>merge.feature</code> option. In that case the function expects a folder
in <code>file</code> and it will import all gtfs located in that folder and merges them according to the <code>merge.feature</code> option.
With the option <code>class.names</code> a vector of prefixes for the merged features can be provided. If this is kept empty, then the 
filenames of the gtf will be used instead (without gtf extension).
</p>
<p>By default the function imprts all features in column 9 as string character. However, for common labels (FPKM and TPM) the class
type is set automatically to numeric. Additional numerical feature names can be defined with the <code>num.feature</code> option.
</p>


<h3>Value</h3>

<p>A gtf object
</p>


<h3>Author(s)</h3>

<p>Daniel Fischer
</p>


<h3>Examples</h3>

<pre><code class="language-R">
 # Define here the location on HDD for the example file
   fpath &lt;- system.file("extdata","example.gtf", package="GenomicTools.fileHandler")
 # Same file, but this time as gzipped version
   fpath.gz &lt;- system.file("extdata","example2.gtf.gz", package="GenomicTools.fileHandler")
   
 # Import the example gtf file  
   importGTF(fpath, level="transcript", features=c("gene_id","FPKM"))
   
 ## Not run: 
 # For the current you need to have zcat installed (should be standard on a Linux system)
   importGTF(fpath.gz, level="transcript", features=c("gene_id","FPKM"))
   
## End(Not run)
 
</code></pre>


</div>