<div class="container">

<table style="width: 100%;"><tr>
<td>add_noise</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Add noise to the VAF values in an F matrix</h2>

<h3>Description</h3>

<p>This function adds noise to the variant allele frequency (VAF) values in an F matrix, simulating the effect of sequencing errors. The noise is modeled as a negative binomial distribution for the depth of the reads and a binomial distribution for both the variant allele counts and the mismatch counts.
</p>


<h3>Usage</h3>

<pre><code class="language-R">add_noise(F_matrix, depth, overdispersion)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>F_matrix</code></td>
<td>
<p>A matrix representing the true VAF values of a series of mutations in a set of samples (F matrix).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>depth</code></td>
<td>
<p>A numeric value representing the mean depth of sequencing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>overdispersion</code></td>
<td>
<p>A numeric value representing the overdispersion parameter for the negative binomial distribution used to simulate the depth of sequencing.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix containing noisy VAF values of a series of mutations in a set of samples.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Calculate the noisy VAF values of a series of mutations in a set of samples, given the true 
# VAF values in the F matrix F_true, a depth of 30 and an overdispersion of 5

# Create an instance of a tumor with 50 clones,
# 10 samples, k = 5, positive selection without noise
F_true &lt;- create_instance(
  n = 50,
  m = 10,
  k = 5,
  selection = "positive", 
  noisy = FALSE)$F

# Then we add the noise using a depth of 30 and an overdispersion of 5.
noisy_F &lt;- add_noise(F_true, 30, 5)

</code></pre>


</div>