<div class="container">

<table style="width: 100%;"><tr>
<td>trim_oneLineage</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Trims the pedigree structure to include one lineage only.
</h2>

<h3>Description</h3>

<p>Trims the families to include only one lineage.
</p>


<h3>Usage</h3>

<pre><code class="language-R">trim_oneLineage(seqSub, pednew)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqSub</code></td>
<td>

<p>A data frame that should include three columns FID (family ID), IID (individual ID), and PHENOTYPE (affection status) for the sequenced subjects in the data. One example is the 1st, 2nd and 6th columns from the plink raw format.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pednew</code></td>
<td>

<p>A data frame includes the complete pedigree structure information for all sequenced families in the dataset. The required column names of this data frame include: FID (family ID), IID (individual ID, must be of class character), faID (father ID, NA if unavailable), moID (mother ID, NA if unavailable), and sex.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For each subject, only the maternal or the paternal family is included, since the rare variant should be present in only the related subjects. The lineage with the maximal set of sequenced cases will be used as the final pedigree.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pedInfoUpdate</code></td>
<td>
<p>the complete pedigrees with only the paternal or maternal lineage </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqSubjUpdate</code></td>
<td>
<p>The sequenced subjects that are in the selected lineage are returned for the rest of the analysis.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>This function can be used for other analysis of family-based data processing. For example, the pre-processing step for PVAAST analysis.
</p>


<h3>Author(s)</h3>

<p>Dandi Qiao
</p>


<h3>References</h3>

<p>Qiao, D. Lange, C., Laird, N.M., Won, S.,  Hersh, C.P., et al. (2017). <em>Gene-based segregation method for identifying rare variants for family-based sequencing studies.</em> Genet Epidemiol 41(4):309-319. DOI:10.1002/gepi.22037.
</p>


<h3>See Also</h3>

<p><code>GESE</code>, <code>trim_unrelated</code></p>


<h3>Examples</h3>

<pre><code class="language-R">data(pednew)
data(mapInfo)
data(dataRaw)
data(database)
subjects &lt;- dataRaw[,c(1:2, 6)]
cat("Trimming the families...\n")
cat("Trimming step 1: keep only one lineage \n")
trim &lt;- trim_oneLineage(seqSub=subjects, pednew)
	
</code></pre>


</div>