<div class="container">

<table style="width: 100%;"><tr>
<td>condSegProbF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Computes conditional segregation probability for the family
</h2>

<h3>Description</h3>

<p>Computes the conditional probability that a variant is segregating in the family conditional on that the variant is present in one of the founders in the family.
</p>


<h3>Usage</h3>

<pre><code class="language-R">condSegProbF(pedTemp, subjInfo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pedTemp</code></td>
<td>

<p>The data frame that includes the complete pedigree structure for the family</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subjInfo</code></td>
<td>

<p>A data frame that contains the subject phenotype information for the sequenced subjects. it should include the columns FID, IID, and PHENOTYPE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>returns the conditional segregating probability of a variant in the family
</p>


<h3>Author(s)</h3>

<p>Dandi Qiao
</p>


<h3>References</h3>

<p>Qiao, D. Lange, C., Laird, N.M., Won, S., Hersh, C.P., et al. (2017). <em>Gene-based segregation method for identifying rare variants for family-based sequencing studies.</em> Genet Epidemiol 41(4):309-319. DOI:10.1002/gepi.22037.
</p>


<h3>See Also</h3>

<p><code>GESE</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(pednew)
data(mapInfo)
data(dataRaw)
data(database)
library(kinship2)
pedigrees = kinship2::pedigree(pednew$IID, pednew$faID, pednew$moID,pednew$sex,famid=pednew$FID)
subjects= dataRaw[,c(1,2,6)]
condSegProbF(pedigrees['93'], subjects)
condSegProbF(pedigrees['412'], subjects)
results2 &lt;- GESE(pednew, database, 1000000, dataRaw, mapInfo, threshold=1e-2)
results2$condSegProb
</code></pre>


</div>