<div class="container">

<table style="width: 100%;"><tr>
<td>GESE-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Gene-Based Segregation Test
</h2>

<h3>Description</h3>

<p>Implements the gene-based segregation test(GESE) and the weighted GESE test for identifying genes with causal variants of large effects for family-based sequencing data. The methods are described in Qiao, D. Lange, C., Laird, N.M., Won, S., Hersh, C.P., et al. (2017). &lt;DOI:10.1002/gepi.22037&gt;. Gene-based segregation method for identifying rare variants for family-based sequencing studies. Genet Epidemiol 41(4):309-319. More details can be found at &lt;http://scholar.harvard.edu/dqiao/gese&gt;.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> GESE</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Title: </td>
<td style="text-align: left;"> Gene-Based Segregation Test</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 2.0.1</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2017-05-17</td>
</tr>
<tr>
<td style="text-align: left;">
Author: </td>
<td style="text-align: left;"> Dandi Qiao, Michael H. Cho</td>
</tr>
<tr>
<td style="text-align: left;">
Maintainer: </td>
<td style="text-align: left;"> Dandi Qiao &lt;redaq@channing.harvard.edu&gt;</td>
</tr>
<tr>
<td style="text-align: left;">
Description: </td>
<td style="text-align: left;"> Implements the gene-based segregation test(GESE) and the weighted GESE test for identifying genes with causal variants of large effects for family-based sequencing data. The methods are described in Qiao, D. Lange, C., Laird, N.M., Won, S., Hersh, C.P., et al. (2017). &lt;DOI:10.1002/gepi.22037&gt;. Gene-based segregation method for identifying rare variants for family-based sequencing studies. Genet Epidemiol 41(4):309-319. More details can be found at &lt;http://scholar.harvard.edu/dqiao/gese&gt;.</td>
</tr>
<tr>
<td style="text-align: left;">
Depends: </td>
<td style="text-align: left;"> kinship2</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
<td style="text-align: left;">
NeedsCompilation: </td>
<td style="text-align: left;"> yes</td>
</tr>
<tr>
<td style="text-align: left;">
Suggests: </td>
<td style="text-align: left;"> knitr,
rmarkdown</td>
</tr>
<tr>
<td style="text-align: left;">
VignetteBuilder: </td>
<td style="text-align: left;"> knitr</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Index of help topics:
</p>
<pre>
GESE                    Gene-Based Segregation Test
GESE-internal functions
                        GESE package internal functions
GESE-package            Gene-Based Segregation Test
condSegProbF            Computes conditional segregation probability
                        for the family
dataRaw                 dataRaw - a data frame containing the pedigree,
                        phenotype and genotype information
database                database file in example
getSegInfo              Computes segregation information for different
                        mode of inheritance.
mapInfo                 mafInfo - example data
pednew                  pednew - an example pedigree structure
trim_oneLineage         Trims the pedigree structure to include one
                        lineage only.
trim_unrelated          Trims the pedigree structure to exclude
                        multiple founder cases
</pre>
<p>computes gene-based segregation tests(GESE and weighted GESE) for family-based sequencing data. The main functions are:
<code>GESE</code>: computes gene-based segregation information and GESE test p-values (unweighted and weighted version).
<code>trim_oneLineage</code>: trims the pedigree so that for any subject, either the paternal family or the maternal family is included. Minimal set of sequenced subjects may be removed to ensure one lineage per pedigree only.
<code>trim_unrelated</code>: trims the pedigree so that only one founder case is kept for each pedigree, and pedigrees with no cases are removed.
<code>condSegProbF</code>: computes the conditional probability that a variant in the gene is segregating in the family specified, conditional on that the variant is present in the family.
</p>


<h3>Author(s)</h3>

<p>Dandi Qiao, Michael H. Cho
</p>
<p>Maintainer: Dandi Qiao &lt;redaq@channing.harvard.edu&gt;
</p>


<h3>References</h3>

<p>Qiao, D. Lange, C., Laird, N.M., Won, S.,  Hersh, C.P., et al. (2017). <em>Gene-based segregation method for identifying rare variants for family-based sequencing studies.</em> Genet Epidemiol 41(4):309-319. DOI:10.1002/gepi.22037.
</p>
<p><a href="http://scholar.harvard.edu/dqiao/gese">http://scholar.harvard.edu/dqiao/gese</a>
</p>


<h3>See Also</h3>

<p><code>GESE</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(pednew)
data(mapInfo)
data(dataRaw)
data(database)
results &lt;- GESE(pednew, database, 1000000, dataRaw, mapInfo, threshold=1e-2)
results

</code></pre>


</div>