<div class="container">

<table style="width: 100%;"><tr>
<td>getmstatistic</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Quantifying Systematic Heterogeneity in Meta-Analysis.</h2>

<h3>Description</h3>

<p><code>getmstatistic</code> computes <em>M</em> statistics to assess the contribution
of each participating study in a meta-analysis. The <em>M</em> statistic 
aggregates heterogeneity information across multiple variants to, identify 
systematic heterogeneity patterns and their direction of effect in 
meta-analysis. It's primary use is to identify outlier studies, which either 
show "null" effects or consistently show stronger or weaker genetic effects 
than average, across the panel of variants examined in a GWAS meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getmstatistic(beta_in, lambda_se_in, study_names_in, variant_names_in, ...)

## Default S3 method:
getmstatistic(
  beta_in,
  lambda_se_in,
  study_names_in,
  variant_names_in,
  save_dir = getwd(),
  tau2_method = "DL",
  x_axis_increment_in = 0.02,
  x_axis_round_in = 2,
  produce_plots = TRUE,
  verbose_output = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>beta_in</code></td>
<td>
<p>A numeric vector of study effect-sizes e.g. log odds-ratios.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda_se_in</code></td>
<td>
<p>A numeric vector of standard errors, genomically 
corrected at study-level.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>study_names_in</code></td>
<td>
<p>A character vector of study names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variant_names_in</code></td>
<td>
<p>A character vector of variant names e.g. rsIDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further arguments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_dir</code></td>
<td>
<p>A character scalar specifying a path to the directory where
plots should be stored (optional). Required if produce_plots = TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau2_method</code></td>
<td>
<p>A character scalar, method to estimate heterogeneity: 
either "DL" or "REML" (Optional). Note: The REML method uses the 
iterative Fisher scoring algorithm (step length = 0.5, maximum 
iterations = 10000) to estimate tau2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x_axis_increment_in</code></td>
<td>
<p>A numeric scalar, value by which x-axis 
of M scatterplot will be incremented (Optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x_axis_round_in</code></td>
<td>
<p>A numeric scalar, value to which x-axis labels of 
M scatterplot will be rounded (Optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>produce_plots</code></td>
<td>
<p>A boolean to generate plots (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose_output</code></td>
<td>
<p>An optional boolean to display intermediate output.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In contrast to conventional heterogeneity metrics (Q-statistic, I-squared and 
tau-squared) which measure random heterogeneity at individual variants, 
<em>M</em> measures systematic (non-random) heterogeneity across multiple 
independently associated variants. 
</p>
<p>Systematic heterogeneity can arise in a meta-analysis due to differences in 
the study characteristics of participating studies. Some of the differences 
may include: ancestry, allele frequencies, phenotype definition, 
age-of-disease onset, family-history, gender, linkage disequilibrium and 
quality control thresholds. See the getmstatistic website for statistical 
theory, documentation and examples.
</p>
<p><code>getmstatistic</code> uses summary data i.e. study effect-sizes and their 
corresponding standard errors to calculate <em>M</em> statistics (One <em>M</em> 
for each study in the meta-analysis).
</p>
<p>In particular, <code>getmstatistic</code> employs the inverse-variance weighted 
random effects regression model provided in the <code>metafor</code> R package 
to extract SPREs (standardized predicted random effects) which are then 
aggregated to formulate <em>M</em> statistics.
</p>


<h3>Value</h3>

<p>Returns a list containing:
</p>

<ul>
<li>
<p> Mstatistic_expected_mean   , A numeric scalar for the expected mean for M
</p>
</li>
<li>
<p> Mstatistic_expected_sd     , A numeric scalar for the expected standard deviation 
for M
</p>
</li>
<li>
<p> number_studies             , A numeric scalar for the number of studies
</p>
</li>
<li>
<p> number_variants            , A numeric scalar for the number of variants
</p>
</li>
<li>
<p> Mstatistic_crit_alpha_0_05 , A numeric scalar of the critical M value at the 5
percent significance level.
</p>
</li>
<li>
<p> M_dataset (dataframe) A dataset of the computed M statistics, which
includes the following fields:
</p>

<ul>
<li>
<p> M           , Mstatistic
</p>
</li>
<li>
<p> M_sd        , standard deviation of M
</p>
</li>
<li>
<p> M_se        , standard error of M
</p>
</li>
<li>
<p> lowerbound  , lowerbound of M 95
</p>
</li>
<li>
<p> upperbound  , upperbound of M 95
</p>
</li>
<li>
<p> bonfpvalue  , 2-sided bonferroni pvalues of M
</p>
</li>
<li>
<p> qvalue      , false discovery rate adjusted pvalues of M
</p>
</li>
<li>
<p> tau2        , tau_squared, DL estimates of between-study 
heterogeneity
</p>
</li>
<li>
<p> I2          , I_squared, proportion of total variation due to 
between study variance
</p>
</li>
<li>
<p> Q           , Cochran's Q
</p>
</li>
<li>
<p> xb          , fitted values excluding random effects
</p>
</li>
<li>
<p> usta        , standardized predicted random effect (SPRE)
</p>
</li>
<li>
<p> xbu         , fitted values including random effects
</p>
</li>
<li>
<p> stdxbu      , standard error of prediction (fitted values) 
including random effects
</p>
</li>
<li>
<p> hat         , diagonal elements of the projection hat matrix
</p>
</li>
<li>
<p> study       , study numbers    
</p>
</li>
<li>
<p> snp         , variant numbers
</p>
</li>
<li>
<p> beta_mean   , average variant effect size
</p>
</li>
<li>
<p> oddsratio   , average variant effect size as oddsratio
</p>
</li>
<li>
<p> beta_n   , number of variants in each study
</p>
</li>
</ul>
</li>
<li>
<p> influential_studies_0_05 (dataframe) A dataset of influential studies 
significant at the 5 percent level.
</p>
</li>
<li>
<p> weaker_studies_0_05 (dataframe) A dataset of under-performing studies 
significant at the 5 percent level.
</p>
</li>
</ul>
<h3>Methods (by class)</h3>


<ul><li> <p><code>default</code>: Computes M statistics
</p>
</li></ul>
<h3>See Also</h3>

<p><code>rma.uni</code> function in <code>metafor</code> for random
effects model, and <a href="https://magosil86.github.io/getmstatistic/">https://magosil86.github.io/getmstatistic/</a> for 
getmstatistic website.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(getmstatistic)
library(gridExtra)


# Basic M analysis using the heartgenes214 dataset.
# heartgenes214 is a multi-ethnic GWAS meta-analysis dataset for coronary artery disease.
# To learn more about the heartgenes214 dataset ?heartgenes214

# Running an M analysis on 20 GWAS significant variants (p &lt; 5e-08) in the first 10 studies

heartgenes44_10studies &lt;- subset(heartgenes214, studies &lt;= 10 &amp; fdr214_gwas46 == 2) 
heartgenes20_10studies &lt;- subset(heartgenes44_10studies, 
    variants %in% unique(heartgenes44_10studies$variants)[1:20])

# Set directory to store plots, this can be a temporary directory
# or a path to a directory of choice e.g. plots_dir &lt;- "~/Downloads"
plots_dir &lt;- tempdir()

getmstatistic_results &lt;- getmstatistic(heartgenes20_10studies$beta_flipped, 
                                        heartgenes20_10studies$gcse, 
                                        heartgenes20_10studies$variants, 
                                        heartgenes20_10studies$studies,
                                        save_dir = plots_dir)
getmstatistic_results

# Explore results generated by getmstatistic function

# Retrieve dataset of M statistics
dframe &lt;- getmstatistic_results$M_dataset



str(dframe)


# Retrieve dataset of stronger than average studies (significant at 5% level)
getmstatistic_results$influential_studies_0_05

# Retrieve dataset of weaker than average studies (significant at 5% level)
getmstatistic_results$weaker_studies_0_05

# Retrieve number of studies and variants
getmstatistic_results$number_studies
getmstatistic_results$number_variants

# Retrieve expected mean, sd and critical M value at 5% significance level
getmstatistic_results$M_expected_mean
getmstatistic_results$M_expected_sd
getmstatistic_results$M_crit_alpha_0_05

# To view plots stored in a temporary directory, call `tempdir()` to view the directory path 
tempdir()


# Additional examples: These take a little bit longer to run

## Not run: 

# Set directory to store plots, this can be a temporary directory
# or a path to a directory of choice e.g. plots_dir &lt;- "~/Downloads"
plots_dir &lt;- tempdir()

# Run M analysis on all 214 lead variants
# heartgenes214 is a multi-ethnic GWAS meta-analysis dataset for coronary artery disease.
getmstatistic_results &lt;- getmstatistic(heartgenes214$beta_flipped, 
                                        heartgenes214$gcse, 
                                        heartgenes214$variants, 
                                        heartgenes214$studies,
                                        save_dir = plots_dir)
getmstatistic_results


# Subset the GWAS significant variants (p &lt; 5e-08) in heartgenes214
heartgenes44 &lt;- subset(heartgenes214, heartgenes214$fdr214_gwas46 == 2)

# Exploring getmstatistic options:
#     Estimate heterogeneity using "REML", default is "DL"
#     Modify x-axis of M scatterplot
#     Run M analysis verbosely
getmstatistic_results &lt;- getmstatistic(heartgenes44$beta_flipped, 
                                        heartgenes44$gcse, 
                                        heartgenes44$variants, 
                                        heartgenes44$studies,
                                        save_dir = plots_dir,
                                        tau2_method = "REML",
                                        x_axis_increment_in = 0.03, 
                                        x_axis_round_in = 3,
                                        produce_plots = TRUE,
                                        verbose_output = TRUE)
getmstatistic_results



## End(Not run)

</code></pre>


</div>