<div class="container">

<table style="width: 100%;"><tr>
<td>GxEscreen</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gene-Environment Interaction: Genome-wide Screen</h2>

<h3>Description</h3>

<p>Function to test for the joint genetic and gene-environment
interaction effects for a set of variant using a LRT model
</p>


<h3>Usage</h3>

<pre><code class="language-R">GxEscreen(dat, geno, nsim = 1e+05, K = 7)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>
<p>a data frame with covariate information. Col 1 should be phenotype,
col 2 should be environmental factor, col 3 and later should be additional covariates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>a genotype matrix with 0-1-2 coding</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsim</code></td>
<td>
<p>the number of replicates in obtaining the p-value (standard 1e5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>the number of knots used to control the flexibility in modeling GxE interaction</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a vector containing the p-value from the LRT associated with each SNP
</p>


<h3>Examples</h3>

<pre><code class="language-R">GxEscreen(cov_example, geno_example, nsim=1e5, K=7)
</code></pre>


</div>