<div class="container">

<table style="width: 100%;"><tr>
<td>def_names</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Default column names and types for defined formats</h2>

<h3>Description</h3>

<p>Intended to be used in <code>readr::read_tsv()</code>-like functions that accept a
<code>col_names</code> and a <code>col_types</code> argument.
</p>


<h3>Usage</h3>

<pre><code class="language-R">def_names(format)

def_types(format)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>specify a format known to gggenomes, such as <code>gff3</code>, <code>gbk</code>, ...</p>
</td>
</tr></table>
<h3>Value</h3>

<p>a vector with default column names for the given format
</p>
<p>a vector with default column types for the given format
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>def_names()</code>: default column names for defined formats
</p>
</li>
<li> <p><code>def_types()</code>: default column types for defined formats
</p>
</li>
</ul>
<h3>Defined formats, column types and names</h3>

<pre>
  gff3       ccciicccc       seq_id,source,type,start,end,score,strand,phase,attributes
  paf        ciiicciiiiid    seq_id,length,start,end,strand,seq_id2,length2,start2,end2,map_match,map_length,map_quality
  blast      ccdiiiiiiidd    seq_id,seq_id2,pident,length,mismatch,gapopen,start,end,start2,end2,evalue,bitscore
  bed        ciicdc          seq_id,start,end,name,score,strand
  fai        ci---           seq_id,seq_desc,length
  seq_len    cci             seq_id,seq_desc,length
  vcf        cicccdccc       seq_id,start,feat_id,ref,alt,qual,filter,info,format
</pre>


<h3>Examples</h3>

<pre><code class="language-R"># read a blast-tabular file with read_tsv
readr::read_tsv(ex("emales/emales-prot-ava.o6"), col_names = def_names("blast"))
</code></pre>


</div>