<div class="container">

<table style="width: 100%;"><tr>
<td>feats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Use tracks inside and outside <code style="white-space: pre;">⁠geom_*⁠</code> calls</h2>

<h3>Description</h3>

<p>Track selection works like <code>dplyr::pull()</code> and supports unquoted ids and
positional arguments. <code>...</code> can be used to subset the data in
<code>dplyr::filter()</code> fashion. <code>pull</code>-prefixed variants return the specified
track from a gggenome object. Unprefixed variants work inside <code style="white-space: pre;">⁠geom_*⁠</code> calls.
</p>


<h3>Usage</h3>

<pre><code class="language-R">feats(.track_id = 1, ..., .ignore = "genes", .geneify = FALSE)

feats0(.track_id = 1, ..., .ignore = NA, .geneify = FALSE)

genes(..., .gene_types = c("CDS", "mRNA", "tRNA", "tmRNA", "ncRNA", "rRNA"))

links(.track_id = 1, ..., .ignore = NULL, .adjacent_only = NULL)

seqs(...)

bins(..., .group = vars())

track(.track_id = 1, ..., .track_type = NULL, .ignore = NULL)

pull_feats(.x, .track_id = 1, ..., .ignore = "genes", .geneify = FALSE)

pull_genes(
  .x,
  ...,
  .gene_types = c("CDS", "mRNA", "tRNA", "tmRNA", "ncRNA", "rRNA")
)

pull_links(.x, .track_id = 1, ..., .ignore = NULL, .adjacent_only = NULL)

pull_seqs(.x, ...)

pull_bins(.x, ..., .group = vars())

## S3 method for class 'gggenomes_layout'
pull_bins(.x, ..., .group = vars())

pull_track(.x, .track_id = 1, ..., .track_type = NULL, .ignore = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>.track_id</code></td>
<td>
<p>The track to pull out, either as a literal variable name or
as a positive/negative integer giving the position from the left/right.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Logical predicates passed on to dplyr::filter. "seqs", "feats",
"links". Affects position-based selection.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.ignore</code></td>
<td>
<p>track names to ignore when selecting by position.
Default is "genes", if using <code>feats0</code> this defaults to <code>NA</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.geneify</code></td>
<td>
<p>add dummy type, introns and geom_id column to play nicely
with geoms supporting multi-level and spliced gene models.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.gene_types</code></td>
<td>
<p>return only feats of this type (<code>type %in% .gene_types</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.adjacent_only</code></td>
<td>
<p>filter for links connecting direct neighbors
(<code style="white-space: pre;">⁠abs(y-yend)==1)⁠</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.group</code></td>
<td>
<p>what variables to use in grouping of bins from seqs in addition
to <code>y</code> and <code>bin_id</code>. Use this to get additional shared variables from the
seqs table into the bins table.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.track_type</code></td>
<td>
<p>restrict to these types of tracks - any combination of
"seqs", "feats", "links".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.x</code></td>
<td>
<p>A gggenomes or gggenomes_layout object.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A function that pulls the specified track from a gggenomes object.
</p>
<p>A function that pulls the specified track from a gggenomes object.
</p>
<p>A function that pulls the specified track from a gggenomes object.
</p>
<p>A function that pulls the specified track from a gggenomes object.
</p>
<p>A function that pulls the specified track from a gggenomes object.
</p>
<p>A function that pulls the specified track from a gggenomes object.
</p>
<p>A function that pulls the specified track from a gggenomes object.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>feats()</code>: by default pulls out the first feat track not named
"genes".
</p>
</li>
<li> <p><code>feats0()</code>: by default pulls out the first feat track.
</p>
</li>
<li> <p><code>genes()</code>: pulls out the first feat track (genes), filtering
for records with <code>type=="CDS"</code>, and adding a dummy <code>gene_id</code> column if missing
to play nice with multi-exon <code>geom</code>s.
</p>
</li>
<li> <p><code>links()</code>: by default pulls out the first link track.
</p>
</li>
<li> <p><code>seqs()</code>: pulls out the seqs track (there is only one).
</p>
</li>
<li> <p><code>bins()</code>: pulls out a binwise summary table of the seqs data powering
<code style="white-space: pre;">⁠geom_bin_*()⁠</code> calls. The bin table is not a real track, but recomputed
on-the-fly.
</p>
</li>
<li> <p><code>track()</code>: pulls from all tracks in order seqs, feats, links.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
gg &lt;- gggenomes(emale_genes, emale_seqs, emale_tirs, emale_ava)
gg %&gt;% track_info() # info about track ids, positions and types

# get first feat track that isn't "genes" (all equivalent)
gg %&gt;% pull_feats() # easiest
gg %&gt;% pull_feats(feats) # by id
gg %&gt;% pull_feats(1) # by position
gg %&gt;% pull_feats(2, .ignore = NULL) # default .ignore="genes"

# get "seqs" track (always track #1)
gg %&gt;% pull_seqs()

# plot integrated transposons and GC content for some viral genomes
gg &lt;- gggenomes(seqs = emale_seqs, feats = list(emale_ngaros, GC = emale_gc))
gg + geom_seq() +
  geom_feat(color = "skyblue") + # defaults to data=feats()
  geom_line(aes(x, y + score - .6, group = y), data = feats(GC), color = "gray60")
</code></pre>


</div>