<div class="container">

<table style="width: 100%;"><tr>
<td>emale_tirs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Terminal inverted repeats of 6 EMALE genomes</h2>

<h3>Description</h3>

<p>Terminal inverted repeats of 6 EMALE genomes
</p>


<h3>Usage</h3>

<pre><code class="language-R">emale_tirs
</code></pre>


<h3>Format</h3>

<p>A data frame with 3 rows and 14 columns
</p>

<dl>
<dt>file_id</dt>
<dd>
<p>name of the file the data was read from</p>
</dd>
<dt>seq_id</dt>
<dd>
<p>identifier of the sequence the feature appears on</p>
</dd>
<dt>start</dt>
<dd>
<p>start of the feature on the sequence</p>
</dd>
<dt>end</dt>
<dd>
<p>end of the feature on the sequence</p>
</dd>
<dt>strand</dt>
<dd>
<p>reading orientation relative to sequence (+ or -)</p>
</dd>
<dt>type</dt>
<dd>
<p>feature type (CDS, mRNA, gene, ...)</p>
</dd>
<dt>feat_id</dt>
<dd>
<p>unique identifier of the feature</p>
</dd>
<dt>introns</dt>
<dd>
<p>a list column with internal intron start/end positions</p>
</dd>
<dt>parent_ids</dt>
<dd>
<p>a list column with parent IDs - feat_id's of parent features</p>
</dd>
<dt>source</dt>
<dd>
<p>source of the annotation</p>
</dd>
<dt>score</dt>
<dd>
<p>score of the annotation</p>
</dd>
<dt>phase</dt>
<dd>
<p>For "CDS" features indicates where the next codon begins relative to the 5' start</p>
</dd>
<dt>name</dt>
<dd>
<p>name of the feature</p>
</dd>
<dt>width</dt>
<dd>
<p>end-start+1</p>
</dd>
<dt>geom_id</dt>
<dd>
<p>an identifier telling the which features should be plotted as on items (usually CDS and mRNA of same gene)</p>
</dd>
</dl>
<h3>Source</h3>


<ul>
<li>
<p> Publication: <a href="https://doi.org/10.1101/2020.11.30.404863">doi:10.1101/2020.11.30.404863</a>
</p>
</li>
<li>
<p> Raw data: <a href="https://github.com/thackl/cb-emales">https://github.com/thackl/cb-emales</a>
</p>
</li>
<li>
<p> Derived &amp; bundled data: <code>ex("emales/emales-tirs.gff")</code>
</p>
</li>
</ul>
</div>