<div class="container">

<table style="width: 100%;"><tr>
<td>gggenomes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot genomes, features and synteny maps</h2>

<h3>Description</h3>

<p><code>gggenomes()</code> initializes a gggenomes-flavored ggplot object.
It is used to declare the input data for gggenomes' track system.
</p>
<p>(<em>See for more details on the track system, gggenomes vignette or the Details/Arguments section</em>)
</p>


<h3>Usage</h3>

<pre><code class="language-R">gggenomes(
  genes = NULL,
  seqs = NULL,
  feats = NULL,
  links = NULL,
  .id = "file_id",
  spacing = 0.05,
  wrap = NULL,
  adjacent_only = TRUE,
  infer_bin_id = seq_id,
  infer_start = min(start, end),
  infer_end = max(start, end),
  infer_length = max(start, end),
  theme = c("clean", NULL),
  .layout = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genes, feats</code></td>
<td>
<p>A data.frame, a list of data.frames, or a character vector
with paths to files containing gene data. Each item is added as feature
track.
</p>
<p>For a single data.frame the track_id will be "genes" and "feats",
respectively. For a list, track_ids are parsed from the list names, or if
names are missing from the name of the variable containing each data.frame.
Data columns:
</p>

<ul>
<li>
<p> required: <code style="white-space: pre;">⁠seq_id,start,end⁠</code>
</p>
</li>
<li>
<p> recognized: <code style="white-space: pre;">⁠strand,bin_id,feat_id,introns⁠</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>A data.frame or a character vector with paths to files containing
sequence data. Data columns:
</p>

<ul>
<li>
<p> required: <code style="white-space: pre;">⁠seq_id,length⁠</code>
</p>
</li>
<li>
<p> recognized: <code style="white-space: pre;">⁠bin_id,start,end,strand⁠</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>links</code></td>
<td>
<p>A data.frame or a character vector with paths to files
containing link data. Each item is added as links track. Data columns:
</p>

<ul>
<li>
<p> required: <code style="white-space: pre;">⁠seq_id,seq_id2⁠</code>
</p>
</li>
<li>
<p> recognized: <code style="white-space: pre;">⁠start,end,bin_id,start2,end2,bin_id2,strand⁠</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.id</code></td>
<td>
<p>The name of the column for file labels that are created when
reading directly from files. Defaults to "file_id". Set to "bin_id" if
every file represents a different bin.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spacing</code></td>
<td>
<p>between sequences in bases (&gt;1) or relative to longest bin (&lt;1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wrap</code></td>
<td>
<p>wrap bins into multiple lines with at most this many nucleotides
per lin.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjacent_only</code></td>
<td>
<p>Indicates whether links should be created between adjacent sequences/chromosomes only.
By default it is set to <code>adjacent_only = TRUE</code>. If <code>FALSE</code>, links will be created between all sequences
</p>
<p>(<em>not recommended for large data sets</em>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>infer_length, infer_start, infer_end, infer_bin_id</code></td>
<td>
<p>used to infer pseudo
seqs if only feats or links are provided, or if no bin_id column was
provided. The expressions are evaluated in the context of the first feat
or link track.
</p>
<p>By default subregions of sequences from the first to the last feat/link
are generated. Set <code>infer_start</code> to 0 to show all sequences from their
true beginning.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theme</code></td>
<td>
<p>choose a gggenomes default theme, NULL to omit.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.layout</code></td>
<td>
<p>a pre-computed layout from <code>layout_genomes()</code>. Useful for
developmental purposes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional parameters, passed to layout</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>gggenomes::gggenomes()</code> resembles the functionality of <code>ggplot2::ggplot()</code>.
It is used to construct the initial plot object, and is often followed by "+" to add components to the plot (<em>e.g. "+ geom_gene()"</em>).
</p>
<p>A big difference between the two is that gggenomes has a multi-track setup (<em><code>'seqs'</code>, <code>'feats'</code>, <code>'genes'</code> and <code>'links'</code></em>).
<code>gggenomes()</code> pre-computes a layout and adds coordinates (<code style="white-space: pre;">⁠y,x,xend⁠</code>) to each data frame prior to the actual plot construction.
This has some implications for the usage of gggenomes:
</p>

<ul>
<li> <p><strong>Data frames for tracks have required variables.</strong> These predefined variables are used during import
to compute x/y coordinates (<em>see arguments</em>).
</p>
</li>
<li> <p><strong>gggenomes' geoms can often be used without explicit <code>aes()</code> mappings</strong>  This works because
we always know the names of the plot variables ahead of time: they originate from the pre-computed layout,
and we can use that information to set sensible default aesthetic mappings for most cases.
</p>
</li>
</ul>
<h3>Value</h3>

<p>gggenomes-flavored ggplot object
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Compare the genomic organization of three viral elements
# EMALEs: endogenous mavirus-like elements (example data shipped with gggenomes)
gggenomes(emale_genes, emale_seqs, emale_tirs, emale_ava) +
  geom_seq() + geom_bin_label() + # chromosomes and labels
  geom_feat(size = 8) + # terminal inverted repeats
  geom_gene(aes(fill = strand), position = "strand") + # genes
  geom_link(offset = 0.15) # synteny-blocks

# with some more information
gggenomes(emale_genes, emale_seqs, emale_tirs, emale_ava) %&gt;%
  add_feats(emale_ngaros, emale_gc) %&gt;%
  add_clusters(emale_cogs) %&gt;%
  sync() +
  geom_link(offset = 0.15, color = "white") + # synteny-blocks
  geom_seq() + geom_bin_label() + # chromosomes and labels
  # thistle4, salmon4, burlywood4
  geom_feat(size = 6, position = "identity") + # terminal inverted repeats
  geom_feat(
    data = feats(emale_ngaros), color = "turquoise4", alpha = .3,
    position = "strand", size = 16
  ) +
  geom_feat_note(aes(label = type),
    data = feats(emale_ngaros),
    position = "strand", nudge_y = .3
  ) +
  geom_gene(aes(fill = cluster_id), position = "strand") + # genes
  geom_wiggle(aes(z = score, linetype = "GC-content"), feats(emale_gc),
    fill = "lavenderblush4", position = position_nudge(y = -.2), height = .2
  ) +
  scale_fill_brewer("Conserved genes", palette = "Dark2", na.value = "cornsilk3")

# initialize plot directly from files
gggenomes(
  ex("emales/emales.gff"),
  ex("emales/emales.gff"),
  ex("emales/emales-tirs.gff"),
  ex("emales/emales.paf")
) + geom_seq() + geom_gene() + geom_feat() + geom_link()

# multi-contig genomes wrap to fixed width
s0 &lt;- read_seqs(list.files(ex("cafeteria"), "Cr.*\\.fa.fai$", full.names = TRUE))
s1 &lt;- s0 %&gt;% dplyr::filter(length &gt; 5e5)
gggenomes(seqs = s1, infer_bin_id = file_id, wrap = 5e6) +
  geom_seq() + geom_bin_label() + geom_seq_label()
</code></pre>


</div>