<div class="container">

<table style="width: 100%;"><tr>
<td>layout_genomes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Layout genomes</h2>

<h3>Description</h3>

<p>Layout genomes
</p>


<h3>Usage</h3>

<pre><code class="language-R">layout_genomes(
  seqs = NULL,
  genes = NULL,
  feats = NULL,
  links = NULL,
  infer_bin_id = seq_id,
  infer_start = min(start, end),
  infer_end = max(start, end),
  infer_length = max(start, end),
  adjacent_only = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>A data.frame or a character vector with paths to files containing
sequence data. Data columns:
</p>

<ul>
<li>
<p> required: <code style="white-space: pre;">⁠seq_id,length⁠</code>
</p>
</li>
<li>
<p> recognized: <code style="white-space: pre;">⁠bin_id,start,end,strand⁠</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes, feats</code></td>
<td>
<p>A data.frame, a list of data.frames, or a character vector
with paths to files containing gene data. Each item is added as feature
track.
</p>
<p>For a single data.frame the track_id will be "genes" and "feats",
respectively. For a list, track_ids are parsed from the list names, or if
names are missing from the name of the variable containing each data.frame.
Data columns:
</p>

<ul>
<li>
<p> required: <code style="white-space: pre;">⁠seq_id,start,end⁠</code>
</p>
</li>
<li>
<p> recognized: <code style="white-space: pre;">⁠strand,bin_id,feat_id,introns⁠</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>links</code></td>
<td>
<p>A data.frame or a character vector with paths to files
containing link data. Each item is added as links track. Data columns:
</p>

<ul>
<li>
<p> required: <code style="white-space: pre;">⁠seq_id,seq_id2⁠</code>
</p>
</li>
<li>
<p> recognized: <code style="white-space: pre;">⁠start,end,bin_id,start2,end2,bin_id2,strand⁠</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>infer_length, infer_start, infer_end, infer_bin_id</code></td>
<td>
<p>used to infer pseudo
seqs if only feats or links are provided, or if no bin_id column was
provided. The expressions are evaluated in the context of the first feat
or link track.
</p>
<p>By default subregions of sequences from the first to the last feat/link
are generated. Set <code>infer_start</code> to 0 to show all sequences from their
true beginning.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjacent_only</code></td>
<td>
<p>Indicates whether links should be created between adjacent sequences/chromosomes only.
By default it is set to <code>adjacent_only = TRUE</code>. If <code>FALSE</code>, links will be created between all sequences
</p>
<p>(<em>not recommended for large data sets</em>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional parameters, passed to layout</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>gggenomes_layout object
</p>


</div>