<div class="container">

<table style="width: 100%;"><tr>
<td>read_paf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read a .paf file (minimap/minimap2).</h2>

<h3>Description</h3>

<p>Read a minimap/minimap2 .paf file including optional tagged extra fields. The
optional fields will be parsed into a tidy format, one column per tag.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_paf(
  file,
  max_tags = 20,
  col_names = def_names("paf"),
  col_types = def_types("paf"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>Either a path to a file, a connection, or literal data
(either a single string or a raw vector).
</p>
<p>Files ending in <code>.gz</code>, <code>.bz2</code>, <code>.xz</code>, or <code>.zip</code> will
be automatically uncompressed. Files starting with <code style="white-space: pre;">⁠http://⁠</code>,
<code style="white-space: pre;">⁠https://⁠</code>, <code style="white-space: pre;">⁠ftp://⁠</code>, or <code style="white-space: pre;">⁠ftps://⁠</code> will be automatically
downloaded. Remote gz files can also be automatically downloaded and
decompressed.
</p>
<p>Literal data is most useful for examples and tests. To be recognised as
literal data, the input must be either wrapped with <code>I()</code>, be a string
containing at least one new line, or be a vector containing at least one
string with a new line.
</p>
<p>Using a value of <code>clipboard()</code> will read from the system clipboard.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_tags</code></td>
<td>
<p>maximum number of optional fields to include</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col_names</code></td>
<td>
<p>column names to use. Defaults to <code>def_names("gff3")</code> (see <code>def_names</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col_types</code></td>
<td>
<p>column types to use. Defaults to <code>def_types("gff3")</code> (see <code>def_types</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional parameters, passed to <code>read_tsv</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Because <code>readr::read_tsv</code> expects a fixed number of columns, but in .paf the
number of optional fields can differ among records, <code>read_paf</code> tries to read
at least as many columns as the longest record has (<code>max_tags</code>). The
resulting warnings for each record with fewer fields of the form "32 columns
expected, only 22 seen" should thus be ignored.
</p>
<p>From the minimap2 manual
</p>
<p>+—-+——–+———————————————————+
|Col |  Type  |                       Description                       |
+—-+——–+———————————————————+
|  1 | string | Query sequence name                                     |
|  2 |  int   | Query sequence length                                   |
|  3 |  int   | Query start coordinate (0-based)                        |
|  4 |  int   | Query end coordinate (0-based)                          |
|  5 |  char  | ‘+’ if query/target on the same strand; ‘-’ if opposite |
|  6 | string | Target sequence name                                    |
|  7 |  int   | Target sequence length                                  |
|  8 |  int   | Target start coordinate on the original strand          |
|  9 |  int   | Target end coordinate on the original strand            |
| 10 |  int   | Number of matching bases in the mapping                 |
| 11 |  int   | Number bases, including gaps, in the mapping            |
| 12 |  int   | Mapping quality (0-255 with 255 for missing)            |
+—-+——–+———————————————————+
</p>
<p>+—-+——+——————————————————-+
|Tag | Type |                      Description                      |
+—-+——+——————————————————-+
| tp |  A   | Type of aln: P/primary, S/secondary and I,i/inversion |
| cm |  i   | Number of minimizers on the chain                     |
| s1 |  i   | Chaining score                                        |
| s2 |  i   | Chaining score of the best secondary chain            |
| NM |  i   | Total number of mismatches and gaps in the alignment  |
| MD |  Z   | To generate the ref sequence in the alignment         |
| AS |  i   | DP alignment score                                    |
| ms |  i   | DP score of the max scoring segment in the alignment  |
| nn |  i   | Number of ambiguous bases in the alignment            |
| ts |  A   | Transcript strand (splice mode only)                  |
| cg |  Z   | CIGAR string (only in PAF)                            |
| cs |  Z   | Difference string                                     |
| dv |  f   | Approximate per-base sequence divergence              |
+—-+——+——————————————————-+
</p>
<p>From https://samtools.github.io/hts-specs/SAMtags.pdf
type may be one of A (character), B (general array), f (real number),
H (hexadecimal array), i (integer), or Z (string).
</p>


<h3>Value</h3>

<p>tibble
</p>


</div>