<div class="container">

<table style="width: 100%;"><tr>
<td>as_sublinks</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute a layout for links linking feats</h2>

<h3>Description</h3>

<p>Reads sublinks connecting feats such as all-vs-all protein blasts into a
tidy dataframe. sublinks need to be associated with an already added feat
track. The sublinks are internally converted into a regular link track by
mapping the feat-based <code>start</code> and <code>end</code> coordinates to coordinates
relative to the sequences underlying the linked feats.
</p>


<h3>Usage</h3>

<pre><code class="language-R">as_sublinks(x, seqs, feats, ..., everything = TRUE)

## S3 method for class 'tbl_df'
as_sublinks(
  x,
  seqs,
  feats,
  ...,
  everything = TRUE,
  transform = c("none", "aa2nuc", "nuc2aa"),
  compute_layout = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>sublink data convertible to a link layout</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>the sequence layout the linked feats map onto.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>feats</code></td>
<td>
<p>the feats the sublinks map onto.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>passed on to <code>layout_seqs()</code>
spaces, i.e. if matching nucleotide-level annotations to protein level
annotations, e.g. genes and protein blast results.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>everything</code></td>
<td>
<p>set to FALSE to drop optional columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transform</code></td>
<td>
<p>use if feats and sublinks are in different coordinate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>compute_layout</code></td>
<td>
<p>set to FALSE to skip layout computation</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The only obligatory columns are <code>feat_id</code> &amp; <code>feat_id2</code>. Also
recognized are <code>start/end</code>, <code>start2/end2</code> and <code>strand</code>.
</p>
<p>Note <code>start</code> and <code>end</code> for every record will be coerced so that <code>start &lt; end</code>. If no <code>strand</code> was provided, <code>strand</code> will be added and set to "+" for
records that initially had <code style="white-space: pre;">⁠start &lt; end == start2 &lt; end2⁠</code> and "-"
otherwise. If <code>strand</code> was provided, <code>start</code> and <code>end</code> will be reorganized to
conform with <code>start &lt; end</code> without any additional effect.
</p>


<h3>Value</h3>

<p>a tbl_df with plot coordinates
</p>


<h3>Methods (by class)</h3>


<ul><li> <p><code>as_sublinks(tbl_df)</code>: Convert a list of tibbles into a link layout
</p>
</li></ul>
</div>