<div class="container">

<table style="width: 100%;"><tr>
<td>add_feats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Add different types of tracks</h2>

<h3>Description</h3>

<p>Add different types of tracks
</p>


<h3>Usage</h3>

<pre><code class="language-R">add_feats(x, ...)

add_links(x, ..., .adjacent_only = TRUE)

add_subfeats(x, ..., .track_id = "genes", .transform = "aa2nuc")

add_sublinks(x, ..., .track_id = "genes", .transform = "aa2nuc")

add_clusters(x, ..., .track_id = "genes")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>object to add the tracks to (e.g. gggenomes, gggenomes_layout)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>named data.frames, i.e. genes=gene_df, snps=snp_df</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.adjacent_only</code></td>
<td>
<p>indicate whether links should be drawn only between vertically adjacent tracks</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.track_id</code></td>
<td>
<p>track_id of the feats that subfeats, sublinks or clusters map to.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.transform</code></td>
<td>
<p>one of "aa2nuc", "none", "nuc2aa"</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>gggenomes object with added features
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>add_feats()</code>: Add feature annotations to sequences
</p>
</li>
<li> <p><code>add_links()</code>: Add links connecting sequences, such as whole-genome
alignment data.
</p>
</li>
<li> <p><code>add_subfeats()</code>: Add features of features, such as gene/protein
domains, blast hits to genes/proteins, etc.
</p>
</li>
<li> <p><code>add_sublinks()</code>: Add links that connect features, such as
protein-protein alignments connecting genes.
</p>
</li>
<li> <p><code>add_clusters()</code>: Add gene clusters or other feature groups. Takes a
data.frame with at least two required columns <code>cluster_id</code> and <code>feat_id</code>. The
data.frame is converted to a link track connecting features belonging to the
same cluster over their entire length. Additionally, the data.frame is joined
to the parent feature track, adding <code>cluster_id</code> and all additional columns
to the parent table.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R"># Add some repeat annotations
gggenomes(seqs = emale_seqs) %&gt;%
  add_feats(repeats = emale_tirs) +
  geom_seq() + geom_feat()

# Add all-vs-all whole-genome alignments
gggenomes(seqs = emale_seqs) %&gt;%
  add_links(links = emale_ava) +
  geom_seq() + geom_link()

# Add domains to genes
genes &lt;- tibble::tibble(seq_id = "A", start = 100, end = 200, feat_id = "gene1")
domains &lt;- tibble::tibble(feat_id = "gene1", start = 40, end = 80)
gggenomes(genes = genes) %&gt;% add_subfeats(domains, .transform = "none") +
  geom_gene() + geom_feat()

# Add protein-protein alignments
gggenomes(emale_genes) %&gt;%
  add_sublinks(emale_prot_ava) +
  geom_gene() + geom_link()

# add clusters
gggenomes(emale_genes, emale_seqs) %&gt;%
  add_clusters(emale_cogs) %&gt;%
  sync() + # works because clusters
  geom_link() + # become links
  geom_seq() +
  # works because cluster info is joined to gene track
  geom_gene(aes(fill = ifelse(is.na(cluster_id), NA,
    stringr::str_glue("{cluster_id} [{cluster_size}]")
  ))) +
  scale_fill_discrete("COGs")

</code></pre>


</div>