<div class="container">

<table style="width: 100%;"><tr>
<td>read_gbk</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read genbank files</h2>

<h3>Description</h3>

<p>Genbank flat files (.gb/.gbk/.gbff) and their ENA and DDBJ equivalents have a
particularly gruesome format. That's why <code>read_gbk()</code> is just a wrapper
around a Perl-based <code>gb2gff</code> converter and <code>read_gff3()</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_gbk(file, sources = NULL, types = NULL, infer_cds_parents = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>Either a path to a file, a connection, or literal data
(either a single string or a raw vector).
</p>
<p>Files ending in <code>.gz</code>, <code>.bz2</code>, <code>.xz</code>, or <code>.zip</code> will
be automatically uncompressed. Files starting with <code style="white-space: pre;">⁠http://⁠</code>,
<code style="white-space: pre;">⁠https://⁠</code>, <code style="white-space: pre;">⁠ftp://⁠</code>, or <code style="white-space: pre;">⁠ftps://⁠</code> will be automatically
downloaded. Remote gz files can also be automatically downloaded and
decompressed.
</p>
<p>Literal data is most useful for examples and tests. To be recognised as
literal data, the input must be either wrapped with <code>I()</code>, be a string
containing at least one new line, or be a vector containing at least one
string with a new line.
</p>
<p>Using a value of <code>clipboard()</code> will read from the system clipboard.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sources</code></td>
<td>
<p>only return features from these sources</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>types</code></td>
<td>
<p>only return features of these types, e.g. gene, CDS, ...</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>infer_cds_parents</code></td>
<td>
<p>infer the mRNA parent for CDS features based on
overlapping coordinates. Default TRUE for gff2/gtf, FALSE for gff3. In most
GFFs this is properly set, but sometimes this information is missing.
Generally, this is not a problem, however, geom_gene calls parse the parent
information to determine which CDS and mRNAs are part of the same gene
model. Without the parent info, mRNA and CDS are plotted as individual
features.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>tibble
</p>


</div>