<div class="container">

<table style="width: 100%;"><tr>
<td>read_tracks</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read files in various standard formats (FASTA, GFF3, GBK, BED, BLAST, ...) into track tables</h2>

<h3>Description</h3>

<p>Convenience functions to read sequences, features or links from various
bioinformatics file formats, such as FASTA, GFF3, Genbank, BLAST tabular
output, etc. See <code>def_formats()</code> for full list. File formats and the
corresponding read-functions are automatically determined based on file
extensions. All these functions can read multiple files in the same format at
once, and combine them into a single table - useful, for example, to read a
folder of gff-files with each file containing genes of a different genome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_feats(files, .id = "file_id", format = NULL, parser = NULL, ...)

read_subfeats(files, .id = "file_id", format = NULL, parser = NULL, ...)

read_links(files, .id = "file_id", format = NULL, parser = NULL, ...)

read_sublinks(files, .id = "file_id", format = NULL, parser = NULL, ...)

read_seqs(
  files,
  .id = "file_id",
  format = NULL,
  parser = NULL,
  parse_desc = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>files</code></td>
<td>
<p>files to reads. Should all be of same format. In many cases,
compressed files (<code>.gz</code>, <code>.bz2</code>, <code>.xz</code>, or <code>.zip</code>) are supported.
Similarly, automatic download of remote files starting with <code style="white-space: pre;">⁠http(s)://⁠</code> or
<code style="white-space: pre;">⁠ftp(s)://⁠</code> works in most cases.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.id</code></td>
<td>
<p>the column with the name of the file a record was read from.
Defaults to "file_id". Set to "bin_id" if every file represents a different
bin.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>specify a format known to gggenomes, such as <code>gff3</code>, <code>gbk</code>, ...
to overwrite automatic determination based on the file extension (see
<code>def_formats()</code> for full list).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parser</code></td>
<td>
<p>specify the name of an R function to overwrite automatic
determination based on format, e.g. <code>parser="read_tsv"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments passed on to the format-specific read
function called down the line.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parse_desc</code></td>
<td>
<p>turn <code style="white-space: pre;">⁠key=some value⁠</code> pairs from <code>seq_desc</code> into <code>key</code>-named
columns and remove them from <code>seq_desc</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A gggenomes-compatible sequence, feature or link tibble
</p>
<p>tibble with features
</p>
<p>tibble with features
</p>
<p>tibble with links
</p>
<p>tibble with links
</p>
<p>tibble with sequence information
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>read_feats()</code>: read files as features mapping onto
sequences.
</p>
</li>
<li> <p><code>read_subfeats()</code>: read files as subfeatures mapping onto other features
</p>
</li>
<li> <p><code>read_links()</code>: read files as links connecting sequences
</p>
</li>
<li> <p><code>read_sublinks()</code>: read files as sublinks connecting features
</p>
</li>
<li> <p><code>read_seqs()</code>: read sequence ID, description and length.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R"># read genes/features from a gff file
read_feats(ex("eden-utr.gff"))


# read all gff files from a directory
read_feats(list.files(ex("emales/"), "*.gff$", full.names = TRUE))


# read remote files

gbk_phages &lt;- c(
  PSSP7 = paste0(
    "ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/",
    "000/858/745/GCF_000858745.1_ViralProj15134/",
    "GCF_000858745.1_ViralProj15134_genomic.gff.gz"
  ),
  PSSP3 = paste0(
    "ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/",
    "000/904/555/GCF_000904555.1_ViralProj195517/",
    "GCF_000904555.1_ViralProj195517_genomic.gff.gz"
  )
)
read_feats(gbk_phages)


# read sequences from a fasta file.
read_seqs(ex("emales/emales.fna"), parse_desc = FALSE)

# read sequence info from a fasta file with `parse_desc=TRUE` (default). `key=value`
# pairs are removed from `seq_desc` and parsed into columns with `key` as name
read_seqs(ex("emales/emales.fna"))

# read sequence info from samtools/seqkit style index
read_seqs(ex("emales/emales.fna.seqkit.fai"))

# read sequence info from multiple gff file
read_seqs(c(ex("emales/emales.gff"), ex("emales/emales-tirs.gff")))

</code></pre>


</div>