<div class="container">

<table style="width: 100%;"><tr>
<td>ggrasp.write</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>ggrasp.write</h2>

<h3>Description</h3>

<p>writes formatted information from a class GGRaSP object to a file. Multiple output options are available.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ggrasp.write(x, file, type, rank.level)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>ggrasp-class object to be written</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>String pointing to file where the data will be saved. If no file is given, the result will be printed out on the screen.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Format of the data printed, either "tree" (New Hampshire extended style), "table" where the medoids or representative are shown in a table format, "list" where the information is shown in a pseudo-fasta format, or "itol" which prints out a file that can be loaded into the itol phylogeny viewer (http://itol.embl.de) which will color the clades of the different clusters</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rank.level</code></td>
<td>
<p>Maximum level of the rank to show. Ignored pre-clustering. After clustering, 0 will show only the medoids, -1 will show all values independent of rank, and any value &gt;= 1 will show all the genomes less than or equal to that rank (including medoids). Default is 0 (only the medoids)</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">#Getting the ggrasp object
Enter.tree &lt;- ggrasp.load(system.file("extdata", "Enter.kSNP.tree", package="ggrasp"), 
file.format = "tree", rank.file =system.file("extdata", "Enter.kSNP.ranks", package="ggrasp"));
Enter.tree.cluster &lt;- ggrasp.cluster(Enter.tree)

#Default examples: using the initizalized ggrasp object will 
#write the newick tree string to "tree.nwk"
ggrasp.write(Enter.tree, type="tree", file=file.path(tempdir(), "tree.nwk"));

# Using the clustered ggrasp object will write a text file with the clusters saved as an ITOL clade
# In conjecture with the phylogeny, this is readable by 
# ITOL web phylogeny visualizer (http://itol.embl.de/) 
ggrasp.write(Enter.tree.cluster, type="itol", file=file.path(tempdir(), "tree.itol.clade.txt"));
</code></pre>


</div>