<div class="container">

<table style="width: 100%;"><tr>
<td>ggrasp.load</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>ggrasp.load</h2>

<h3>Description</h3>

<p>ggrasp.load() initializes a class GGRaSP object from a file containing either a tree, a distance matrix or a multi-fasta alignment. The returned object can subsequently be clustered using ggrasp.cluster().
</p>


<h3>Usage</h3>

<pre><code class="language-R">ggrasp.load(file, file.format, rank.file, offset, tree.method = "complete")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>File containing the tree, matrix or sequence alignment used to initialize the ggrasp object. Required.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file.format</code></td>
<td>
<p>The format the file is in, with tree, fasta and matrix accepted. If not given the program makes a guess.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rank.file</code></td>
<td>
<p>File containing the ranks of genomes in a tab-delineated file with the genome in column 1 and the rank in column 2. The rank is a non-negative number.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>offset</code></td>
<td>
<p>Numeric representing a perfect match. Default is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree.method</code></td>
<td>
<p>The method used to make the tree from a distance matrix. "Complete" (Default), "Average", "Single", and "nj" (Neighbor Joining) are currently available.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a class GGRaSP variable
</p>


<h3>Examples</h3>

<pre><code class="language-R">#The following data is from Chavda et al 2016 which phylotyped Enterobacter genomes
# Our example uses the data underpinning the tree shown in Figure 2
# Also included is a ranking file to prioritize closed Enterobactor genomes

library(ggrasp);
tree.file &lt;- system.file("extdata", "Enter.kSNP.tree", package="ggrasp")
rank.file.in &lt;- system.file("extdata", "Enter.kSNP.ranks", package="ggrasp")
Enter.tree &lt;- ggrasp.load(tree.file, file.format = "tree", rank.file = rank.file.in);

# Other options include loading by fasta file:
fasta.file &lt;- system.file("extdata", "Enter.kSNP2.fasta", package="ggrasp")
rank.file.in &lt;- system.file("extdata", "Enter.kSNP.ranks", package="ggrasp")
Enter.tree &lt;- ggrasp.load(fasta.file, file.format = "fasta", rank.file =rank.file.in)

# and by distance matrix. Since this distance matrix is actually percent identity,
# we will us an offset of 100
mat.file &lt;- system.file("extdata", "Enter.ANI.mat", package="ggrasp")
rank.file.in &lt;- system.file("extdata", "Enter.kSNP.ranks", package="ggrasp")
Enter.in &lt;- ggrasp.load(mat.file, file.format = "matrix", rank.file =rank.file.in, offset = 100)

# Use summary() to examine the data loaded
summary(Enter.in)

#Use plot() to see the tree
plot(Enter.in)
</code></pre>


</div>