<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.loadhaplo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Load haplotype genotype data
</h2>

<h3>Description</h3>

<p>This function loads HapGenotypes generated by <code>ghap.haplotyping</code> and converts them to a native GHap.haplo object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  ghap.loadhaplo(input.file = NULL,
                 hapsamples.file = NULL,
                 hapalleles.file = NULL,
                 hapgenotypesb.file = NULL,
                 verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input.file</code></td>
<td>

<p>Prefix for input files.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hapsamples.file</code></td>
<td>

<p>Individual information file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hapalleles.file</code></td>
<td>

<p>Haplotype alleles information file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hapgenotypesb.file</code></td>
<td>

<p>Binary haplotype genotype matrix file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>A logical value specifying whether log messages should be printed (default = TRUE).
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The returned GHap.haplo object is a list with components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>nsamples</code></td>
<td>

<p>An integer value for the sample size.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nalleles</code></td>
<td>

<p>An integer value for the number of haplotype alleles.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsamples.in</code></td>
<td>

<p>An integer value for the number of active samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nalleles.in</code></td>
<td>

<p>An integer value for the number of active haplotype alleles.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>

<p>A character vector relating samples to populations. This information is obtained from the first column of the hapsamples file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>

<p>A character vector mapping genotypes to samples. This information is obtained from the second column of the hapsamples file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id.in</code></td>
<td>

<p>A logical vector indicating active samples. By default, all samples are set to TRUE.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>

<p>A character vector mapping haplotype alleles to chromosomes. This information is obtained from the second column of the hapalleles file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block</code></td>
<td>

<p>A character vector containing block names. This information is obtained from the first column of the hapalleles file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp1</code></td>
<td>

<p>A numeric vector with haplotype allele start positions. This information is obtained from the third column of the hapalleles file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp2</code></td>
<td>

<p>A numeric vector with haplotype allele end positions. This information is obtained from the fourth column of the hapalleles file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele</code></td>
<td>

<p>A character vector with haplotype allele identity. This information is obtained from the fifth column of the hapalleles file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele.in</code></td>
<td>

<p>A logical vector indicating active haplotype alleles. By default, all alleles are set to TRUE.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypes</code></td>
<td>

<p>A character value giving the pathway to the binary haplotype genotype matrix.
</p>
</td>
</tr>
</table>
<p>The input format is described in <code>ghap.haplotyping</code>.
</p>


<h3>Author(s)</h3>

<p>Yuri Tani Utsunomiya &lt;ytutsunomiya@gmail.com&gt; <br>
Marco Milanesi &lt;marco.milanesi.mm@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# #### DO NOT RUN IF NOT NECESSARY ###
# 
# # Copy phase data in the current working directory
# exfiles &lt;- ghap.makefile(dataset = "example",
#                          format = "phase",
#                          verbose = TRUE)
# file.copy(from = exfiles, to = "./")
# 
# # Load data
# phase &lt;- ghap.loadphase("example")
# 
# # Generate blocks of 5 markers sliding 5 markers at a time
# blocks &lt;- ghap.blockgen(phase, windowsize = 5,
#                         slide = 5, unit = "marker")
# 
# # Haplotyping
# ghap.haplotyping(phase = phase, blocks = blocks, outfile = "example",
#                  binary = T, ncores = 1)
# 
# ### RUN ###
# 
# # Load haplotype genotypes using prefix
# haplo &lt;- ghap.loadhaplo("example")
# 
# # Load haplotype genotypes using file names
# haplo &lt;- ghap.loadhaplo(hapsamples.file = "example.hapsamples",
#                         hapalleles.file = "example.hapalleles",
#                         hapgenotypesb.file = "example.hapgenotypesb")

</code></pre>


</div>