<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.inbcoef</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Compute measures of inbreeding
</h2>

<h3>Description</h3>

<p>This function computes genomic measures of inbreeding.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ghap.inbcoef(object, freq, batchsize=NULL,
             only.active.samples=TRUE,
             only.active.variants=TRUE,
             ncores=1, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A valid GHap object (phase, haplo or plink).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>

<p>A named numeric vector providing allele frequencies.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batchsize</code></td>
<td>

<p>A numeric value controlling the number of variants to be processed at a time (default = nalleles/10).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only.active.samples</code></td>
<td>

<p>A logical value specifying whether only active samples should be included in the output (default = TRUE).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only.active.variants</code></td>
<td>

<p>A logical value specifying whether only active variants should be included in the output (default = TRUE).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>

<p>A numeric value specifying the number of cores to be used in parallel computations (default = 1).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>A logical value specfying whether log messages should be printed (default = TRUE).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The inbreeding measures are computed as k + s*sum(f)/d, where k is a constant, s is a sign shifting scalar, f is a per-variant function and d is a scaling denominator. Four different measures of inbreeding are currently available:
</p>
<p>Type = 1 (based on genomic relationship)<br>
k = -1<br>
s = 1<br>
f = ((m - 2*p)^2)/(2*p*(1-p))<br>
d = n<br></p>
<p>Type = 2 (excess homozygosity)<br>
k = 1<br>
s = -1<br>
f = m*(2-m)/(2*p*(1-p))<br>
d = n<br></p>
<p>Type = 3 (correlation between uniting gametes)<br>
k = 0<br>
s = 1<br>
f = (m^2 - (1+2*p) + 2*p^2)/(2*p*(1-p))<br>
d = n<br></p>
<p>Type = 4 (method-of-moments)<br>
k = 1<br>
s = -1<br>
f = length(het)<br>
d = sum(2*p*(1-p))<br></p>
<p>In the expressions above, m is the genotype coded as 0, 1 or 2 copies of A1, p is the frequency of A1, n is the number of variants (only non-monomorphic ones are considered), and het is the number of heterozygous genotypes.
</p>


<h3>Value</h3>

<p>The function returns a dataframe containing population name, id and inbreeding measures for each individual.
</p>


<h3>Author(s)</h3>

<p>Yuri Tani Utsunomiya &lt;ytutsunomiya@gmail.com&gt;
</p>


<h3>References</h3>

<p>J. Yang et al. GCTA: A Tool for Genome-wide Complex Trait Analysis. Am. J. Hum. Genet. 2011. 88:76â€“82.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# #### DO NOT RUN IF NOT NECESSARY ###
# 
# # Copy plink data in the current working directory
# exfiles &lt;- ghap.makefile(dataset = "example",
#                          format = "plink",
#                          verbose = TRUE)
# file.copy(from = exfiles, to = "./")
# 
# # Load plink data
# plink &lt;- ghap.loadplink("example")
# 
# ### RUN ###
# 
# # Subset individuals from the pure1 population
# pure1 &lt;- plink$id[which(plink$pop == "Pure1")]
# plink &lt;- ghap.subset(object = plink, ids = pure1, variants = plink$marker)
# 
# # Subset markers with MAF &gt; 0.05
# freq &lt;- ghap.freq(plink)
# mkr &lt;- names(freq)[which(freq &gt; 0.05)]
# plink &lt;- ghap.subset(object = plink, ids = pure1, variants = mkr)
# 
# # Compute A1 allele frequencies
# p &lt;- ghap.freq(plink, type = "A1")
# 
# # Compute inbreeding coefficients
# ibc &lt;- ghap.inbcoef(object = plink, freq = p)

</code></pre>


</div>