<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.slice</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Get a slice of a GHap object
</h2>

<h3>Description</h3>

<p>This function parses a binary PLINK, phased or HapGenotypes matrix and returns the slice as an R matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ghap.slice(object, ids, variants, index=FALSE,
           transposed=FALSE, sparse=TRUE,
           unphase=FALSE, impute=FALSE,
           ncores=1, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A valid GHap object (phase, haplo or plink).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>

<p>A character or numeric vector indicating individuals to parse.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variants</code></td>
<td>

<p>A character or numeric vector indicating variants to parse. If a "GHap.haplo" object is provided, the vector must be numeric.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index</code></td>
<td>

<p>A logical value specfying if values provided for ids and variants are indices (see details).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transposed</code></td>
<td>

<p>A logical value specfying if genotypes should be transposed. If FALSE (default), the matrix is returned as variants by individuals. Otherwise, the retrieved matrix is organized as individuals by variants.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sparse</code></td>
<td>

<p>A logical value specfying if the returned matrix should be formatted as a sparse matrix (default TRUE).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unphase</code></td>
<td>

<p>A logical value specfying if phased genotypes should be retrieved as unphased allele counts. Only meaningful for "GHap.phase" objects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute</code></td>
<td>

<p>A logical value specfying if missing genotypes should be replaced by 0 (i.e., A0/A0 genotypes, default = FALSE). Only meaningful for "GHap.plink" objects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>

<p>A numeric value specifying the number of cores to be used in parallel computations (default = 1).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>A logical value specfying whether log messages should be printed (default = TRUE).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function parses the binary input file and returns an R matrix with the requested list of variants (markers or alleles) and individuals. The argument index allows the user to specify individuals either by name (index = FALSE) or by indices as stored in the GHap object (index = TRUE). In the case of "GHap.haplo" objects, HapAlleles can only by parsed via indices.
</p>


<h3>Value</h3>

<p>An R matrix with variants in rows and individuals in columns (this is inverted if transposed = TRUE).
</p>


<h3>Author(s)</h3>

<p>Yuri Tani Utsunomiya &lt;ytutsunomiya@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# #### DO NOT RUN IF NOT NECESSARY ###
# 
# # Copy phase data in the current working directory
# exfiles &lt;- ghap.makefile(dataset = "example",
#                          format = "phase",
#                          verbose = TRUE)
# file.copy(from = exfiles, to = "./")
# 
# # Load data
# phase &lt;- ghap.loadphase("example")
# 
# ### RUN ###
# 
# # Select random individuals and markers
# ind &lt;- sample(x = unique(phase$id), size = 5)
# mkr &lt;- sample(x = phase$marker, size = 10)
# 
# # Generate slice of the data
# ghap.slice(object = phase, ids = ind, variants = mkr)
# 
# # Import as unphased data
# ghap.slice(object = phase, ids = ind, variants = mkr,
#            unphase = TRUE)
# 
# # Return transposed matrix
# ghap.slice(object = phase, ids = ind, variants = mkr,
#            unphase = TRUE, transposed = TRUE)
# 
# # Display data as non-sparse matrix
# ghap.slice(object = phase, ids = ind, variants = mkr,
#            unphase = TRUE, transposed = TRUE,
#            sparse = FALSE)

</code></pre>


</div>