<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.fast2phase</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Convert fastPHASE data into the GHap phase format
</h2>

<h3>Description</h3>

<p>This function takes phased genotype data in fastPHASE format and converts them into a GHap plan and compressed binary format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  ghap.fast2phase(input.files = NULL, switchout.files = NULL,
                  map.files = NULL, fam.file = NULL,
                  out.file = NULL, overwrite = FALSE,
                  ncores = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<p>If all input files share the same prefix, the user can use the following shortcut options:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>input.files</code></td>
<td>

<p>Character vector with the list of prefixes for input files.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.file</code></td>
<td>

<p>Character value for the output file name.
</p>
</td>
</tr>
</table>
<p>The user can also opt to point to input files separately:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>switchout.files</code></td>
<td>

<p>Character vector containing the list of fastPHASE files.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map.files</code></td>
<td>

<p>Character vector containing the list of map files.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fam.file</code></td>
<td>

<p>Name of the file containing the population and individual ids.
</p>
</td>
</tr>
</table>
<p>The function avoids overwritting existing files by default. To change this behavior, please use:
</p>
<table><tr style="vertical-align: top;">
<td><code>overwrite</code></td>
<td>

<p>A logical value controling if existing files with the same name as the selected output should be overwritten (default = FALSE).
</p>
</td>
</tr></table>
<p>To turn conversion progress-tracking on or off or set the number of cores please use:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>

<p>A numeric value specfying the number of cores (default = 1).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>A logical value specfying whether log messages should be printed (default = TRUE).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Currently this function handles _switch.out files from fastPHASE v1.4.0 or later. The map files should contain the following 5 space-delimited columns: chromosome, marker, position, allele 0 and allele 1 (no header). The fam file can have an arbitrary number of columns (with no header), but by default only the first two are read and should be space-delimited with the following data: population and individual name.
</p>


<h3>Author(s)</h3>

<p>Mario Barbato &lt;mario.barbato@unicatt.it&gt;
</p>


<h3>References</h3>

<p>Scheet, P., Stephens, M., 2006. A Fast and Flexible Statistical Model for Large-Scale Population Genotype Data: Applications to Inferring Missing Genotypes and Haplotypic Phase. Am. J. Hum. Genet. 78, 629-644. https://doi.org/10.1086/502802.
</p>


<h3>See Also</h3>

<p><code>ghap.compress</code>, <code>ghap.loadphase</code>, <code>ghap.oxford2phase</code>, <code>ghap.vcf2phase</code>
</p>


</div>