<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.vcf2phase</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Convert VCF data into GHap phase
</h2>

<h3>Description</h3>

<p>This function takes phased genotype data in the Variant Call Format (VCF) and converts them into the GHap phase format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  ghap.vcf2phase(input.files = NULL, vcf.files = NULL,
                 sample.files = NULL, out.file,
                 ncores = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<p>If all input files share the same prefix, the user can use the following shortcut options:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>input.files</code></td>
<td>

<p>Character vector with the list of prefixes for input files.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.file</code></td>
<td>

<p>Character value for the output file name.
</p>
</td>
</tr>
</table>
<p>The user can also opt to point to input files separately:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>vcf.files</code></td>
<td>

<p>Character vector containing the list of VCF files.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.files</code></td>
<td>

<p>Character vector containing the list of SAMPLE files.
</p>
</td>
</tr>
</table>
<p>To turn conversion progress-tracking on or off or set the number of cores please use:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>

<p>A numeric value specfying the number of cores to use (default = 1).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>A logical value specfying whether log messages should be printed (default = TRUE).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The Variant Call Format (VCF) - as described in https://github.com/samtools/hts-specs - is here manipulated to obtain the GHap phase format. Important: the function does not apply filters to the data, except for skipping multi-allelic variants. Should variants be filtered, the user is advised to pre-process the VCF files with third-party software (such as BCFTools). The FORMAT field should also follow the "GT:..." specification, with genotypes placed first in each sample column. Finally, all genotypes should be phased and take one of the following values: "0|0", "0|1", "1|0" or "1|1".
Warning: this function is not optimized for very large datasets.
</p>


<h3>Author(s)</h3>

<p>Yuri Tani Utsunomiya &lt;ytutsunomiya@gmail.com&gt;
</p>


<h3>References</h3>

<p>H. Li et al. The Sequence alignment/map (SAM) format and SAMtools. Bioinformatics. 2009. 25:2078-2079.
</p>
<p>H. Li. A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. Bioinformatics. 2011. 27(21):2987-2993.
</p>


<h3>See Also</h3>

<p><code>ghap.compress</code>, <code>ghap.loadphase</code>, <code>ghap.fast2phase</code>, <code>ghap.oxford2phase</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# #### DO NOT RUN IF NOT NECESSARY ###
# 
# # Copy the example data in the current working directory
# exfiles &lt;- ghap.makefile(dataset = "example",
#                          format = "vcf",
#                          verbose = TRUE)
# file.copy(from = exfiles, to = "./")
# 
# ### RUN ###
# 
# # Convert from a single genome-wide file
# ghap.vcf2phase(input.files = "example",
#                out.file = "example")
# 
# # Convert from a list of chromosome files
# ghap.vcf2phase(input.files = paste0("example_chr",1:10),
#                out.file = "example")
# 
# # Convert using separate lists for file extensions
# ghap.vcf2phase(vcf.files = paste0("example_chr",1:10,".vcf"),
#                sample.files = paste0("example_chr",1:10,".sample"),
#                out.file = "example")

</code></pre>


</div>