<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.haplotyping</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Haplotype genotypes
</h2>

<h3>Description</h3>

<p>Generate matrix of HapGenotypes for user-defined blocks.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ghap.haplotyping(object, blocks, outfile,
                 freq = c(0, 1), drop.minor = FALSE,
                 only.active.samples = TRUE,
                 only.active.markers = TRUE,
                 batchsize = NULL, binary = TRUE,
                 ncores = 1, verbose = TRUE)

</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A GHap.phase object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blocks</code></td>
<td>

<p>A data frame containing block boundaries, such as supplied by the <code>ghap.blockgen</code> function.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>

<p>A character value specifying the name for the output files.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>

<p>A numeric vector of length 2 specifying the range of haplotype allele frequency to be included in the output. Default is c(0,1), which includes all alleles.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drop.minor</code></td>
<td>

<p>A logical value specfying whether the minor allele should be excluded from the output (default = FALSE).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only.active.samples</code></td>
<td>

<p>A logical value specifying whether only active samples should be included in the output (default = TRUE).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only.active.markers</code></td>
<td>

<p>A logical value specifying whether only active markers should be used for haplotyping (default = TRUE).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batchsize</code></td>
<td>

<p>A numeric value controlling the number of haplotype blocks to be processed and written to output at a time (default = nblocks/10).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>binary</code></td>
<td>

<p>A logical value specfying whether the output file should be binary (default = TRUE).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>

<p>A numeric value specifying the number of cores to be used in parallel computations (default = 1).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>A logical value specfying whether log messages should be printed (default = TRUE).
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The function outputs three files with suffix:
</p>

<ul>
<li> <p><strong>.hapsamples</strong>: space-delimited file without header containing two columns: Population and Individual ID.
</p>
</li>
<li> <p><strong>.hapalleles</strong>: space-delimited file without header containing five columns: Block Name, Chromosome, Start and End Position (in bp), and HapAllele.
</p>
</li>
<li> <p><strong>.hapgenotypes</strong>: if binary = FALSE, a space-delimited file without header containing the HapGenotype matrix (coded as 0, 1 or 2 copies of the HapAllele). The dimension of the matrix is <em>m x n</em>, where <em>m</em> is the number of HapAlleles and <em>n</em> is the number of individuals.
</p>
</li>
<li> <p><strong>.hapgenotypesb</strong>: if binary = TRUE (default), the same matrix as described above compressed into bits. For seamless compatibility with softwares that use PLINK binary files, the compression is performed using the SNP-major bed format.
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Yuri Tani Utsunomiya &lt;ytutsunomiya@gmail.com&gt;
</p>
<p>Marco Milanesi &lt;marco.milanesi.mm@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# #### DO NOT RUN IF NOT NECESSARY ###
# 
# # Copy phase data in the current working directory
# exfiles &lt;- ghap.makefile(dataset = "example",
#                          format = "phase",
#                          verbose = TRUE)
# file.copy(from = exfiles, to = "./")
# 
# # Load data
# phase &lt;- ghap.loadphase("example")
# 
# ### RUN ###
# 
# # Generate blocks
# blocks &lt;- ghap.blockgen(phase, windowsize = 5,
#                         slide = 5, unit = "marker")
# 
# # Haplotyping
# ghap.haplotyping(phase, blocks = blocks,
#                  outfile = "example",
#                  binary = T, ncores = 1)

</code></pre>


</div>