<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.oxford2phase</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Convert Oxford data into GHap phase
</h2>

<h3>Description</h3>

<p>This function takes phased genotype data in Oxford HAPS/SAMPLES format and converts them into the GHap phase format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  ghap.oxford2phase(input.files = NULL, haps.files = NULL,
                    sample.files = NULL, out.file,
                    ncores = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<p>If all input files share the same prefix, the user can use the following shortcut options:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>input.files</code></td>
<td>

<p>Character vector with the list of prefixes for input files.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.file</code></td>
<td>

<p>Character value for the output file name.
</p>
</td>
</tr>
</table>
<p>The user can also opt to point to input files separately:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>haps.files</code></td>
<td>

<p>Character vector containing the list of Oxford HAPS files.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.files</code></td>
<td>

<p>Character vector containing the list of Oxford SAMPLES files.
</p>
</td>
</tr>
</table>
<p>To turn conversion progress-tracking on or off or use multiple cores please use:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>

<p>A numeric value specfying the number of cores to use (default = 1).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>A logical value specfying whether log messages should be printed (default = TRUE).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The Oxford HAPS/SAMPLE format output of widely used phasing software such as SHAPEIT2 (O'Connell et al., 2014) or Eagle (Loh et al., 2106) is here manipulated to obtain the GHap phase format.
</p>


<h3>Author(s)</h3>

<p>Mario Barbato &lt;mario.barbato@unicatt.it&gt;, Yuri Tani Utsunomiya &lt;ytutsunomiya@gmail.com&gt;
</p>


<h3>References</h3>

<p>R-R. Loh P-R et al. Reference-based phasing using the Haplotype Reference Consortium panel. Nat Genet. 2016. 48(11):1443-1448.
J. O'Connell et al. A general approach for haplotype phasing across the full spectrum of relatedness. PLOS Genet. 2014. 10:e1004234.
</p>


<h3>See Also</h3>

<p><code>ghap.compress</code>, <code>ghap.loadphase</code>, <code>ghap.fast2phase</code>, <code>ghap.vcf2phase</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# #### DO NOT RUN IF NOT NECESSARY ###
# 
# # Copy the example data in the current working directory
# exfiles &lt;- ghap.makefile(dataset = "example",
#                          format = "oxford",
#                          verbose = TRUE)
# file.copy(from = exfiles, to = "./")
# 
# ### RUN ###
# 
# # Convert from a single genome-wide file
# ghap.oxford2phase(input.files = "example",
#                   out.file = "example")
# 
# # Convert from a list of chromosome files
# ghap.oxford2phase(input.files = paste0("example_chr",1:10),
#                   out.file = "example")
# 
# # Convert using separate lists for file extensions
# ghap.oxford2phase(haps.files = paste0("example_chr",1:10,".haps"),
#                   sample.files = paste0("example_chr",1:10,".sample"),
#                   out.file = "example")
# 
# # A more efficient alternative for *nix system users
# # Note: replace "cat" by "zcat" if files are gzipped
# haps.files = paste("example_chr",1:10,".haps",sep="")
# command &lt;- "tail -n+3 example_chr1.sample | cut -d' ' -f1,2 &gt; example.samples"
# system(command)
# for(i in 1:10){
#   command &lt;- paste("cat",haps.files[i],"| cut -d' ' -f1-5 &gt;&gt; example.markers")
#   system(command)
#   command &lt;- paste("cat",haps.files[i],"| cut -d' ' -f1-5 --complement &gt;&gt; example.phase")
#   system(command)
# }

</code></pre>


</div>