<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.blockstats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
HapBlock statistics
</h2>

<h3>Description</h3>

<p>Generate HapBlock summary statistics from pre-computed HapAlleles statistics.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ghap.blockstats(hapstats, ncores = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hapstats</code></td>
<td>

<p>A data.frame containing HapAllele statistics, as generated by the <code>ghap.hapstats</code> function.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>

<p>A numeric value specifying the number of cores to be used in parallel computing (default = 1).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>A logical value specfying whether log messages should be printed (default = TRUE).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For each HapBlock, the function counts the number of unique HapAlleles and computes the expected heterozygosity <code class="reqn"> 1 - \sum {p_{i}}^2 </code>, where <code class="reqn">p_{i}</code> is the frequency of HapAllele <em>i</em>. Please notice that when HapAlleles are prunned out by frequency the block statistics can retrieve high expected heterozygosity for blocks with small number of HapAlleles.
</p>


<h3>Value</h3>

<p>A data frame with columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>BLOCK</code></td>
<td>

<p>Block alias.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CHR</code></td>
<td>

<p>Chromosome name.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BP1</code></td>
<td>

<p>Block start position.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BP2</code></td>
<td>

<p>Block end position.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EXP.H</code></td>
<td>

<p>Block expected heterozygosity.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N.ALLELES</code></td>
<td>

<p>Number of HapAlleles per block.
</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Yuri Tani Utsunomiya &lt;ytutsunomiya@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# #### DO NOT RUN IF NOT NECESSARY ###
# 
# # Copy phase data in the current working directory
# exfiles &lt;- ghap.makefile(dataset = "example",
#                          format = "phase",
#                          verbose = TRUE)
# file.copy(from = exfiles, to = "./")
# 
# # Load data
# phase &lt;- ghap.loadphase("example")
# 
# # Generate blocks of 5 markers
# blocks &lt;- ghap.blockgen(phase, windowsize = 5,
#                         slide = 5, unit = "marker")
# 
# # Haplotyping
# ghap.haplotyping(phase = phase, blocks = blocks, outfile = "example",
#                  binary = T, ncores = 1)
# 
# # Load haplotype genotypes using prefix
# haplo &lt;- ghap.loadhaplo("example")
# 
# # Subset
# ids &lt;- which(haplo$pop == "Pure1")
# haplo &lt;- ghap.subset(haplo, ids = ids,
#                      variants = haplo$allele.in,
#                      index = TRUE)
# 
# # Compute haplotype statistics
# hapstats &lt;- ghap.hapstats(haplo)
# 
# ### RUN ###
#
# # Compute block statistics
# blockstats &lt;- ghap.blockstats(hapstats)

</code></pre>


</div>