<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.hapstats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Haplotype allele statistics
</h2>

<h3>Description</h3>

<p>Summary statistics for HapAlleles.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ghap.hapstats(object,
              alpha = c(1, 1),
              batchsize = NULL,
              only.active.samples = TRUE,
              only.active.alleles = TRUE,
              ncores = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A GHap.haplo object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>

<p>A numeric vector of size 2 specifying the shrinkage parameters for the expected-to-observed homozygotes ratio. Default is c(1, 1).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batchsize</code></td>
<td>

<p>A numeric value controlling the number of HapAlleles to be processed at a time (default = nalleles/10).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only.active.samples</code></td>
<td>

<p>A logical value specifying whether only active samples should be included in the output (default = TRUE).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only.active.alleles</code></td>
<td>

<p>A logical value specifying whether only active haplotype alleles should be included in the output (default = TRUE).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>

<p>A numeric value specifying the number of cores to be used in parallel computations (default = 1).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>A logical value specfying whether log messages should be printed (default = TRUE).
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame with columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>BLOCK</code></td>
<td>

<p>Block alias.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CHR</code></td>
<td>

<p>Chromosome name.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BP1</code></td>
<td>

<p>Block start position.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BP2</code></td>
<td>

<p>Block end position.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ALLELE</code></td>
<td>

<p>Haplotype allele identity.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>

<p>Number of observations for the haplotype.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FREQ</code></td>
<td>

<p>Haplotype frequency.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>O.HOM</code></td>
<td>

<p>Observed number of homozygotes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>O.HET</code></td>
<td>

<p>Observed number of heterozygotes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>E.HOM</code></td>
<td>

<p>Expected number of homozygotes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>RATIO</code></td>
<td>

<p>Shrinkage expected-to-observed ratio for the number of homozygotes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BIN.logP</code></td>
<td>

<p>log10(1/P) or -log10(P) for Hardy-Weinberg equilibrium assuming number of homozygotes follows a Binomial distribution.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>POI.logP</code></td>
<td>

<p>log10(1/P) or -log10(P) for Hardy-Weinberg equilibrium assuming number of homozygotes follows a Poisson distribution.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TYPE</code></td>
<td>

<p>Category of the HapAllele: "SINGLETON" = single allele of its block; "ABSENT" = the frequency of the allele is 0; "MINOR" = the least frequent allele of its block (in the case of ties, only the first allele is marked); "MAJOR" = the most frequent allele of its block (ties are also resolved by marking the first allele); "REGULAR" = the allele does not fall in any of the previous categories. Categories "SINGLETON", "MINOR" and "MAJOR" only apply for blocks where frequencies sum to 1.
</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Yuri Tani Utsunomiya &lt;ytutsunomiya@gmail.com&gt; <br>
Marco Milanesi &lt;marco.milanesi.mm@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# #### DO NOT RUN IF NOT NECESSARY ###
# 
# # Copy phase data in the current working directory
# exfiles &lt;- ghap.makefile(dataset = "example",
#                          format = "phase",
#                          verbose = TRUE)
# file.copy(from = exfiles, to = "./")
# 
# ### RUN ###
# 
# # Load data
# phase &lt;- ghap.loadphase("example")
# 
# # Generate blocks of 5 markers
# blocks &lt;- ghap.blockgen(phase, windowsize = 5,
#                         slide = 5, unit = "marker")
# 
# # Haplotyping
# ghap.haplotyping(phase = phase, blocks = blocks, outfile = "example",
#                  binary = T, ncores = 1)
# 
# # Load haplotype genotypes using prefix
# haplo &lt;- ghap.loadhaplo("example")
# 
# ### RUN ###
# 
# # Subset
# ids &lt;- which(haplo$pop == "Pure1")
# haplo &lt;- ghap.subset(haplo, ids = ids,
#                      variants = haplo$allele.in,
#                      index = TRUE)
# 
# # Compute haplotype statistics
# hapstats &lt;- ghap.hapstats(haplo)

</code></pre>


</div>