<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.loadphase</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Load binary phased genotype data
</h2>

<h3>Description</h3>

<p>This function loads binary phased genotype data and converts them into a native GHap.phase object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  ghap.loadphase(input.file = NULL,
                 samples.file = NULL,
                 markers.file = NULL,
                 phaseb.file = NULL,
                 ncores = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<p>If all input files share the same prefix, the user can use the following shortcut option:
</p>
<table><tr style="vertical-align: top;">
<td><code>input.file</code></td>
<td>

<p>Prefix for input files.
</p>
</td>
</tr></table>
<p>For backward compatibility, the user can still point to input files separately:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>samples.file</code></td>
<td>

<p>Individual information.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markers.file</code></td>
<td>

<p>Variant map information.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phaseb.file</code></td>
<td>

<p>Binary phased genotype matrix, such as supplied by the <code>ghap.compress</code> function.
</p>
</td>
</tr>
</table>
<p>To turn loading progress-tracking on or off, or use multiple cores, please use:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>

<p>A numerical value specfying the number of cores to use while loading the input files (default = 1).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>A logical value specfying whether log messages should be printed (default = TRUE).
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The returned GHap.phase object is a list with components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>nsamples</code></td>
<td>

<p>An integer value for the sample size.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nmarkers</code></td>
<td>

<p>An integer value for the number of markers.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsamples.in</code></td>
<td>

<p>An integer value for the number of active samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nmarkers.in</code></td>
<td>

<p>An integer value for the number of active markers.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>

<p>A character vector relating chromosome alleles to populations. This information is obtained from the first column of the sample file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>

<p>A character vector mapping chromosome alleles to samples. This information is obtained from the second column of the sample file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id.in</code></td>
<td>

<p>A logical vector indicating active chromosome alleles. By default, all chromosomes are set to TRUE.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sire</code></td>
<td>

<p>A character vector indicating sire names, as provided in the third column of the sample file (optional).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dam</code></td>
<td>

<p>A character vector indicating dam names, as provided in the fourth column of the sample file (optional).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex</code></td>
<td>

<p>A character vector indicating individual sex, as provided in the fifth column of the sample file (optional). Codes are converted as follows: 0 = NA, 1 = Male and 2 = Female.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>

<p>A character vector indicating chromosome identity for each marker.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker</code></td>
<td>

<p>A character vector containing marker names. This information is obtained from the second column of the marker map file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker.in</code></td>
<td>

<p>A logical vector indicating active markers. By default, all markers are set to TRUE.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cm</code></td>
<td>

<p>A numeric vector with genetic positions for markers. This information is obtained from the third column of the marker map file if it contains 6 columns. Otherwise, if the map file contains only 5 columns, genetic positions are considered absent and approximated from physical positions (in this case assumed to be the third column) as 1 Mb ~ 1 cM.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp</code></td>
<td>

<p>A numeric vector with marker positions. This information is obtained from the third column of the marker map file if it contains 5 columns, or from the fourth column if it contains 6 columns.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>A0</code></td>
<td>

<p>A character vector with reference alleles. This information is obtained from the fourth column of the marker map file in case it contains 5 columns, or from the fifth column if it contains 6 columns.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>A1</code></td>
<td>

<p>A character vector with alternative alleles. This information is obtained from the last column of the marker map file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phase</code></td>
<td>

<p>A character value giving the pathway to the binary phased genotype matrix.
</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Yuri Tani Utsunomiya &lt;ytutsunomiya@gmail.com&gt; <br>
Marco Milanesi &lt;marco.milanesi.mm@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# #### DO NOT RUN IF NOT NECESSARY ###
# 
# # Copy the example data in the current working directory
# exfiles &lt;- ghap.makefile(dataset = "example",
#                          format = "phase",
#                          verbose = TRUE)
# file.copy(from = exfiles, to = "./")
# 
# ### RUN ###
# 
# # Load data using prefix
# phase &lt;- ghap.loadphase(input.file = "example")
# 
# # Load data using file names
# phase &lt;- ghap.loadphase(samples.file = "example.samples",
#                         markers.file = "example.markers",
#                         phaseb.file = "example.phaseb")

</code></pre>


</div>