<div class="container">

<table style="width: 100%;"><tr>
<td>ghap.fst</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Haplotype-based Fst
</h2>

<h3>Description</h3>

<p>Multi-alleleic Fst computed using block summary statistics generated from <code>ghap.blockstats</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R"> ghap.fst(blockstats.pop1,
          blockstats.pop2,
          blockstats.tot)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>blockstats.pop1</code></td>
<td>

<p>A data.frame containing block statistics computed on population 1.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blockstats.pop2</code></td>
<td>

<p>A data.frame containing block statistics computed on population 2.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blockstats.tot</code></td>
<td>

<p>A data.frame containing block statistics computed on population 1 + population 2.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calculates Fst (Nei, 1973) based on the formula for multi-allelic markers:
</p>
<p style="text-align: center;"><code class="reqn">Fst = (Ht - Hs) / Ht</code>
</p>

<p>where <em>Ht</em> is the total gene diversity (i.e., expected heterozygosity in the population) and <em>Hs</em> is the subpopulation gene diversity (i.e., the average expected heterozygosity in the subpopulations).
</p>


<h3>Value</h3>

<p>The function returns a data.frame with the following columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>BLOCK</code></td>
<td>

<p>Block alias.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CHR</code></td>
<td>

<p>Chromosome name.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BP1</code></td>
<td>

<p>Block start position.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BP2</code></td>
<td>

<p>Block end position.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EXP.H.pop1</code></td>
<td>

<p>Expected heterozygosity in population 1.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EXP.H.pop2</code></td>
<td>

<p>Expected heterozygosity in population 2.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EXP.H.tot</code></td>
<td>

<p>Expected heterozygosity in the total population.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FST</code></td>
<td>

<p>Fst value.
</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Yuri Tani Utsunomiya &lt;ytutsunomiya@gmail.com&gt; <br>
Marco Milanesi &lt;marco.milanesi.mm@gmail.com&gt;
</p>


<h3>References</h3>

<p>M. Nei. Analysis of Gene Diversity in Subdivided Populations. PNAS. 1973. 70, 3321-3323.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# #### DO NOT RUN IF NOT NECESSARY ###
# 
# # Copy phase data in the current working directory
# exfiles &lt;- ghap.makefile(dataset = "example",
#                          format = "phase",
#                          verbose = TRUE)
# file.copy(from = exfiles, to = "./")
# 
# # Load data
# phase &lt;- ghap.loadphase("example")
# 
# # Generate blocks of 5 markers
# blocks &lt;- ghap.blockgen(phase, windowsize = 5,
#                         slide = 5, unit = "marker")
# 
# # Haplotyping
# ghap.haplotyping(phase = phase, blocks = blocks, outfile = "example",
#                  binary = T, ncores = 1)
# 
# # Load haplotype genotypes using prefix
# haplo &lt;- ghap.loadhaplo("example")
# 
# ### RUN ###
# 
# # Compute block statistics for population 1
# ids &lt;- which(haplo$pop == "Pure1")
# haplo &lt;- ghap.subset(haplo, ids = ids,
#                      variants = haplo$allele.in,
#                      index = TRUE)
# hapstats1 &lt;- ghap.hapstats(haplo)
# blockstats1 &lt;- ghap.blockstats(hapstats1)
# 
# # Compute block statistics for population 2
# ids &lt;- which(haplo$pop == "Pure2")
# haplo &lt;- ghap.subset(haplo, ids = ids,
#                      variants = haplo$allele.in,
#                      index = TRUE)
# hapstats2 &lt;- ghap.hapstats(haplo)
# blockstats2 &lt;- ghap.blockstats(hapstats2)
# 
# # Compute block statistics for combined populations
# ids &lt;- which(haplo$pop %in% c("Pure1","Pure2"))
# haplo &lt;- ghap.subset(haplo, ids = ids,
#                      variants = haplo$allele.in,
#                      index = TRUE)
# hapstats12 &lt;- ghap.hapstats(haplo)
# blockstats12 &lt;- ghap.blockstats(hapstats12)
# 
# # Compute FST
# fst &lt;- ghap.fst(blockstats1, blockstats2, blockstats12)
# ghap.manhattan(data = fst, chr = "CHR", bp = "BP1",
#                y = "FST", type = "h")

</code></pre>


</div>