<div class="container">

<table style="width: 100%;"><tr>
<td>GhostKnockoff.fit</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Feature importance score generator
</h2>

<h3>Description</h3>

<p>Generate original and knockoff feature importance scores given original Z-scores from multiple studies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GhostKnockoff.fit(Zscore_0,n.study,fit.prelim,gamma=1,weight.study=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Zscore_0</code></td>
<td>

<p>A p*K Z-score matrix, where p is the number of variants and K is the number of studies. Variants not observed in the study should be coded as NA.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.study</code></td>
<td>

<p>A vector of length K, where each element is the study sample size.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fit.prelim</code></td>
<td>

<p>The output of function "GhostKnockoff.prelim".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gamma</code></td>
<td>

<p>The estimated study dependency. See functon "GhostKnockoff.GetGamma".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weight.study</code></td>
<td>

<p>The weights to combine multiple studies. The default is based on sample size. The optimal weights can be estimated using function "GhostKnockoff.prelim.Meta".
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>GK.Zscore_0</code></td>
<td>

<p>A vector of length p, where each element is weighted combination of original Z-scores
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GK.Zscore_k</code></td>
<td>

<p>A p*M matrix, where each column is a knockoff copy of GK.Zscore_0.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>T_0</code></td>
<td>

<p>Feature importance score of original data: square of GK.Zscore_0.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>T_k</code></td>
<td>

<p>Feature importance score of knockoff copies: square of GK.Zscore_k.
</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">
# We use genetic data as an example
library(GhostKnockoff)

# load example vcf file from package "seqminer", this serves as the reference panel
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])
example.G &lt;- example.G[,apply(example.G,2,sd)!=0]
example.G &lt;- example.G[,1:100]

# compute correlation among variants
cor.G&lt;-matrix(as.numeric(corpcor::cor.shrink(example.G)), nrow=ncol(example.G))

# fit null model
fit.prelim&lt;-GhostKnockoff.prelim(cor.G,M=5,method='asdp',max.size=500)

### if only one study is involved
Zscore_0&lt;-as.matrix(rnorm(nrow(cor.G))) # hypothetical Z-scores
Zscore_0&lt;-Zscore_0+rbinom(nrow(cor.G),size=2,0.1) # set causal
n.study&lt;-c(5000)

# knockoff scores for each block, this can be run in parallel too
GK.stat&lt;-GhostKnockoff.fit(Zscore_0,n.study,fit.prelim,gamma=1,weight.study=NULL)

### if multiple studies are involved, for a meta-analysis

# compute study correlation
Zscore_0&lt;-cbind(rnorm(nrow(cor.G)),rnorm(nrow(cor.G))) # hypothetical Z-scores
Zscore_0&lt;-Zscore_0+rbinom(nrow(cor.G),size=2,0.1) # set causal
cor.study&lt;-GhostKnockoff.GetCorStudy(Zscore_0,fit.prelim)

# note that all steps above can be run in parallel for nonoverlapping blocks of the genome.
# Then the overall study correlation can be computed by averaging cor.study of all blocks.

# compute optimal weights and study dependency using overall study correlaton
n.study&lt;-c(5000,7500)
Meta.prelim&lt;-GhostKnockoff.prelim.Meta(cor.study, n.study)
gamma&lt;-Meta.prelim$gamma
weight.study&lt;-Meta.prelim$w_opt

# knockoff scores for each block, this can be run in parallel too
GK.stat&lt;-GhostKnockoff.fit(Zscore_0,n.study,fit.prelim,gamma=gamma,weight.study=weight.study)

</code></pre>


</div>