<div class="container">

<table style="width: 100%;"><tr>
<td>GhostKnockoff.filter</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
GhostKnockoff filter
</h2>

<h3>Description</h3>

<p>This function calculates Q-values to select variants associated with the outcome, given the feature importance scores
</p>


<h3>Usage</h3>

<pre><code class="language-R">GhostKnockoff.filter(T_0,T_k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>T_0</code></td>
<td>

<p>A p*1 matrix. The feature importance score for the original variants.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>T_k</code></td>
<td>

<p>A p*M matrix where M is the number of hypothetical knockoffs. The knockoff feature importance scores.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>q</code></td>
<td>

<p>A vector of length p, where each element is a Q-value. Variants with q &lt;= target FDR will be selected.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kappa</code></td>
<td>

<p>A vector of length p. Multiple knockoff statistics kappa.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau</code></td>
<td>

<p>A vector of length p. Multiple knockoff statistics tau.
</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">
# We use genetic data as an example
library(GhostKnockoff)

# load example vcf file from package "seqminer", this serves as the reference panel
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])
example.G &lt;- example.G[,apply(example.G,2,sd)!=0]
example.G &lt;- example.G[,1:100]

# compute correlation among variants
cor.G&lt;-matrix(as.numeric(corpcor::cor.shrink(example.G)), nrow=ncol(example.G))

# fit null model
fit.prelim&lt;-GhostKnockoff.prelim(cor.G,M=5,method='asdp',max.size=500)

### if only one study is involved
Zscore_0&lt;-as.matrix(rnorm(nrow(cor.G))) # hypothetical Z-scores
Zscore_0&lt;-Zscore_0+rbinom(nrow(cor.G),size=2,0.1) # set causal
n.study&lt;-c(5000)

# knockoff scores for each block, this can be run in parallel too
GK.stat&lt;-GhostKnockoff.fit(Zscore_0,n.study,fit.prelim,gamma=1,weight.study=NULL)

# knockoff filter
GK.filter&lt;-GhostKnockoff.filter(GK.stat$T_0,GK.stat$T_k)
GK.filter$q

### if multiple studies are involved, for a meta-analysis

# compute study correlation
Zscore_0&lt;-cbind(rnorm(nrow(cor.G)),rnorm(nrow(cor.G))) # hypothetical Z-scores
Zscore_0&lt;-Zscore_0+rbinom(nrow(cor.G),size=2,0.1) # set causal
cor.study&lt;-GhostKnockoff.GetCorStudy(Zscore_0,fit.prelim)

# note that all steps above can be run in parallel for nonoverlapping blocks of the genome.
# Then the overall study correlation can be computed by averaging cor.study of all blocks.

# compute optimal weights and study dependency using overall study correlaton
n.study&lt;-c(5000,7500)
Meta.prelim&lt;-GhostKnockoff.prelim.Meta(cor.study, n.study)
gamma&lt;-Meta.prelim$gamma
weight.study&lt;-Meta.prelim$w_opt

# knockoff scores for each block, this can be run in parallel too
GK.stat&lt;-GhostKnockoff.fit(Zscore_0,n.study,fit.prelim,gamma=gamma,weight.study=weight.study)

# knockoff filter
GK.filter&lt;-GhostKnockoff.filter(GK.stat$T_0,GK.stat$T_k)
GK.filter$q

</code></pre>


</div>