<div class="container">

<table style="width: 100%;"><tr>
<td>leveneTests_per_SNP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Levene's test for variance homogeneity by SNP genotypes (sex-specific p-values)</h2>

<h3>Description</h3>

<p>This function takes as input the genotype of a SNP (<code>geno_one</code>), the genetic sex (<code>SEX</code>), a quantitative trait (<code>Y</code>) in a sample population. The function then returns the variance heterogeneity <em>p</em>-values for each sex and the overall variance heterogeneity signal using Fisher's method by combining the sex-specific results.
</p>


<h3>Usage</h3>

<pre><code class="language-R">leveneTests_per_SNP(
  geno_one,
  SEX = NULL,
  Y,
  centre = "median",
  transformed = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>geno_one</code></td>
<td>
<p>the genotype of a bi-allelic SNP, must be a vector of 0, 1, 2's coded for the number of reference allele. Alternatively, for imputed genotypes, it could be a matrix/vector of dosage values, numerically between 0 and 2. The length/dimension of <code>geno_one</code> should match that of <code>Y</code>, and/or <code>SEX</code> and <code>COVAR</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SEX</code></td>
<td>
<p>optional: the genetic sex of individuals in the sample population, must be a vector of 1 and 2 following the default sex code is 1 for males and 2 for females in PLINK.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>a vector of quantitative traits, such as human height.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>centre</code></td>
<td>
<p>a character indicating whether the absolute deviation should be calculated with respect to "median" or "mean", the default option is "median".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transformed</code></td>
<td>
<p>a logical indicating whether the quantitative response <code>Y</code> should be transformed using a rank-based method to resemble a normal distribution; recommended for traits with non-symmetric distribution. The default option is <code>TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a vector of Levene's test p-values according to levels specified by <code>geno_one</code> in each sex and the Fisher's method to combine the sex-specific Levene's test <em>p</em>-values.
</p>


<h3>Note</h3>

<p>We recommend to quantile-normally transform <code>Y</code> to avoid ‘scale-effect’ where
the variance values tend to be proportional to mean values when stratified by <code>G</code>.
</p>


<h3>Author(s)</h3>

<p>Wei Q. Deng <a href="mailto:deng@utstat.toronto.edu">deng@utstat.toronto.edu</a>, Lei Sun <a href="mailto:sun@utstat.toronto.edu">sun@utstat.toronto.edu</a>
</p>


<h3>References</h3>

<p>Levene H. (1960) Robust tests for equality of variances. In <em>Contributions to Probability and Statistics: Essays in Honor of Harold Hotelling</em> eds:I. Olkin, S.G. Ghurye, W. Hoeffding, W.G. Madow &amp; H.B.Mann, pp.278-292. Stanford: Stanford University Press.
</p>


<h3>Examples</h3>

<pre><code class="language-R">N &lt;- 5000
sex &lt;- rbinom(N, 1, 0.5)+1
genDAT &lt;- rbinom(N, 2, 0.3)
y &lt;- rnorm(N);

genDAT[sex==2] &lt;- rbinom(sum(sex==2), 1, 0.3)
table(genDAT, sex)
leveneTests_per_SNP(geno_one=genDAT, SEX=sex, Y=y^2, transform=TRUE)

genDAT[sex==2] &lt;- rbinom(sum(sex==2), 1, 0.01)
table(genDAT, sex)
leveneTests_per_SNP(geno_one=genDAT, SEX=sex, Y=y^2, transform=FALSE)

leveneTests_per_SNP(geno_one=rep(0, N), SEX=sex, Y=y^2, transform=TRUE)
leveneTests_per_SNP(geno_one=rep(0, N), Y=y^2, transform=TRUE)


</code></pre>


</div>