<div class="container">

<table style="width: 100%;"><tr>
<td>leveneRegA_per_SNP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The generalized Levene's test via a two-stage regression for variance homogeneity by SNP genotype (autosomes)</h2>

<h3>Description</h3>

<p>This function takes as input the genotype of a SNP (<code>GENO</code>), a quantitative trait (<code>Y</code>) in a sample population, and possibly additional covariates, such as principal components. The function returns the scale association <em>p</em>-values for each autosomal SNP using the generalized Levene's test.
</p>


<h3>Usage</h3>

<pre><code class="language-R">leveneRegA_per_SNP(
  geno_one,
  Y,
  COVAR = NULL,
  transformed = TRUE,
  loc_alg = "LAD",
  related = FALSE,
  cov.structure = "corCompSymm",
  clust = NULL,
  genotypic = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>geno_one</code></td>
<td>
<p>the genotype of a biallelic SNP, must be a vector of 0, 1, 2's coded for the number of reference allele. Alternatively, for imputed genotypes, it could be a matrix/vector of dosage values, numerically between 0 and 2. The length/dimension of <code>geno_one</code> should match that of <code>Y</code>, and/or <code>SEX</code> and <code>COVAR</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>a vector of quantitative traits, such as human height.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>COVAR</code></td>
<td>
<p>optional: a vector or matrix of covariates that are used to reduce bias due to confounding, such as age.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transformed</code></td>
<td>
<p>a logical indicating whether the quantitative response <code>Y</code> should be transformed using a rank-based method to resemble a normal distribution; recommended for traits with non-symmetric distribution. The default option is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loc_alg</code></td>
<td>
<p>a character indicating the type of algorithm to compute the centre in stage 1; the value is either "OLS", corresponding to an ordinary linear regression under Gaussian assumptions to compute the mean, or "LAD", corresponding to a quantile regression to compute the median. The recommended default option is "LAD". For the quantile regression, the function calls <code>quantreg::rq</code> and the median is estimated using either the "fn" (smaller samples) or "sfn" (larger samples and sparse problems) algorithm depending the sample size, for more details see <code>?quantreg::rq</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>related</code></td>
<td>
<p>optional: a logical indicating whether the samples should be treated as related; if <code>TRUE</code> while no relatedness covariance information is given, it is then estimated under a <code>cov.structure</code> and assumes this structure among all within-group errors pertaining to the same pair/cluster if specified using <code>clust</code>. This option currently only applies to autosomal SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cov.structure</code></td>
<td>
<p>optional: should be one of standard classes of correlation structures listed in <code>corClasses</code> from <span class="pkg">R</span> package <span class="pkg">nlme</span>. See <code>?corClasses</code>. The most commonly used option is <code>corCompSymm</code> for a compound symmetric correlation structure. This option currently only applies to autosomal SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clust</code></td>
<td>
<p>optional: a factor indicating the grouping of samples; it should have at least two distinct values. It could be the family ID (FID) for family studies. This option currently only applies to autosomal SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypic</code></td>
<td>
<p>optional: a logical indicating whether the variance homogeneity should be tested with respect to an additively (linearly) coded or non-additively coded <code>geno_one</code>. The former has one less degree of freedom than the latter and is the default option. For dosage data without genotypic probabilities, <code>genotypic</code> is forced to be <code>FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Levene's test regression p-values for autosomal SNPs according to the model specified.
</p>


<h3>Note</h3>

<p>We recommend to quantile-normally transform <code>Y</code> to avoid ‘scale-effect’ where
the variance values tend to be proportional to mean values when stratified by <code>geno_one</code>.
</p>
<p>When the relatedness option is used, the computational time is expected to be longer for larger sample size ($$n &gt; 1000$$), thus we recommend this option for smaller studies rather than large population based studies.
</p>
<p>There is no explicit argument to supply <code>SEX</code> for autosomal SNPs, the user can choose to include the genetic sex of individuals as a column of the <code>COVAR</code> argument.
</p>


<h3>Author(s)</h3>

<p>Wei Q. Deng <a href="mailto:deng@utstat.toronto.edu">deng@utstat.toronto.edu</a>, Lei Sun <a href="mailto:sun@utstat.toronto.edu">sun@utstat.toronto.edu</a>
</p>


<h3>References</h3>

<p>Soave D, Sun L. (2017). A generalized Levene's scale test for variance heterogeneity in the presence of sample correlation and group uncertainty. <em>Biometrics</em>. <strong>73</strong>(3):960-971. doi: <a href="https://doi.org/10.1111/biom.12651">10.1111/biom.12651</a>. PMID: 28099998.
</p>
<p>Gastwirth JL, Gel YR, Miao W. (2009). The Impact of Levene's Test of Equality of Variances on Statistical Theory and Practice." <em>Statistical Science</em>. <strong>24</strong>(3) 343 - 360, doi: <a href="https://doi.org/10.1214/09-STS301">10.1214/09-STS301</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">N &lt;- 100
genDAT &lt;- rbinom(N, 2, 0.3)
Y &lt;- rnorm(N)
covar &lt;- matrix(rnorm(N*10), ncol=10)

# vanilla example:
leveneRegA_per_SNP(geno_one=genDAT, Y=Y, COVAR=covar)

# relatedness samples:
leveneRegA_per_SNP(geno_one=genDAT, Y=Y, COVAR=covar,
related=TRUE)
leveneRegA_per_SNP(geno_one=genDAT, Y=Y, COVAR=covar,
related=TRUE, clust = factor(rbinom(N, 2, 0.6)))


# dosage genotypes example:
library("MCMCpack")
a &lt;- 0.3
geno &lt;- rbinom(N, 2, 0.3)
a &lt;- 0.3 ## uncertainty
genPP &lt;- rbind(rdirichlet(sum(geno==0),c(a,(1-a)/2,(1-a)/2)),
               rdirichlet(sum(geno==1),c((1-a)/2,a,(1-a)/2)),
               rdirichlet(sum(geno==2),c((1-a)/2,(1-a)/2,a)))

leveneRegA_per_SNP(geno_one=genPP, Y=Y, COVAR=covar)
leveneRegA_per_SNP(geno_one=genPP, Y=Y, COVAR=covar,
genotypic=TRUE)

# dosage and related samples:
leveneRegA_per_SNP(geno_one=genPP, Y=Y, COVAR=covar,
related=TRUE, clust = factor(rbinom(N, 1, 0.3)))
leveneRegA_per_SNP(geno_one=genPP, Y=Y, COVAR=covar,
related=TRUE, clust = factor(rbinom(N, 1, 0.3)), genotypic=TRUE)



</code></pre>


</div>