<div class="container">

<table style="width: 100%;"><tr>
<td>scaleReg</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Scale (variance-based association) test</h2>

<h3>Description</h3>

<p>This function takes as input the genotype of SNPs (<code>GENO</code>), the SEX (<code>SEX</code>), and a quantitative trait (<code>Y</code>) in a sample population, and possibly additional covariates, such as principal components. The function returns the scale association <em>p</em>-values for each SNP.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scaleReg(
  GENO,
  Y,
  COVAR = NULL,
  SEX = NULL,
  Xchr = FALSE,
  transformed = FALSE,
  loc_alg = "LAD",
  related = FALSE,
  cov.structure = "corCompSymm",
  clust = NULL,
  genotypic = FALSE,
  origLev = FALSE,
  centre = "median"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>GENO</code></td>
<td>
<p>a list of a genotype matrix/vector of SNPs, must contain values 0, 1, 2's coded for the number of reference allele. Alternatively, for imputed genotypes, it could either be a vector of dosage values between 0 and 2, or a list of matrix of genotype probabilities, numerically between 0 and 1 for each genotype. The length/dimension of <code>GENO</code> should match that of <code>Y</code>, and/or <code>SEX</code> and <code>COVAR</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>a vector of quantitative traits, such as human height.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>COVAR</code></td>
<td>
<p>optional: a vector or matrix of covariates that are used to reduce bias due to confounding, such as age.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SEX</code></td>
<td>
<p>optional: the genetic sex of individuals in the sample population, must be a vector of 1 and 2 following the default sex code is 1 for males and 2 for females in PLINK.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Xchr</code></td>
<td>
<p>a logical indicator for whether the analysis is for X-chromosome SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transformed</code></td>
<td>
<p>a logical indicating whether the quantitative response <code>Y</code> should be transformed using a rank-based method to resemble a normal distribution; recommended for traits with non-symmetric distribution. The default option is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loc_alg</code></td>
<td>
<p>a character indicating the type of algorithm to compute the centre in stage 1; the value is either "OLS", corresponding to an ordinary linear regression under Gaussian assumptions to compute the mean, or "LAD", corresponding to a quantile regression to compute the median. The recommended default option is "LAD". For the quantile regression, the function calls <code>quantreg::rq</code> and the median is estimated using either the "br" (smaller samples) or "sfn" (larger samples and sparse problems) algorithm depending the sample size, for more details see <code>?quantreg::rq</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>related</code></td>
<td>
<p>optional: a logical indicating whether the samples should be treated as related; if <code>TRUE</code> while no relatedness covariance information is given, it is then estimated under a <code>cov.structure</code> and assumes this structure among all within-group errors pertaining to the same pair/cluster if specified using <code>clust</code>. This option currently only applies to autosomal SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cov.structure</code></td>
<td>
<p>optional: should be one of standard classes of correlation structures listed in <code>corClasses</code> from <span class="pkg">R</span> package <span class="pkg">nlme</span>. See <code>?corClasses</code>. The most commonly used option is <code>corCompSymm</code> for a compound symmetric correlation structure. This option currently only applies to autosomal SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clust</code></td>
<td>
<p>optional: a factor indicating the grouping of samples; it should have at least two distinct values. It could be the family ID (FID) for family studies. This option currently only applies to autosomal SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypic</code></td>
<td>
<p>a logical indicating whether the variance homogeneity should be tested with respect to an additively (linearly) coded or non-additively coded <code>geno_one</code>. The former has one less degree of freedom than the latter and is the default option. For dosage genotypes without genotypic probabilities, <code>genotypic</code> is forced to be <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>origLev</code></td>
<td>
<p>a logical indicator for whether the reported p-values should also include original Levene's test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>centre</code></td>
<td>
<p>a character indicating whether the absolute deviation should be calculated with respect to "median" or "mean" in the traditional sex-specific and Fisher combined Levene's test p-values (three tests) for X-chromosome. The default value is "median". This option applies to sex-specific analysis using original Levene's test (i.e. when <code>regression</code>$$=$$<code>TRUE</code>).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a vector of Levene's test regression p-values according to the models
specified.
</p>


<h3>Note</h3>

<p>We recommend to quantile-normally transform <code>Y</code> to avoid ‘scale-effect’ where
the variance values tend to be proportional to mean values when stratified by <code>GENO</code>.
</p>


<h3>Author(s)</h3>

<p>Wei Q. Deng <a href="mailto:deng@utstat.toronto.edu">deng@utstat.toronto.edu</a>, Lei Sun <a href="mailto:sun@utstat.toronto.edu">sun@utstat.toronto.edu</a>
</p>


<h3>References</h3>

<p>Deng WQ, Mao S, Kalnapenkis A, Esko T, Magi R, Pare G, Sun L. (2019) Analytical strategies to include the X-chromosome in variance heterogeneity analyses: Evidence for trait-specific polygenic variance structure. <em>Genet Epidemiol</em>. <strong>43</strong>(7):815-830. doi: <a href="https://doi.org/10.1002/gepi.22247">10.1002/gepi.22247</a>. PMID:31332826.
</p>
<p>Gastwirth JL, Gel YR, Miao W. (2009). The Impact of Levene's Test of Equality of Variances on Statistical Theory and Practice." <em>Statistical Science</em>. <strong>24</strong>(3) 343-360, doi: <a href="https://doi.org/10.1214/09-STS301">10.1214/09-STS301</a>.
</p>
<p>Soave D, Sun L. (2017). A generalized Levene's scale test for variance heterogeneity in the presence of sample correlation and group uncertainty. <em>Biometrics</em>. <strong>73</strong>(3):960-971. doi: <a href="https://doi.org/10.1111/biom.12651">10.1111/biom.12651</a>. PMID:28099998.
</p>


<h3>Examples</h3>

<pre><code class="language-R">N &lt;- 1000
genoDAT &lt;- rbinom(N, 2, 0.3)
sex &lt;- rbinom(N, 1, 0.5)+1
Y &lt;- rnorm(N)
covar &lt;- matrix(rnorm(N*10), ncol=10)

# vanilla example:
scaleReg(GENO=list(genoDAT, genoDAT), Y=Y, COVAR=covar)
scaleReg(GENO=list(genoDAT, genoDAT), Y=Y, COVAR=covar, genotypic=TRUE)
scaleReg(GENO=list(genoDAT, genoDAT), Y=Y, COVAR=covar, origLev = TRUE)
scaleReg(GENO=list(genoDAT, genoDAT), Y=Y, COVAR=covar, origLev = TRUE, SEX=sex)

</code></pre>


</div>