<div class="container">

<table style="width: 100%;"><tr>
<td>gkmsvm_delta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Calculating deltaSVM scores 
</h2>

<h3>Description</h3>

<p>Given support vectors SVs and corresponding coefficients alphas and a pair of file test sequence files (one for reference allele, and one for alternate allele), calculates the deltaSVM scores for the sequences.</p>


<h3>Usage</h3>

<pre><code class="language-R">gkmsvm_delta(seqfile_allele1, seqfile_allele2, svmfnprfx, outfile, L = 10, 
K = 6, maxnmm = 3, maxseqlen = 10000, maxnumseq = 1e+06, useTgkm = 1, alg = 2, 
addRC = TRUE, usePseudocnt = FALSE, batchSize = 1e+05, wildcardLambda = 1, 
wildcardMismatchM = 2, alphabetFN = "NULL", svseqfile = NA,alphafile = NA, 
outfile_allele1 = NA, outfile_allele2 = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqfile_allele1</code></td>
<td>

<p>fasta file containing the test sequences (reference allele)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqfile_allele2</code></td>
<td>

<p>fasta file containing the test sequences (alternate allele). The sequences in this file should be in the exact same order as in seqfile_allele1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>svmfnprfx</code></td>
<td>
<p>SVM model file name prefix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p>output file name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>L</code></td>
<td>
<p>word length, default=10</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>number of informative columns, default=6</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxnmm</code></td>
<td>
<p>maximum number of mismatches to consider, default=3</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxseqlen</code></td>
<td>
<p>maximum sequence length in the sequence files, default=10000</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxnumseq</code></td>
<td>
<p>maximum number of sequences in the sequence files, default=1000000</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useTgkm</code></td>
<td>
<p>filter type: 0(use full filter), 1(use truncated filter: this gaurantees non-negative counts for all L-mers), 2(use h[m], gkm count vector), 3(wildcard), 4(mismatch), default=1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alg</code></td>
<td>
<p>algorithm type: 0(auto), 1(XOR Hashtable), 2(tree), default=0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addRC</code></td>
<td>
<p>adds reverse complement sequences, default=TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>usePseudocnt</code></td>
<td>
<p>adds a constant to count estimates, default=FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batchSize</code></td>
<td>
<p>number of sequences to compute scores for in batch, default=100000</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wildcardLambda</code></td>
<td>
<p>lambda for wildcard kernel, defaul=0.9</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wildcardMismatchM</code></td>
<td>
<p>max mismatch for Mismatch kernel or wildcard kernel, default=2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphabetFN</code></td>
<td>
<p>alphabets file name, if not specified, it is assumed the inputs are DNA sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>svseqfile</code></td>
<td>
<p>SVM support vectors sequence file name (not needed if svmfnprfx is provided)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphafile</code></td>
<td>
<p>SVM support vectors weights file name (not needed if svmfnprfx is provided)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile_allele1</code></td>
<td>

<p>output filename for gkmSVM scores for the reference sequences (optional)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile_allele2</code></td>
<td>

<p>output filename for gkmSVM scores for the alternate sequences (optional)
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>predicting the effect of variants using gkmSVM model:
gkmsvm_delta can be used to predict the effect of sequence variants. The sequences corresponding to reference allele and alternate alleles are given in two separate files. gkmSVM is used to score each set of sequences, and the difference in the gkmSVM score for the reference and alternate allele is reported. 
Note that the same set of parameters (L, K, maxnmm) used in the gkmsvm_kernel should be specified for
optimal scoring.
</p>
<p>gkmsvm_kernel(seqfile_allele1, seqfile_allele2, svmfnprfx, outfn);            #scores test sequences </p>


<h3>Value</h3>

<p>deltaSVM scores
</p>


<h3>Author(s)</h3>

<p>Mahmoud Ghandi
</p>


<h3>References</h3>

<p>Ghandi M, Mohammad-Noori M, Ghareghani N, Lee D, Garraway LA, and Beer MA. gkmSVM: an R package for gapped-kmer SVM, Bioinformatics 2016.
</p>
<p>Ghandi M, Lee D, Mohammad-Noori M, Beer MA. 2014. Enhanced Regulatory Sequence Prediction Using Gapped k-mer Features. PLoS Comput Biol 10: e1003711. 
</p>
<p>Lee D, Gorkin DU, Baker M, Strober BJ, Asoni AL, McCallion AS, and Beer MA. A method to predict the impact of regulatory variants from DNA sequence. Nature Genetics 2015. 
</p>


<h3>Examples</h3>

<pre><code class="language-R">
  #Input file names:  
  posfn= 'test_positives.fa'   #positive set (FASTA format)
  negfn= 'test_negatives.fa'   #negative set (FASTA format)
  testfn_ref= 'test_testsetRef.fa'    #test set (reference allele) (FASTA format)
  testfn_alt= 'test_testsetAlt.fa'    #test set (alternate allele) (FASTA format)
  
  #Output file names:  
  kernelfn= 'test_kernel.txt' #kernel matrix
  svmfnprfx= 'test_svmtrain'  #SVM files 
  outfn =   'output.txt'      #output delta svm scores for sequences in the test set       

#  gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
#  gkmsvm_train(kernelfn,posfn, negfn, svmfnprfx);       #trains SVM
#  gkmsvm_delta(testfn_ref,testfn_alt, svmfnprfx, outfn);            #scores test sequences 

</code></pre>


</div>