<div class="container">

<table style="width: 100%;"><tr>
<td>gkmSVM-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Gapped-Kmer Support Vector Machine
</h2>

<h3>Description</h3>

<p>Imports the 'gkmSVM' v2.0 functionalities into R &lt;http://www.beerlab.org/gkmsvm/&gt; . It also uses the 'kernlab' library (separate R package by different authors) for various SVM algorithms.   
</p>


<h3>Details</h3>

<p>The gkm-SVM provides implementation of a new SVM kernel method using gapped
k-mers as features for DNA or Protein sequences.
</p>
<p>There are three main functions in the gkmSVM package: 
</p>
<p>gkmsvm_kernel: computes the kernel matrix
</p>
<p>gkmsvm_train:  computes the SVM coefficients
</p>
<p>gkmsvm_classify: scores new sequences using the SVM model  
</p>
<p>Tutorial
</p>
<p>========
</p>
<p>We introduce the users to the basic workflow of our gkmSVM step-by-step.
Please refer to help messages for more detailed information of each function.
</p>
<p>1) making a kernel matrix
</p>
<p>First of all, we should calculate a full kernel matrix before training SVM
classifiers. In this tutorial, we are going to use test_positives.fa
as a positive set, and test_negatives.fa as a negative set. 
</p>
<p>#Input file names:  
</p>
<p>posfn= 'test_positives.fa'   #positive set (FASTA format)
</p>
<p>negfn= 'test_negatives.fa'   #negative set (FASTA format)
</p>
<p>testfn= 'test_testset.fa'    #test set (FASTA format)
</p>
<p>Alternatively if the negative set is not available, and positive set is provided as a bed file, genNullSeqs function could be used to generate the negative set and positive set sequences. 
</p>
<p>#Output file names:  
</p>
<p>kernelfn= 'test_kernel.txt' #kernel matrix
</p>
<p>svmfnprfx= 'test_svmtrain'  #SVM files 
</p>
<p>outfn =   'output.txt'      #output scores for sequences in the test set       
</p>
<p>gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
</p>
<p>2) training SVM
</p>
<p>We can now train a SVM classifier using the kernel matrix generated above. For that we use gkmsvm_train function It takes four arguments; kernel file, positive sequences file, negative sequences file, and prefix of output file names for the svm model.
</p>
<p>gkmsvm_train(kernelfn, posfn, negfn, svmfnprfx);        #trains SVM
</p>
<p>It will generate two files, test_svmtrain_svalpha.out and
test_svmtrain_svseq.fa, which will then be used for classification/scoring
of test sequences as described below.
</p>
<p>3) classification using SVM
</p>
<p>gkmsvm_classify can be used to score any set of sequences. Here, we will 
score the test sequences which are given in test_testset.fa. Note that the same
set of parameters used in the gkmsvm_kernel should always be specified for
optimal classification (here we used default parameters).
</p>
<p>gkmsvm_classify(testfn, svmfnprfx, outfn);            #scores test sequences 
</p>
<p>In a more advanced example, we set the word length L=18, and the number of non-gapped positions K=7, and maximum number of mismatches maxnmm=4:
</p>
<p>gkmsvm_kernel(posfn, negfn, kernelfn, L=18, K=7, maxnmm=4);     #computes kernel 
</p>
<p>gkmsvm_train(kernelfn,posfn, negfn, svmfnprfx);                 #trains SVM
</p>
<p>gkmsvm_classify(testfn, svmfnprfx, outfn, L=18, K=7, maxnmm=4); #scores test sequences 
</p>
<p>In another example, we run a 5-fold cross validation to plot the ROC curves: 
</p>
<p>gkmsvm_kernel(posfn, negfn, kernelfn);     #computes kernel 
</p>
<p>cvres = gkmsvm_trainCV(kernelfn, posfn, negfn, svmfnprfx, outputPDFfn='ROC.pdf', outputCVpredfn='cvpred.out');       #trains SVM, plots ROC and PRC curves, and outputs model predictions.
</p>


<h3>Author(s)</h3>

<p>Mahmoud Ghandi
</p>
<p>Maintainer: Mike Beer &lt;mbeer@jhu.edu&gt;
</p>


<h3>References</h3>

<p>Ghandi M, Lee D, Mohammad-Noori M, Beer MA. 2014. Enhanced Regulatory Sequence Prediction Using Gapped k-mer Features. PLoS Comput Biol 10: e1003711. 
</p>
<p>Ghandi M, Mohammad-Noori M, Ghareghani N, Lee D, Garraway LA, and Beer MA. 2016. gkmSVM an R package for gapped-kmer SVM, Bioinformatics 32 (14), 2205-2207.
</p>


<h3>Examples</h3>

<pre><code class="language-R">  #Input file names:  
  posfn= 'test_positives.fa'   #positive set (FASTA format)
  negfn= 'test_negatives.fa'   #negative set (FASTA format)
  testfn= 'test_testset.fa'    #test set (FASTA format)
  
  #Output file names:  
  kernelfn= 'test_kernel.txt' #kernel matrix
  svmfnprfx= 'test_svmtrain'  #SVM files 
  outfn =   'output.txt'      #output scores for sequences in the test set       

#  gkmsvm_kernel(posfn, negfn, kernelfn);                #computes kernel 
#  gkmsvm_train(kernelfn, posfn, negfn, svmfnprfx);       #trains SVM
#  gkmsvm_classify(testfn, svmfnprfx, outfn);            #scores test sequences 

#  using L=18, K=7, maxnmm=4

#  gkmsvm_kernel(posfn, negfn, kernelfn, L=18, K=7, maxnmm=4);     #computes kernel 
#  gkmsvm_train(kernelfn, posfn, negfn, svmfnprfx);                 #trains SVM
#  gkmsvm_classify(testfn, svmfnprfx, outfn, L=18, K=7, maxnmm=4); #scores test sequences 

</code></pre>


</div>