<div class="container">

<table style="width: 100%;"><tr>
<td>plot.egger</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
A function used to draw Egger plot
</h2>

<h3>Description</h3>

<p>It is used to draw MR-Egger regression plot.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'egger'
plot(x,exposure_coeff,qcutoff=0.2,xlab="Genetic association with the exposure",
                       ylab="Genetic association with the outcome",...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>x is a dataframe, it is the outcome of the glide function.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exposure_coeff</code></td>
<td>

<p>A named numeric vector or a one-column dataframe for the external regression coefficients
for genotype-exposure associations. These coefficients are typically derived from previous
genome-wide association studies.
All the names or rownames must be found in "genotype_columns" (specified below) columns of the
input data.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qcutoff</code></td>
<td>

<p>qcutoff is the q-value cutoff.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>

<p>xlab is the default x label.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>

<p>ylab is the default y label.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>the rest of plot arguments.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>No return value, the Egger plot will be generated.</p>


<h3>Author(s)</h3>

<p>James Y. Dai and X. Wang
</p>


<h3>Examples</h3>

<pre><code class="language-R">
##first generate output using the glide function
#load an example data
data(simdata)
simdat=simdata$simdat
coeff=simdata$coeff

formula=as.formula("outcome~age+sex+pc1+pc2+pc3")
genotype_columns=which(grepl("^SNP",colnames(simdat)))

#run glide
out=glide(formula=formula,exposure_coeff=coeff,genotype_columns,data=simdat,np=100000,
qcutoff=0.2,parallel=TRUE,corenumber=1,verbose=TRUE)

##draw the Egger plot
plot.egger(out,exposure_coeff=coeff)

</code></pre>


</div>