<div class="container">

<table style="width: 100%;"><tr>
<td>glmmSeq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GLMM with negative binomial distribution for sequencing count data</h2>

<h3>Description</h3>

<p>Fits many generalised linear mixed effects models (GLMM) with negative
binomial distribution for analysis of overdispersed count data with random
effects. Designed for longitudinal analysis of RNA-Sequencing count data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">glmmSeq(
  modelFormula,
  countdata,
  metadata,
  id = NULL,
  dispersion = NA,
  sizeFactors = NULL,
  reduced = NULL,
  modelData = NULL,
  designMatrix = NULL,
  method = c("lme4", "glmmTMB"),
  control = NULL,
  family = nbinom2,
  cores = 1,
  removeSingles = FALSE,
  zeroCount = 0.125,
  verbose = TRUE,
  returnList = FALSE,
  progress = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>modelFormula</code></td>
<td>
<p>the model formula. This must be of the form <code>"~ ..."</code>
where the structure is assumed to be <code>"counts ~ ..."</code>. The formula must
include a random effects term. For more information on formula structure
for random effects see <code>lme4::glmer()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>countdata</code></td>
<td>
<p>the sequencing count data matrix with genes in rows and
samples in columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metadata</code></td>
<td>
<p>a dataframe of sample information with variables in columns
and samples in rows</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>Optional. Used to specify the column in metadata which contains the
sample IDs to be used in repeated samples for random effects. If not
specified, the function defaults to using the variable after the "|" in the
random effects term in the formula.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dispersion</code></td>
<td>
<p>a numeric vector of gene dispersion. Not required for
<code>glmmTMB</code> models, as these determine and fit dispersion for each gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sizeFactors</code></td>
<td>
<p>size factors (default = NULL). If provided the <code>glmer</code>
offset is set to log(sizeFactors). For more information seeâ€œ
<code>lme4::glmer()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reduced</code></td>
<td>
<p>Optional reduced model formula. If this is chosen, a
likelihood ratio test is used to calculate p-values instead of the default
Wald type 2 Chi-squared test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>modelData</code></td>
<td>
<p>Optional dataframe. Default is generated by call to
<code>expand.grid</code> using levels of variables in the formula. Used to calculate
model predictions (estimated means &amp; 95% CI) for plotting via modelPlot.
It can therefore be used to add/remove points in modelPlot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>designMatrix</code></td>
<td>
<p>custom design matrix, used only for prediction</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Specifies which package to use for fitting GLMM models. Either
"lme4" or "glmmTMB" depending on whether to use lme4::glmer or
glmmTMB::glmmTMB to fit GLMM models.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>control</code></td>
<td>
<p>the <code>glmer</code> optimizer control. If <code>method = "lme4"</code> default is
<code>glmerControl(optimizer = "bobyqa")</code>). If
<code>method = "glmmTMB"</code> default is <code>glmmTMBControl()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>family</code></td>
<td>
<p>Only used with <code>glmmTMB</code> models. Default is <code>nbinom2</code>. See
glmmTMB::nbinom2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p>number of cores to use. Default = 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>removeSingles</code></td>
<td>
<p>whether to remove individuals without repeated measures
(default = FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zeroCount</code></td>
<td>
<p>numerical value to offset zeroes for the purpose of log
(default = 0.125)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical whether to display messaging (default = TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnList</code></td>
<td>
<p>Logical whether to return results as a list or <code>glmmSeq</code>
object (default = FALSE). Useful for debugging.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>progress</code></td>
<td>
<p>Logical whether to display a progress bar</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other parameters to pass to
<code>lme4::glmer()</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is a wrapper for <code>lme4::glmer()</code>. By default, p-values for each
model term are computed using Wald type 2 Chi-squared test as per
<code>car::Anova()</code>. The underlying code for this has been optimised for speed.
However, if a reduced model formula is specified by setting <code>reduced</code>, then a
likelihood ratio test is performed instead using stats::anova. This will
double computation time since two GLMM have to be fitted.
</p>
<p>Parallelisation is provided using parallel::mclapply on Unix/Mac or
parallel::parLapply on PC.
</p>
<p>Setting <code>method = "glmmTMB"</code> enables an alternative method of fitting GLMM
using the <code>glmmTMB</code> package. This gives access to a variety of alternative
GLM family functions. Note, <code>glmmTMB</code> negative binomial models are
substantially slower to fit than <code>glmer</code> models with known dispersion due to
the extra time taken by <code>glmmTMB</code> to optimise the dispersion parameter.
</p>
<p>The <code>id</code> argument is usually optional. By default the <code>id</code> column in the
metadata is determined as the term after the bar in the random effects term
of the model. Note that <code>id</code> is not passed to <code>glmer</code> or <code>glmmTMB</code>. It is
only really used to remove singletons from the <code>countdata</code> matrix and
<code>metadata</code> dataframe. The <code>id</code> is also stored in the output from <code>glmmSeq</code>
and used by plotting function <code>modelPlot()</code>. However, due to its flexible
nature, in theory <code>glmmSeq</code> should allow for more than one random effect
term, although this has not been tested formally. In this case, it is
probably prudent to specify a value for <code>id</code>.
</p>


<h3>Value</h3>

<p>Returns an S4 class <code>GlmmSeq</code> object with results for gene-wise
general linear mixed models. A list of results is returned if <code>returnList</code>
is <code>TRUE</code> which is useful for debugging. If all genes return errors from
<code>glmer</code>, then an error message is shown and a character vector containing
error messages for all genes is returned.
</p>


<h3>See Also</h3>

<p>lme4::glmer lme4::glmerControl glmmTMB::glmmTMB
glmmTMB::nbinom2 glmmTMB::glmmTMBControl car::Anova
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(PEAC_minimal_load)
disp &lt;- apply(tpm, 1, function(x) {
(var(x, na.rm = TRUE)-mean(x, na.rm = TRUE))/(mean(x, na.rm = TRUE)**2)
})
MS4A1glmm &lt;- glmmSeq(~ Timepoint * EULAR_6m + (1 | PATID),
                     countdata = tpm[1:2, ],
                     metadata = metadata,
                     dispersion = disp,
                     verbose = FALSE)
names(attributes(MS4A1glmm))

</code></pre>


</div>