<div class="container">

<table style="width: 100%;"><tr>
<td>glmpca</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GLM-PCA</h2>

<h3>Description</h3>

<p>Generalized principal components analysis for 
dimension reduction of non-normally distributed data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">glmpca(
  Y,
  L,
  fam = c("poi", "nb", "nb2", "binom", "mult", "bern"),
  minibatch = c("none", "stochastic", "memoized"),
  optimizer = c("avagrad", "fisher"),
  ctl = list(),
  sz = NULL,
  nb_theta = NULL,
  X = NULL,
  Z = NULL,
  init = list(factors = NULL, loadings = NULL),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>matrix-like object of count or binary data with features as rows 
and observations as columns. Sparse matrices from the <code>Matrix</code> 
package are supported. Column-oriented sparsity is preferred.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>L</code></td>
<td>
<p>desired number of latent dimensions (positive integer).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fam</code></td>
<td>
<p>string describing the likelihood to use for the data. Families
include Poisson ('<code>poi</code>'), negative binomial with global 
overdispersion ('<code>nb</code>'), negative binomial with feature-specific 
overdispersion ('<code>nb2</code>'), or binomial ('<code>binom</code>'). Families 
'<code>mult</code>' and '<code>bern</code>' are deprecated as both are special cases of
'<code>binom</code>' with <code>sz</code> set to NULL and 1, respectively. They are 
provided only for backward compatibility. Family '<code>nb2</code>' has not been
thoroughly tested and is considered experimental.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minibatch</code></td>
<td>
<p>string describing whether gradients should be computed with
all observations ('<code>none</code>', the default) or a subset of observations, 
which is useful for larger datasets. Option '<code>stochastic</code>' computes
a noisy estimate of the full gradient using a random sample of observations
at each iteration. Option '<code>memoized</code>' computes the full data 
gradient under memory constraints by caching summary statistics across
batches of observations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>optimizer</code></td>
<td>
<p>string describing whether to use the fast AvaGrad method
('<code>avagrad</code>', the default) or the slower diagonal Fisher scoring 
method ('<code>fisher</code>') that was used in the original glmpca 
implementation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ctl</code></td>
<td>
<p>a list of control parameters. See 'Details'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sz</code></td>
<td>
<p>numeric vector of size factors for each observation. If NULL 
(the default), colSums are used for family '<code>binom</code>', and 
colMeans are used for families '<code>poi</code>','<code>nb</code>', and '<code>nb2</code>'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nb_theta</code></td>
<td>
<p>initial value for negative binomial overdispersion 
parameter(s). Small values lead to more overdispersion. Default: 100. See
<code>negative.binomial</code>. (<code>nb_theta</code>-&gt;<code class="reqn">\infty</code>
equivalent to Poisson).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>a matrix of column (observations) covariates. Any column with all
same values (eg. 1 for intercept) will be removed. This is because we force
a feature-specific intercept and want to avoid collinearity.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>
<p>a matrix of row (feature) covariates, usually not needed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>init</code></td>
<td>
<p>a list containing initial estimates for the factors (<code>U</code>) 
and loadings (<code>V</code>) matrices.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional named arguments. Provided only for backward 
compatibility.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The basic model is <code class="reqn">R = AX'+ZG'+VU'</code>, where <code class="reqn">E[Y] = M
  = linkinv(R)</code>. Regression coefficients are <code>A</code> and <code>G</code>, latent
factors are <code>U</code> and loadings are <code>V</code>. 
The objective is to minimize the deviance between <code>Y</code> 
and <code>M</code>. The deviance quantifies the goodness-of-fit of the GLM-PCA
model to the data (smaller=better). 
Note that <code>glmpca</code> uses a random initialization, 
so for fully reproducible results one may use <code>set.seed</code>.
</p>
<p>The <code>ctl</code> argument accepts any of the following optional components:
</p>

<dl>
<dt>verbose</dt>
<dd>
<p>Logical. Should detailed status messages be printed 
during the optimization run? Default: <code>FALSE</code>.</p>
</dd>
<dt>batch_size</dt>
<dd>
<p>Positive integer. How many observations should be
included in a minibatch? Larger values use more memory but lead to 
more accurate gradient estimation. Ignored if <code>minibatch='none'</code>.
Default: 1000.</p>
</dd>
<dt>lr</dt>
<dd>
<p>Positive scalar. The AvaGrad learning rate. Large values
enable faster convergence but can lead to numerical instability.
Default: 0.1. If a numerical divergence occurs, <code>glmpca</code>
will restart the optimization <code>maxTry</code> times (see below)
and reduce the learning rate by a factor of five each time.</p>
</dd>
<dt>penalty</dt>
<dd>
<p>Non-negative scalar. The L2 penalty for the latent 
factors. Default: 1. Regression coefficients are not penalized. Only
used by the Fisher scoring optimizer. Larger values improve numerical
stability but bias the parameter estimates. If a numerical divergence 
occurs, <code>glmpca</code> will restart the optimization <code>maxTry</code> times
(see below) and increase the penalty by a factor of five each time.</p>
</dd>
<dt>maxTry</dt>
<dd>
<p>Positive integer. In case of numerical divergence, how
many times should optimization be restarted with a more stable penalty
or learning rate? Default: 10.</p>
</dd>
<dt>minIter</dt>
<dd>
<p>Positive integer. Minimum number of iterations (full
passes through the dataset) before checking for numerical convergence.
Default: 30.</p>
</dd>
<dt>maxIter</dt>
<dd>
<p>Positive integer. Maximum number of iterations. If
numerical convergence is not achieved by this point, the results may
not be reliable and a warning is issued. Default: 1000.</p>
</dd>
<dt>tol</dt>
<dd>
<p>Positive scalar. Relative tolerance for assessing convergence.
Convergence is determined by comparing the deviance at the previous
iteration to the current iteration. Default: 1e-4.</p>
</dd>
<dt>epsilon</dt>
<dd>
<p>Positive scalar. AvaGrad hyperparameter. See Savarese et 
al (2020). Default: 0.1.</p>
</dd>
<dt>betas</dt>
<dd>
<p>Numeric vector of length two. AvaGrad hyperparameters. 
See Savarese et al (2020). Default: <code>c(0.9, 0.999)</code>.</p>
</dd>
<dt>minDev</dt>
<dd>
<p>Scalar. Minimum deviance threshold at which optimization
is terminated. Useful for comparing different algorithms as it avoids
the need to determine numerical convergence. Default: NULL</p>
</dd>
</dl>
<h3>Value</h3>

<p>An S3 object of class <code>glmpca</code> with copies of input components 
<code>optimizer</code>, <code>minibatch</code>, <code>ctl</code>,<code>X</code>, and <code>Z</code>,
along with the following additional fitted components:
</p>

<dl>
<dt>factors</dt>
<dd>
<p>a matrix <code>U</code> whose rows match the columns 
(observations) of <code>Y</code>. It is analogous to the principal components
in PCA. Each column of the factors matrix is a different latent 
dimension.</p>
</dd>
<dt>loadings</dt>
<dd>
<p>a matrix <code>V</code> whose rows match the rows 
(features/dimensions) of <code>Y</code>. It is analogous to loadings in PCA. 
Each column of the loadings matrix is a different latent dimension.</p>
</dd>
<dt>coefX</dt>
<dd>
<p>a matrix <code>A</code> of coefficients for the 
observation-specific covariates matrix <code>X</code>. Each row of coefX 
corresponds to a row of <code>Y</code> and each column corresponds to a 
column of <code>X</code>. The first column of coefX contains feature-specific 
intercepts which are included by default.</p>
</dd>
<dt>coefZ</dt>
<dd>
<p>a matrix <code>G</code> of coefficients for the feature-specific 
covariates matrix <code>Z</code>. Each row of coefZ corresponds to a column 
of <code>Y</code> and each column corresponds to a column of <code>Z</code>. By 
default no such covariates are included and this is returned as NULL.</p>
</dd>
<dt>dev</dt>
<dd>
<p>a vector of deviance values. The length of the vector is the 
number of iterations it took for GLM-PCA's optimizer to converge. 
The deviance should generally decrease over time. 
If it fluctuates wildly, this often indicates numerical instability, 
which can be improved by decreasing the learning rate or increasing the 
penalty, see <code>ctl</code>.</p>
</dd>
<dt>dev_smooth</dt>
<dd>
<p>a locally smoothed version of <code>dev</code> that may be
easier to visualize when <code>minibatch='stochastic'</code>.</p>
</dd>
<dt>glmpca_family</dt>
<dd>
<p>an S3 object of class glmpca_family. This is a minor
extension to the family or negative.binomial
object used by functions like glm and 
glm.nb. It is basically a list with various internal 
functions and parameters needed to optimize the GLM-PCA objective 
function. For the negative binomial case, it also contains the final 
estimated value of the overdispersion parameter (<code>nb_theta</code>).</p>
</dd>
<dt>offsets</dt>
<dd>
<p>For Poisson and negative binomial families, the offsets
are the logarithmically transformed size factors. These are needed to
compute the predicted mean values.</p>
</dd>
</dl>
<h3>References</h3>

<p>Savarese P, McAllester D, Babu S, and Maire M (2020).
Domain-independent Dominance of Adaptive Methods. <em>arXiv</em>
<a href="https://arxiv.org/abs/1912.01823">https://arxiv.org/abs/1912.01823</a>
</p>
<p>Townes FW (2019). Generalized Principal Component Analysis. <em>arXiv</em>
<a href="https://arxiv.org/abs/1907.02647">https://arxiv.org/abs/1907.02647</a>
</p>
<p>Townes FW, Hicks SC, Aryee MJ, and Irizarry RA (2019).
Feature Selection and Dimension Reduction for Single Cell RNA-Seq based on a 
Multinomial Model. <em>Genome Biology</em>
<a href="https://doi.org/10.1186/s13059-019-1861-6">https://doi.org/10.1186/s13059-019-1861-6</a>
</p>


<h3>See Also</h3>

<p><code>predict.glmpca</code>, 
<code>prcomp</code>, <code>glm</code>,
<code>logisticSVD</code>,
<code>scry::devianceFeatureSelection</code>, 
<code>scry::nullResiduals</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#create a simple dataset with two clusters
mu&lt;-rep(c(.5,3),each=10)
mu&lt;-matrix(exp(rnorm(100*20)),nrow=100)
mu[,1:10]&lt;-mu[,1:10]*exp(rnorm(100))
clust&lt;-rep(c("red","black"),each=10)
Y&lt;-matrix(rpois(prod(dim(mu)),mu),nrow=nrow(mu))
#visualize the latent structure
res&lt;-glmpca(Y, 2)
factors&lt;-res$factors
plot(factors[,1],factors[,2],col=clust,pch=19)

</code></pre>


</div>