<div class="container">

<table style="width: 100%;"><tr>
<td>kappa_calc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genetic correlation effects</h2>

<h3>Description</h3>

<p>This function estimates the correlation effect of a genetic variant on a pair of quantitative traits.
The effect (kappa) is measured on Fisher transformed correlation values.
The genotype is coded as 0 (non-carrier), 1 (single copy of effect allele) and 2 (two copies of effect allele).
We use a likelihood ratio test with 1 degree of freedom:
* H0: z_0 = z_1 = z_2,
* H1: z_j = z_null +kappa*j,
where z_j is the Fisher-transformed correlation between the traits amongst subjects of genotype j.
</p>
<p>If the traits follow a joint normal distribution, then (thoretically) the Fisher-transformed sample correlation (z) is approximately normally distributed with mean 0 and
standard error 1/sqrt(n-3).
Otherwise, we have to correct for possible bias in the estimators and scale the weights we assign to them appropriately
</p>
<p>Note, that even though each trait follows a normal distribution individually, that does not necessarily imply that the pair of them follow a joint normal distirbution.
</p>


<h3>Usage</h3>

<pre><code class="language-R">kappa_calc(qt1, qt2, g, weight_scale = 1, bias_scale = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>qt1</code></td>
<td>
<p>A numeric vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qt2</code></td>
<td>
<p>A numeric vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>An integer vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weight_scale</code></td>
<td>
<p>Used to appropriately scale the weight assigned to the correlation estimators</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bias_scale</code></td>
<td>
<p>Used to appropriately scale the bias of the correlation estimators</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with the values:
</p>
<p>* kappa, the estimated correlation effect
* pval, the p-value of the likelihood ratio test
</p>


<h3>Examples</h3>

<pre><code class="language-R">Sigma0 &lt;- matrix(c(1,0,0,1),nrow=2,ncol=2)
Sigma1 &lt;- matrix(c(1,0.3,0.3,1),nrow=2,ncol=2)
Sigma2 &lt;- matrix(c(1,0.6,0.6,1),nrow=2,ncol=2)
geno_vec &lt;- c(rep(0,10000),rep(1,1000),rep(2,100))
Q0 &lt;- MASS::mvrnorm(n = 10000, mu = c(0,0), Sigma = Sigma0)
Q1 &lt;- MASS::mvrnorm(n = 1000, mu = c(0,0), Sigma = Sigma1)
Q2 &lt;- MASS::mvrnorm(n = 100, mu = c(0,0), Sigma = Sigma2)
Q &lt;- rbind(Q0,Q1)
Q &lt;- rbind(Q,Q2)
qt1_val &lt;- Q[,1]
qt2_val &lt;- Q[,2]
res &lt;- kappa_calc(qt1_val, qt2_val, geno_vec)
</code></pre>


</div>