<div class="container">

<table style="width: 100%;"><tr>
<td>alpha.calc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genetic variance effects</h2>

<h3>Description</h3>

<p>This function estimates the variance effect of a genetic variant on a quantitative trait.
The genotype is coded as 0 (non-carrier), 1 (single copy of effect allele) and 2 (two copies of effect allele).
The variance effect (alpha) is modelled to be multiplicitive.
We use a likelihood ratio test with 1 degree of freedom,
* H0: y~N(mu_g,sigma^2)
* H1: y~N(mu_g,sigma^2\*alpha^g)
Under the alternative model, the variance of the trait changes multiplicatively
with genotype:
Var(y|g=j)=alpha^j*sigma^2, j in 0,1,2
</p>


<h3>Usage</h3>

<pre><code class="language-R">alpha.calc(qt, g)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>qt</code></td>
<td>
<p>A numeric vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>An integer vector.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with the values:
</p>
<p>* alpha, the estimated variance effect
* sigma2_alt, The variance estimated for non-carriers
* pval, the p-value of the likelihood ratio test
* chi2, the chi squared statistics (with one degree of freedom) corrisponding to the p-value
</p>


<h3>Examples</h3>

<pre><code class="language-R">n_val &lt;- 50000L
geno_vec &lt;- sample(c(0, 1, 2), size = n_val, replace = TRUE)
qt_vec &lt;- rnorm(n_val) * (1.1^geno_vec)
res &lt;- alpha.calc(qt_vec, geno_vec)
</code></pre>


</div>