<div class="container">

<table style="width: 100%;"><tr>
<td>dominance.calc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genetic dominance effects</h2>

<h3>Description</h3>

<p>This function estimates the dominance effect of a genetic variant on a quantitatvie trait
Nothing fancy here. We apply a simple linear regression model to estimate dominance effects.
We include a linear term, coded as 0,1 and 2 for non-carriers, heterozygotes and homozygous carriers of the effect allele.
We also include a dominance term, coded as 1 for homozygous carriers and 0 for others.
Effect size and significance is based on the dominance term.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dominance.calc(
  qt,
  g,
  round_imputed = FALSE,
  covariates = as.data.frame(matrix(0, nrow = 0, ncol = 0))
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>qt</code></td>
<td>
<p>A numeric vector</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>A vector with (possibly imputed) genotype values. All entries should be larger than 0 and smaller than 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>round_imputed</code></td>
<td>
<p>A boolian variable determining whether imputed genotype values should be rounded to the nearest integer in the analysis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covariates</code></td>
<td>
<p>A dataframe containing any covariates that should be used; one column per covariate.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with the dominanc effect and corresponding standard error, t statistic and p-value
</p>


<h3>Examples</h3>

<pre><code class="language-R">g_vec &lt;- rbinom(100000, 2, 0.3)
qt_vec &lt;- rnorm(100000) + 0.2 * g_vec^2
res &lt;- dominance.calc(qt_vec, g_vec)
</code></pre>


</div>