<div class="container">

<table style="width: 100%;"><tr>
<td>GOVenn</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Venn diagram of differentially expressed genes.</h2>

<h3>Description</h3>

<p>The function compares lists of differentially expressed genes 
and illustrates possible relations.Additionally it represents the variety 
of gene expression patterns within the intersection in small pie charts 
with three segements. Clockwise are shown the number of commonly up- 
regulated, commonly down- regulated and contra- regulated genes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GOVenn(data1, data2, data3, title, label, lfc.col, circle.col, plot = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data1</code></td>
<td>
<p>A data frame consisting of two columns: ID, logFC</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data2</code></td>
<td>
<p>A data frame consisting of two columns: ID, logFC</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data3</code></td>
<td>
<p>A data frame consisting of two columns: ID, logFC</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>The title of the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>A character vector to define the legend keys</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lfc.col</code></td>
<td>
<p>A character vector determining the background colors of the 
pie segments representing up- and down- regulated genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>circle.col</code></td>
<td>
<p>A character vector to assign clockwise colors for the 
circles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>If TRUE only the venn diagram is plotted. Otherwise the function 
returns a list with two items: the actual plot and a list containing the 
overlap entries (default= TRUE)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The <code>plot</code> argument can be used to adjust the amount of 
information that is returned by calling the function. If you are only 
interested in the actual plot of the venn diagram, <code>plot</code> should be 
set to TRUE. Sometimes you also want to know the elements of the 
intersections. In this case <code>plot</code> should be set to FALSE and the 
function call will return a list of two items. The first item, that can be 
accessed by $plot, contains the plotting information. Additionally, a list
($table) will be returned containing the elements of the various overlaps.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
#Load the included dataset
data(EC)

#Generating the circ object
circ&lt;-circular_dat(EC$david, EC$genelist)

#Selecting terms of interest
l1&lt;-subset(circ,term=='heart development',c(genes,logFC))
l2&lt;-subset(circ,term=='plasma membrane',c(genes,logFC))
l3&lt;-subset(circ,term=='tissue morphogenesis',c(genes,logFC))

GOVenn(l1,l2,l3, label=c('heart development','plasma membrane','tissue morphogenesis'))

## End(Not run)
</code></pre>


</div>