<div class="container">

<table style="width: 100%;"><tr>
<td>pltr.glm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Partially tree-based regression model function
</h2>

<h3>Description</h3>

<p>The <code>pltr.glm</code> function is designed to fit an hybrid glm model with  an additive tree part on a glm scale.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pltr.glm(data, Y.name, X.names, G.names, family = "binomial", 
    args.rpart = list(cp = 0, minbucket = 20, maxdepth = 10), 
    epsi = 0.001, iterMax = 5, iterMin = 3, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>

<p>a data frame containing the variables in the model
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y.name</code></td>
<td>

<p>the name  of the dependent variable
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X.names</code></td>
<td>

<p>the names of independent variables to consider in the linear part of the <code>glm</code>
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G.names</code></td>
<td>

<p>the names of independent variables to consider in the tree part of the hybrid <code>glm</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>family</code></td>
<td>

<p>the <code>glm</code> family considered depending on the type of the dependent variable.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>args.rpart</code></td>
<td>

<p>a list of options that control details of the rpart algorithm. <code>minbucket</code>: the minimum number of observations in any terminal &lt;leaf&gt; node; <code>cp</code>: complexity parameter (Any split that does not decrease the overall lack of fit by a factor of cp is not attempted); <code>maxdepth</code>:  the maximum depth of any node of the final tree, with the root node counted as depth 0. ... 
See <code>rpart.control</code> for further details
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epsi</code></td>
<td>

<p>a treshold value  to check the convergence of the algorithm
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iterMax</code></td>
<td>

<p>the maximal number of iteration to consider
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iterMin</code></td>
<td>

<p>the minimum number of iteration to consider
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical; TRUE for printing progress during the computation (helpful for debugging)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The <code>pltr.glm</code> function use an itterative procedure to fit the linear part of the <code>glm</code> and the tree part. The tree obtained at the convergence of the procedure is a maximal tree which overfits the data. It's then mandatory to prunned back this tree by using one of the proposed criteria (<code>BIC</code>, <code>AIC</code> and <code>CV</code>).
</p>


<h3>Value</h3>

<p>A list with four elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>fit </code></td>
<td>
<p>the glm fitted on the confounding factors at the end of the iterative algorithm</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree </code></td>
<td>
<p>the maximal tree obtained at the end of the algorithm</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nber_iter</code></td>
<td>
<p>the number of iterations used by the algorithm</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Timediff</code></td>
<td>
<p>The execution time of the iterative procedure</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p> The tree obtained at the end of these itterative procedure usually overfits the data. It's therefore mendatory to use either <code>best.tree.BIC.AIC</code> or <code>best.tree.CV</code> to prunne back the tree.
</p>


<h3>Author(s)</h3>

<p>Cyprien Mbogning and Wilson Toussile
</p>


<h3>References</h3>

<p>Mbogning, C., Perdry, H., Toussile, W., Broet, P.: A novel tree-based procedure for deciphering the genomic spectrum of clinical disease entities. Journal of Clinical Bioinformatics 4:6, (2014)
</p>
<p>Terry M. Therneau, Elizabeth J. Atkinson (2013) An Introduction to Recursive Partitioning Using the <code>RPART</code> Routines. Mayo Foundation.
</p>
<p>Chen, J., Yu, K., Hsing, A., Therneau, T.M.: A partially linear tree-based regression model for assessing complex joint gene-gene and gene-environment effects. Genetic Epidemiology
31, 238-251 (2007)
</p>


<h3>See Also</h3>

<p><code>rpart</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(burn)

args.rpart &lt;- list(minbucket = 10, maxdepth = 4, cp = 0, maxcompete = 0, 
                    maxsurrogate = 0)
 family &lt;- "binomial"
 X.names = "Z2"
 Y.name = "D2"
 G.names = c('Z1','Z3','Z4','Z5','Z6','Z7','Z8','Z9','Z10','Z11')
 
pltr.burn &lt;- pltr.glm(burn, Y.name, X.names, G.names, args.rpart = args.rpart,
                   family = family, iterMax = 4, iterMin = 3, verbose = FALSE)


## Not run: 
## load the data set

data(data_pltr)

## set the parameters 

args.rpart &lt;- list(minbucket = 40, maxdepth = 10, cp = 0)
family &lt;- "binomial"
Y.name &lt;- "Y"
X.names &lt;- "G1"
G.names &lt;- paste("G", 2:15, sep="")

## build a maximal tree 

fit_pltr &lt;- pltr.glm(data_pltr, Y.name, X.names, G.names, args.rpart = args.rpart, 
                    family = family,iterMax = 5, iterMin = 3)
                    
plot(fit_pltr$tree, main = 'MAXIMAL TREE')
text(fit_pltr$tree, minlength = 0L, xpd = TRUE, cex = .6)

## End(Not run)
</code></pre>


</div>