<div class="container">

<table style="width: 100%;"><tr>
<td>gorth</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find orthologs.</h2>

<h3>Description</h3>

<p>Interface to the g:Orth tool. Organism names are constructed by concatenating
the first letter of the name and the family name. Example: human - 'hsapiens',
mouse - 'mmusculus'.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gorth(query, source_organism = "hsapiens",
  target_organism = "mmusculus", region_query = F, numeric_ns = "",
  mthreshold = Inf, filter_na = T, df = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>query</code></td>
<td>
<p>list of gene IDs to be translated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>source_organism</code></td>
<td>
<p>name of the source organism.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>target_organism</code></td>
<td>
<p>name of the target organism.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region_query</code></td>
<td>
<p>interpret query as chromosomal ranges.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numeric_ns</code></td>
<td>
<p>namespace to use for fully numeric IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mthreshold</code></td>
<td>
<p>maximum number of ortholog names per gene to show.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_na</code></td>
<td>
<p>logical indicating whether to filter out results without a
corresponding target name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>logical indicating whether the output will be a data.frame or list.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>To alleviate the problem of having many orthologs per gene (most of them
uninformative) one can set a threshold for the number of results. The program
tries to find the most informative by selecting the most popular ones.
</p>


<h3>Value</h3>

<p>The output can be either a list or a data.frame. The list has an entry
for every input gene. The data frame is a table closely corresponding to the
web interface output.
</p>


<h3>Author(s)</h3>

<p>Raivo Kolde &lt;rkolde@gmail.com&gt;, Juri Reimand &lt;juri.reimand@ut.ee&gt;,
Tambet Arak &lt;tambet.arak@gmail.com&gt;
</p>


<h3>References</h3>

<p>J. Reimand, M. Kull, H. Peterson, J.  Hansen, J. Vilo: g:Profiler
â€“ a web-based toolset for functional profiling of gene lists from large-scale
experiments (2007) NAR 35 W193-W200
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
 gorth(c("Klf4","Pax5","Sox2","Nanog"), source_organism="mmusculus", target_organism="hsapiens")

## End(Not run)
</code></pre>


</div>