<div class="container">

<table style="width: 100%;"><tr>
<td>GeneticAlg.int</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> 
Genetic Algorithm with Integer Chromosomes
</h2>

<h3>Description</h3>

<p>Uses genetic algorithm to find the minima of a given cost function.
It evolves chromosomes with limited-range integers as codons.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GeneticAlg.int(genomeLen, codonMin, codonMax, 
    genomeMin = rep.int(codonMin, genomeLen), 
    genomeMax = rep.int(codonMax, genomeLen), 
    suggestions = NULL, popSize = 50, 
    iterations = 100, terminationCost = NA, 
    mutationChance = 1/(genomeLen+1), elitism = floor(popSize/10), 
    geneCrossoverPoints = NULL,
    monitorFunc = NULL, evalFunc, allowrepeat = TRUE, 
    showSettings = FALSE, verbose = FALSE, plapply = lapply)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genomeLen</code></td>
<td>

<p>Number of integers (i.e, codons) in chromosome.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>codonMin</code></td>
<td>

<p>Minimum integer value range for all codons.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>codonMax</code></td>
<td>

<p>Maximum integer value range for all codons.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomeMin</code></td>
<td>

<p>A vector of length <code>genomeLen</code> containing fine-grained control
over each codon's minimum. Overrides <code>codonMin</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomeMax</code></td>
<td>

<p>A vector of length <code>genomeLen</code> containing fine-grained control
over each codon's maximum. Overrides <code>codonMax</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestions</code></td>
<td>

<p>A list of suggested chromosomes to be used in the initial population.
Alternatively, an m-by-n matrix, where m is the number of suggestions and n is the chromosome length.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>popSize</code></td>
<td>

<p>Size of the population.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iterations</code></td>
<td>

<p>Number of generations to evolve the population.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>terminationCost</code></td>
<td>

<p>Target cost. If the best chromosome's cost reaches this value
the algorithm terminates.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutationChance</code></td>
<td>

<p>The chance of a codon being mutated. It must be between 0 and 1.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneCrossoverPoints</code></td>
<td>

<p>Codon groupings (genes) to be considered while crossover occurs. 
If given, odd and even codon groups are exchanged between parents.
Otherwise random points are selected and a classic single-point crossover is performed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>elitism</code></td>
<td>

<p>Number of top ranking chromosomes that are directly transfered to next generation without
going through evolutionary operations.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>monitorFunc</code></td>
<td>

<p>A function that is called at each generation. Can be used to monitor evolution of population.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>evalFunc</code></td>
<td>

<p>The cost function.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allowrepeat</code></td>
<td>

<p>Allows or forbids repeated integers in the chromosome.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showSettings</code></td>
<td>

<p>Enables printing GA settings.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>Enables verbose debugging info.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plapply</code></td>
<td>

<p><code>lapply</code> function used for mapping chromosomes to cost function.
See details for parallelization tips.
</p>
</td>
</tr>
</table>
<h3>Details</h3>


<p><code>GeneticAlg.int</code> implements evolutionary algorithms with
chromosomes created from integer values in the range of <code>codonMin</code> to
<code>codonMax</code>. <code>genomeMin</code> and <code>genomeMax</code> allow fine-grained
control of range for individual codons.
It first creates an initial population, using suggested inputs
<code>suggestions</code> and randomly generated chromosomes. 
Cost of each chromosome is evaluated using the cost function
<code>evalFunc</code>. If one of the chromosomes reaches
<code>terminationCost</code>, the algorithm terminates; 
Otherwise evolutionary operators including selection, cross-over
and mutation are applied to the population to create a new generation.
This iteration is continued until the required cost is reached
or the number of generations exceeds <code>iterations</code>.
</p>
<p>At each generation, the supplied <code>monitorFunc</code> is called with a
list similar to <code>GeneticAlg.int</code> returning value as its argument. 
</p>
<p>The <code>evalFunc</code> receives integer sequences and must return a numeric value.
The goal of optimization would be to find a chromosome which minimizes this value.
</p>
<p>To parallelize cost function evaluation, set <code>plapply</code> to a parallelized
<code>lapply</code>, such as <code>mclapply</code> from package <code>parallel</code>. 
In functions that do not handle data dependencies such as <code>parLapply</code>, 
variables and functions required for correct execution of <code>evalFunc</code> 
must be exported to worker nodes before invoking <code>GeneticAlg.int</code>.
</p>


<h3>Value</h3>

<p>A list containing information about 
<code>settings</code>, <code>population</code>, and the <code>best</code> chromosome.
</p>
<table>
<tr style="vertical-align: top;">
<td><code>settings$genomeMin</code></td>
<td>

<p>Minimum of each codon. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Settings$genomeMax</code></td>
<td>

<p>Maximum of each codon. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>settings$popSize</code></td>
<td>

<p>Size of the population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>settings$elitism</code></td>
<td>

<p>Number of elite individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>settings$iterations</code></td>
<td>

<p>Number of maximum generations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>settings$suggestions</code></td>
<td>

<p>Suggested chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>settings$mutationChance</code></td>
<td>

<p>Mutation chance.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>settings$geneCrossoverPoints</code></td>
<td>

<p>Cross-over points.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>population$population</code></td>
<td>

<p>The genomic data of the current population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>population$evaluations</code></td>
<td>

<p>Cost of the latest generation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>population$best</code></td>
<td>

<p>Historical cost of the best chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>population$mean</code></td>
<td>

<p>Historical mean cost of population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>population$currentIteration</code></td>
<td>

<p>Number of generations evolved until now.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>best$genome</code></td>
<td>

<p>The best chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>best$cost</code></td>
<td>

<p>The cost of the best chromosome.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>This function is partially inspired by <code>genalg</code> package
by Egon Willighagen. See <a href="https://cran.r-project.org/package=genalg">https://cran.r-project.org/package=genalg</a>.
</p>


<h3>See Also</h3>

<p><code>GrammaticalEvolution</code>, 
<code>EvolutionStrategy.int</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# define the evaluate function
evalfunc &lt;- function(l) {
    # maximize the odd indices and minimize the even indices
    # no repeated values are allowed
    odd &lt;- seq(1, 20, 2)
    even &lt;- seq(2, 20, 2)
    err &lt;- sum(l[even]) - sum(l[odd]);

    stopifnot(!any(duplicated(l))) # no duplication allowed

    return (err)
}

monitorFunc &lt;- function(result) {
    cat("Best of gen: ", min(result$best$cost), "\n")
}

x &lt;- GeneticAlg.int(genomeLen = 20, codonMin = 0, codonMax = 20,
                allowrepeat = FALSE, terminationCost = -110,
                monitorFunc = monitorFunc, evalFunc = evalfunc)

print(x)

best.result &lt;- x$best$genome
print("Odds:")
print(sort(best.result[seq(1, 20, 2)]))
print("Evens:")
print(sort(best.result[seq(2, 20, 2)]))
</code></pre>


</div>