<div class="container">

<table style="width: 100%;"><tr>
<td>PlotMismatchPositionForType</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Diagnostic plot for mismatch position for columns (by mismatch type)</h2>

<h3>Description</h3>

<p>This belongs to the first diagnostic plots (raw mismatches) generated by GRAND3.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PlotMismatchPositionForType(
  data,
  genomic,
  read,
  orientation = NULL,
  category = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a grandR object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomic</code></td>
<td>
<p>the nucleotide as it occurs in the genome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>read</code></td>
<td>
<p>the nucleotide as it occurs in the read</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>orientation</code></td>
<td>
<p>restrict to either Sense or Antisense; can be NULL</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>category</code></td>
<td>
<p>restrict to a specific category (see GetDiagnosticParameters); can be NULL</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For all positions along the reads (x axis; potentially paired end, shown left and right),
show the percentage of a specific mismatch type for all samples.
Positions outside of shaded areas are clipped. Uncorrected and Retained means before and
after correcting multiply sequenced bases. Sense/Antisense means reads (first read for paired end)
that are (based on the annotation) oriented in sense or antisense direction to a gene
(i.e. this is only relevant for sequencing protocols that do not preserve strand information).
</p>


<h3>Value</h3>

<p>a list with a ggplot object, a description, and the desired size for the plot
</p>


</div>