<div class="container">

<table style="width: 100%;"><tr>
<td>SimulateTimeCourse</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate a complete time course of metabolic labeling - nucleotide conversion RNA-seq data.</h2>

<h3>Description</h3>

<p>This function takes a vector of <em>true</em> synthesis rates and RNA half-lives, and then simulates
data for multiple time points and replicates. Both synthesis rate and RNA half-lives are assumed to be constant,
but the system might not be in steady-state.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SimulateTimeCourse(
  condition,
  gene.info,
  s,
  d,
  f0 = s/d,
  s.variation = 1,
  d.variation = 1,
  dispersion,
  num.reads = 1e+07,
  timepoints = c(0, 0, 0, 1, 1, 1, 2, 2, 2, 4, 4, 4),
  beta.approx = FALSE,
  conversion.reads = FALSE,
  verbose = TRUE,
  seed = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>condition</code></td>
<td>
<p>A user-defined condition name (which is placed into the <code>Coldata</code> of the final grandR object)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.info</code></td>
<td>
<p>either a data frame containing gene annotation or a vector of gene names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s</code></td>
<td>
<p>a vector of synthesis rates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>a vector of degradation rates (to get a specific half-life HL, use d=log(2)/HL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>f0</code></td>
<td>
<p>the abundance at time t=0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s.variation</code></td>
<td>
<p>biological variability of s among all samples (see details)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d.variation</code></td>
<td>
<p>biological variability of d among all samples (see details)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dispersion</code></td>
<td>
<p>a vector of dispersion parameters (estimate from data using DESeq2, e.g. by the estimate.dispersion utility function)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.reads</code></td>
<td>
<p>a vector representing the number of reads for each sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>timepoints</code></td>
<td>
<p>a vector representing the labeling duration (in h) for each sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.approx</code></td>
<td>
<p>should the beta approximation of the NTR posterior be computed?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conversion.reads</code></td>
<td>
<p>also output the number of reads with conversion</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Print status updates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>seed value for the random number generator (set to make it deterministic!)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>provided to <code>SimulateReadsForSample</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If <em>s.variation</em> or <em>d.variation</em> are &gt; 1, then for each gene a random gaussian is added to s (or d)
such that 90
of the gaussian is log2(s.variation).
</p>


<h3>Value</h3>

<p>a grandR object containing the simulated data in its data slots and the true parameters in the gene annotation table
</p>


</div>