<div class="container">

<table style="width: 100%;"><tr>
<td>ReadGRAND3</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read the output of GRAND-SLAM 3.0 into a grandR object.</h2>

<h3>Description</h3>

<p>Metabolic labeling - nucleotide conversion RNA-seq data (such as generated by SLAM-seq,TimeLapse-seq or TUC-seq)
must be carefully analyzed to remove bias due to incomplete labeling. GRAND-SLAM is a software package that
employs a binomial mixture modeling approach to obtain precise estimates of the new-to-total RNA ratio (NTR) per gene and sample (or cell).
This function directly reads the output of GRAND-SLAM 3.0 into a grandR object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ReadGRAND3(
  prefix,
  design = NULL,
  label = "4sU",
  estimator = "Binom",
  classify.genes = ClassifyGenes(),
  read.posterior = NULL,
  rename.sample = NULL,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>prefix</code></td>
<td>
<p>the prefix used to call GRAND-SLAM</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>design</code></td>
<td>
<p>Either a design vector (see details), or a data.frame providing metadata for all columns (samples/cells),
or a function that is called with the condition name vector and is supposed to return this data.frame. if NULL, a
library,sample,barcode design is used for sparse data, and a condition,replicate design for dense data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>which nucleoside analog</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimator</code></td>
<td>
<p>which estimator to use (one of Binom,TbBinom,TbBinomShape)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classify.genes</code></td>
<td>
<p>A function that is used to add the <em>type</em> column to the gene annotation table, always a call to ClassifyGenes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>read.posterior</code></td>
<td>
<p>also read the posterior parameters alpha and beta? if NULL, TRUE for dense data, FALSE for sparse data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rename.sample</code></td>
<td>
<p>function that is applied to each sample name before parsing (or NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Print status updates</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If columns (samples/cells) are named systematically in a particular way, the design vector provides
a powerful and easy way to create the column annotations.
</p>
<p>The column names have to contain dots (.) to separate the fields for the column annotation table.
E.g. the name <em>Mock.4h.A</em> will be split into the fields <em>Mock</em>, <em>4h</em> and  <em>A</em>.
For such names, a design vector of length 3 has to be given, that describes the meaning of each field.
A reasonable design vector for the example would be <code>c("Treatment","Time","Replicate")</code>.
Some names are predefined in the list Design.
</p>
<p>The names given in the design vector might even have additional semantics:
E.g. for the name <em>duration.4sU</em> the values are interpreted (e.g. 4h is converted into the number 4,
or 30min into 0.5, or no4sU into 0). Semantics can be user-defined by calling <code>MakeColdata</code>
and using the return value as the design parameter, or a function that calls MakeColdata.
In most cases it is easier to manipulate the <code>Coldata</code> table after loading data instead of using this mechanism;
the build-in semantics simply provide a convenient way to reduce this kind of manipulation in most cases.
</p>
<p>Sometimes you might have forgotten to name all samples consistently (or you simply messed something up).
In this case, the rename.sample parameter can be handy (e.g. to rename a particular misnamed sample).
</p>


<h3>Value</h3>

<p>A grandR object containing the read counts, NTRs, information on the NTR posterior distribution (alpha,beta)
and potentially additional information of all genes detected by GRAND-SLAM
</p>


<h3>See Also</h3>

<p>ReadGRAND,ClassifyGenes,MakeColdata,DesignSemantics
</p>


</div>