<div class="container">

<table style="width: 100%;"><tr>
<td>ReadNewTotal</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read sparse new/total matrices</h2>

<h3>Description</h3>

<p>This function can be used to load matrix market data in case genes were quantified by
(i) counting all reads (for total RNA) and (ii) counting T-to-C mismatch reads (for new RNA)
</p>


<h3>Usage</h3>

<pre><code class="language-R">ReadNewTotal(
  genes,
  cells,
  new.matrix,
  total.matrix,
  detection.rate = 1,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>csv file (or URL) containing gene information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cells</code></td>
<td>
<p>csv file (or URL) containing cell information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>new.matrix</code></td>
<td>
<p>Matrix market file of new counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>total.matrix</code></td>
<td>
<p>Matrix market file of total counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>detection.rate</code></td>
<td>
<p>the detection rate of T-to-C mismatch reads (see details)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>verbose output</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Metabolic labeling - nucleotide conversion RNA-seq data (such as generated by SLAM-seq,TimeLapse-seq or TUC-seq)
must be carefully analyzed to remove bias due to incomplete labeling. We advice against counting read with and without T-to-C mismatches
for quantification, and encourage using a statistical method such as GRAND-SLAM that properly deals with incomplete labeling.
</p>
<p>To correct for some bias, a detection rate (as suggested by Cao et al., Nature Biotech 2020) should be provided. This detection rate
defines, how much new RNA is detected on average using the T-to-C mismatch reads.
</p>


<h3>Value</h3>

<p>a grandR object
</p>


</div>