<div class="container">

<table style="width: 100%;"><tr>
<td>SimulateTimeCourseNonConstant</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate a complete time course of metabolic labeling - nucleotide conversion RNA-seq data.</h2>

<h3>Description</h3>

<p>This function takes a vector of <em>true</em> synthesis rates and RNA half-lives, and then simulates
data for multiple time points and replicates. Both synthesis rate and RNA half-lives are assumed to be constant,
but the system might not be in steady-state.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SimulateTimeCourseNonConstant(
  condition,
  gene.info,
  s,
  d,
  dispersion,
  num.reads = 1e+07,
  t = 2,
  replicates = 3,
  beta.approx = FALSE,
  conversion.reads = FALSE,
  verbose = TRUE,
  seed = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>condition</code></td>
<td>
<p>A user-defined condition name (which is placed into the <code>Coldata</code> of the final grandR object)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.info</code></td>
<td>
<p>either a data frame containing gene annotation or a vector of gene names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s</code></td>
<td>
<p>a vector of synthesis rates (see details)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>a vector of degradation rates (see details)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dispersion</code></td>
<td>
<p>a vector of dispersion parameters (estimate from data using DESeq2, e.g. by the estimate.dispersion utility function)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.reads</code></td>
<td>
<p>a vector representing the number of reads for each sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>t</code></td>
<td>
<p>a single number denoting the time</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replicates</code></td>
<td>
<p>a single number denoting the number of replicates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.approx</code></td>
<td>
<p>should the beta approximation of the NTR posterior be computed?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conversion.reads</code></td>
<td>
<p>also output the number of reads with conversion</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Print status updates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>seed value for the random number generator (set to make it deterministic!)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>provided to <code>SimulateReadsForSample</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Both rates can be either (i) a single number (constant rate), (ii) a data frame with names "offset",
"factor" and "exponent" (for linear functions, see ComputeNonConstantParam; only one row allowed) or
(iii) a unary function time-&gt;rate. Functions
</p>


<h3>Value</h3>

<p>a grandR object containing the simulated data in its data slots and the true parameters in the gene annotation table
</p>


<h3>See Also</h3>

<p>SimulateTimeCourse
</p>


</div>