<div class="container">

<table style="width: 100%;"><tr>
<td>ComputeSteadyStateHalfLives</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Steady state half-lives for each sample</h2>

<h3>Description</h3>

<p>Transforms each NTR to a half-life value (assuming steady state gene expression) and puts them into a new slot or adds an analysis
</p>


<h3>Usage</h3>

<pre><code class="language-R">ComputeSteadyStateHalfLives(
  data,
  time = Design$dur.4sU,
  name = "HL",
  columns = NULL,
  max.HL = 48,
  CI.size = 0.95,
  compute.CI = FALSE,
  as.analysis = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>the grandR object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>time</code></td>
<td>
<p>either a number indicating the labeling time, or a name of the Coldata table</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>the name of the new slot/analysis to put half-life values in</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>columns</code></td>
<td>
<p>which columns (i.e. samples or cells) to return; sets as.analysis to TRUE (see details)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.HL</code></td>
<td>
<p>all values above this will be set to this</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CI.size</code></td>
<td>
<p>A number between 0 and 1 representing the size of the credible interval</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>compute.CI</code></td>
<td>
<p>it TRUE, credible intervals are computed, this also sets as.analysis to TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>as.analysis</code></td>
<td>
<p>if TRUE add the results as analysis and not as data slot</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>An NTR value p can be transformed into an RNA half-live using the equation
log(2)/(-1/t*log(1-p))
This is described in our GRAND-SLAM paper (Juerges et al., Bioinformatics 2018).
</p>
<p>Columns can be given as a logical, integer or character vector representing a selection of the columns (samples or cells).
The expression is evaluated in an environment havin the <code>Coldata</code>, i.e. you can use names of <code>Coldata</code> as variables to
conveniently build a logical vector (e.g., columns=Condition=="x").
</p>


<h3>Value</h3>

<p>a new grandR object with an additional slot or analysis
</p>


</div>