<div class="container">

<table style="width: 100%;"><tr>
<td>mat_gen_dist</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute a pairwise matrix of genetic distances between populations</h2>

<h3>Description</h3>

<p>The function computes a pairwise matrix of genetic distances
between populations and allows to implement several formula.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mat_gen_dist(x, dist = "basic", null_val = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>genind</code> that contains the multilocus
genotypes (format 'locus') of the individuals as well as their populations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dist</code></td>
<td>
<p>A character string indicating the method used to compute the
multilocus genetic distance between populations
</p>

<ul>
<li>
<p>If 'dist = 'basic” (default), then the multilocus genetic distance is
computed using a formula of Euclidean genetic
distance (Excoffier et al., 1992)
</p>
</li>
<li>
<p>If 'dist = 'weight”, then the multilocus genetic distance is computed
as in Fortuna et al. (2009). It is a Euclidean genetic distance giving more
weight to rare alleles
</p>
</li>
<li>
<p>If 'dist = 'PG”, then the multilocus genetic distance is computed as
in popgraph::popgraph function, following several steps of PCA and SVD
(Dyer et Nason, 2004).
</p>
</li>
<li>
<p>If 'dist = 'DPS”, then the genetic distance used is equal to
1 - the proportion of shared alleles (Bowcock, 1994)
</p>
</li>
<li>
<p>If 'dist = 'FST”, then the genetic distance used is the pairwise
FST (Weir et Cockerham, 1984)
</p>
</li>
<li>
<p>If 'dist = 'FST_lin”, then the genetic distance used is the linearised
pairwise FST (Weir et Cockerham, 1984)(FST_lin = FST/(1-FST))
</p>
</li>
<li>
<p>If 'dist = 'PCA”, then the genetic distance is computed following a
PCA of the matrix of allelic frequencies by population. It is a
Euclidean genetic distance between populations in the multidimensional
space defined by all the independent principal components.
</p>
</li>
<li>
<p>If 'dist = 'GST”, then the genetic distance used is the
G'ST (Hedrick, 2005). See graph4lg &lt;= 1.6.0 only, because it used diveRsity
</p>
</li>
<li>
<p>If 'dist = 'D”, then the genetic distance used is
Jost's D (Jost, 2008). See graph4lg &lt;= 1.6.0 only, because it used diveRsity
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>null_val</code></td>
<td>
<p>(optional) Logical. Should negative and null FST, FST_lin,
GST or D values be replaced by half the minimum positive value?
This option allows to compute Gabriel graphs from these "distances".
Default is null_val = FALSE.
This option only works if 'dist = 'FST” or 'FST_lin' or 'GST' or 'D'</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Negative values are converted into 0.
Euclidean genetic distance <code class="reqn">d_{ij}</code> between population i and j
is computed as follows:
</p>
<p style="text-align: center;"><code class="reqn">d_{ij}^{2} = \sum_{k=1}^{n} (x_{ki} - x_{kj})^{2} </code>
</p>
<p> where
<code class="reqn">x_{ki}</code> is the allelic frequency of allele k in population i and n is
the total number of alleles. Note that when 'dist = 'weight”, the formula
becomes </p>
<p style="text-align: center;"><code class="reqn">d_{ij}^{2} = \sum_{k=1}^{n} (1/(K*p_{k}))(x_{ki} - x_{kj})^{2}</code>
</p>

<p>where K is the number of alleles at the locus of the allele k and <code class="reqn">p_{k}</code>
is the frequency of the allele k in all populations.
Note that when 'dist = 'PCA”, n is the number of conserved independent
principal components and <code class="reqn">x_{ki}</code> is the value taken by the principal
component k in population i.
</p>


<h3>Value</h3>

<p>An object of class <code>matrix</code>
</p>


<h3>Author(s)</h3>

<p>P. Savary
</p>


<h3>References</h3>

<p>Bowcock AM, Ruiz-Linares A, Tomfohrde J, Minch E, Kidd JR, Cavalli-Sforza LL (1994).
“High resolution of human evolutionary trees with polymorphic microsatellites.”
<em>nature</em>, <b>368</b>(6470), 455–457.
Excoffier L, Smouse PE, Quattro JM (1992).
“Analysis of molecular variance inferred from metric distances among DNA haplotypes: application to human mitochondrial DNA restriction data.”
<em>Genetics</em>, <b>131</b>(2), 479–491.
Dyer RJ, Nason JD (2004).
“Population graphs: the graph theoretic shape of genetic structure.”
<em>Molecular ecology</em>, <b>13</b>(7), 1713–1727.
Fortuna MA, Albaladejo RG, Fernández L, Aparicio A, Bascompte J (2009).
“Networks of spatial genetic variation across species.”
<em>Proceedings of the National Academy of Sciences</em>, <b>106</b>(45), 19044–19049.
Weir BS, Cockerham CC (1984).
“Estimating F-statistics for the analysis of population structure.”
<em>evolution</em>, <b>38</b>(6), 1358–1370.
Hedrick PW (2005).
“A standardized genetic differentiation measure.”
<em>Evolution</em>, <b>59</b>(8), 1633–1638.
Jost L (2008).
“GST and its relatives do not measure differentiation.”
<em>Molecular ecology</em>, <b>17</b>(18), 4015–4026.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(data_ex_genind)
x &lt;- data_ex_genind
D &lt;- mat_gen_dist(x = x, dist = "basic")
</code></pre>


</div>