<div class="container">

<table style="width: 100%;"><tr>
<td>gen_graph_indep</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create an independence graph of genetic differentiation
from genetic data of class genind</h2>

<h3>Description</h3>

<p>The function allows to create genetic graphs from genetic data
by applying the conditional independence principle. Populations whose allelic
frequencies covary significantly once the covariance with the other
populations has been taken into account are linked on the graphs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gen_graph_indep(
  x,
  dist = "basic",
  cov = "sq",
  pcor = "magwene",
  alpha = 0.05,
  test = "EED",
  adj = "none",
  output = "igraph"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>genind</code> that contains the multilocus
genotype (format 'locus') of the individuals as well as their population
and their geographical coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dist</code></td>
<td>
<p>A character string indicating the method used to compute the
multilocus genetic distance between populations
</p>

<ul>
<li>
<p>If 'dist = 'basic” (default), then the multilocus genetic distance is
computed using a Euclidean genetic distance formula (Excoffier et al., 1992)
</p>
</li>
<li>
<p>If 'dist = 'weight”, then the multilocus genetic distance is computed
as in Fortuna et al. (2009). It is a Euclidean genetic distance giving more
weight to rare alleles
</p>
</li>
<li>
<p>If 'dist = 'PG”, then the multilocus genetic distance is computed as
in popgraph::popgraph function, following several steps of PCA and SVD
(Dyer et Nason, 2004).
</p>
</li>
<li>
<p>If 'dist = 'PCA”, then the genetic distance is computed following a
PCA of the matrix of allelic frequencies by population. It is a Euclidean
genetic distance between populations in the multidimensional space defined
by all the independent principal components.
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cov</code></td>
<td>
<p>A character string indicating the formula used to compute the
covariance matrix from the distance matrix
</p>

<ul>
<li>
<p>If 'cov = 'sq” (default), then the covariance matrix is calculated
from the matrix of squared distances as in Everitt et Hothorn (2011)
</p>
</li>
<li>
<p>If 'cov = 'dist”, then the covariance matrix is calculated from the
matrix of distances as in Dyer et Nason (2004) and popgraph function
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pcor</code></td>
<td>
<p>A character string indicating the way the partial correlation
matrix is computed from the covariance matrix.
</p>

<ul>
<li>
<p>If 'pcor = 'magwene”, the steps followed are the same as in
Magwene (2001) and in popgraph::popgraph function. It is the recommended
option as it meets mathematical requirements.
</p>
</li>
<li>
<p>If 'pcor = 'other”, the steps followed are the same as used
by Fortuna et al. (2009). They are not consistent with the approach
of Magwene (2001).
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>A numeric value corresponding to the statistical tolerance
threshold used to test the difference from 0 of the partial correlation
coefficients. By default, 'alpha=0.05'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test</code></td>
<td>
<p>A character string indicating the method used to test the
significance of the partial correlation coefficients.
</p>

<ul><li>
<p>If 'test = 'EED” (default), then the Edge Exclusion Deviance
criterion is used (Whittaker, 2009). Although other methods exist, this is
the most common and thus the only one implemented here.
</p>
</li></ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adj</code></td>
<td>
<p>A character string indicating the way of adjusting p-values to
assess the significance of the p-values
</p>

<ul>
<li>
<p>If 'adj = 'none” (default), there is no p-value adjustment correction
</p>
</li>
<li>
<p>If 'adj = 'holm”, p-values are adjusted using the sequential
Bonferroni correction (Holm, 1979)
</p>
</li>
<li>
<p>If 'adj = 'bonferroni”, p-values are adjusted using the classic
Bonferroni correction
</p>
</li>
<li>
<p>If 'adj = 'BH”, p-values are adjusted using Benjamini et Hochberg
(1995) correction controlling false discovery rate
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>A character string indicating the matrices included in
the output list.
</p>

<ul>
<li>
<p>If 'output = 'all” (default), then D (distance matrix),
C (covariance matrix), Rho (partial correlation matrix),
M (graph incidence matrix) and S (strength matrix) are included
</p>
</li>
<li>
<p>If 'output = 'dist_graph”, then the distance matrix D is returned
only with the values corresponding to the graph edges
</p>
</li>
<li>
<p>If 'output = 'str_graph”, then the strength values matrix S is
returned only with the values corresponding to the graph edges
</p>
</li>
<li>
<p>If 'output = 'inc”, then the binary adjacency matrix M is returned
</p>
</li>
<li>
<p>If 'output = 'igraph”, then a graph of class <code>igraph</code>
is returned
</p>
</li>
</ul>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function allows to vary many parameters such as the genetic
distance used, the formula used to compute the covariance, the statistical
tolerance threshold, the p-values adjustment, among others.
</p>


<h3>Value</h3>

<p>A <code>list</code> of objects of class <code>matrix</code>, an object of
class <code>matrix</code> or a graph object of class <code>igraph</code>
</p>


<h3>Author(s)</h3>

<p>P. Savary
</p>


<h3>References</h3>

<p>Dyer RJ, Nason JD (2004).
“Population graphs: the graph theoretic shape of genetic structure.”
<em>Molecular ecology</em>, <b>13</b>(7), 1713–1727.
Benjamini Y, Hochberg Y (1995).
“Controlling the false discovery rate: a practical and powerful approach to multiple testing.”
<em>Journal of the royal statistical society. Series B (Methodological)</em>, 289–300.
Bowcock AM, Ruiz-Linares A, Tomfohrde J, Minch E, Kidd JR, Cavalli-Sforza LL (1994).
“High resolution of human evolutionary trees with polymorphic microsatellites.”
<em>nature</em>, <b>368</b>(6470), 455–457.
Everitt B, Hothorn T (2011).
<em>An introduction to applied multivariate analysis with R</em>.
Springer.
Excoffier L, Smouse PE, Quattro JM (1992).
“Analysis of molecular variance inferred from metric distances among DNA haplotypes: application to human mitochondrial DNA restriction data.”
<em>Genetics</em>, <b>131</b>(2), 479–491.
Fortuna MA, Albaladejo RG, Fernández L, Aparicio A, Bascompte J (2009).
“Networks of spatial genetic variation across species.”
<em>Proceedings of the National Academy of Sciences</em>, <b>106</b>(45), 19044–19049.
Holm S (1979).
“A simple sequentially rejective multiple test procedure.”
<em>Scandinavian journal of statistics</em>, 65–70.
Magwene PM (2001).
“New tools for studying integration and modularity.”
<em>Evolution</em>, <b>55</b>(9), 1734–1745.
Wermuth N, Scheidt E (1977).
“Algorithm AS 105: fitting a covariance selection model to a matrix.”
<em>Journal of the Royal Statistical Society. Series C (Applied Statistics)</em>, <b>26</b>(1), 88–92.
Whittaker J (2009).
<em>Graphical models in applied multivariate statistics</em>.
Wiley Publishing.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(data_ex_genind)
dist_graph_test &lt;- gen_graph_indep(x = data_ex_genind, dist = "basic",
                             cov = "sq", pcor = "magwene",
                             alpha = 0.05, test = "EED",
                             adj = "none", output = "igraph")
</code></pre>


</div>