<div class="container">

<table style="width: 100%;"><tr>
<td>GROAN.run</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compare Genomic Regressors on a Noisy Dataset</h2>

<h3>Description</h3>

<p>This function runs the experiment described in a GROAN.Workbench object,
training regressor(s) on the data contained in a GROAN.NoisyDataSet object
via parameter <code>nds</code>. The prediction accuracy is estimated either through crossvalidation
or on separate test dataset supplied via parameter <code>nds.test</code>.
It returns a <code>GROAN.Result</code> object, which have a summary
function for quick inspection and can be fed to plotResult for visual comparisons.
In case of crossvalidation the test dataset in the result object will report the <code>[CV]</code>
suffix.<br>
The experiment statistics are computed via measurePredictionPerformance.<br>
Each time this function is invoked it will refer to a <code>runId</code> - an alphanumeric string identifying
each specific run. The <code>runId</code> is usually generated internally, but it is possible to pass it if
the intention is to join results from different runs for analysis purposes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GROAN.run(nds, wb, nds.test = NULL, run.id = createRunId())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nds</code></td>
<td>
<p>a GROAN.NoisyDataSet object, containing the data (genotypes, phenotypes and so forth) plus a <code>noiseInjector</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wb</code></td>
<td>
<p>a GROAN.Workbench object, containing the regressors to be tested together with the description of the experiment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nds.test</code></td>
<td>
<p>either a GROAN.NoisyDataSet or a list of GROAN.NoisyDataSet. The regression model(s) trained
on <code>nds</code> will be tested on <code>nds.test</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>run.id</code></td>
<td>
<p>an alphanumeric string identifying this specific run. If not passed it is generated using createRunId</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a <code>GROAN.Result</code> object
</p>


<h3>See Also</h3>

<p>measurePredictionPerformance
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
#Complete examples are found in the vignette
vignette('GROAN.vignette', package='GROAN')

#Minimal example
#1) creating a noisy dataset with normal noise
nds = createNoisyDataset(
  name = 'PEA KI, normal noise',
  genotypes = GROAN.KI$SNPs,
  phenotypes = GROAN.KI$yield,
  noiseInjector = noiseInjector.norm,
  mean = 0,
  sd = sd(GROAN.KI$yield) * 0.5
)

#2) creating a GROAN.WorkBench using default regressor and crossvalidation preset
wb = createWorkbench()

#3) running the experiment
res = GROAN.run(nds, wb)

#4) examining results
summary(res)
plotResult(res)

## End(Not run)
</code></pre>


</div>