<div class="container">

<table style="width: 100%;"><tr>
<td>GROAN.KI</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Example data for pea KI lines</h2>

<h3>Description</h3>

<p>This list contains all data required to run GROAN examples. It refers to a pea experiment
with 103 lines coming from a biparental Kaspa x Isard cross.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GROAN.KI
</code></pre>


<h3>Format</h3>

<p>A list with the following fields:
</p>

<ul>
<li>
<p><em>"GROAN.KI$yield"</em>: named array with 103 slots, containing data on grain yield [t/ha]
</p>
</li>
<li>
<p><em>"GROAN.KI$SNPs"</em>: data frame with 103 rows and 647 variables. Each row is
a pea KI line, each column a SNP marker. Values can either be 0, 1, or 2,
representing the three possible genotypes (AA, Aa, and aa, respectively).
</p>
</li>
<li>
<p><em>"GROAN.KI$kinship"</em>: square dataframe containing the realized kinships between
all pairs of each of the 103 pea KI lines. Values were computed following the
<a href="https://www.jstor.org/stable/25681325">Astle &amp; Balding metric</a>.
Higher values represent a higher degree of genetic similarity between
lines. This metric mainly accounts for additive genetic contributions
(as an alternative to dominant contributions).
</p>
</li>
</ul>
<h3>Source</h3>

<p>Annicchiarico et al., <em>GBS-Based Genomic Selection
for Pea Grain Yield under Severe Terminal Drought</em>, The Plant Genome,
Volume 10. doi: <a href="https://doi.org/10.3835/plantgenome2016.07.0072">10.3835/plantgenome2016.07.0072</a>
</p>


</div>