<div class="container">

<table style="width: 100%;"><tr>
<td>GSAfisher.multiple</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GSAfisher.multiple</h2>

<h3>Description</h3>

<p>Multiple method of GSAfisher function that computes the p-value for a joint test of association between a phenotype and multiple gene-sets of genetic variants (SNPs) by combining multiple marginal p-values data vectors using the Fisher method.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'multiple'
GSAfisher(...)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Set of multiple marginal p-values data vectors or matrix.</p>
</td>
</tr></table>
<h3>Value</h3>

<table><tr style="vertical-align: top;">
<td><code>p.multiple</code></td>
<td>
<p>Set of multiple Fisher method combined p-values for each data vector.</p>
</td>
</tr></table>
<h3>Author(s)</h3>

<p>Carlos Garcia Prieto</p>


<h3>Examples</h3>

<pre><code class="language-R">#Generate 3 vectors with 1000 random values from an Uniform distribution U (0, 0.25). 
data.vector1&lt;-runif(1000, 0, 0.25)
data.vector2&lt;-runif(1000, 0, 0.25)
data.vector3&lt;-runif(1000, 0, 0.25)
#Set data class to "multiple".
class(data.vector1)&lt;-"multiple"
#Apply the GSAfisher.multiple function to these data vectors.
GSAfisher(data.vector1, data.vector2, data.vector3)

#You can also try:
#Generate 5000 vectors with 1000 random values from an Uniform distribution U(0,0.25).
data.vectors&lt;-sapply(1:5000, function (x) runif(1000, 0, 0.25))
#Set data class to "multiple".
class(data.vectors)&lt;-"multiple"
#Apply the GSAfisher.multiple function to these data vectors.
GSAfisher(data.vectors)
</code></pre>


</div>