<div class="container">

<table style="width: 100%;"><tr>
<td>plot_simulated_chromomsome_segments</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot the simulated chromosomes of an individual</h2>

<h3>Description</h3>

<p>This function uses the information in the tibble about segments
dropped down a genome simulation pedigree to plot the
chromomomes of an individual colored by
the population of origin of each segment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_simulated_chromomsome_segments(
  Segs,
  RR = NULL,
  fill_by_group_origin = FALSE,
  rel_heights = c(chrom_ht = 4, chrom_gap = 0.8, spark_gap = 0.2 * !is.null(RR),
    spark_box = 2.6 * !is.null(RR), unit_gap = 4),
  bottom_gap = 0.3,
  spark_thick = 0.2,
  spark_splat = 0.25
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Segs</code></td>
<td>
<p>a tibble of segments</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>RR</code></td>
<td>
<p>a tibble of recombination rates in bins in the format
of RecRates. If this is
included, the recombination rates in cM/Mb are plotted atop the chromosomes
as a little sparkline. If it is not included, then the there are
no little sparklines above the chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fill_by_group_origin</code></td>
<td>
<p>If FALSE (the default) the fill color of segments
is mapped to the pop_origin, which is where the founder haplotypes came from according
to the <code>hpop1</code> and <code>hpop2</code> columns in the GSP specification. If you set this
to TRUE, then we map the "group" column of the reppop to fill.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rel_heights</code></td>
<td>
<p>a vector the the relative heights of the different
elements of each chromosomal unit of the plot.  This is a named vector
with the following elements, listed in order of the bottom of each
unit to the top:
</p>

<ul>
<li> <p><code>chrom_ht</code>: the height of the bars for each of the two chromosomes of the
pair in a chromosome unit.
</p>
</li>
<li> <p><code>chrom_gap</code>: The gap between the two homologous chromosomes of the individual.
</p>
</li>
<li> <p><code>spark_gap</code>: the gap between the top chromosome and the sparkline box
for recombination rates.
</p>
</li>
<li> <p><code>spark_box</code>: height of the box within which the sparkline goes.  Note that
the sparkline itself will be scaled so that the highest rate anywhere within
the genome will correspond to the top of the spark box.
</p>
</li>
<li> <p><code>unit_gap</code>: The relative height of the gap between one chromosome unit and
the next.
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bottom_gap</code></td>
<td>
<p>the y value of the bottom chromosome unit.  Basically the
absolute distance between the y=0 line and the start of the plotted material.
Should typically be between 0 and 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spark_thick</code></td>
<td>
<p>thickness of the line that draws the recombination rate
sparkline.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spark_splat</code></td>
<td>
<p>fraction by which the unit gap should be reduced when
there are sparklines being drawn.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function returns a ggplot object.  Each facet of the plot shows
the chromosomes of a different sampled individual from a particular replicate
simulation from a particular genome simulation pedigree.  The facets are titled
like: <code style="white-space: pre;">⁠GSP 1, Idx 2, ID 8[3]⁠</code>, which means that the chromosomes shown in the panel
are from the third sampled set of chromosomes from the individual with ID 8 from the
simulation from genome simulation pedigree 1 with index 2.
</p>


<h3>Examples</h3>

<pre><code class="language-R">s &lt;- example_segments
rr &lt;- RecRates
g &lt;- plot_simulated_chromomsome_segments(s)
g_with_sparklines &lt;- plot_simulated_chromomsome_segments(s, rr)
</code></pre>


</div>