<div class="container">

<table style="width: 100%;"><tr>
<td>GSDA-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Gene Set Distance Analysis (GSDA)
</h2>

<h3>Description</h3>

<p>The gene-set distance analysis of omic data is implemented by generalizing distance correlations to evaluate the association of a gene set with categorical and censored event-time variables.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> GSDA</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Title: </td>
<td style="text-align: left;"> Gene Set Distance Analysis (GSDA)</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 1.0</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2021-01-014</td>
</tr>
<tr>
<td style="text-align: left;">
Authors@R: </td>
<td style="text-align: left;"> c(person("Xueyuan", "Cao", email = "xcao12@uthsc.edu", role = c("aut", "cre")),
             person("Stanley", "Pounds", email = "stanley.pounds@stjude.org", role = c("aut"))) </td>
</tr>
<tr>
<td style="text-align: left;">
Description: </td>
<td style="text-align: left;"> The gene-set distance analysis of omic data is implemented by generalizing distance correlations to evaluate the association of a gene set with categorical and censored event-time variables.</td>
</tr>
<tr>
<td style="text-align: left;">
Depends: </td>
<td style="text-align: left;"> R (&gt;= 3.5.0),msigdbr</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
<td style="text-align: left;">
biocViews: </td>
<td style="text-align: left;"> Microarray, Bioinformatics, Gene expression</td>
</tr>
<tr>
<td style="text-align: left;">
Suggests: </td>
<td style="text-align: left;"> knitr, rmarkdown</td>
</tr>
<tr>
<td style="text-align: left;">
VignetteBuilder: </td>
<td style="text-align: left;"> knitr</td>
</tr>
<tr>
<td style="text-align: left;">
LazyLoad: </td>
<td style="text-align: left;"> yes</td>
</tr>
<tr>
<td style="text-align: left;">
Author: </td>
<td style="text-align: left;"> Xueyuan Cao [aut, cre],
  Stanley Pounds [aut]</td>
</tr>
<tr>
<td style="text-align: left;">
Maintainer: </td>
<td style="text-align: left;"> Xueyuan Cao &lt;xcao12@uthsc.edu&gt;</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Index of help topics:
</p>
<pre>
GSDA-package            Gene Set Distance Analysis (GSDA)
U.center                U Centering
best.dist.corr          Best Distance Correlattion
cat.dist                Distance for a Categorical Variable
dist.corr               Distance Correlattion
gsda                    Gene-Set Distance Analsysis (GSDA)
kegg.ml.gsets           KEGG gene set data for the AML and CML pathways
prep.gsda               GSDA Data Preparation
prep.msigdb             Preparation of MSigDB for GSDA
print.bdc               Print Method for Best Distance Correlation
print.dcor              Print Method for Distance Correlation
print.gsda.result       Print Method for GSDA Result
surv.dist               Distance of a Survival Endpoint
target.aml.clin         Clinical outcomes for AML TARGET Project
target.aml.expr         RNA-seq expression from the AML TARGET project
uc.dist                 U-centered Distance Matrix
write.gsda.csv.file     Write GSDA Result to a Comma Delimited File
</pre>

<p>Further information is available in the following vignettes:<br><br></p>

<table>
<tr>
<td style="text-align: left;">
<code>GSDA</code> </td>
<td style="text-align: left;"> An Introduction to the GSDA Package (source, pdf)</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>NA
</p>
<p>Maintainer: NA
</p>


<h3>References</h3>

<p>Cao X and Pounds S (2021)  Gene-Set Distance Associations (GSDA): A Powerful Tool for Gene-Set Association Analysis.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(target.aml.clin)
data(target.aml.expr)
data(kegg.ml.gsets)
res=gsda(target.aml.expr,
         target.aml.clin,
         kegg.ml.gsets,
         "Chloroma","oe","ct")
</code></pre>


</div>