<div class="container">

<table style="width: 100%;"><tr>
<td>GS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genotypic and phenotypic simulated dataset</h2>

<h3>Description</h3>

<p>This dataset is simulated with the R package "qtlbim" where 10 are true features associated with the trait of study and remaining 100 are random markers. we consider 10 chromosomes each containing 10 markers. Each chromosome have 1 qtl which is the true feature.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("GS")</code></pre>


<h3>Format</h3>

<p>A data frame with 60 rows as genotypes with 111 columns (i.e. contains information of genotyped markers and phenotypic traits).</p>


<h3>Details</h3>

<p>It has total 60 rows which represents 200 individuals genotypes and a total of 111 of columns, in which first 110 columns contain information of genotyped markers and last column represents value of phenotypic trait associated with genotype under study.
</p>


<h3>Source</h3>

<p>Yandell, B. S., Mehta, T., Banerjee, S., Shriner, D., Venkataraman, R. et al. (2007). R/qtlbim: QTL with Bayesian Interval Mapping in experimental crosses. <em>Bioinformatics</em>, 23, 641-643.
<br> Yandell, B. S., Nengjun, Y., Mehta, T., Banerjee, S., Shriner, D. et al. (2012). qtlbim: QTL Bayesian Interval Mapping. R package version 2.0.5. http://CRAN.R-project.org/package=qtlbim
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(GSelection)
data(GS)
X&lt;-GS[,1:110]   ## Extracting Genotype
Y&lt;-GS[,111] ## Extracting Phenotype
</code></pre>


</div>