<div class="container">

<table style="width: 100%;"><tr>
<td>flatten_normal_tiss</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Flatten normal tissues</h2>

<h3>Description</h3>

<p>Given a matrix containing the expression values of
<code>n</code> healthy tissue samples, it produces the flattened vector matrix
as reported in "Disease-specific genomic analysis: identifying
the signature of pathology biology".
</p>


<h3>Usage</h3>

<pre><code class="language-R">flatten_normal_tiss(normal_tiss)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>normal_tiss</code></td>
<td>
<p>A normal tissue data gene expression matrix.
The columns should be the samples and the rows should be the genes.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>A gene expression matrix containing the flattened
version of the vectors.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
normal_tissue_matrix &lt;- matrix(stats::rnorm(36), nrow=6)
flatten_normal_tiss(normal_tissue_matrix)

</code></pre>


</div>