<div class="container">

<table style="width: 100%;"><tr>
<td>calculate_ieqtls</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate Ieqtls</h2>

<h3>Description</h3>

<p>Calculate your own Cell Specific eQTLs.
</p>

<ul>
<li>
<p> This service calculates the gene-variant association for any given pair of gene and variant, which may or may not be significant.
</p>
</li>
<li>
<p> This requires as input a GENCODE ID, GTEx variant ID, and tissue site detail ID.
</p>
</li>
</ul>
<p>By default, the calculation is based on the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Dynamic-Association-Endpoints/operation/calculate_ieqtls_api_v2_association_dynieqtl_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">calculate_ieqtls(
  cellType,
  tissueSiteDetailId,
  gencodeId,
  variantId,
  datasetId = "gtex_v8"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cellType</code></td>
<td>
<p>String. "Adipocytes", "Epithelial_cells", "Hepatocytes",
"Keratinocytes", "Myocytes", "Neurons", "Neutrophils".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. "Whole_Blood"; use <code>get_tissue_site_detail()</code>
to see valid values) or an Ontology ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
"ENSG00000065613.9".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Dynamic Association Endpoints: 
<code>calculate_expression_quantitative_trait_loci()</code>,
<code>calculate_isqtls()</code>,
<code>calculate_splicing_quantitative_trait_loci()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# perform request
calculate_ieqtls(cellType = "Adipocytes",
                 tissueSiteDetailId = "Adipose_Subcutaneous",
                 gencodeId = "ENSG00000203782.5",
                 variantId = "chr1_1099341_T_C_b38")

## End(Not run)
</code></pre>


</div>