<div class="container">

<table style="width: 100%;"><tr>
<td>get_significant_single_tissue_eqtls</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Significant Single Tissue Eqtls</h2>

<h3>Description</h3>

<p>Find significant single tissue
eQTLs.
</p>

<ul>
<li>
<p> This service returns precomputed significant single tissue eQTLs.
</p>
</li>
<li>
<p> Results may be filtered by tissue, gene, variant or dataset.
</p>
</li>
<li>
<p> To search by gene, use the versioned GENCODE ID.
</p>
</li>
<li>
<p> To search by variant, use the dbSNP rs ID (snpId).
</p>
</li>
</ul>
<p>By default, the service queries the latest GTEx release and the retrieved data is split into pages with <code>items_per_page</code> entries per page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_significant_single_tissue_eqtls_api_v2_association_singleTissueEqtl_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_significant_single_tissue_eqtls(
  gencodeIds = NULL,
  variantIds = NULL,
  tissueSiteDetailIds = NULL,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c("ENSG00000132693.12", "ENSG00000203782.5").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variantIds</code></td>
<td>
<p>Character vector. Gtex variant IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. "Whole_Blood"; use
<code>get_tissue_site_detail()</code> to see valid values) or Ontology IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>page</code></td>
<td>
<p>Integer (default = 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><strong>Note:</strong> although the GTEx Portal API documentation says to use the dbSNP rsID
when searching by variant, this returns no results. Instead use gtex variant
IDs e.g. use "chr1_153209640_C_A_b38" instead of "rs1410858".
</p>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code>get_eqtl_genes()</code>,
<code>get_fine_mapping()</code>,
<code>get_independent_eqtl()</code>,
<code>get_multi_tissue_eqtls()</code>,
<code>get_significant_single_tissue_eqtls_by_location()</code>,
<code>get_significant_single_tissue_ieqtls()</code>,
<code>get_significant_single_tissue_isqtls()</code>,
<code>get_significant_single_tissue_sqtls()</code>,
<code>get_sqtl_genes()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# search by gene
get_significant_single_tissue_eqtls(gencodeIds = c("ENSG00000132693.12",
                                                  "ENSG00000203782.5"))

# search by variant - must be variantId (not rsid)
get_significant_single_tissue_eqtls(variantIds = "chr1_153209640_C_A_b38")

# filter by gene/variant and tissue site - either `gencodeIds` or `variantIds`
# should be supplied as a minimum
get_significant_single_tissue_eqtls(gencodeIds = c("ENSG00000132693.12",
                                                  "ENSG00000203782.5"),
                                    variantIds = "chr1_153209640_C_A_b38",
                                    tissueSiteDetailIds = "Whole_Blood")

## End(Not run)
</code></pre>


</div>