<div class="container">

<table style="width: 100%;"><tr>
<td>get_multi_tissue_eqtls</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Multi Tissue Eqtls</h2>

<h3>Description</h3>

<p>Find multi-tissue eQTL <code>Metasoft</code> results.
</p>

<ul>
<li>
<p> This service returns multi-tissue eQTL Metasoft results for a given gene and variant in a specified dataset.
</p>
</li>
<li>
<p> A Versioned GENCODE ID must be provided.
</p>
</li>
<li>
<p> For each tissue, the results include: m-value (mValue), normalized effect size (nes), p-value (pValue), and standard error (se).
</p>
</li>
<li>
<p> The m-value is the posterior probability that an eQTL effect exists in each tissue tested in the cross-tissue meta-analysis (Han and Eskin, PLoS Genetics 8(3): e1002555, 2012).
</p>
</li>
<li>
<p> The normalized effect size is the slope of the linear regression of normalized expression data versus the three genotype categories using single-tissue eQTL analysis, representing eQTL effect size.
</p>
</li>
<li>
<p> The p-value is from a t-test that compares observed NES from single-tissue eQTL analysis to a null NES of 0.
</p>
</li>
</ul>
<p>By default, the service queries the latest GTEx release. The retrieved data is split into pages with <code>items_per_page</code> entries per page
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_multi_tissue_eqtls_api_v2_association_metasoft_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_multi_tissue_eqtls(
  gencodeIds,
  variantId = NULL,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c("ENSG00000132693.12", "ENSG00000203782.5").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>page</code></td>
<td>
<p>Integer (default = 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code>get_eqtl_genes()</code>,
<code>get_fine_mapping()</code>,
<code>get_independent_eqtl()</code>,
<code>get_significant_single_tissue_eqtls()</code>,
<code>get_significant_single_tissue_eqtls_by_location()</code>,
<code>get_significant_single_tissue_ieqtls()</code>,
<code>get_significant_single_tissue_isqtls()</code>,
<code>get_significant_single_tissue_sqtls()</code>,
<code>get_sqtl_genes()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# search by gene
get_multi_tissue_eqtls(gencodeId = c("ENSG00000132693.12",
                                     "ENSG00000203782.5"))

## End(Not run)
</code></pre>


</div>