<div class="container">

<table style="width: 100%;"><tr>
<td>get_genomic_features</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Genomic Features</h2>

<h3>Description</h3>

<p><a href="https://gtexportal.org/api/v2/redoc#tag/Reference-Genome-Endpoints/operation/get_genomic_features_api_v2_reference_features__featureId__get">GTEx API Portal documentation</a>
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_genomic_features(.featureId, datasetId = "gtex_v8")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>.featureId</code></td>
<td>
<p>String. A genomic feature e.g. GENCODE ID, RSID or
GTEx Variant ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This endpoint takes a path parameter "featureId".
</p>


<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Reference Genome Endpoints: 
<code>get_exons()</code>,
<code>get_gene_search()</code>,
<code>get_genes()</code>,
<code>get_gwas_catalog_by_location()</code>,
<code>get_neighbor_gene()</code>,
<code>get_transcripts()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# gene symbol
get_genomic_features("brca1")

# GENCODE ID
get_genomic_features("ENSG00000132693.12")

# RSID
get_genomic_features("rs1815739")

# GTEx variant ID
get_genomic_features("chr11_66561023_G_GTTA_b38")

## End(Not run)
</code></pre>


</div>