<div class="container">

<table style="width: 100%;"><tr>
<td>gtexr_arguments</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>gtexr arguments</h2>

<h3>Description</h3>

<p>Internal function that documents all arguments for exported gtexr functions
that wrap GTEx Portal API endpoints (using roxygen <code style="white-space: pre;">⁠@inheritParams⁠</code> tag).
</p>


<h3>Usage</h3>

<pre><code class="language-R">gtexr_arguments(
  ageBrackets = NULL,
  aliquotIds = NULL,
  attributeSubset = NULL,
  autolysisScores = NULL,
  cellType = NULL,
  bp_window = NULL,
  chromosome = NULL,
  datasetId = NULL,
  dataTypes = NULL,
  draw = NULL,
  end = NULL,
  excludeDataArray = NULL,
  .featureId = NULL,
  filterMtGene = NULL,
  gencodeId = NULL,
  gencodeIds = NULL,
  gencodeVersion = NULL,
  geneId = NULL,
  geneIds = NULL,
  genomeBuild = NULL,
  hardyScales = NULL,
  hasExpressionData = NULL,
  hasGenotype = NULL,
  itemsPerPage = NULL,
  ischemicTime = NULL,
  ischemicTimeGroups = NULL,
  materialTypes = NULL,
  organizationName = NULL,
  page = NULL,
  pathCategory = NULL,
  phenotypeId = NULL,
  pos = NULL,
  project_id = NULL,
  rin = NULL,
  sampleId = NULL,
  sampleIds = NULL,
  searchTerm = NULL,
  sex = NULL,
  snpId = NULL,
  sortBy = NULL,
  sortDirection = NULL,
  start = NULL,
  subjectIds = NULL,
  tissueSampleIds = NULL,
  tissueSiteDetailId = NULL,
  tissueSiteDetailIds = NULL,
  uberonIds = NULL,
  variantId = NULL,
  variantIds = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ageBrackets</code></td>
<td>
<p>The age bracket(s) of the donors of interest. Options:
"20-29", "30-39", "40-49", "50-59", "60-69", "70-79".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aliquotIds</code></td>
<td>
<p>Character vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>attributeSubset</code></td>
<td>
<p>String. Examples include but are not limited to "sex", "ageBracket"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>autolysisScores</code></td>
<td>
<p>Character vector. Options: "None", "Mild", "Moderate",
"Severe".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cellType</code></td>
<td>
<p>String. "Adipocytes", "Epithelial_cells", "Hepatocytes",
"Keratinocytes", "Myocytes", "Neurons", "Neutrophils".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp_window</code></td>
<td>
<p>Integer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome</code></td>
<td>
<p>String. One of "chr1", "chr2", "chr3", "chr4", "chr5",
"chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13",
"chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21",
"chr22", "chrM", "chrX", "chrY".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dataTypes</code></td>
<td>
<p>Character vector. Options: "RNASEQ", "WGS", "WES", "OMNI",
"EXCLUDE".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>draw</code></td>
<td>
<p>Integer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>Integer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>excludeDataArray</code></td>
<td>
<p>String. Options are TRUE or FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.featureId</code></td>
<td>
<p>String. A genomic feature e.g. GENCODE ID, RSID or
GTEx Variant ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filterMtGene</code></td>
<td>
<p>Logical. Exclude mitochondrial genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
"ENSG00000065613.9".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c("ENSG00000132693.12", "ENSG00000203782.5").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gencodeVersion</code></td>
<td>
<p>String (default = "v26"). GENCODE annotation release.
Either "v26" or "v19".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneId</code></td>
<td>
<p>String. A gene symbol, a gencode ID, or an Ensemble ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneIds</code></td>
<td>
<p>A character vector of gene symbols, versioned gencodeIds, or
unversioned gencodeIds.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomeBuild</code></td>
<td>
<p>String. Options: "GRCh38/hg38", "GRCh37/hg19". Default =
"GRCh38/hg38".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hardyScales</code></td>
<td>
<p>Character vector. A list of Hardy Scale(s) of interest.
Options: "Ventilator case", "Fast death - violent", "Fast death - natural
causes", "Intermediate death", "Slow death".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hasExpressionData</code></td>
<td>
<p>Logical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hasGenotype</code></td>
<td>
<p>Logical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ischemicTime</code></td>
<td>
<p>Integer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ischemicTimeGroups</code></td>
<td>
<p>Character vector. Options:
"&lt;= 0", "1 - 300", "301 - 600", "601 - 900", "901 - 1200", "1201 - 1500", "&gt; 1500".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>materialTypes</code></td>
<td>
<p>String, vector. Options: "Cells:Cell Line Viable",
"DNA:DNA Genomic", "DNA:DNA Somatic", "RNA:Total RNA", "Tissue:PAXgene
Preserved", "Tissue:PAXgene Preserved Paraffin-embedded", "Tissue:Fresh
Frozen Tissue".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organizationName</code></td>
<td>
<p>String. Options: "GTEx Consortium" "Kid's First".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>page</code></td>
<td>
<p>Integer (default = 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pathCategory</code></td>
<td>
<p>Character vector. Options: "adenoma", "amylacea",
"atelectasis", "atherosclerosis", "atherosis", "atrophy", "calcification",
"cirrhosis", "clean_specimens", "congestion", "corpora_albicantia", "cyst",
"desquamation", "diabetic", "dysplasia", "edema", "emphysema",
"esophagitis", "fibrosis", "gastritis", "glomerulosclerosis", "goiter",
"gynecomastoid", "hashimoto", "heart_failure_cells", "hemorrhage",
"hepatitis", "hyalinization", "hypereosinophilia", "hyperplasia",
"hypertrophy", "hypoxic", "infarction", "inflammation", "ischemic_changes",
"macrophages", "mastopathy", "metaplasia", "monckeberg", "necrosis",
"nephritis", "nephrosclerosis", "no_abnormalities", "nodularity",
"pancreatitis", "pigment", "pneumonia", "post_menopausal", "prostatitis",
"saponification", "scarring", "sclerotic", "solar_elastosis",
"spermatogenesis", "steatosis", "sweat_glands", "tma".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenotypeId</code></td>
<td>
<p>String. See
<a href="https://www.gtexportal.org/home/faq#splicingPhenotypeId">GTEx portal FAQs</a>
for further details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>Integer, vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>project_id</code></td>
<td>
<p>String. Options: "gtex", "adult-gtex", "egtex".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rin</code></td>
<td>
<p>Integer vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampleId</code></td>
<td>
<p>String. <code style="white-space: pre;">⁠^GTEX-[A-Z0-9]{5}-[0-9]{4}-SM-[A-Z0-9]{5}$⁠</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampleIds</code></td>
<td>
<p>Character vector. GTEx sample ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>searchTerm</code></td>
<td>
<p>String.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex</code></td>
<td>
<p>String. Options: "male", "female".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpId</code></td>
<td>
<p>String</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sortBy</code></td>
<td>
<p>String. Options: "sampleId", "ischemicTime", "aliquotId",
"tissueSampleId", "hardyScale", "pathologyNotes", "ageBracket",
"tissueSiteDetailId", "sex".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sortDirection</code></td>
<td>
<p>String. Options: "asc", "desc". Default = "asc".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>Integer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subjectIds</code></td>
<td>
<p>Character vector. GTEx subject ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tissueSampleIds</code></td>
<td>
<p>Array of strings. A list of Tissue Sample ID(s).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tissueSiteDetailId</code></td>
<td>
<p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. "Whole_Blood"; use <code>get_tissue_site_detail()</code>
to see valid values) or an Ontology ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. "Whole_Blood"; use
<code>get_tissue_site_detail()</code> to see valid values) or Ontology IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>uberonIds</code></td>
<td>
<p>Character vector of Uberon IDs (e.g. "UBERON:EFO_0000572";
use <code>get_tissue_site_detail()</code> to see valid values).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variantId</code></td>
<td>
<p>String. A gtex variant ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variantIds</code></td>
<td>
<p>Character vector. Gtex variant IDs.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns <code>NULL</code> invisibly. Used for documentation only.
</p>


</div>