<div class="container">

<table style="width: 100%;"><tr>
<td>get_clustered_median_transcript_expression</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Clustered Median Transcript Expression</h2>

<h3>Description</h3>

<p>Find median transcript expression data of all known transcripts of a gene along with hierarchical clusters.
</p>

<ul>
<li>
<p> Returns median normalized expression in tissues of all known transcripts of a given gene along with the hierarchical clustering results of tissues and genes, based on expression, in Newick format.
</p>
</li>
<li>
<p> Results may be filtered by dataset, gene or tissue, but at least one gene must be provided.
</p>
</li>
<li>
<p> The hierarchical clustering is performed by calculating Euclidean distances and using the average linkage method.
</p>
</li>
<li> <p><strong>This endpoint is not paginated.</strong>
</p>
</li>
</ul>
<p>By default, this service queries the latest GTEx release.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_median_transcript_expression_api_v2_expression_medianTranscriptExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_clustered_median_transcript_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailIds = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c("ENSG00000132693.12", "ENSG00000203782.5").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. "Whole_Blood"; use
<code>get_tissue_site_detail()</code> to see valid values) or Ontology IDs.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A tibble, with clustering data stored as an attribute, "clusters".
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code>get_clustered_median_exon_expression()</code>,
<code>get_clustered_median_gene_expression()</code>,
<code>get_clustered_median_junction_expression()</code>,
<code>get_expression_pca()</code>,
<code>get_gene_expression()</code>,
<code>get_median_exon_expression()</code>,
<code>get_median_gene_expression()</code>,
<code>get_median_junction_expression()</code>,
<code>get_median_transcript_expression()</code>,
<code>get_single_nucleus_gex()</code>,
<code>get_single_nucleus_gex_summary()</code>,
<code>get_top_expressed_genes()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
get_clustered_median_transcript_expression(gencodeIds = c("ENSG00000203782.5",
                                                    "ENSG00000132693.12"))

# clustering data is stored as an attribute "clusters"
result &lt;- get_clustered_median_transcript_expression(c("ENSG00000203782.5",
                                                 "ENSG00000132693.12"))
attr(result, "clusters")

# process clustering data with the ape package
# install.packages("ape")
# phylo_tree &lt;- ape::read.tree(text = attr(result, "clusters")$tissue)
# plot(phylo_tree)
# print(phylo_tree)

## End(Not run)
</code></pre>


</div>