<div class="container">

<table style="width: 100%;"><tr>
<td>get_exons</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Exons</h2>

<h3>Description</h3>

<p>This service returns exons from all known transcripts of the given gene.
</p>

<ul>
<li>
<p> A versioned GENCODE ID is required to ensure that all exons are from a single gene.
</p>
</li>
<li>
<p> A dataset ID or both GENCODE version and genome build must be provided.
</p>
</li>
<li>
<p> Although annotated exons are not dataset dependent, specifying a dataset here is equivalent to specifying the GENCODE version and genome build used by that dataset.
</p>
</li>
</ul>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Reference-Genome-Endpoints/operation/get_exons_api_v2_reference_exon_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_exons(
  gencodeId,
  gencodeVersion = NULL,
  genomeBuild = NULL,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gencodeId</code></td>
<td>
<p>String. A Versioned GENCODE ID of a gene, e.g.
"ENSG00000065613.9".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gencodeVersion</code></td>
<td>
<p>String (default = "v26"). GENCODE annotation release.
Either "v26" or "v19".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomeBuild</code></td>
<td>
<p>String. Options: "GRCh38/hg38", "GRCh37/hg19". Default =
"GRCh38/hg38".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>page</code></td>
<td>
<p>Integer (default = 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Reference Genome Endpoints: 
<code>get_gene_search()</code>,
<code>get_genes()</code>,
<code>get_genomic_features()</code>,
<code>get_gwas_catalog_by_location()</code>,
<code>get_neighbor_gene()</code>,
<code>get_transcripts()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
get_exons(gencodeId = "ENSG00000203782.5")

## End(Not run)
</code></pre>


</div>