<div class="container">

<table style="width: 100%;"><tr>
<td>get_eqtl_genes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Eqtl Genes</h2>

<h3>Description</h3>

<p>Retrieve eGenes (eQTL Genes).
</p>

<ul>
<li>
<p> This service returns eGenes (eQTL Genes) from the specified dataset.
</p>
</li>
<li>
<p> eGenes are genes that have at least one significant cis-eQTL acting upon them.
</p>
</li>
<li>
<p> Results may be filtered by tissue. By default, the service queries the latest GTEx release.
</p>
</li>
</ul>
<p>For each eGene, the results include the allelic fold change
(log2AllelicFoldChange), p-value (pValue), p-value threshold
(pValueThreshold), empirical p-value (empiricalPValue), and q-value
(qValue).
</p>

<ul>
<li>
<p> The log2AllelicFoldChange is the allelic fold change (in log2 scale) of the most significant eQTL.
</p>
</li>
<li>
<p> The pValue is the nominal p-value of the most significant eQTL.
</p>
</li>
<li>
<p> The pValueThreshold is the p-value threshold used to determine whether a cis-eQTL for this gene is significant. For more details see https://gtexportal.org/home/documentationPage#staticTextAnalysisMethods.
</p>
</li>
<li>
<p> The empiricalPValue is the beta distribution-adjusted empirical p-value from FastQTL.
</p>
</li>
<li>
<p> The qValues were calculated based on the empirical p-values. A false discovery rate (FDR) threshold of &lt;= 0.05 was applied to identify genes with a significant eQTL.
</p>
</li>
</ul>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_eqtl_genes_api_v2_association_egene_get">GTEx Portal API documentation</a>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_eqtl_genes(
  tissueSiteDetailIds,
  datasetId = "gtex_v8",
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. "Whole_Blood"; use
<code>get_tissue_site_detail()</code> to see valid values) or Ontology IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>page</code></td>
<td>
<p>Integer (default = 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A tibble.
</p>


<h3>See Also</h3>

<p>Other Static Association Endpoints: 
<code>get_fine_mapping()</code>,
<code>get_independent_eqtl()</code>,
<code>get_multi_tissue_eqtls()</code>,
<code>get_significant_single_tissue_eqtls()</code>,
<code>get_significant_single_tissue_eqtls_by_location()</code>,
<code>get_significant_single_tissue_ieqtls()</code>,
<code>get_significant_single_tissue_isqtls()</code>,
<code>get_significant_single_tissue_sqtls()</code>,
<code>get_sqtl_genes()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
get_eqtl_genes(c("Whole_Blood", "Artery_Aorta"))

## End(Not run)
</code></pre>


</div>