<div class="container">

<table style="width: 100%;"><tr>
<td>get_single_nucleus_gex</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Single Nucleux Gex</h2>

<h3>Description</h3>

<p>Retrieve Single Nucleus Gene Expression Data for a given Gene.
</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_single_nucleus_gex_api_v2_expression_singleNucleusGeneExpression_get">GTEx Portal API documentation</a>
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_single_nucleus_gex(
  gencodeIds,
  datasetId = "gtex_snrnaseq_pilot",
  tissueSiteDetailIds = NULL,
  excludeDataArray = TRUE,
  page = 0,
  itemsPerPage = 250
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gencodeIds</code></td>
<td>
<p>A character vector of Versioned GENCODE IDs, e.g.
c("ENSG00000132693.12", "ENSG00000203782.5").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>datasetId</code></td>
<td>
<p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tissueSiteDetailIds</code></td>
<td>
<p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. "Whole_Blood"; use
<code>get_tissue_site_detail()</code> to see valid values) or Ontology IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>excludeDataArray</code></td>
<td>
<p>String. Options are TRUE or FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>page</code></td>
<td>
<p>Integer (default = 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>itemsPerPage</code></td>
<td>
<p>Integer (default = 250).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A tibble
</p>


<h3>See Also</h3>

<p>Other Expression Data Endpoints: 
<code>get_clustered_median_exon_expression()</code>,
<code>get_clustered_median_gene_expression()</code>,
<code>get_clustered_median_junction_expression()</code>,
<code>get_clustered_median_transcript_expression()</code>,
<code>get_expression_pca()</code>,
<code>get_gene_expression()</code>,
<code>get_median_exon_expression()</code>,
<code>get_median_gene_expression()</code>,
<code>get_median_junction_expression()</code>,
<code>get_median_transcript_expression()</code>,
<code>get_single_nucleus_gex_summary()</code>,
<code>get_top_expressed_genes()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Search for one or more genes - returns a tibble with one row per tissue.
# Column "cellTypes" now contains a tibble of expression summary data, with
# one row for each cell type
get_single_nucleus_gex(gencodeIds = c("ENSG00000203782.5",
                                      "ENSG00000132693.12"))

# `excludeDataArray = FALSE` - expression values are stored under "celltypes"
# in an additional column called "data"
response &lt;- get_single_nucleus_gex(gencodeIds = "ENSG00000132693.12",
                                   excludeDataArray = FALSE,
                                   itemsPerPage = 2)

response

# "cellTypes" contains a tibble of data with one row for each
# cell type e.g. for Breast_Mammary_Tissue
response$cellTypes[[2]]

# when `excludeDataArray = FALSE`, expression values are stored in "data"
# e.g. for Breast_Mammary_Tissue, Epithelial cell (luminal):
response$cellTypes[[2]]$data[[1]]

## End(Not run)
</code></pre>


</div>