<div class="container">

<table style="width: 100%;"><tr>
<td>man_plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Manhattan Plot</h2>

<h3>Description</h3>

<p>Creates a Manhattan plot
</p>


<h3>Usage</h3>

<pre><code class="language-R">man_plot(
  data,
  chromCol = c("gray44", "black"),
  genomewideline = -log10(5e-08),
  suggestiveline = -log10(1e-05),
  chromosome = "ALL",
  annotatePval = FALSE,
  annotateSNP = NULL,
  annotateCol = "red",
  highlight = NULL,
  highlightCol = "green3",
  title = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>PLINK assoc output, tab-delimited, or a data.frame with "SNP", "CHR", "BP", and "P" columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromCol</code></td>
<td>
<p>A character vector indicating which colors to alternate for the chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomewideline</code></td>
<td>
<p>Where to draw the "genome-wide significant" line. Default
-log10(5e-8). Set to FALSE or F to disable</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestiveline</code></td>
<td>
<p>Where to draw the "suggestive" line. Default
-log10(1e-5). Set to FALSE or F to disable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome</code></td>
<td>
<p>An integer indicating which chromosome to plot. Default is "ALL".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotatePval</code></td>
<td>
<p>If set, SNPs with p-value less than or equal to this p-value will be annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotateSNP</code></td>
<td>
<p>A character vector of SNPs in your dataset to annotate.
If some of the SNPs are not in your dataset, gwaRs will throw a warning message.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotateCol</code></td>
<td>
<p>A string denoting the color to use for the annotations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight</code></td>
<td>
<p>A character vector of SNPs in the dataset to highlight.
If some of the SNPs are not in your dataset, gwaRs will throw a warning message. Default is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlightCol</code></td>
<td>
<p>A string denoting the color to use to highlight the SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>A string denoting the title to use for the plot. Default is 'Manhattan Plot'</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Creates a Manhattan plot from PLINK assoc output (or any tab-delimited file or data frame with
"SNP", "CHR", "BP", and "P" columns).
</p>


<h3>Value</h3>

<p>A Manhattan plot.
</p>


<h3>Author(s)</h3>

<p>Lindokuhle Nkambule
</p>


<h3>Examples</h3>

<pre><code class="language-R">man_plot(gwasData)

</code></pre>


</div>