<div class="container">

<table style="width: 100%;"><tr>
<td>mirrored_man_plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Mirrored Manhattan Plot</h2>

<h3>Description</h3>

<p>Creates a Mirrored Manhattan Plot for two traits
</p>


<h3>Usage</h3>

<pre><code class="language-R">mirrored_man_plot(
  data,
  trait1 = NULL,
  trait2 = NULL,
  trait1_chromCols = c("gray66", "grey36"),
  trait2_chromCols = c("steelblue1", "steelblue4"),
  xlab = "Genomic Position (chromosome)",
  title = "Manhattan Plot",
  annotate_trait1_pval = FALSE,
  annotate_trait1_color = "red",
  annotate_trait2_pval = FALSE,
  annotate_trait2_color = "red",
  annotateSNP = NULL,
  annotateSNPcolor = "red",
  highlight = NULL,
  highlightcolor = "green3",
  genomewideline_trait1 = NULL,
  genomewideline_trait2 = NULL,
  genomewideline_type = "dashed",
  genomewideline_color = "red",
  suggestiveline_trait1 = NULL,
  suggestiveline_trait2 = NULL,
  suggestiveline_type = "dashed",
  suggestiveline_color = "blue"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>A tab-delimited or data frame with the compulsory columns:
"CHR", "SNP", "BP", "P", "Trait".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait1</code></td>
<td>
<p>A character string of the trait1 as it appears in the input data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait2</code></td>
<td>
<p>A character string of the trait2 as it appears in the input data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait1_chromCols</code></td>
<td>
<p>A character vector indicating which colors to alternate
for trait1 chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait2_chromCols</code></td>
<td>
<p>A character vector indicating which colors to alternate
for trait2 chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>A character string to be used as the x-axis label.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>A character string to be used as the plot title</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate_trait1_pval</code></td>
<td>
<p>If set, trait1 SNPs with p-value less than or equal to this
p-value will be annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate_trait1_color</code></td>
<td>
<p>A character string indicating the color to be used for annotating
trait1 SNPs by p-value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate_trait2_pval</code></td>
<td>
<p>If set, trait2 SNPs with p-value less than or equal to this
p-value will be annotated on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate_trait2_color</code></td>
<td>
<p>A character string indicating the color to be used for annotating
trait2 SNPs by p-value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotateSNP</code></td>
<td>
<p>A character vector of SNPs in your dataset to annotate.
If some of the SNPs are not in your dataset, gwaRs will throw a warning message.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotateSNPcolor</code></td>
<td>
<p>A character string denoting the color to use for the annotations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight</code></td>
<td>
<p>A character vector of SNPs in the dataset to highlight.
If some of the SNPs are not in your dataset, gwaRs will throw a warning message. Default is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlightcolor</code></td>
<td>
<p>A character string denoting the color to use to highlight the SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomewideline_trait1</code></td>
<td>
<p>Where to draw the "genome-wide significant" line for trait1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomewideline_trait2</code></td>
<td>
<p>Where to draw the "genome-wide significant" line for trait2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomewideline_type</code></td>
<td>
<p>A character string denoting the type of line to be used for the
"genome-wide significant" line. This is the same for both traits. Default is dashed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomewideline_color</code></td>
<td>
<p>A character string denoting the color to be used for the
"genome-wide significant" line. This is the same for both traits. Default is red.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestiveline_trait1</code></td>
<td>
<p>Where to draw the "suggestive" line for trait1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestiveline_trait2</code></td>
<td>
<p>Where to draw the "suggestive" line for trait2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestiveline_type</code></td>
<td>
<p>A character string denoting the type of line to be used for the
"suggestive" line. This is the same for both traits. Default is dashed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestiveline_color</code></td>
<td>
<p>A character string denoting the color to be used for the
"suggestive" line. This is the same for both traits. Default is blue.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Create a Mirrored Manhattan Plot from a tab-delimited file or data frame with the compulsory
columns: "CHR", "SNP", "BP", "P", "Trait" .
</p>


<h3>Value</h3>

<p>A Mirrored Manhattan plot for two traits.
</p>


<h3>Author(s)</h3>

<p>Lindokuhle Nkambule
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
mirrored_man_plot(inputData)

## End(Not run)

</code></pre>


</div>