<div class="container">

<table style="width: 100%;"><tr>
<td>filter_variants_by_standard_chromosomes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Filter variants by standard human chromosomes.</h2>

<h3>Description</h3>

<p>This function filters a variants object by standard human chromosomes, i.e.,
1â€“22, X and Y. In addition to these chromosomes, some variants retrieved
from the GWAS Catalog might be also mapped to non-standard locations, such as
GRC assembly patches, haplotype (HAPs) or pseudo autosomal regions (PARs).
When this happens the main table <code>variants</code> includes rows for these
cases too. This function removes these.
</p>


<h3>Usage</h3>

<pre><code class="language-R">filter_variants_by_standard_chromosomes(
  s4_variants,
  chromosomes = c(seq_len(22), "X", "Y", "MT")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>s4_variants</code></td>
<td>
<p>An object of class variants.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosomes</code></td>
<td>
<p>A character vector of valid chromosome names. Default is
autosomal chromosomes 1 thru 22 and, X, Y, and MT.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class variants.
</p>


</div>