<div class="container">

<table style="width: 100%;"><tr>
<td>studies-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>An S4 class to represent a set of GWAS Catalog studies</h2>

<h3>Description</h3>

<p>The studies object consists of eight slots, each a table
(<code>tibble</code>), that combined form a relational database of a
subset of GWAS Catalog studies. Each study is an observation (row) in the
<code>studies</code> table â€” main table. All tables have the column
<code>study_id</code> as primary key.
</p>


<h3>Slots</h3>


<dl>
<dt><code>studies</code></dt>
<dd>
<dl>
<dt>study_id</dt>
<dd>
<p>GWAS Catalog study accession identifier, e.g.,
<code>"GCST002735"</code>.</p>
</dd>
<dt>reported_trait</dt>
<dd>
<p>Phenotypic trait as reported by the authors of the
study, e.g. <code>"Breast cancer"</code>.</p>
</dd>
<dt>initial_sample_size</dt>
<dd>
<p>Free text description of the initial cohort sample
size.</p>
</dd>
<dt>replication_sample_size</dt>
<dd>
<p>Free text description of the replication
cohort sample size.</p>
</dd>
<dt>gxe</dt>
<dd>
<p>Whether the study investigates a gene-environment interaction.</p>
</dd>
<dt>gxg</dt>
<dd>
<p>Whether the study investigates a gene-gene interaction.</p>
</dd>
<dt>snp_count</dt>
<dd>
<p>Number of variants passing quality control.</p>
</dd>
<dt>qualifier</dt>
<dd>
<p>Qualifier of number of variants passing quality control.</p>
</dd>
<dt>imputed</dt>
<dd>
<p>Whether variants were imputed.</p>
</dd>
<dt>pooled</dt>
<dd>
<p>Whether samples were pooled.</p>
</dd>
<dt>study_design_comment</dt>
<dd>
<p>Any other relevant study design information.</p>
</dd>
<dt>full_pvalue_set</dt>
<dd>
<p>Whether full summary statistics are available for this
study.</p>
</dd>
<dt>user_requested</dt>
<dd>
<p>Whether the addition of this study to the GWAS Catalog
was requested by a user.</p>
</dd>
</dl>
</dd>
<dt><code>genotyping_techs</code></dt>
<dd>
<p>A <code>tibble</code> listing genotyping
technologies employed in each study. Columns:
</p>

<dl>
<dt>study_id</dt>
<dd>
<p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>genotyping_technology</dt>
<dd>
<p>Genotyping technology employed, e.g.
<code>"Exome genotyping array"</code>, <code>"Exome-wide sequencing"</code>,
<code>"Genome-wide genotyping array"</code>, <code>"Genome-wide sequencing"</code>, or
<code>"Targeted genotyping array"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>platforms</code></dt>
<dd>
<p>A <code>tibble</code> listing platforms used per
study.
</p>

<dl>
<dt>study_id</dt>
<dd>
<p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>manufacturer</dt>
<dd>
<p>Platform manufacturer, e.g., <code>"Affymetrix"</code>,
<code>"Illumina"</code>, or <code>"Perlegen"</code>.</p>
</dd>
</dl>
</dd>
<dt><code>ancestries</code></dt>
<dd>
<p>A <code>tibble</code> listing ancestry of samples
used in each study.
</p>

<dl>
<dt>study_id</dt>
<dd>
<p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>ancestry_id</dt>
<dd>
<p>Ancestry identifier.</p>
</dd>
<dt>type</dt>
<dd>
<p>Stage of the ancestry sample: either <code>'initial'</code> or
<code>'replication'</code>.</p>
</dd>
<dt>number_of_individuals</dt>
<dd>
<p>Number of individuals comprising this ancestry
sample.</p>
</dd>
</dl>
</dd>
<dt><code>ancestral_groups</code></dt>
<dd>
<p>A <code>tibble</code> listing ancestral
groups used in each ancestry.
</p>

<dl>
<dt>study_id</dt>
<dd>
<p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>ancestry_id</dt>
<dd>
<p>Ancestry identifier.</p>
</dd>
<dt>ancestral_group</dt>
<dd>
<p>Genetic ancestry groups present in the sample.</p>
</dd>
</dl>
</dd>
<dt><code>countries_of_origin</code></dt>
<dd>
<p>A <code>tibble</code> listing countries of
origin of samples.
</p>

<dl>
<dt>study_id</dt>
<dd>
<p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>ancestry_id</dt>
<dd>
<p>Ancestry identifier.</p>
</dd>
<dt>country_name</dt>
<dd>
<p>Country name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</dd>
<dt>major_area</dt>
<dd>
<p>Region name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</dd> <dt>region</dt>
<dd>
<p>Sub-region name,
according to <a href="https://unstats.un.org/unsd/methodology/m49/overview/">The
United Nations M49 Standard of Geographic Regions.</a></p>
</dd>
</dl>
</dd>
<dt><code>countries_of_recruitment</code></dt>
<dd>
<p>A <code>tibble</code> listing
countries of recruitment of samples.
</p>

<dl>
<dt>study_id</dt>
<dd>
<p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>ancestry_id</dt>
<dd>
<p>Ancestry identifier.</p>
</dd>
<dt>country_name</dt>
<dd>
<p>Country name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</dd>
<dt>major_area</dt>
<dd>
<p>Region name, according to
<a href="https://unstats.un.org/unsd/methodology/m49/overview/">The United
Nations M49 Standard of Geographic Regions.</a></p>
</dd> <dt>region</dt>
<dd>
<p>Sub-region name,
according to <a href="https://unstats.un.org/unsd/methodology/m49/overview/">The
United Nations M49 Standard of Geographic Regions.</a></p>
</dd>
</dl>
</dd>
<dt><code>publications</code></dt>
<dd>
<p>A <code>tibble</code> listing publications
associated with each study.
</p>

<dl>
<dt>study_id</dt>
<dd>
<p>GWAS Catalog study accession identifier.</p>
</dd>
<dt>pubmed_id</dt>
<dd>
<p><a href="https://en.wikipedia.org/wiki/PubMed">PubMed</a>
identifier.</p>
</dd>
<dt>publication_date</dt>
<dd>
<p>Publication date (online date if available) formatted
as <code>ymd</code>.</p>
</dd>
<dt>publication</dt>
<dd>
<p>Abbreviated journal name.</p>
</dd>
<dt>title</dt>
<dd>
<p>Publication title.</p>
</dd>
<dt>author_fullname</dt>
<dd>
<p>Last name and initials of first author.</p>
</dd>
<dt>author_orcid</dt>
<dd>
<p>Author's <a href="https://en.wikipedia.org/wiki/ORCID">ORCID
iD</a> (Open Researcher and Contributor ID).</p>
</dd>
</dl>
</dd>
</dl>
</div>