<div class="container">

<table style="width: 100%;"><tr>
<td>variants_tbl</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates a variants table.</h2>

<h3>Description</h3>

<p>Creates a variants table.
</p>


<h3>Usage</h3>

<pre><code class="language-R">variants_tbl(
  variant_id = character(),
  merged = integer(),
  functional_class = character(),
  chromosome_name = character(),
  chromosome_position = integer(),
  chromosome_region = character(),
  last_update_date = lubridate::ymd_hms()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>variant_id</code></td>
<td>
<p>A character vector of variant identifiers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>merged</code></td>
<td>
<p>A logical vector indicating if a SNP has been merged with
another SNP in a newer genome build.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>functional_class</code></td>
<td>
<p>A character vector of functional classes, see
<code>functional_class</code> in slot <code>variants</code> of variants.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome_name</code></td>
<td>
<p>A character vector of chromosome names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome_position</code></td>
<td>
<p>An integer vector of chromosome positions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome_region</code></td>
<td>
<p>A character vector of cytogenetic regions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>last_update_date</code></td>
<td>
<p>A <code>POSIXct</code> object
indicating the last time the variants have been updated.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>tibble</code> whose columns are the named arguments
to the function.
</p>


</div>