<div class="container">

<table style="width: 100%;"><tr>
<td>associations-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>An S4 class to represent a set of GWAS Catalog associations</h2>

<h3>Description</h3>

<p>The association object consists of six slots, each a table
(<code>tibble</code>), that combined form a relational database of a
subset of GWAS Catalog associations. Each association is an observation (row)
in the <code>associations</code> table â€” main table. All tables have the column
<code>association_id</code> as primary key.
</p>


<h3>Slots</h3>


<dl>
<dt><code>associations</code></dt>
<dd>
<p>A <code>tibble</code> listing associations.
Columns:
</p>

<dl>
<dt>association_id</dt>
<dd>
<p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>pvalue</dt>
<dd>
<p>Reported p-value for strongest variant risk or effect allele.</p>
</dd>
<dt>pvalue_description</dt>
<dd>
<p>Information describing context of p-value.</p>
</dd>
<dt>pvalue_mantissa</dt>
<dd>
<p>Mantissa of p-value.</p>
</dd>
<dt>pvalue_exponent</dt>
<dd>
<p>Exponent of p-value.</p>
</dd>
<dt>multiple_snp_haplotype</dt>
<dd>
<p>Whether the association is for a multi-SNP
haplotype.</p>
</dd>
<dt>snp_interaction</dt>
<dd>
<p>Whether the association is for a SNP-SNP interaction.</p>
</dd>
<dt>snp_type</dt>
<dd>
<p>Whether the SNP has previously been reported. Either
<code>'known'</code> or <code>'novel'</code>.</p>
</dd>
<dt>risk_frequency</dt>
<dd>
<p>Reported risk/effect allele frequency associated with
strongest SNP in controls.</p>
</dd>
<dt>standard_error</dt>
<dd>
<p>Standard error of the effect size.</p>
</dd>
<dt>range</dt>
<dd>
<p>Reported 95% confidence interval associated with strongest SNP
risk allele, along with unit in the case of beta coefficients. If 95% CIs
have not been not reported, these are estimated using the standard error,
when available.</p>
</dd>
<dt>or_per_copy_number</dt>
<dd>
<p>Reported odds ratio (OR) associated
with strongest SNP risk allele. Note that all ORs included in the
Catalog are &gt;1.</p>
</dd>
<dt>beta_number</dt>
<dd>
<p>Beta coefficient associated with strongest SNP risk
allele.</p>
</dd>
<dt>beta_unit</dt>
<dd>
<p>Beta coefficient unit.</p>
</dd>
<dt>beta_direction</dt>
<dd>
<p>Beta coefficient direction, either <code>'decrease'</code> or
<code>'increase'</code>.</p>
</dd>
<dt>beta_description</dt>
<dd>
<p>Additional beta coefficient comment.</p>
</dd>
<dt>last_mapping_date</dt>
<dd>
<p>Last time this association was mapped to Ensembl.</p>
</dd>
<dt>last_update_date</dt>
<dd>
<p>Last time this association was updated.</p>
</dd>
</dl>
</dd>
<dt><code>loci</code></dt>
<dd>
<p>A <code>tibble</code> listing loci. Columns:
</p>

<dl>
<dt>association_id</dt>
<dd>
<p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>locus_id</dt>
<dd>
<p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p>
</dd>
<dt>haplotype_snp_count</dt>
<dd>
<p>Number of variants per locus. Most loci are
single-SNP loci, i.e., there is a one to one relationship between a variant
and a <code>locus_id</code> (<code>haplotype_snp_count == NA</code>). There are however
cases of associations involving multiple loci at once, such as SNP-SNP
interactions and multi-SNP haplotypes. This is signalled in the columns:
<code>multiple_snp_haplotype</code> and <code>snp_interaction</code> with value
<code>TRUE</code>.</p>
</dd>
<dt>description</dt>
<dd>
<p>Description of the locus identifier, e.g.,
<code>'Single variant'</code>, <code>SNP x SNP interaction</code>, or <code>3-SNP
Haplotype</code>.</p>
</dd>
</dl>
</dd>
<dt><code>risk_alleles</code></dt>
<dd>
<p>A <code>tibble</code> listing risk alleles.
Columns:
</p>

<dl>
<dt>association_id</dt>
<dd>
<p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>locus_id</dt>
<dd>
<p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p>
</dd>
<dt>variant_id</dt>
<dd>
<p>Variant identifier, e.g., <code>'rs1333048'</code>.</p>
</dd>
<dt>risk_allele</dt>
<dd>
<p>Risk allele or effect allele.</p>
</dd>
<dt>risk_frequency</dt>
<dd>
<p>Reported risk/effect allele frequency associated with
strongest SNP in controls (if not available among all controls, among the
control group with the largest sample size). If the associated locus is a
haplotype the haplotype frequency will be extracted.</p>
</dd>
<dt>genome_wide</dt>
<dd>
<p>Whether this variant allele has been part of a genome-wide
study or not.</p>
</dd>
<dt>limited_list</dt>
<dd>
<p>Undocumented.</p>
</dd>
</dl>
</dd>
<dt><code>genes</code></dt>
<dd>
<p>A <code>tibble</code> listing author reported genes.
Columns:
</p>

<dl>
<dt>association_id</dt>
<dd>
<p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>locus_id</dt>
<dd>
<p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p>
</dd>
<dt>gene_name</dt>
<dd>
<p>Gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</dd> </dl>
</dd>
<dt><code>ensembl_ids</code></dt>
<dd>
<p>A <code>tibble</code> listing Ensembl gene
identifiers. Columns:
</p>

<dl>
<dt>association_id</dt>
<dd>
<p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>locus_id</dt>
<dd>
<p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p>
</dd>
<dt>gene_name</dt>
<dd>
<p>Gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</dd>
<dt>ensembl_id</dt>
<dd>
<p>The Ensembl identifier of an Ensembl gene, see Section
<a href="https://www.ensembl.org/info/genome/genebuild/index.html">Gene
annotation in Ensembl</a> for more information.</p>
</dd> </dl>
</dd>
<dt><code>entrez_ids</code></dt>
<dd>
<p>A <code>tibble</code> listing Entrez gene
identifiers. Columns:
</p>

<dl>
<dt>association_id</dt>
<dd>
<p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p>
</dd>
<dt>locus_id</dt>
<dd>
<p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p>
</dd>
<dt>gene_name</dt>
<dd>
<p>Gene symbol according to
<a href="https://www.genenames.org/">HUGO Gene Nomenclature (HGNC)</a>.</p>
</dd>
<dt>entrez_id</dt>
<dd>
<p>The Entrez identifier of a gene, see ref.
<a href="https://doi.org/10.1093/nar/gkq1237">doi:10.1093/nar/gkq1237</a> for
more information.</p>
</dd>
</dl>
</dd>
</dl>
</div>