<div class="container">

<table style="width: 100%;"><tr>
<td>get_variants_by_genomic_range</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get GWAS Catalog variants by genomic range</h2>

<h3>Description</h3>

<p>Gets variants by genomic range.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_variants_by_genomic_range(
  chromosome = NULL,
  start = NULL,
  end = NULL,
  verbose = FALSE,
  warnings = TRUE,
  page_size = 20L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>chromosome</code></td>
<td>
<p>A character vector of human chromosome names: autosomal
and sexual chromosomes only, i.e., 1â€“22, X and Y.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>Start position of range (starts at 1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>End position of range (inclusive).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Whether the function should be verbose about the different
queries or not.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>warnings</code></td>
<td>
<p>Whether to print warnings.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>page_size</code></td>
<td>
<p>An integer scalar indicating the
<a href="https://www.ebi.ac.uk/gwas/rest/docs/api#_paging_resources">page</a>
value to be used in the JSON requests, can be between <code>1</code> and
<code>1000</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A variants object.
</p>


</div>