<div class="container">

<table style="width: 100%;"><tr>
<td>simGenoFuncDiffPriors</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Simulate Genotype Data from a Mixture of 3 Bayesian Hierarchical Models
</h2>

<h3>Description</h3>

<p>Simulate Genotype Data from a Mixture of 3 Bayesian Hierarchical Models.
The minor allele frequency (MAF) of cases has different priors
than that of controls. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">simGenoFuncDiffPriors(
    nCases = 100, 
    nControls = 100, 
    nSNPs = 1000, 
    alpha.p.ca = 2, 
    beta.p.ca = 3, 
    alpha.p.co = 2, 
    beta.p.co = 8, 
    pi.p = 0.1, 
    alpha0 = 2, 
    beta0 = 5, 
    pi0 = 0.8, 
    alpha.n.ca = 2, 
    beta.n.ca = 8, 
    alpha.n.co = 2, 
    beta.n.co = 3, 
    pi.n = 0.1, 
    low = 0.02, 
    upp = 0.5, 
    verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nCases</code></td>
<td>

<p>integer. Number of cases.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nControls</code></td>
<td>

<p>integer. Number of controls.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nSNPs</code></td>
<td>

<p>integer. Number of SNPs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha.p.ca</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">+</code> for cases.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.p.ca</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">+</code> for cases. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha.p.co</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">+</code> for controls.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.p.co</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">+</code> for controls. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi.p</code></td>
<td>

<p>numeric. Mixture proportion for cluster <code class="reqn">+</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha0</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">0</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta0</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">0</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi0</code></td>
<td>

<p>numeric. Mixture proportion for cluster <code class="reqn">0</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha.n.ca</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">-</code> for cases.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.n.ca</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">-</code> for cases.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha.n.co</code></td>
<td>

<p>numeric. The first shape parameter of Beta prior in cluster <code class="reqn">-</code> for controls.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.n.co</code></td>
<td>

<p>numeric. The second shape parameter of Beta prior in cluster <code class="reqn">-</code> for controls.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi.n</code></td>
<td>

<p>numeric. Mixture proportion for cluster <code class="reqn">-</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>low</code></td>
<td>

<p>numeric. A small positive value. If a MAF generated from half-flat shape
bivariate prior is smaller than <code>low</code>, we will delete the SNP to be generated.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upp</code></td>
<td>

<p>numeric. A positive value. If a MAF generated from half-flat shape
bivariate prior is greater than <code>upp</code>, we will delete the SNP to be generated.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>logical. Indicating if intermediate results or final results should be output 
to output screen.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In this simulation, we generate additive-coded genotypes for 3 clusters of SNPs
based on a mixture of 3 Bayesian hierarchical models. 
</p>
<p>In cluster <code class="reqn">+</code>, the minor allele frequency
(MAF) <code class="reqn">\theta_{x+}</code> of cases is greater than the MAF <code class="reqn">\theta_{y+}</code> of
controls.
</p>
<p>In cluster <code class="reqn">0</code>, the MAF <code class="reqn">\theta_{0}</code> of cases is equal to 
the MAF of controls.
</p>
<p>In cluster <code class="reqn">-</code>, the MAF <code class="reqn">\theta_{x-}</code> of cases is smaller than 
the MAF <code class="reqn">\theta_{y-}</code> of controls.
</p>
<p>The proportions of the 3 clusters of SNPs are <code class="reqn">\pi_{+}</code>,
<code class="reqn">\pi_{0}</code>, and <code class="reqn">\pi_{-}</code>, respectively.
</p>
<p>We assume a “half-flat shape” bivariate prior for the MAF in
cluster <code class="reqn">+</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{+}\left(\theta_{x+}\right)h_{+}\left(\theta_{y+}\right)
I\left(\theta_{x+}&gt;\theta_{y+}\right),
</code>
</p>

<p>where <code class="reqn">I(a)</code> is hte indicator function taking value <code class="reqn">1</code> if the event 
<code class="reqn">a</code> is true, and value <code class="reqn">0</code> otherwise.
The function <code class="reqn">h_{+}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{+}, \beta_{+}\right)</code>.
</p>
<p>We assume <code class="reqn">\theta_{0}</code> has the beta prior <code class="reqn">Beta(\alpha_0, \beta_0)</code>. 
</p>
<p>We also assume a “half-flat shape” bivariate prior for the MAF in
cluster <code class="reqn">-</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{-}\left(\theta_{x-}\right)h_{-}\left(\theta_{y-}\right)
I\left(\theta_{x-}&gt;\theta_{y-}\right).
</code>
</p>

<p>The function <code class="reqn">h_{-}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{-}, \beta_{-}\right)</code>.
</p>
<p>Given a SNP, we assume Hardy-Weinberg equilibrium holds for its genotypes. 
That is, given MAF <code class="reqn">\theta</code>, the probabilities of genotypes are
</p>
<p style="text-align: center;"><code class="reqn">Pr(geno=2) = \theta^2</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=1) = 2\theta\left(1-\theta\right)</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=0) = \left(1-\theta\right)^2</code>
</p>

<p>We also assume the genotypes <code class="reqn">0</code> (wild-type), <code class="reqn">1</code> (heterozygote), and 
<code class="reqn">2</code> (mutation) follows a multinomial distribution
<code class="reqn">Multinomial\left\{1, \left[
\theta^2, 2\theta\left(1-\theta\right),
\left(1-\theta\right)^2
\right]\right\}</code>
</p>
<p>Note that when we generate MAFs from the half-flat shape bivariate priors, 
we might get very small MAFs or get MAFs <code class="reqn">&gt;0.5</code>. In these cased,
we then delete this SNP.
</p>
<p>So the final number of SNPs generated might be less than the initially-set 
number of SNPs.
</p>


<h3>Value</h3>

<p>An ExpressionSet object stores genotype data.
</p>


<h3>Author(s)</h3>

<p>Yan Xu &lt;yanxu@uvic.ca&gt;, 
Li Xing &lt;sfulxing@gmail.com&gt;, 
Jessica Su &lt;rejas@channing.harvard.edu&gt;, 
Xuekui Zhang &lt;xuekui@uvic.ca&gt;, 
Weiliang Qiu &lt;Weiliang.Qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Yan X, Xing L, Su J, Zhang X, Qiu W. 
Model-based clustering for identifying disease-associated SNPs in case-control genome-wide association studies. 
Scientific Reports 9, Article number: 13686 (2019) https://www.nature.com/articles/s41598-019-50229-6.
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(2)

esSimDiffPriors = simGenoFuncDiffPriors(
  nCases = 100,
  nControls = 100,
  nSNPs = 500,
  alpha.p.ca = 2, beta.p.ca = 3,
  alpha.p.co = 2, beta.p.co = 8, pi.p = 0.1,
  alpha0 = 2, beta0 = 5, pi0 = 0.8,
  alpha.n.ca = 2, beta.n.ca = 8,
  alpha.n.co = 2, beta.n.co = 3, pi.n = 0.1,
  low = 0.02, upp = 0.5, verbose = FALSE
)

print(esSimDiffPriors)

pDat = pData(esSimDiffPriors)
print(pDat[1:2,])
print(table(pDat$memSubjs))

fDat = fData(esSimDiffPriors)
print(fDat[1:2,])
print(table(fDat$memGenes))
print(table(fDat$memGenes2))
</code></pre>


</div>