<div class="container">

<table style="width: 100%;"><tr>
<td>esSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
An ExpressionSet Object Storing Simulated Genotype Data
</h2>

<h3>Description</h3>

<p>An ExpressionSet object storing simulated genotype data.
The minor allele frequency (MAF) of cases has the same prior
as that of controls. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("esSim")</code></pre>


<h3>Details</h3>

<p>In this simulation, we generate additive-coded genotypes for 3 clusters of SNPs
based on a mixture of 3 Bayesian hierarchical models. 
</p>
<p>In cluster <code class="reqn">+</code>, the minor allele frequency
(MAF) <code class="reqn">\theta_{x+}</code> of cases is greater than the MAF <code class="reqn">\theta_{y+}</code> of
controls.
</p>
<p>In cluster <code class="reqn">0</code>, the MAF <code class="reqn">\theta_{0}</code> of cases is equal to 
the MAF of controls.
</p>
<p>In cluster <code class="reqn">-</code>, the MAF <code class="reqn">\theta_{x-}</code> of cases is smaller than 
the MAF <code class="reqn">\theta_{y-}</code> of controls.
</p>
<p>The proportions of the 3 clusters of SNPs are <code class="reqn">\pi_{+}</code>,
<code class="reqn">\pi_{0}</code>, and <code class="reqn">\pi_{-}</code>, respectively.
</p>
<p>We assume a “half-flat shape” bivariate prior for the MAF in
cluster <code class="reqn">+</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{+}\left(\theta_{x+}\right)h_{+}\left(\theta_{y+}\right)
I\left(\theta_{x+}&gt;\theta_{y+}\right),
</code>
</p>

<p>where <code class="reqn">I(a)</code> is hte indicator function taking value <code class="reqn">1</code> if the event 
<code class="reqn">a</code> is true, and value <code class="reqn">0</code> otherwise.
The function <code class="reqn">h_{+}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{+}, \beta_{+}\right)</code>.
</p>
<p>We assume <code class="reqn">\theta_{0}</code> has the beta prior <code class="reqn">Beta(\alpha_0, \beta_0)</code>. 
</p>
<p>We also assume a “half-flat shape” bivariate prior for the MAF in
cluster <code class="reqn">-</code>
</p>
<p style="text-align: center;"><code class="reqn">2h_{-}\left(\theta_{x-}\right)h_{-}\left(\theta_{y-}\right)
I\left(\theta_{x-}&gt;\theta_{y-}\right).
</code>
</p>

<p>The function <code class="reqn">h_{-}</code> is the probability density function of the 
beta distribution <code class="reqn">Beta\left(\alpha_{-}, \beta_{-}\right)</code>.
</p>
<p>Given a SNP, we assume Hardy-Weinberg equilibrium holds for its genotypes. 
That is, given MAF <code class="reqn">\theta</code>, the probabilities of genotypes are
</p>
<p style="text-align: center;"><code class="reqn">Pr(geno=2) = \theta^2</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=1) = 2\theta\left(1-\theta\right)</code>
</p>

<p style="text-align: center;"><code class="reqn">Pr(geno=0) = \left(1-\theta\right)^2</code>
</p>

<p>We also assume the genotypes <code class="reqn">0</code> (wild-type), <code class="reqn">1</code> (heterozygote), and 
<code class="reqn">2</code> (mutation) follows a multinomial distribution
<code class="reqn">Multinomial\left\{1, \left[
\theta^2, 2\theta\left(1-\theta\right),
\left(1-\theta\right)^2
\right]\right\}</code>
</p>
<p>We set the number of cases as <code class="reqn">100</code>, the number of controls as 
<code class="reqn">100</code>,
and the number of SNPs as <code class="reqn">1000</code>.
</p>
<p>The hyperparameters are
<code class="reqn">\alpha_{+}=2</code>, <code class="reqn">\beta_{+}=5</code>, <code class="reqn">\pi_{+}=0.1</code>,
<code class="reqn">\alpha_{0}=2</code>, <code class="reqn">\beta_{0}=5</code>, <code class="reqn">\pi_{0}=0.8</code>,
<code class="reqn">\alpha_{-}=2</code>, <code class="reqn">\beta_{-}=5</code>, <code class="reqn">\pi_{-}=0.1</code>.
</p>
<p>Note that when we generate MAFs from the half-flat shape bivariate priors, 
we might get very small MAFs or get MAFs <code class="reqn">&gt;0.5</code>. In these cased,
we then delete this SNP.
</p>
<p>So the final number of SNPs generated might be less than the initially-set 
number <code class="reqn">1000</code> of SNPs.
</p>
<p>For the dataset stored in <code>esSim</code>, there are <code class="reqn">872</code> SNPs.
<code class="reqn">83</code> SNPs are in cluster -, <code class="reqn">714</code> SNPs are in cluster <code class="reqn">0</code>,
and <code class="reqn">75</code> SNPs are in cluster <code class="reqn">+</code>.
</p>


<h3>References</h3>

<p>Yan X, Xing L, Su J, Zhang X, Qiu W. 
Model-based clustering for identifying disease-associated SNPs in case-control genome-wide association studies. 
Scientific Reports 9, Article number: 13686 (2019) https://www.nature.com/articles/s41598-019-50229-6.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(esSim)
print(esSim)

pDat=pData(esSim)
print(pDat[1:2,])
print(table(pDat$memSubjs))

fDat=fData(esSim)
print(fDat[1:2,])
print(table(fDat$memGenes))
print(table(fDat$memGenes2))

</code></pre>


</div>