<div class="container">

<table style="width: 100%;"><tr>
<td>risk_alleles_tbl</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates a risk alleles table.</h2>

<h3>Description</h3>

<p>Creates a risk alleles table.
</p>


<h3>Usage</h3>

<pre><code class="language-R">risk_alleles_tbl(
  association_id = character(),
  locus_id = integer(),
  variant_id = character(),
  risk_allele = character(),
  risk_frequency = double(),
  genome_wide = logical(),
  limited_list = logical()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>association_id</code></td>
<td>
<p>A character vector of association identifiers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus_id</code></td>
<td>
<p>An integer vector of locus identifiers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variant_id</code></td>
<td>
<p>A character vector of variant identifiers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>risk_allele</code></td>
<td>
<p>A character vector of risk or effect allele names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>risk_frequency</code></td>
<td>
<p>A numeric vector of the frequency of risk or effect
alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_wide</code></td>
<td>
<p>A logical vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limited_list</code></td>
<td>
<p>A logical vector.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>tibble</code> whose columns are the named arguments
to the function.
</p>


</div>