<div class="container">

<table style="width: 100%;"><tr>
<td>fisher_test.gwid</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fisher's Exact Test for gwid count data</h2>

<h3>Description</h3>

<p>Fisher's Exact Test for gwid count data
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'gwid'
fisher_test(
  obj,
  caco,
  snp_start,
  snp_end,
  reference,
  alternative = c("two.sided", "greater", "less"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>An object of class gwid. Output of <code>build_gwid</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>caco</code></td>
<td>
<p>An object of class caco. Output of <code>case_control</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp_start</code></td>
<td>
<p>select starting position of snps.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp_end</code></td>
<td>
<p>select ending position of snp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>reference group of subjects in which we want to perform fisher test</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternative</code></td>
<td>
<p>indicates the alternative hypothesis and must be one of "two.sided", "greater" or "less".
You can specify just the initial letter. Only used in the 2 by 2 case</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>optional arguments to fisher.test</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>the output will be a test_snps (data.table) object including 3 columns:
“snp_pos”, “case_control”, and “value” which is
a p-values.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
piggyback::pb_download(repo = "soroushmdg/gwid",tag = "v0.0.1",dest = tempdir())
ibd_data_file &lt;- paste0(tempdir(),"//chr3.ibd")
genome_data_file &lt;- paste0(tempdir(),"//chr3.gds")
phase_data_file &lt;- paste0(tempdir(),"//chr3.vcf")
case_control_data_file &lt;- paste0(tempdir(),"//case-cont-RA.withmap.Rda")
# case-control data
case_control &lt;- gwid::case_control(case_control_rda = case_control_data_file)
names(case_control) #cases and controls group
summary(case_control) # in here, we only consider cases,cont1,cont2,cont3 groups in the study
case_control$cases[1:3] # first three subject names of cases group
# read SNP data (use SNPRelate to convert it to gds) and count number of minor alleles
snp_data_gds &lt;- gwid::build_gwas(gds_data = genome_data_file,
caco = case_control,gwas_generator = TRUE)
class(snp_data_gds)
names(snp_data_gds)
head(snp_data_gds$snps) # it has information about counts of minor alleles in each location.
# read haplotype data (output of beagle)
haplotype_data &lt;- gwid::build_phase(phased_vcf = phase_data_file,caco = case_control)
class(haplotype_data)
names(haplotype_data)
dim(haplotype_data$Hap.1) #22302 SNP and 1911 subjects
# read IBD data (output of Refined-IBD)
ibd_data &lt;- gwid::build_gwid(ibd_data = ibd_data_file,gwas = snp_data_gds)
class(ibd_data)
ibd_data$ibd # refined IBD output
ibd_data$res # count number of IBD for each SNP location
# plot count of IBD in chromosome 3
plot(ibd_data,y = c("cases","cont1"),ly = FALSE)
# Further investigate location between 117M and 122M
# significant number of IBD's in group cases, compare to cont1, cont2 and cont3.
plot(ibd_data,y = c("cases","cont1"),snp_start = 119026294,snp_end = 120613594,ly = FALSE)
model_fisher &lt;- gwid::fisher_test(ibd_data,case_control,reference = "cases",
snp_start = 119026294,snp_end = 120613594)
class(model_fisher)
plot(model_fisher, y = c("cases","cont1"),ly = FALSE)
hap_str &lt;- gwid::haplotype_structure(ibd_data,phase = haplotype_data,w = 10,
snp_start = 119026294,snp_end = 120613594)
haplo_freq &lt;- gwid::haplotype_frequency(hap_str)
plot(haplo_freq,y = c("cases", "cont1"),plot_type = "haplotype_structure_frequency",
nwin = 1, type = "version1",ly = FALSE)

</code></pre>


</div>