<div class="container">

<table style="width: 100%;"><tr>
<td>GWAS_quantitative</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GWAS_quantitative function
This function performs GWAS using plink2 and outputs the GWAS summary statistics with additive SNP effects. Users may save the output in a user-specified file (see example).</h2>

<h3>Description</h3>

<p>GWAS_quantitative function
This function performs GWAS using plink2 and outputs the GWAS summary statistics with additive SNP effects. Users may save the output in a user-specified file (see example).
</p>


<h3>Usage</h3>

<pre><code class="language-R">GWAS_quantitative(
  plink_path,
  b_file,
  Qphe_discovery,
  Qcov_discovery,
  thread = 20
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>plink_path</code></td>
<td>
<p>Path to the PLINK executable application</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b_file</code></td>
<td>
<p>Prefix of the binary files, where all .fam, .bed and .bim files have a common prefix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Qphe_discovery</code></td>
<td>
<p>Name (with file extension) of the phenotype file containing family ID, individual ID and phenotype of the discovery dataset as columns, without heading</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Qcov_discovery</code></td>
<td>
<p>Name (with file extension) of the covariate file containing family ID, individual ID, standardized covariate, square of standardized covariate, and/or confounders of the discovery dataset as columns, without heading</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thread</code></td>
<td>
<p>Number of threads used</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function will perform GWAS and output
</p>
<table><tr style="vertical-align: top;">
<td><code>Q_out.trd.sum</code></td>
<td>
<p>GWAS summary statistics with additive SNP effects</p>
</td>
</tr></table>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
x &lt;- GWAS_quantitative(plink_path, DummyData, Qphe_discovery, Qcov_discovery, 
thread = 20)
sink("Q_out.trd.sum") #to create a file in the working directory
write.table(x[c("ID", "A1", "BETA")], sep = " ", 
row.names = FALSE, quote = FALSE) #to write the output
sink() #to save the output
head(x) #to obtain the head of GWAS summary statistics of additive SNP effects
x$CHROM #to extract the chromosome number 
x$POS #to extract the base pair position 
x$ID #to extract the SNP ID 
x$REF #to extract the reference allele 
x$ALT #to extract the alternate allele 
x$A1 #to extract the minor allele 
x$OBS_CT #to extract the number of allele observations 
x$BETA #to extract the SNP effects
x$SE #to extract the standard errors of the SNP effects
x$T_STAT #to extract the test statistics 
x$P #to extract the p values 

## End(Not run)
</code></pre>


</div>